{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# <a id='toc1_'></a>[ClinVar Variant Analysis](#toc0_)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The clinvar_variation_analysis notebook contains an analysis on ClinVar variant data"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Table of contents**<a id='toc0_'></a>    \n",
    "- [ClinVar Variant Analysis](#toc1_)    \n",
    "  - [Initialize](#toc1_1_)    \n",
    "    - [Import necessary libraries](#toc1_1_1_)    \n",
    "    - [Create output directory](#toc1_1_2_)    \n",
    "    - [Import variant information file](#toc1_1_3_)    \n",
    "  - [Add Supported Status of Variant based on in.vrs_xform_plan.policy](#toc1_2_)    \n",
    "  - [Add Normalization Status of Variant based on out.errors](#toc1_3_)    \n",
    "    - [Set Normalize Status of Variant as T/F](#toc1_3_1_)    \n",
    "      - [Summary Table](#toc1_3_1_1_)    \n",
    "  - [Create subgroups based on Variant Status](#toc1_4_)    \n",
    "    - [Supported and Normalized Variants](#toc1_4_1_)    \n",
    "    - [Supported and Not Normalized Variants](#toc1_4_2_)    \n",
    "    - [Not Supported Variants](#toc1_4_3_)    \n",
    "  - [Counting variants from each group](#toc1_5_)    \n",
    "  - [Counting variant types for each group](#toc1_6_)    \n",
    "\n",
    "<!-- vscode-jupyter-toc-config\n",
    "\tnumbering=false\n",
    "\tanchor=true\n",
    "\tflat=false\n",
    "\tminLevel=1\n",
    "\tmaxLevel=6\n",
    "\t/vscode-jupyter-toc-config -->\n",
    "<!-- THIS CELL WILL BE REPLACED ON TOC UPDATE. DO NOT WRITE YOUR TEXT IN THIS CELL -->"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## <a id='toc1_1_'></a>[Initialize](#toc0_)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### <a id='toc1_1_1_'></a>[Import necessary libraries](#toc0_)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import ndjson\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "from pathlib import Path\n",
    "import gzip\n",
    "import os\n",
    "import sys\n",
    "\n",
    "analysis_dir = os.path.abspath(os.path.join(os.getcwd(), '..'))\n",
    "sys.path.append(analysis_dir)\n",
    "\n",
    "from utils import get_errors  # noqa: E402"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### <a id='toc1_1_2_'></a>[Create output directory](#toc0_)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "path = Path(\"variation_analysis_output\")\n",
    "path.mkdir(exist_ok=True)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### <a id='toc1_1_3_'></a>[Import variant information file](#toc0_)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "with gzip.open(\"vi-normalized-with-liftover.jsonl.gz\", \"rb\") as f:\n",
    "    file_content = ndjson.load(f)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.json_normalize(file_content)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## <a id='toc1_2_'></a>[Add Supported Status of Variant based on in.issue](#toc0_)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Checking for blanks"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "in.issue\n",
       "No viable variation members identified.                       2095\n",
       "haplotype and genotype variations are not supported.          1540\n",
       "sequence for accession not supported by vrs-python release     669\n",
       "repeat expressions are not supported.                          474\n",
       "unsupported hgvs expression.                                   457\n",
       "intronic positions are not resolvable in sequence.             271\n",
       "range copies are not supported.                                 95\n",
       "expression contains unbalaned paretheses.                        3\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df[\"in.issue\"].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "df[\"in.issue\"] = df[\"in.issue\"].fillna(\"None\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "df[\"support_status\"] = df[\"in.issue\"].copy()\n",
    "\n",
    "df.loc[df[\"support_status\"] == \"No viable variation members identified.\", \"support_status\"] = False\n",
    "df.loc[\n",
    "    df[\"support_status\"] == \"haplotype and genotype variations are not supported.\",\n",
    "    \"support_status\",\n",
    "] = False\n",
    "df.loc[df[\"support_status\"] == \"sequence for accession not supported by vrs-python release\", \"support_status\"] = False\n",
    "df.loc[df[\"support_status\"] == \"repeat expressions are not supported.\", \"support_status\"] = False\n",
    "df.loc[df[\"support_status\"] == \"unsupported hgvs expression.\", \"support_status\"] = False\n",
    "df.loc[df[\"support_status\"] == \"intronic positions are not resolvable in sequence.\", \"support_status\"] = False\n",
    "df.loc[df[\"support_status\"] == \"range copies are not supported.\", \"support_status\"] = False\n",
    "df.loc[\n",
    "    df[\"support_status\"] == \"expression contains unbalaned paretheses.\", \"support_status\"\n",
    "] = False\n",
    "df[\"support_status\"] = df[\"support_status\"].fillna(False).astype(bool)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "support_status\n",
       "True     3751863\n",
       "False       5604\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df[\"support_status\"].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "out.errors\n",
       "[\"Unable to get GRCh37/GRCh38 assembly for: NC_000023.9\"]                                  422\n",
       "[\"Unable to get GRCh37/GRCh38 assembly for: NC_000024.8\"]                                  392\n",
       "[\"Unable to get GRCh37/GRCh38 assembly for: NC_000002.10\"]                                 305\n",
       "[\"Unable to get GRCh37/GRCh38 assembly for: NC_000009.10\"]                                 279\n",
       "[\"Unable to get GRCh37/GRCh38 assembly for: NC_000015.8\"]                                  275\n",
       "                                                                                          ... \n",
       "[\"Unable to find classification for: NC_000001.10:g.(161224742_?)_(?_161429290)dup\"]         1\n",
       "[\"Unable to find classification for: NC_000008.10:g.(68912432_?)_(?_146295771)dup\"]          1\n",
       "[\"Unable to find classification for: NC_000009.10:g.(?_2934)_(140241935_140273252)dup\"]      1\n",
       "[\"Unable to find classification for: NC_000010.10:g.(46292022_?)_(?_48174779)dup\"]           1\n",
       "[\"Unable to find classification for: NC_000023.10:g.(77670699_?)_(?_155233731)del\"]          1\n",
       "Name: count, Length: 3338, dtype: int64"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df[\"out.errors\"].value_counts()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## <a id='toc1_3_'></a>[Add Normalization Status of Variant based on out.errors](#toc0_)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The errors are stored as a list of values, some of which are strings and other of which are dictionaries (determined by whether error was handled at the level of Variation Normalizer or after the normalizer)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The \"get_errors\" function extracts the text error responses for better readability and ease string processing"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "df[\"error_string\"] = df[\"out.errors\"].apply(get_errors)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "error_string\n",
       "Success                                                                            3748697\n",
       "Unable to get GRCh37/GRCh38 assembly for: NC_000023.9                                  422\n",
       "Unable to get GRCh37/GRCh38 assembly for: NC_000024.8                                  392\n",
       "Unable to get GRCh37/GRCh38 assembly for: NC_000002.10                                 305\n",
       "Unable to get GRCh37/GRCh38 assembly for: NC_000009.10                                 279\n",
       "                                                                                    ...   \n",
       "Unable to find classification for: NC_000010.9:g.(?_62842)_(637331_658595)del            1\n",
       "Unable to find classification for: NC_000012.10:g.(?_18636)_(136086_140673)del           1\n",
       "Unable to find classification for: NC_000014.8:g.(66881928_?)_(?_67493801)del            1\n",
       "Unable to find classification for: NC_000016.9:g.(21596299_?)_(?_21946045)del            1\n",
       "Unable to find classification for: NC_000023.10:g.(77670699_?)_(?_155233731)del          1\n",
       "Name: count, Length: 3339, dtype: int64"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df[\"error_string\"].value_counts()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Check through to update `in.issue` and `support_status` for variants described on NCBI 36"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "mask = df[\"error_string\"].str.contains(\"Unable to get GRCh37/GRCh38 assembly\", na=False)\n",
    "\n",
    "df.loc[mask, \"in.issue\"] = \"Variant not described on GRCh37/GRCh38 assembly\"\n",
    "df.loc[mask, \"support_status\"] = False"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "There are Not Supported variants that have no error (marked as success inaccurately) because they were labeled \"Not Supported\" manually."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "An error (\"Not Supported\") is entered manually for those variants so that they are not categorized as normalized"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "df.loc[\n",
    "    (~df[\"support_status\"]) & (df[\"error_string\"] == \"Success\"),\n",
    "    \"error_string\",\n",
    "] = \"Not Supported\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### <a id='toc1_3_1_'></a>[Set Normalize Status of Variant as T/F](#toc0_)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "If an error is present, the variant was not normalized and therefore has a False Normalize Status"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>out</th>\n",
       "      <th>in.variation_id</th>\n",
       "      <th>in.name</th>\n",
       "      <th>in.vrs_class</th>\n",
       "      <th>in.range_copies</th>\n",
       "      <th>in.issue</th>\n",
       "      <th>in.variation_type</th>\n",
       "      <th>in.subclass_type</th>\n",
       "      <th>in.cytogenetic</th>\n",
       "      <th>in.mappings</th>\n",
       "      <th>...</th>\n",
       "      <th>out.state.type</th>\n",
       "      <th>out.state.length</th>\n",
       "      <th>out.state.sequence</th>\n",
       "      <th>out.state.repeatSubunitLength</th>\n",
       "      <th>out.extensions</th>\n",
       "      <th>in.absolute_copies</th>\n",
       "      <th>out.copies</th>\n",
       "      <th>support_status</th>\n",
       "      <th>error_string</th>\n",
       "      <th>normalize_status</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NaN</td>\n",
       "      <td>561260</td>\n",
       "      <td>46,XX,der(9)(q21.2,q21.2).seq[GRCh37/hg19]der(...</td>\n",
       "      <td>Not Available</td>\n",
       "      <td>[]</td>\n",
       "      <td>No viable variation members identified.</td>\n",
       "      <td>Complex</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>9q21.2</td>\n",
       "      <td>[]</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NaN</td>\n",
       "      <td>221359</td>\n",
       "      <td>chr1:17555508-24706269 complex variant</td>\n",
       "      <td>Not Available</td>\n",
       "      <td>[]</td>\n",
       "      <td>No viable variation members identified.</td>\n",
       "      <td>Complex</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>1p36.13-36.11</td>\n",
       "      <td>[]</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NaN</td>\n",
       "      <td>3897886</td>\n",
       "      <td>NM_001003722.2:c.[1807C&gt;T];[1705C&gt;T]</td>\n",
       "      <td>Not Available</td>\n",
       "      <td>[]</td>\n",
       "      <td>haplotype and genotype variations are not supp...</td>\n",
       "      <td>CompoundHeterozygote</td>\n",
       "      <td>Genotype</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[{'system': 'ClinVar', 'code': '3897886', 'rel...</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NaN</td>\n",
       "      <td>982523</td>\n",
       "      <td>NM_025114.4(CEP290):c.[4655AAG[2]];[6012-12T&gt;A]</td>\n",
       "      <td>Not Available</td>\n",
       "      <td>[]</td>\n",
       "      <td>haplotype and genotype variations are not supp...</td>\n",
       "      <td>CompoundHeterozygote</td>\n",
       "      <td>Genotype</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[{'system': 'ClinVar', 'code': '982523', 'rela...</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NaN</td>\n",
       "      <td>427839</td>\n",
       "      <td>NM_023073.3(CPLANE1):c.[2080A&gt;G];[6968A&gt;C]</td>\n",
       "      <td>Not Available</td>\n",
       "      <td>[]</td>\n",
       "      <td>haplotype and genotype variations are not supp...</td>\n",
       "      <td>CompoundHeterozygote</td>\n",
       "      <td>Genotype</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[{'system': 'ClinVar', 'code': '427839', 'rela...</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3757462</th>\n",
       "      <td>NaN</td>\n",
       "      <td>693214</td>\n",
       "      <td>NC_012920.1(MT-CO3):m.9739C&gt;T</td>\n",
       "      <td>Allele</td>\n",
       "      <td>[]</td>\n",
       "      <td>None</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[]</td>\n",
       "      <td>...</td>\n",
       "      <td>LiteralSequenceExpression</td>\n",
       "      <td>NaN</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>Success</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3757463</th>\n",
       "      <td>NaN</td>\n",
       "      <td>693231</td>\n",
       "      <td>NC_012920.1(MT-CO3):m.9813T&gt;C</td>\n",
       "      <td>Allele</td>\n",
       "      <td>[]</td>\n",
       "      <td>None</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[]</td>\n",
       "      <td>...</td>\n",
       "      <td>LiteralSequenceExpression</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>Success</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3757464</th>\n",
       "      <td>NaN</td>\n",
       "      <td>693257</td>\n",
       "      <td>NC_012920.1(MT-CO3):m.9984G&gt;A</td>\n",
       "      <td>Allele</td>\n",
       "      <td>[]</td>\n",
       "      <td>None</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[]</td>\n",
       "      <td>...</td>\n",
       "      <td>LiteralSequenceExpression</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>Success</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3757465</th>\n",
       "      <td>NaN</td>\n",
       "      <td>3281075</td>\n",
       "      <td>NM_080615.1:c.1078G&gt;A</td>\n",
       "      <td>Allele</td>\n",
       "      <td>[]</td>\n",
       "      <td>None</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[]</td>\n",
       "      <td>...</td>\n",
       "      <td>LiteralSequenceExpression</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>Success</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3757466</th>\n",
       "      <td>NaN</td>\n",
       "      <td>4028921</td>\n",
       "      <td>NM_080615.1:c.50T&gt;C</td>\n",
       "      <td>Allele</td>\n",
       "      <td>[]</td>\n",
       "      <td>None</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[]</td>\n",
       "      <td>...</td>\n",
       "      <td>LiteralSequenceExpression</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>Success</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3757467 rows × 40 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         out in.variation_id  \\\n",
       "0        NaN          561260   \n",
       "1        NaN          221359   \n",
       "2        NaN         3897886   \n",
       "3        NaN          982523   \n",
       "4        NaN          427839   \n",
       "...      ...             ...   \n",
       "3757462  NaN          693214   \n",
       "3757463  NaN          693231   \n",
       "3757464  NaN          693257   \n",
       "3757465  NaN         3281075   \n",
       "3757466  NaN         4028921   \n",
       "\n",
       "                                                   in.name   in.vrs_class  \\\n",
       "0        46,XX,der(9)(q21.2,q21.2).seq[GRCh37/hg19]der(...  Not Available   \n",
       "1                   chr1:17555508-24706269 complex variant  Not Available   \n",
       "2                     NM_001003722.2:c.[1807C>T];[1705C>T]  Not Available   \n",
       "3          NM_025114.4(CEP290):c.[4655AAG[2]];[6012-12T>A]  Not Available   \n",
       "4               NM_023073.3(CPLANE1):c.[2080A>G];[6968A>C]  Not Available   \n",
       "...                                                    ...            ...   \n",
       "3757462                      NC_012920.1(MT-CO3):m.9739C>T         Allele   \n",
       "3757463                      NC_012920.1(MT-CO3):m.9813T>C         Allele   \n",
       "3757464                      NC_012920.1(MT-CO3):m.9984G>A         Allele   \n",
       "3757465                              NM_080615.1:c.1078G>A         Allele   \n",
       "3757466                                NM_080615.1:c.50T>C         Allele   \n",
       "\n",
       "        in.range_copies                                           in.issue  \\\n",
       "0                    []            No viable variation members identified.   \n",
       "1                    []            No viable variation members identified.   \n",
       "2                    []  haplotype and genotype variations are not supp...   \n",
       "3                    []  haplotype and genotype variations are not supp...   \n",
       "4                    []  haplotype and genotype variations are not supp...   \n",
       "...                 ...                                                ...   \n",
       "3757462              []                                               None   \n",
       "3757463              []                                               None   \n",
       "3757464              []                                               None   \n",
       "3757465              []                                               None   \n",
       "3757466              []                                               None   \n",
       "\n",
       "                 in.variation_type in.subclass_type in.cytogenetic  \\\n",
       "0                          Complex     SimpleAllele         9q21.2   \n",
       "1                          Complex     SimpleAllele  1p36.13-36.11   \n",
       "2             CompoundHeterozygote         Genotype            NaN   \n",
       "3             CompoundHeterozygote         Genotype            NaN   \n",
       "4             CompoundHeterozygote         Genotype            NaN   \n",
       "...                            ...              ...            ...   \n",
       "3757462  single nucleotide variant     SimpleAllele            NaN   \n",
       "3757463  single nucleotide variant     SimpleAllele            NaN   \n",
       "3757464  single nucleotide variant     SimpleAllele            NaN   \n",
       "3757465  single nucleotide variant     SimpleAllele            NaN   \n",
       "3757466  single nucleotide variant     SimpleAllele            NaN   \n",
       "\n",
       "                                               in.mappings  ...  \\\n",
       "0                                                       []  ...   \n",
       "1                                                       []  ...   \n",
       "2        [{'system': 'ClinVar', 'code': '3897886', 'rel...  ...   \n",
       "3        [{'system': 'ClinVar', 'code': '982523', 'rela...  ...   \n",
       "4        [{'system': 'ClinVar', 'code': '427839', 'rela...  ...   \n",
       "...                                                    ...  ...   \n",
       "3757462                                                 []  ...   \n",
       "3757463                                                 []  ...   \n",
       "3757464                                                 []  ...   \n",
       "3757465                                                 []  ...   \n",
       "3757466                                                 []  ...   \n",
       "\n",
       "                    out.state.type out.state.length out.state.sequence  \\\n",
       "0                              NaN              NaN                NaN   \n",
       "1                              NaN              NaN                NaN   \n",
       "2                              NaN              NaN                NaN   \n",
       "3                              NaN              NaN                NaN   \n",
       "4                              NaN              NaN                NaN   \n",
       "...                            ...              ...                ...   \n",
       "3757462  LiteralSequenceExpression              NaN                  T   \n",
       "3757463  LiteralSequenceExpression              NaN                  C   \n",
       "3757464  LiteralSequenceExpression              NaN                  A   \n",
       "3757465  LiteralSequenceExpression              NaN                  A   \n",
       "3757466  LiteralSequenceExpression              NaN                  C   \n",
       "\n",
       "        out.state.repeatSubunitLength out.extensions in.absolute_copies  \\\n",
       "0                                 NaN            NaN                NaN   \n",
       "1                                 NaN            NaN                NaN   \n",
       "2                                 NaN            NaN                NaN   \n",
       "3                                 NaN            NaN                NaN   \n",
       "4                                 NaN            NaN                NaN   \n",
       "...                               ...            ...                ...   \n",
       "3757462                           NaN            NaN                NaN   \n",
       "3757463                           NaN            NaN                NaN   \n",
       "3757464                           NaN            NaN                NaN   \n",
       "3757465                           NaN            NaN                NaN   \n",
       "3757466                           NaN            NaN                NaN   \n",
       "\n",
       "        out.copies support_status   error_string normalize_status  \n",
       "0              NaN          False  Not Supported            False  \n",
       "1              NaN          False  Not Supported            False  \n",
       "2              NaN          False  Not Supported            False  \n",
       "3              NaN          False  Not Supported            False  \n",
       "4              NaN          False  Not Supported            False  \n",
       "...            ...            ...            ...              ...  \n",
       "3757462        NaN           True        Success             True  \n",
       "3757463        NaN           True        Success             True  \n",
       "3757464        NaN           True        Success             True  \n",
       "3757465        NaN           True        Success             True  \n",
       "3757466        NaN           True        Success             True  \n",
       "\n",
       "[3757467 rows x 40 columns]"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df[\"normalize_status\"] = df[\"error_string\"] == \"Success\"\n",
    "df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### <a id='toc1_3_1_1_'></a>[Summary Table](#toc0_)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "In the table below, the cells show the number of variants with each expected behavior and how they actually ended up performing."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "If a variant was in an \"expected to pass\" category and ends up as text, that is an instance of a normalizer failure on a supported variant"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "summary_df = (\n",
    "    df[[\"in.variation_id\", \"support_status\", \"in.issue\", \"out.type\"]]\n",
    "    .fillna(\"NONE\")\n",
    "    .groupby([\"support_status\", \"in.issue\", \"out.type\"])\n",
    "    .count()\n",
    "    .unstack(level=2)\n",
    "    .fillna(0)\n",
    "    .astype(int)[\"in.variation_id\"]\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "summary_df[\"VariantSum\"] = summary_df.sum(axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "summary_df[\"NormalizedSum\"] = summary_df[\n",
    "    [\"Allele\", \"CopyNumberChange\", \"CopyNumberCount\"]\n",
    "].sum(axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "summary_df[\"NormalizedPercent\"] = (\n",
    "    summary_df[\"NormalizedSum\"] / summary_df[\"VariantSum\"]\n",
    ").apply(lambda x: f\"{round(x * 100, 2)}%\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>out.type</th>\n",
       "      <th>Allele</th>\n",
       "      <th>CopyNumberChange</th>\n",
       "      <th>CopyNumberCount</th>\n",
       "      <th>NONE</th>\n",
       "      <th>NormalizedPercent</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>support_status</th>\n",
       "      <th>in.issue</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th rowspan=\"9\" valign=\"top\">False</th>\n",
       "      <th>No viable variation members identified.</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2095</td>\n",
       "      <td>0.0%</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Variant not described on GRCh37/GRCh38 assembly</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4631</td>\n",
       "      <td>0.0%</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>expression contains unbalaned paretheses.</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>0.0%</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>haplotype and genotype variations are not supported.</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1540</td>\n",
       "      <td>0.0%</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>intronic positions are not resolvable in sequence.</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>271</td>\n",
       "      <td>0.0%</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>range copies are not supported.</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>95</td>\n",
       "      <td>0.0%</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>repeat expressions are not supported.</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>474</td>\n",
       "      <td>0.0%</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>sequence for accession not supported by vrs-python release</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>669</td>\n",
       "      <td>0.0%</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>unsupported hgvs expression.</th>\n",
       "      <td>0</td>\n",
       "      <td>113</td>\n",
       "      <td>0</td>\n",
       "      <td>344</td>\n",
       "      <td>24.73%</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>True</th>\n",
       "      <th>None</th>\n",
       "      <td>3673493</td>\n",
       "      <td>32241</td>\n",
       "      <td>39379</td>\n",
       "      <td>2119</td>\n",
       "      <td>99.94%</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "out.type                                                            Allele  \\\n",
       "support_status in.issue                                                      \n",
       "False          No viable variation members identified.                   0   \n",
       "               Variant not described on GRCh37/GRCh38 assembly           0   \n",
       "               expression contains unbalaned paretheses.                 0   \n",
       "               haplotype and genotype variations are not suppo...        0   \n",
       "               intronic positions are not resolvable in sequence.        0   \n",
       "               range copies are not supported.                           0   \n",
       "               repeat expressions are not supported.                     0   \n",
       "               sequence for accession not supported by vrs-pyt...        0   \n",
       "               unsupported hgvs expression.                              0   \n",
       "True           None                                                3673493   \n",
       "\n",
       "out.type                                                           CopyNumberChange  \\\n",
       "support_status in.issue                                                               \n",
       "False          No viable variation members identified.                            0   \n",
       "               Variant not described on GRCh37/GRCh38 assembly                    0   \n",
       "               expression contains unbalaned paretheses.                          0   \n",
       "               haplotype and genotype variations are not suppo...                 0   \n",
       "               intronic positions are not resolvable in sequence.                 0   \n",
       "               range copies are not supported.                                    0   \n",
       "               repeat expressions are not supported.                              0   \n",
       "               sequence for accession not supported by vrs-pyt...                 0   \n",
       "               unsupported hgvs expression.                                     113   \n",
       "True           None                                                           32241   \n",
       "\n",
       "out.type                                                           CopyNumberCount  \\\n",
       "support_status in.issue                                                              \n",
       "False          No viable variation members identified.                           0   \n",
       "               Variant not described on GRCh37/GRCh38 assembly                   0   \n",
       "               expression contains unbalaned paretheses.                         0   \n",
       "               haplotype and genotype variations are not suppo...                0   \n",
       "               intronic positions are not resolvable in sequence.                0   \n",
       "               range copies are not supported.                                   0   \n",
       "               repeat expressions are not supported.                             0   \n",
       "               sequence for accession not supported by vrs-pyt...                0   \n",
       "               unsupported hgvs expression.                                      0   \n",
       "True           None                                                          39379   \n",
       "\n",
       "out.type                                                           NONE  \\\n",
       "support_status in.issue                                                   \n",
       "False          No viable variation members identified.             2095   \n",
       "               Variant not described on GRCh37/GRCh38 assembly     4631   \n",
       "               expression contains unbalaned paretheses.              3   \n",
       "               haplotype and genotype variations are not suppo...  1540   \n",
       "               intronic positions are not resolvable in sequence.   271   \n",
       "               range copies are not supported.                       95   \n",
       "               repeat expressions are not supported.                474   \n",
       "               sequence for accession not supported by vrs-pyt...   669   \n",
       "               unsupported hgvs expression.                         344   \n",
       "True           None                                                2119   \n",
       "\n",
       "out.type                                                          NormalizedPercent  \n",
       "support_status in.issue                                                              \n",
       "False          No viable variation members identified.                         0.0%  \n",
       "               Variant not described on GRCh37/GRCh38 assembly                 0.0%  \n",
       "               expression contains unbalaned paretheses.                       0.0%  \n",
       "               haplotype and genotype variations are not suppo...              0.0%  \n",
       "               intronic positions are not resolvable in sequence.              0.0%  \n",
       "               range copies are not supported.                                 0.0%  \n",
       "               repeat expressions are not supported.                           0.0%  \n",
       "               sequence for accession not supported by vrs-pyt...              0.0%  \n",
       "               unsupported hgvs expression.                                  24.73%  \n",
       "True           None                                                          99.94%  "
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "summary_df = summary_df.drop([\"VariantSum\", \"NormalizedSum\"], axis=1)\n",
    "summary_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "summary_df.to_csv(\"variation_analysis_output/variant_analysis_summary_df.csv\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## <a id='toc1_4_'></a>[Create subgroups based on Variant Status](#toc0_)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### <a id='toc1_4_1_'></a>[Supported and Normalized Variants](#toc0_)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [],
   "source": [
    "supported_df = df.copy()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>out</th>\n",
       "      <th>in.variation_id</th>\n",
       "      <th>in.name</th>\n",
       "      <th>in.vrs_class</th>\n",
       "      <th>in.range_copies</th>\n",
       "      <th>in.issue</th>\n",
       "      <th>in.variation_type</th>\n",
       "      <th>in.subclass_type</th>\n",
       "      <th>in.cytogenetic</th>\n",
       "      <th>in.mappings</th>\n",
       "      <th>...</th>\n",
       "      <th>out.state.type</th>\n",
       "      <th>out.state.length</th>\n",
       "      <th>out.state.sequence</th>\n",
       "      <th>out.state.repeatSubunitLength</th>\n",
       "      <th>out.extensions</th>\n",
       "      <th>in.absolute_copies</th>\n",
       "      <th>out.copies</th>\n",
       "      <th>support_status</th>\n",
       "      <th>error_string</th>\n",
       "      <th>normalize_status</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>NaN</td>\n",
       "      <td>3247636</td>\n",
       "      <td>NC_000001.10:g.(?_100381624)_(100382277_?)del</td>\n",
       "      <td>CopyNumberChange</td>\n",
       "      <td>[]</td>\n",
       "      <td>None</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>1p21.2</td>\n",
       "      <td>[]</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>Success</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>NaN</td>\n",
       "      <td>2425513</td>\n",
       "      <td>NC_000001.10:g.(?_153963273)_(154580482_?)del</td>\n",
       "      <td>CopyNumberChange</td>\n",
       "      <td>[]</td>\n",
       "      <td>None</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>1q21.3</td>\n",
       "      <td>[]</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>Success</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>NaN</td>\n",
       "      <td>1067607</td>\n",
       "      <td>NC_000001.10:g.(?_156104586)_(156108907_?)del</td>\n",
       "      <td>CopyNumberChange</td>\n",
       "      <td>[]</td>\n",
       "      <td>None</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>1q22</td>\n",
       "      <td>[]</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>Success</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>NaN</td>\n",
       "      <td>3247983</td>\n",
       "      <td>NC_000001.10:g.(?_169446833)_(169454047_?)del</td>\n",
       "      <td>CopyNumberChange</td>\n",
       "      <td>[]</td>\n",
       "      <td>None</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>1q24.2</td>\n",
       "      <td>[]</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>Success</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>NaN</td>\n",
       "      <td>3247768</td>\n",
       "      <td>NC_000001.10:g.(?_183538257)_(183543776_?)del</td>\n",
       "      <td>CopyNumberChange</td>\n",
       "      <td>[]</td>\n",
       "      <td>None</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>1q25.3</td>\n",
       "      <td>[]</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>Success</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3757462</th>\n",
       "      <td>NaN</td>\n",
       "      <td>693214</td>\n",
       "      <td>NC_012920.1(MT-CO3):m.9739C&gt;T</td>\n",
       "      <td>Allele</td>\n",
       "      <td>[]</td>\n",
       "      <td>None</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[]</td>\n",
       "      <td>...</td>\n",
       "      <td>LiteralSequenceExpression</td>\n",
       "      <td>NaN</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>Success</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3757463</th>\n",
       "      <td>NaN</td>\n",
       "      <td>693231</td>\n",
       "      <td>NC_012920.1(MT-CO3):m.9813T&gt;C</td>\n",
       "      <td>Allele</td>\n",
       "      <td>[]</td>\n",
       "      <td>None</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[]</td>\n",
       "      <td>...</td>\n",
       "      <td>LiteralSequenceExpression</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>Success</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3757464</th>\n",
       "      <td>NaN</td>\n",
       "      <td>693257</td>\n",
       "      <td>NC_012920.1(MT-CO3):m.9984G&gt;A</td>\n",
       "      <td>Allele</td>\n",
       "      <td>[]</td>\n",
       "      <td>None</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[]</td>\n",
       "      <td>...</td>\n",
       "      <td>LiteralSequenceExpression</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>Success</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3757465</th>\n",
       "      <td>NaN</td>\n",
       "      <td>3281075</td>\n",
       "      <td>NM_080615.1:c.1078G&gt;A</td>\n",
       "      <td>Allele</td>\n",
       "      <td>[]</td>\n",
       "      <td>None</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[]</td>\n",
       "      <td>...</td>\n",
       "      <td>LiteralSequenceExpression</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>Success</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3757466</th>\n",
       "      <td>NaN</td>\n",
       "      <td>4028921</td>\n",
       "      <td>NM_080615.1:c.50T&gt;C</td>\n",
       "      <td>Allele</td>\n",
       "      <td>[]</td>\n",
       "      <td>None</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[]</td>\n",
       "      <td>...</td>\n",
       "      <td>LiteralSequenceExpression</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>Success</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3745114 rows × 40 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         out in.variation_id                                        in.name  \\\n",
       "30       NaN         3247636  NC_000001.10:g.(?_100381624)_(100382277_?)del   \n",
       "32       NaN         2425513  NC_000001.10:g.(?_153963273)_(154580482_?)del   \n",
       "33       NaN         1067607  NC_000001.10:g.(?_156104586)_(156108907_?)del   \n",
       "34       NaN         3247983  NC_000001.10:g.(?_169446833)_(169454047_?)del   \n",
       "35       NaN         3247768  NC_000001.10:g.(?_183538257)_(183543776_?)del   \n",
       "...      ...             ...                                            ...   \n",
       "3757462  NaN          693214                  NC_012920.1(MT-CO3):m.9739C>T   \n",
       "3757463  NaN          693231                  NC_012920.1(MT-CO3):m.9813T>C   \n",
       "3757464  NaN          693257                  NC_012920.1(MT-CO3):m.9984G>A   \n",
       "3757465  NaN         3281075                          NM_080615.1:c.1078G>A   \n",
       "3757466  NaN         4028921                            NM_080615.1:c.50T>C   \n",
       "\n",
       "             in.vrs_class in.range_copies in.issue          in.variation_type  \\\n",
       "30       CopyNumberChange              []     None                   Deletion   \n",
       "32       CopyNumberChange              []     None                   Deletion   \n",
       "33       CopyNumberChange              []     None                   Deletion   \n",
       "34       CopyNumberChange              []     None                   Deletion   \n",
       "35       CopyNumberChange              []     None                   Deletion   \n",
       "...                   ...             ...      ...                        ...   \n",
       "3757462            Allele              []     None  single nucleotide variant   \n",
       "3757463            Allele              []     None  single nucleotide variant   \n",
       "3757464            Allele              []     None  single nucleotide variant   \n",
       "3757465            Allele              []     None  single nucleotide variant   \n",
       "3757466            Allele              []     None  single nucleotide variant   \n",
       "\n",
       "        in.subclass_type in.cytogenetic in.mappings  ...  \\\n",
       "30          SimpleAllele         1p21.2          []  ...   \n",
       "32          SimpleAllele         1q21.3          []  ...   \n",
       "33          SimpleAllele           1q22          []  ...   \n",
       "34          SimpleAllele         1q24.2          []  ...   \n",
       "35          SimpleAllele         1q25.3          []  ...   \n",
       "...                  ...            ...         ...  ...   \n",
       "3757462     SimpleAllele            NaN          []  ...   \n",
       "3757463     SimpleAllele            NaN          []  ...   \n",
       "3757464     SimpleAllele            NaN          []  ...   \n",
       "3757465     SimpleAllele            NaN          []  ...   \n",
       "3757466     SimpleAllele            NaN          []  ...   \n",
       "\n",
       "                    out.state.type out.state.length out.state.sequence  \\\n",
       "30                             NaN              NaN                NaN   \n",
       "32                             NaN              NaN                NaN   \n",
       "33                             NaN              NaN                NaN   \n",
       "34                             NaN              NaN                NaN   \n",
       "35                             NaN              NaN                NaN   \n",
       "...                            ...              ...                ...   \n",
       "3757462  LiteralSequenceExpression              NaN                  T   \n",
       "3757463  LiteralSequenceExpression              NaN                  C   \n",
       "3757464  LiteralSequenceExpression              NaN                  A   \n",
       "3757465  LiteralSequenceExpression              NaN                  A   \n",
       "3757466  LiteralSequenceExpression              NaN                  C   \n",
       "\n",
       "        out.state.repeatSubunitLength out.extensions in.absolute_copies  \\\n",
       "30                                NaN            NaN                NaN   \n",
       "32                                NaN            NaN                NaN   \n",
       "33                                NaN            NaN                NaN   \n",
       "34                                NaN            NaN                NaN   \n",
       "35                                NaN            NaN                NaN   \n",
       "...                               ...            ...                ...   \n",
       "3757462                           NaN            NaN                NaN   \n",
       "3757463                           NaN            NaN                NaN   \n",
       "3757464                           NaN            NaN                NaN   \n",
       "3757465                           NaN            NaN                NaN   \n",
       "3757466                           NaN            NaN                NaN   \n",
       "\n",
       "        out.copies support_status error_string normalize_status  \n",
       "30             NaN           True      Success             True  \n",
       "32             NaN           True      Success             True  \n",
       "33             NaN           True      Success             True  \n",
       "34             NaN           True      Success             True  \n",
       "35             NaN           True      Success             True  \n",
       "...            ...            ...          ...              ...  \n",
       "3757462        NaN           True      Success             True  \n",
       "3757463        NaN           True      Success             True  \n",
       "3757464        NaN           True      Success             True  \n",
       "3757465        NaN           True      Success             True  \n",
       "3757466        NaN           True      Success             True  \n",
       "\n",
       "[3745114 rows x 40 columns]"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "supported_df = supported_df.loc[\n",
    "    (supported_df[\"support_status\"] & supported_df[\"normalize_status\"])\n",
    "]\n",
    "supported_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>in.variation_type</th>\n",
       "      <th>in.issue</th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>None</td>\n",
       "      <td>3401625</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Deletion</td>\n",
       "      <td>None</td>\n",
       "      <td>160232</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Duplication</td>\n",
       "      <td>None</td>\n",
       "      <td>73474</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Microsatellite</td>\n",
       "      <td>None</td>\n",
       "      <td>36095</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>copy number gain</td>\n",
       "      <td>None</td>\n",
       "      <td>21484</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>copy number loss</td>\n",
       "      <td>None</td>\n",
       "      <td>20367</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>Indel</td>\n",
       "      <td>None</td>\n",
       "      <td>17038</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>Insertion</td>\n",
       "      <td>None</td>\n",
       "      <td>13019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>Inversion</td>\n",
       "      <td>None</td>\n",
       "      <td>1401</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>Variation</td>\n",
       "      <td>None</td>\n",
       "      <td>379</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "           in.variation_type in.issue    count\n",
       "0  single nucleotide variant     None  3401625\n",
       "1                   Deletion     None   160232\n",
       "2                Duplication     None    73474\n",
       "3             Microsatellite     None    36095\n",
       "4           copy number gain     None    21484\n",
       "5           copy number loss     None    20367\n",
       "6                      Indel     None    17038\n",
       "7                  Insertion     None    13019\n",
       "8                  Inversion     None     1401\n",
       "9                  Variation     None      379"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "variation_type_count_supported_df = supported_df.value_counts(\n",
    "    [\"in.variation_type\", \"in.issue\"]\n",
    ").reset_index()\n",
    "variation_type_count_supported_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "variation_type_count_supported_df.to_csv(\n",
    "    \"variation_analysis_output/variation_type_count_supported_df.csv\"\n",
    ")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### <a id='toc1_4_2_'></a>[Supported and Not Normalized Variants](#toc0_)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [],
   "source": [
    "supported_not_normalized_df = df.copy()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>out</th>\n",
       "      <th>in.variation_id</th>\n",
       "      <th>in.name</th>\n",
       "      <th>in.vrs_class</th>\n",
       "      <th>in.range_copies</th>\n",
       "      <th>in.issue</th>\n",
       "      <th>in.variation_type</th>\n",
       "      <th>in.subclass_type</th>\n",
       "      <th>in.cytogenetic</th>\n",
       "      <th>in.mappings</th>\n",
       "      <th>...</th>\n",
       "      <th>out.state.type</th>\n",
       "      <th>out.state.length</th>\n",
       "      <th>out.state.sequence</th>\n",
       "      <th>out.state.repeatSubunitLength</th>\n",
       "      <th>out.extensions</th>\n",
       "      <th>in.absolute_copies</th>\n",
       "      <th>out.copies</th>\n",
       "      <th>support_status</th>\n",
       "      <th>error_string</th>\n",
       "      <th>normalize_status</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>NaN</td>\n",
       "      <td>3068946</td>\n",
       "      <td>NC_000001.10:g.(?_1447541)_(1454371_1455520)del</td>\n",
       "      <td>CopyNumberChange</td>\n",
       "      <td>[]</td>\n",
       "      <td>None</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>1p36.33</td>\n",
       "      <td>[]</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>Unable to find classification for: NC_000001.1...</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>572</th>\n",
       "      <td>NaN</td>\n",
       "      <td>3768691</td>\n",
       "      <td>NC_000003.11:g.(?_47021155)_(47021403_47030158...</td>\n",
       "      <td>CopyNumberChange</td>\n",
       "      <td>[]</td>\n",
       "      <td>None</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>3p21.31</td>\n",
       "      <td>[]</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>Unable to find classification for: NC_000003.1...</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1140</th>\n",
       "      <td>NaN</td>\n",
       "      <td>3233599</td>\n",
       "      <td>NC_000007.13:g.(?_117120078)_(117120202_117144...</td>\n",
       "      <td>CopyNumberChange</td>\n",
       "      <td>[]</td>\n",
       "      <td>None</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>7q31.2</td>\n",
       "      <td>[]</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>Unable to find classification for: NC_000007.1...</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1383</th>\n",
       "      <td>NaN</td>\n",
       "      <td>1704584</td>\n",
       "      <td>NC_000009.11:g.(108510472_108536145)_(10853889...</td>\n",
       "      <td>CopyNumberChange</td>\n",
       "      <td>[]</td>\n",
       "      <td>None</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>9q31.2</td>\n",
       "      <td>[]</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>Unable to find classification for: NC_000009.1...</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1838</th>\n",
       "      <td>NaN</td>\n",
       "      <td>1065232</td>\n",
       "      <td>NC_000012.11:g.(6204751_6219539)_(6233842_?)del</td>\n",
       "      <td>CopyNumberChange</td>\n",
       "      <td>[]</td>\n",
       "      <td>None</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>12p13.31</td>\n",
       "      <td>[]</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>Unable to find classification for: NC_000012.1...</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3689344</th>\n",
       "      <td>NaN</td>\n",
       "      <td>226245</td>\n",
       "      <td>GRCh37/hg19 15q11.2(chr15:22665220-23147419)</td>\n",
       "      <td>CopyNumberChange</td>\n",
       "      <td>[]</td>\n",
       "      <td>None</td>\n",
       "      <td>copy number loss</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>15q11.2</td>\n",
       "      <td>[]</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>Unable to find a GRCh38 accession for: NC_0000...</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3689346</th>\n",
       "      <td>NaN</td>\n",
       "      <td>807326</td>\n",
       "      <td>GRCh37/hg19 15q11.2(chr15:23006221-23021336)x1</td>\n",
       "      <td>CopyNumberCount</td>\n",
       "      <td>[]</td>\n",
       "      <td>None</td>\n",
       "      <td>copy number loss</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>15q11.2</td>\n",
       "      <td>[]</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>Unable to find a GRCh38 accession for: NC_0000...</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3689368</th>\n",
       "      <td>NaN</td>\n",
       "      <td>1706519</td>\n",
       "      <td>GRCh37/hg19 16q22.1-23.1(chr16:68971067-748235...</td>\n",
       "      <td>CopyNumberCount</td>\n",
       "      <td>[]</td>\n",
       "      <td>None</td>\n",
       "      <td>copy number loss</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>16q22.1-23.1</td>\n",
       "      <td>[]</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>Unable to find classification for: NC_000016.9...</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3689450</th>\n",
       "      <td>NaN</td>\n",
       "      <td>1706520</td>\n",
       "      <td>GRCh37/hg19 Xq21.1-28(chrX:77670699-155233731)x1</td>\n",
       "      <td>CopyNumberCount</td>\n",
       "      <td>[]</td>\n",
       "      <td>None</td>\n",
       "      <td>copy number loss</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>Xq21.1-28</td>\n",
       "      <td>[]</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>Unable to find classification for: NC_000023.1...</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3689493</th>\n",
       "      <td>NaN</td>\n",
       "      <td>2639135</td>\n",
       "      <td>Single allele</td>\n",
       "      <td>Allele</td>\n",
       "      <td>[]</td>\n",
       "      <td>None</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>1q21.2</td>\n",
       "      <td>[]</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>Unable to translate variation</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2118 rows × 40 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         out in.variation_id  \\\n",
       "31       NaN         3068946   \n",
       "572      NaN         3768691   \n",
       "1140     NaN         3233599   \n",
       "1383     NaN         1704584   \n",
       "1838     NaN         1065232   \n",
       "...      ...             ...   \n",
       "3689344  NaN          226245   \n",
       "3689346  NaN          807326   \n",
       "3689368  NaN         1706519   \n",
       "3689450  NaN         1706520   \n",
       "3689493  NaN         2639135   \n",
       "\n",
       "                                                   in.name      in.vrs_class  \\\n",
       "31         NC_000001.10:g.(?_1447541)_(1454371_1455520)del  CopyNumberChange   \n",
       "572      NC_000003.11:g.(?_47021155)_(47021403_47030158...  CopyNumberChange   \n",
       "1140     NC_000007.13:g.(?_117120078)_(117120202_117144...  CopyNumberChange   \n",
       "1383     NC_000009.11:g.(108510472_108536145)_(10853889...  CopyNumberChange   \n",
       "1838       NC_000012.11:g.(6204751_6219539)_(6233842_?)del  CopyNumberChange   \n",
       "...                                                    ...               ...   \n",
       "3689344       GRCh37/hg19 15q11.2(chr15:22665220-23147419)  CopyNumberChange   \n",
       "3689346     GRCh37/hg19 15q11.2(chr15:23006221-23021336)x1   CopyNumberCount   \n",
       "3689368  GRCh37/hg19 16q22.1-23.1(chr16:68971067-748235...   CopyNumberCount   \n",
       "3689450   GRCh37/hg19 Xq21.1-28(chrX:77670699-155233731)x1   CopyNumberCount   \n",
       "3689493                                      Single allele            Allele   \n",
       "\n",
       "        in.range_copies in.issue          in.variation_type in.subclass_type  \\\n",
       "31                   []     None                   Deletion     SimpleAllele   \n",
       "572                  []     None                   Deletion     SimpleAllele   \n",
       "1140                 []     None                   Deletion     SimpleAllele   \n",
       "1383                 []     None                   Deletion     SimpleAllele   \n",
       "1838                 []     None                   Deletion     SimpleAllele   \n",
       "...                 ...      ...                        ...              ...   \n",
       "3689344              []     None           copy number loss     SimpleAllele   \n",
       "3689346              []     None           copy number loss     SimpleAllele   \n",
       "3689368              []     None           copy number loss     SimpleAllele   \n",
       "3689450              []     None           copy number loss     SimpleAllele   \n",
       "3689493              []     None  single nucleotide variant     SimpleAllele   \n",
       "\n",
       "        in.cytogenetic in.mappings  ... out.state.type out.state.length  \\\n",
       "31             1p36.33          []  ...            NaN              NaN   \n",
       "572            3p21.31          []  ...            NaN              NaN   \n",
       "1140            7q31.2          []  ...            NaN              NaN   \n",
       "1383            9q31.2          []  ...            NaN              NaN   \n",
       "1838          12p13.31          []  ...            NaN              NaN   \n",
       "...                ...         ...  ...            ...              ...   \n",
       "3689344        15q11.2          []  ...            NaN              NaN   \n",
       "3689346        15q11.2          []  ...            NaN              NaN   \n",
       "3689368   16q22.1-23.1          []  ...            NaN              NaN   \n",
       "3689450      Xq21.1-28          []  ...            NaN              NaN   \n",
       "3689493         1q21.2          []  ...            NaN              NaN   \n",
       "\n",
       "        out.state.sequence out.state.repeatSubunitLength out.extensions  \\\n",
       "31                     NaN                           NaN            NaN   \n",
       "572                    NaN                           NaN            NaN   \n",
       "1140                   NaN                           NaN            NaN   \n",
       "1383                   NaN                           NaN            NaN   \n",
       "1838                   NaN                           NaN            NaN   \n",
       "...                    ...                           ...            ...   \n",
       "3689344                NaN                           NaN            NaN   \n",
       "3689346                NaN                           NaN            NaN   \n",
       "3689368                NaN                           NaN            NaN   \n",
       "3689450                NaN                           NaN            NaN   \n",
       "3689493                NaN                           NaN            NaN   \n",
       "\n",
       "        in.absolute_copies out.copies support_status  \\\n",
       "31                     NaN        NaN           True   \n",
       "572                    NaN        NaN           True   \n",
       "1140                   NaN        NaN           True   \n",
       "1383                   NaN        NaN           True   \n",
       "1838                   NaN        NaN           True   \n",
       "...                    ...        ...            ...   \n",
       "3689344                NaN        NaN           True   \n",
       "3689346                  1        NaN           True   \n",
       "3689368                  1        NaN           True   \n",
       "3689450                  1        NaN           True   \n",
       "3689493                NaN        NaN           True   \n",
       "\n",
       "                                              error_string normalize_status  \n",
       "31       Unable to find classification for: NC_000001.1...            False  \n",
       "572      Unable to find classification for: NC_000003.1...            False  \n",
       "1140     Unable to find classification for: NC_000007.1...            False  \n",
       "1383     Unable to find classification for: NC_000009.1...            False  \n",
       "1838     Unable to find classification for: NC_000012.1...            False  \n",
       "...                                                    ...              ...  \n",
       "3689344  Unable to find a GRCh38 accession for: NC_0000...            False  \n",
       "3689346  Unable to find a GRCh38 accession for: NC_0000...            False  \n",
       "3689368  Unable to find classification for: NC_000016.9...            False  \n",
       "3689450  Unable to find classification for: NC_000023.1...            False  \n",
       "3689493                      Unable to translate variation            False  \n",
       "\n",
       "[2118 rows x 40 columns]"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "supported_not_normalized_df = supported_not_normalized_df.loc[\n",
    "    (\n",
    "        supported_not_normalized_df[\"support_status\"]\n",
    "        & ~supported_not_normalized_df[\"normalize_status\"]\n",
    "    )\n",
    "]\n",
    "supported_not_normalized_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>in.variation_type</th>\n",
       "      <th>in.issue</th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Deletion</td>\n",
       "      <td>None</td>\n",
       "      <td>711</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>copy number loss</td>\n",
       "      <td>None</td>\n",
       "      <td>538</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>copy number gain</td>\n",
       "      <td>None</td>\n",
       "      <td>438</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Duplication</td>\n",
       "      <td>None</td>\n",
       "      <td>357</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>None</td>\n",
       "      <td>71</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>Indel</td>\n",
       "      <td>None</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>Insertion</td>\n",
       "      <td>None</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "           in.variation_type in.issue  count\n",
       "0                   Deletion     None    711\n",
       "1           copy number loss     None    538\n",
       "2           copy number gain     None    438\n",
       "3                Duplication     None    357\n",
       "4  single nucleotide variant     None     71\n",
       "5                      Indel     None      2\n",
       "6                  Insertion     None      1"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "variation_type_count_supported_not_normalized_df = (\n",
    "    supported_not_normalized_df.value_counts(\n",
    "        [\"in.variation_type\", \"in.issue\"]\n",
    "    ).reset_index()\n",
    ")\n",
    "variation_type_count_supported_not_normalized_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [],
   "source": [
    "variation_type_count_supported_not_normalized_df.to_csv(\n",
    "    \"variation_analysis_output/variation_type_count_supported_not_normalized_df.csv\"\n",
    ")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### <a id='toc1_4_3_'></a>[Not Supported Variants](#toc0_)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [],
   "source": [
    "not_supported_df = df.copy()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>out</th>\n",
       "      <th>in.variation_id</th>\n",
       "      <th>in.name</th>\n",
       "      <th>in.vrs_class</th>\n",
       "      <th>in.range_copies</th>\n",
       "      <th>in.issue</th>\n",
       "      <th>in.variation_type</th>\n",
       "      <th>in.subclass_type</th>\n",
       "      <th>in.cytogenetic</th>\n",
       "      <th>in.mappings</th>\n",
       "      <th>...</th>\n",
       "      <th>out.state.type</th>\n",
       "      <th>out.state.length</th>\n",
       "      <th>out.state.sequence</th>\n",
       "      <th>out.state.repeatSubunitLength</th>\n",
       "      <th>out.extensions</th>\n",
       "      <th>in.absolute_copies</th>\n",
       "      <th>out.copies</th>\n",
       "      <th>support_status</th>\n",
       "      <th>error_string</th>\n",
       "      <th>normalize_status</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NaN</td>\n",
       "      <td>561260</td>\n",
       "      <td>46,XX,der(9)(q21.2,q21.2).seq[GRCh37/hg19]der(...</td>\n",
       "      <td>Not Available</td>\n",
       "      <td>[]</td>\n",
       "      <td>No viable variation members identified.</td>\n",
       "      <td>Complex</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>9q21.2</td>\n",
       "      <td>[]</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NaN</td>\n",
       "      <td>221359</td>\n",
       "      <td>chr1:17555508-24706269 complex variant</td>\n",
       "      <td>Not Available</td>\n",
       "      <td>[]</td>\n",
       "      <td>No viable variation members identified.</td>\n",
       "      <td>Complex</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>1p36.13-36.11</td>\n",
       "      <td>[]</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NaN</td>\n",
       "      <td>3897886</td>\n",
       "      <td>NM_001003722.2:c.[1807C&gt;T];[1705C&gt;T]</td>\n",
       "      <td>Not Available</td>\n",
       "      <td>[]</td>\n",
       "      <td>haplotype and genotype variations are not supp...</td>\n",
       "      <td>CompoundHeterozygote</td>\n",
       "      <td>Genotype</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[{'system': 'ClinVar', 'code': '3897886', 'rel...</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NaN</td>\n",
       "      <td>982523</td>\n",
       "      <td>NM_025114.4(CEP290):c.[4655AAG[2]];[6012-12T&gt;A]</td>\n",
       "      <td>Not Available</td>\n",
       "      <td>[]</td>\n",
       "      <td>haplotype and genotype variations are not supp...</td>\n",
       "      <td>CompoundHeterozygote</td>\n",
       "      <td>Genotype</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[{'system': 'ClinVar', 'code': '982523', 'rela...</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NaN</td>\n",
       "      <td>427839</td>\n",
       "      <td>NM_023073.3(CPLANE1):c.[2080A&gt;G];[6968A&gt;C]</td>\n",
       "      <td>Not Available</td>\n",
       "      <td>[]</td>\n",
       "      <td>haplotype and genotype variations are not supp...</td>\n",
       "      <td>CompoundHeterozygote</td>\n",
       "      <td>Genotype</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[{'system': 'ClinVar', 'code': '427839', 'rela...</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3689488</th>\n",
       "      <td>NaN</td>\n",
       "      <td>41607</td>\n",
       "      <td>NM_000245.2:c.-14-3312A&gt;T</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>[]</td>\n",
       "      <td>No viable variation members identified.</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>7q31.2</td>\n",
       "      <td>[]</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3689489</th>\n",
       "      <td>NaN</td>\n",
       "      <td>3764520</td>\n",
       "      <td>G317D</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>[]</td>\n",
       "      <td>No viable variation members identified.</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[]</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3689490</th>\n",
       "      <td>NaN</td>\n",
       "      <td>16606</td>\n",
       "      <td>EPHX1, -4238T-A</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>[]</td>\n",
       "      <td>No viable variation members identified.</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>1q42.1</td>\n",
       "      <td>[]</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3689491</th>\n",
       "      <td>NaN</td>\n",
       "      <td>3901242</td>\n",
       "      <td>MTTH, 12148T-C</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>[]</td>\n",
       "      <td>No viable variation members identified.</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[]</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3689492</th>\n",
       "      <td>NaN</td>\n",
       "      <td>3903429</td>\n",
       "      <td>IKZF2, ARG106TRP</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>[]</td>\n",
       "      <td>No viable variation members identified.</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[]</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>10235 rows × 40 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         out in.variation_id  \\\n",
       "0        NaN          561260   \n",
       "1        NaN          221359   \n",
       "2        NaN         3897886   \n",
       "3        NaN          982523   \n",
       "4        NaN          427839   \n",
       "...      ...             ...   \n",
       "3689488  NaN           41607   \n",
       "3689489  NaN         3764520   \n",
       "3689490  NaN           16606   \n",
       "3689491  NaN         3901242   \n",
       "3689492  NaN         3903429   \n",
       "\n",
       "                                                   in.name   in.vrs_class  \\\n",
       "0        46,XX,der(9)(q21.2,q21.2).seq[GRCh37/hg19]der(...  Not Available   \n",
       "1                   chr1:17555508-24706269 complex variant  Not Available   \n",
       "2                     NM_001003722.2:c.[1807C>T];[1705C>T]  Not Available   \n",
       "3          NM_025114.4(CEP290):c.[4655AAG[2]];[6012-12T>A]  Not Available   \n",
       "4               NM_023073.3(CPLANE1):c.[2080A>G];[6968A>C]  Not Available   \n",
       "...                                                    ...            ...   \n",
       "3689488                          NM_000245.2:c.-14-3312A>T        Unknown   \n",
       "3689489                                              G317D        Unknown   \n",
       "3689490                                    EPHX1, -4238T-A        Unknown   \n",
       "3689491                                     MTTH, 12148T-C        Unknown   \n",
       "3689492                                   IKZF2, ARG106TRP        Unknown   \n",
       "\n",
       "        in.range_copies                                           in.issue  \\\n",
       "0                    []            No viable variation members identified.   \n",
       "1                    []            No viable variation members identified.   \n",
       "2                    []  haplotype and genotype variations are not supp...   \n",
       "3                    []  haplotype and genotype variations are not supp...   \n",
       "4                    []  haplotype and genotype variations are not supp...   \n",
       "...                 ...                                                ...   \n",
       "3689488              []            No viable variation members identified.   \n",
       "3689489              []            No viable variation members identified.   \n",
       "3689490              []            No viable variation members identified.   \n",
       "3689491              []            No viable variation members identified.   \n",
       "3689492              []            No viable variation members identified.   \n",
       "\n",
       "                 in.variation_type in.subclass_type in.cytogenetic  \\\n",
       "0                          Complex     SimpleAllele         9q21.2   \n",
       "1                          Complex     SimpleAllele  1p36.13-36.11   \n",
       "2             CompoundHeterozygote         Genotype            NaN   \n",
       "3             CompoundHeterozygote         Genotype            NaN   \n",
       "4             CompoundHeterozygote         Genotype            NaN   \n",
       "...                            ...              ...            ...   \n",
       "3689488  single nucleotide variant     SimpleAllele         7q31.2   \n",
       "3689489  single nucleotide variant     SimpleAllele            NaN   \n",
       "3689490  single nucleotide variant     SimpleAllele         1q42.1   \n",
       "3689491  single nucleotide variant     SimpleAllele            NaN   \n",
       "3689492  single nucleotide variant     SimpleAllele            NaN   \n",
       "\n",
       "                                               in.mappings  ...  \\\n",
       "0                                                       []  ...   \n",
       "1                                                       []  ...   \n",
       "2        [{'system': 'ClinVar', 'code': '3897886', 'rel...  ...   \n",
       "3        [{'system': 'ClinVar', 'code': '982523', 'rela...  ...   \n",
       "4        [{'system': 'ClinVar', 'code': '427839', 'rela...  ...   \n",
       "...                                                    ...  ...   \n",
       "3689488                                                 []  ...   \n",
       "3689489                                                 []  ...   \n",
       "3689490                                                 []  ...   \n",
       "3689491                                                 []  ...   \n",
       "3689492                                                 []  ...   \n",
       "\n",
       "        out.state.type out.state.length out.state.sequence  \\\n",
       "0                  NaN              NaN                NaN   \n",
       "1                  NaN              NaN                NaN   \n",
       "2                  NaN              NaN                NaN   \n",
       "3                  NaN              NaN                NaN   \n",
       "4                  NaN              NaN                NaN   \n",
       "...                ...              ...                ...   \n",
       "3689488            NaN              NaN                NaN   \n",
       "3689489            NaN              NaN                NaN   \n",
       "3689490            NaN              NaN                NaN   \n",
       "3689491            NaN              NaN                NaN   \n",
       "3689492            NaN              NaN                NaN   \n",
       "\n",
       "        out.state.repeatSubunitLength out.extensions in.absolute_copies  \\\n",
       "0                                 NaN            NaN                NaN   \n",
       "1                                 NaN            NaN                NaN   \n",
       "2                                 NaN            NaN                NaN   \n",
       "3                                 NaN            NaN                NaN   \n",
       "4                                 NaN            NaN                NaN   \n",
       "...                               ...            ...                ...   \n",
       "3689488                           NaN            NaN                NaN   \n",
       "3689489                           NaN            NaN                NaN   \n",
       "3689490                           NaN            NaN                NaN   \n",
       "3689491                           NaN            NaN                NaN   \n",
       "3689492                           NaN            NaN                NaN   \n",
       "\n",
       "        out.copies support_status   error_string normalize_status  \n",
       "0              NaN          False  Not Supported            False  \n",
       "1              NaN          False  Not Supported            False  \n",
       "2              NaN          False  Not Supported            False  \n",
       "3              NaN          False  Not Supported            False  \n",
       "4              NaN          False  Not Supported            False  \n",
       "...            ...            ...            ...              ...  \n",
       "3689488        NaN          False  Not Supported            False  \n",
       "3689489        NaN          False  Not Supported            False  \n",
       "3689490        NaN          False  Not Supported            False  \n",
       "3689491        NaN          False  Not Supported            False  \n",
       "3689492        NaN          False  Not Supported            False  \n",
       "\n",
       "[10235 rows x 40 columns]"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "not_supported_df = not_supported_df.loc[\n",
    "    ~not_supported_df[\"support_status\"] & ~not_supported_df[\"normalize_status\"]\n",
    "]\n",
    "not_supported_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>in.variation_type</th>\n",
       "      <th>in.issue</th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>copy number gain</td>\n",
       "      <td>Variant not described on GRCh37/GRCh38 assembly</td>\n",
       "      <td>2807</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>copy number loss</td>\n",
       "      <td>Variant not described on GRCh37/GRCh38 assembly</td>\n",
       "      <td>1706</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Haplotype</td>\n",
       "      <td>haplotype and genotype variations are not supp...</td>\n",
       "      <td>617</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Diplotype</td>\n",
       "      <td>haplotype and genotype variations are not supp...</td>\n",
       "      <td>596</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Deletion</td>\n",
       "      <td>No viable variation members identified.</td>\n",
       "      <td>593</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>Microsatellite</td>\n",
       "      <td>repeat expressions are not supported.</td>\n",
       "      <td>456</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>Deletion</td>\n",
       "      <td>sequence for accession not supported by vrs-py...</td>\n",
       "      <td>336</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>CompoundHeterozygote</td>\n",
       "      <td>haplotype and genotype variations are not supp...</td>\n",
       "      <td>297</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>No viable variation members identified.</td>\n",
       "      <td>290</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>Translocation</td>\n",
       "      <td>No viable variation members identified.</td>\n",
       "      <td>282</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>Insertion</td>\n",
       "      <td>No viable variation members identified.</td>\n",
       "      <td>264</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>Deletion</td>\n",
       "      <td>intronic positions are not resolvable in seque...</td>\n",
       "      <td>220</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>Deletion</td>\n",
       "      <td>unsupported hgvs expression.</td>\n",
       "      <td>193</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>Duplication</td>\n",
       "      <td>sequence for accession not supported by vrs-py...</td>\n",
       "      <td>157</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>Variation</td>\n",
       "      <td>No viable variation members identified.</td>\n",
       "      <td>157</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>Insertion</td>\n",
       "      <td>sequence for accession not supported by vrs-py...</td>\n",
       "      <td>137</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>Duplication</td>\n",
       "      <td>No viable variation members identified.</td>\n",
       "      <td>116</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>Indel</td>\n",
       "      <td>No viable variation members identified.</td>\n",
       "      <td>102</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>protein only</td>\n",
       "      <td>No viable variation members identified.</td>\n",
       "      <td>93</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>Complex</td>\n",
       "      <td>No viable variation members identified.</td>\n",
       "      <td>87</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>copy number gain</td>\n",
       "      <td>range copies are not supported.</td>\n",
       "      <td>76</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>Insertion</td>\n",
       "      <td>unsupported hgvs expression.</td>\n",
       "      <td>69</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>Inversion</td>\n",
       "      <td>No viable variation members identified.</td>\n",
       "      <td>69</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>Deletion</td>\n",
       "      <td>Variant not described on GRCh37/GRCh38 assembly</td>\n",
       "      <td>68</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>Microsatellite</td>\n",
       "      <td>unsupported hgvs expression.</td>\n",
       "      <td>67</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>Duplication</td>\n",
       "      <td>Variant not described on GRCh37/GRCh38 assembly</td>\n",
       "      <td>50</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>Indel</td>\n",
       "      <td>unsupported hgvs expression.</td>\n",
       "      <td>43</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>Duplication</td>\n",
       "      <td>unsupported hgvs expression.</td>\n",
       "      <td>42</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>Microsatellite</td>\n",
       "      <td>No viable variation members identified.</td>\n",
       "      <td>33</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>Duplication</td>\n",
       "      <td>intronic positions are not resolvable in seque...</td>\n",
       "      <td>33</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>Inversion</td>\n",
       "      <td>unsupported hgvs expression.</td>\n",
       "      <td>32</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>Indel</td>\n",
       "      <td>sequence for accession not supported by vrs-py...</td>\n",
       "      <td>22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>Haplotype, single variant</td>\n",
       "      <td>haplotype and genotype variations are not supp...</td>\n",
       "      <td>21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>copy number loss</td>\n",
       "      <td>range copies are not supported.</td>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>Microsatellite</td>\n",
       "      <td>sequence for accession not supported by vrs-py...</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>Phase unknown</td>\n",
       "      <td>haplotype and genotype variations are not supp...</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>Insertion</td>\n",
       "      <td>intronic positions are not resolvable in seque...</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>Indel</td>\n",
       "      <td>intronic positions are not resolvable in seque...</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>Variation</td>\n",
       "      <td>unsupported hgvs expression.</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>fusion</td>\n",
       "      <td>No viable variation members identified.</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>Insertion</td>\n",
       "      <td>repeat expressions are not supported.</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>Deletion</td>\n",
       "      <td>repeat expressions are not supported.</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>Variation</td>\n",
       "      <td>repeat expressions are not supported.</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>Variation</td>\n",
       "      <td>sequence for accession not supported by vrs-py...</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>copy number gain</td>\n",
       "      <td>No viable variation members identified.</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>Deletion</td>\n",
       "      <td>expression contains unbalaned paretheses.</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>Complex</td>\n",
       "      <td>unsupported hgvs expression.</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>unsupported hgvs expression.</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>Variation</td>\n",
       "      <td>intronic positions are not resolvable in seque...</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>Duplication</td>\n",
       "      <td>repeat expressions are not supported.</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50</th>\n",
       "      <td>Distinct chromosomes</td>\n",
       "      <td>haplotype and genotype variations are not supp...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51</th>\n",
       "      <td>copy number gain</td>\n",
       "      <td>repeat expressions are not supported.</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52</th>\n",
       "      <td>Translocation</td>\n",
       "      <td>unsupported hgvs expression.</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53</th>\n",
       "      <td>Microsatellite</td>\n",
       "      <td>intronic positions are not resolvable in seque...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54</th>\n",
       "      <td>Complex</td>\n",
       "      <td>sequence for accession not supported by vrs-py...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55</th>\n",
       "      <td>Indel</td>\n",
       "      <td>repeat expressions are not supported.</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56</th>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>intronic positions are not resolvable in seque...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>57</th>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>sequence for accession not supported by vrs-py...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>58</th>\n",
       "      <td>Tandem duplication</td>\n",
       "      <td>No viable variation members identified.</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            in.variation_type  \\\n",
       "0            copy number gain   \n",
       "1            copy number loss   \n",
       "2                   Haplotype   \n",
       "3                   Diplotype   \n",
       "4                    Deletion   \n",
       "5              Microsatellite   \n",
       "6                    Deletion   \n",
       "7        CompoundHeterozygote   \n",
       "8   single nucleotide variant   \n",
       "9               Translocation   \n",
       "10                  Insertion   \n",
       "11                   Deletion   \n",
       "12                   Deletion   \n",
       "13                Duplication   \n",
       "14                  Variation   \n",
       "15                  Insertion   \n",
       "16                Duplication   \n",
       "17                      Indel   \n",
       "18               protein only   \n",
       "19                    Complex   \n",
       "20           copy number gain   \n",
       "21                  Insertion   \n",
       "22                  Inversion   \n",
       "23                   Deletion   \n",
       "24             Microsatellite   \n",
       "25                Duplication   \n",
       "26                      Indel   \n",
       "27                Duplication   \n",
       "28             Microsatellite   \n",
       "29                Duplication   \n",
       "30                  Inversion   \n",
       "31                      Indel   \n",
       "32  Haplotype, single variant   \n",
       "33           copy number loss   \n",
       "34             Microsatellite   \n",
       "35              Phase unknown   \n",
       "36                  Insertion   \n",
       "37                      Indel   \n",
       "38                  Variation   \n",
       "39                     fusion   \n",
       "40                  Insertion   \n",
       "41                   Deletion   \n",
       "42                  Variation   \n",
       "43                  Variation   \n",
       "44           copy number gain   \n",
       "45                   Deletion   \n",
       "46                    Complex   \n",
       "47  single nucleotide variant   \n",
       "48                  Variation   \n",
       "49                Duplication   \n",
       "50       Distinct chromosomes   \n",
       "51           copy number gain   \n",
       "52              Translocation   \n",
       "53             Microsatellite   \n",
       "54                    Complex   \n",
       "55                      Indel   \n",
       "56  single nucleotide variant   \n",
       "57  single nucleotide variant   \n",
       "58         Tandem duplication   \n",
       "\n",
       "                                             in.issue  count  \n",
       "0     Variant not described on GRCh37/GRCh38 assembly   2807  \n",
       "1     Variant not described on GRCh37/GRCh38 assembly   1706  \n",
       "2   haplotype and genotype variations are not supp...    617  \n",
       "3   haplotype and genotype variations are not supp...    596  \n",
       "4             No viable variation members identified.    593  \n",
       "5               repeat expressions are not supported.    456  \n",
       "6   sequence for accession not supported by vrs-py...    336  \n",
       "7   haplotype and genotype variations are not supp...    297  \n",
       "8             No viable variation members identified.    290  \n",
       "9             No viable variation members identified.    282  \n",
       "10            No viable variation members identified.    264  \n",
       "11  intronic positions are not resolvable in seque...    220  \n",
       "12                       unsupported hgvs expression.    193  \n",
       "13  sequence for accession not supported by vrs-py...    157  \n",
       "14            No viable variation members identified.    157  \n",
       "15  sequence for accession not supported by vrs-py...    137  \n",
       "16            No viable variation members identified.    116  \n",
       "17            No viable variation members identified.    102  \n",
       "18            No viable variation members identified.     93  \n",
       "19            No viable variation members identified.     87  \n",
       "20                    range copies are not supported.     76  \n",
       "21                       unsupported hgvs expression.     69  \n",
       "22            No viable variation members identified.     69  \n",
       "23    Variant not described on GRCh37/GRCh38 assembly     68  \n",
       "24                       unsupported hgvs expression.     67  \n",
       "25    Variant not described on GRCh37/GRCh38 assembly     50  \n",
       "26                       unsupported hgvs expression.     43  \n",
       "27                       unsupported hgvs expression.     42  \n",
       "28            No viable variation members identified.     33  \n",
       "29  intronic positions are not resolvable in seque...     33  \n",
       "30                       unsupported hgvs expression.     32  \n",
       "31  sequence for accession not supported by vrs-py...     22  \n",
       "32  haplotype and genotype variations are not supp...     21  \n",
       "33                    range copies are not supported.     19  \n",
       "34  sequence for accession not supported by vrs-py...     11  \n",
       "35  haplotype and genotype variations are not supp...      8  \n",
       "36  intronic positions are not resolvable in seque...      7  \n",
       "37  intronic positions are not resolvable in seque...      7  \n",
       "38                       unsupported hgvs expression.      6  \n",
       "39            No viable variation members identified.      5  \n",
       "40              repeat expressions are not supported.      5  \n",
       "41              repeat expressions are not supported.      5  \n",
       "42              repeat expressions are not supported.      4  \n",
       "43  sequence for accession not supported by vrs-py...      4  \n",
       "44            No viable variation members identified.      3  \n",
       "45          expression contains unbalaned paretheses.      3  \n",
       "46                       unsupported hgvs expression.      2  \n",
       "47                       unsupported hgvs expression.      2  \n",
       "48  intronic positions are not resolvable in seque...      2  \n",
       "49              repeat expressions are not supported.      2  \n",
       "50  haplotype and genotype variations are not supp...      1  \n",
       "51              repeat expressions are not supported.      1  \n",
       "52                       unsupported hgvs expression.      1  \n",
       "53  intronic positions are not resolvable in seque...      1  \n",
       "54  sequence for accession not supported by vrs-py...      1  \n",
       "55              repeat expressions are not supported.      1  \n",
       "56  intronic positions are not resolvable in seque...      1  \n",
       "57  sequence for accession not supported by vrs-py...      1  \n",
       "58            No viable variation members identified.      1  "
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "variation_type_count_not_supported_df = not_supported_df.value_counts(\n",
    "    [\"in.variation_type\", \"in.issue\"]\n",
    ").reset_index()\n",
    "variation_type_count_not_supported_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [],
   "source": [
    "variation_type_count_not_supported_df.to_csv(\n",
    "    \"variation_analysis_output/variation_type_count_not_supported_df.csv\"\n",
    ")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Sanity check: making sure there are no supported variants that have been marked as normalized"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [],
   "source": [
    "not_supported_but_normalized_df = df.copy()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>out</th>\n",
       "      <th>in.variation_id</th>\n",
       "      <th>in.name</th>\n",
       "      <th>in.vrs_class</th>\n",
       "      <th>in.range_copies</th>\n",
       "      <th>in.issue</th>\n",
       "      <th>in.variation_type</th>\n",
       "      <th>in.subclass_type</th>\n",
       "      <th>in.cytogenetic</th>\n",
       "      <th>in.mappings</th>\n",
       "      <th>...</th>\n",
       "      <th>out.state.type</th>\n",
       "      <th>out.state.length</th>\n",
       "      <th>out.state.sequence</th>\n",
       "      <th>out.state.repeatSubunitLength</th>\n",
       "      <th>out.extensions</th>\n",
       "      <th>in.absolute_copies</th>\n",
       "      <th>out.copies</th>\n",
       "      <th>support_status</th>\n",
       "      <th>error_string</th>\n",
       "      <th>normalize_status</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>0 rows × 40 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [out, in.variation_id, in.name, in.vrs_class, in.range_copies, in.issue, in.variation_type, in.subclass_type, in.cytogenetic, in.mappings, in.accession, in.fmt, in.source, in.precedence, out.errors, in.copy_change_type, in.assembly_version, in.chr, in.variant_length, out.id, out.type, out.digest, out.location.id, out.location.type, out.location.digest, out.location.sequenceReference.type, out.location.sequenceReference.refgetAccession, out.location.start, out.location.end, out.copyChange, out.state.type, out.state.length, out.state.sequence, out.state.repeatSubunitLength, out.extensions, in.absolute_copies, out.copies, support_status, error_string, normalize_status]\n",
       "Index: []\n",
       "\n",
       "[0 rows x 40 columns]"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "not_supported_but_normalized_df = not_supported_but_normalized_df.loc[\n",
    "    (\n",
    "        ~not_supported_but_normalized_df[\"support_status\"]\n",
    "        & not_supported_but_normalized_df[\"normalize_status\"]\n",
    "    )\n",
    "]\n",
    "not_supported_but_normalized_df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## <a id='toc1_5_'></a>[Counting variants from each group](#toc0_)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [],
   "source": [
    "num_supported = len(supported_df)\n",
    "num_supported_not_normalized = len(supported_not_normalized_df)\n",
    "num_not_supported_but_normalized = len(not_supported_but_normalized_df)\n",
    "num_not_supported = len(not_supported_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [],
   "source": [
    "summary_df2 = pd.DataFrame(\n",
    "    {\n",
    "        \"Supported\": [num_supported, num_supported_not_normalized],\n",
    "        \"Not Supported\": [num_not_supported_but_normalized, num_not_supported],\n",
    "    }\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Supported</th>\n",
       "      <th>Not Supported</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Normalized</th>\n",
       "      <td>3745114</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Not Normalized</th>\n",
       "      <td>2118</td>\n",
       "      <td>10235</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                Supported  Not Supported\n",
       "Normalized        3745114              0\n",
       "Not Normalized       2118          10235"
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "summary_df2.index = [\"Normalized\", \"Not Normalized\"]\n",
    "summary_df2"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## <a id='toc1_6_'></a>[Counting variant types for each group](#toc0_)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [],
   "source": [
    "variation_type_count_summary_df = pd.merge(\n",
    "    pd.merge(\n",
    "        variation_type_count_supported_df,\n",
    "        variation_type_count_supported_not_normalized_df,\n",
    "        on=\"in.variation_type\",\n",
    "        how=\"left\",\n",
    "    ),\n",
    "    variation_type_count_not_supported_df,\n",
    "    on=\"in.variation_type\",\n",
    "    how=\"right\",\n",
    ")\n",
    "variation_type_count_summary_df = variation_type_count_summary_df.replace(\n",
    "    np.nan, 0, regex=True\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [],
   "source": [
    "variation_type_count_summary_df = variation_type_count_summary_df.rename(\n",
    "    columns={\n",
    "        \"in.id_x\": \"supported\",\n",
    "        \"in.id_y\": \"supported_not_normalized\",\n",
    "        \"in.id\": \"not_supported\",\n",
    "    }\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>in.variation_type</th>\n",
       "      <th>in.issue_x</th>\n",
       "      <th>count_x</th>\n",
       "      <th>in.issue_y</th>\n",
       "      <th>count_y</th>\n",
       "      <th>in.issue</th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>copy number gain</td>\n",
       "      <td>None</td>\n",
       "      <td>21484.0</td>\n",
       "      <td>None</td>\n",
       "      <td>438.0</td>\n",
       "      <td>Variant not described on GRCh37/GRCh38 assembly</td>\n",
       "      <td>2807</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>copy number loss</td>\n",
       "      <td>None</td>\n",
       "      <td>20367.0</td>\n",
       "      <td>None</td>\n",
       "      <td>538.0</td>\n",
       "      <td>Variant not described on GRCh37/GRCh38 assembly</td>\n",
       "      <td>1706</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Haplotype</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>haplotype and genotype variations are not supp...</td>\n",
       "      <td>617</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Diplotype</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>haplotype and genotype variations are not supp...</td>\n",
       "      <td>596</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Deletion</td>\n",
       "      <td>None</td>\n",
       "      <td>160232.0</td>\n",
       "      <td>None</td>\n",
       "      <td>711.0</td>\n",
       "      <td>No viable variation members identified.</td>\n",
       "      <td>593</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>Microsatellite</td>\n",
       "      <td>None</td>\n",
       "      <td>36095.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>repeat expressions are not supported.</td>\n",
       "      <td>456</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>Deletion</td>\n",
       "      <td>None</td>\n",
       "      <td>160232.0</td>\n",
       "      <td>None</td>\n",
       "      <td>711.0</td>\n",
       "      <td>sequence for accession not supported by vrs-py...</td>\n",
       "      <td>336</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>CompoundHeterozygote</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>haplotype and genotype variations are not supp...</td>\n",
       "      <td>297</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>None</td>\n",
       "      <td>3401625.0</td>\n",
       "      <td>None</td>\n",
       "      <td>71.0</td>\n",
       "      <td>No viable variation members identified.</td>\n",
       "      <td>290</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>Translocation</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>No viable variation members identified.</td>\n",
       "      <td>282</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>Insertion</td>\n",
       "      <td>None</td>\n",
       "      <td>13019.0</td>\n",
       "      <td>None</td>\n",
       "      <td>1.0</td>\n",
       "      <td>No viable variation members identified.</td>\n",
       "      <td>264</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>Deletion</td>\n",
       "      <td>None</td>\n",
       "      <td>160232.0</td>\n",
       "      <td>None</td>\n",
       "      <td>711.0</td>\n",
       "      <td>intronic positions are not resolvable in seque...</td>\n",
       "      <td>220</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>Deletion</td>\n",
       "      <td>None</td>\n",
       "      <td>160232.0</td>\n",
       "      <td>None</td>\n",
       "      <td>711.0</td>\n",
       "      <td>unsupported hgvs expression.</td>\n",
       "      <td>193</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>Duplication</td>\n",
       "      <td>None</td>\n",
       "      <td>73474.0</td>\n",
       "      <td>None</td>\n",
       "      <td>357.0</td>\n",
       "      <td>sequence for accession not supported by vrs-py...</td>\n",
       "      <td>157</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>Variation</td>\n",
       "      <td>None</td>\n",
       "      <td>379.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>No viable variation members identified.</td>\n",
       "      <td>157</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>Insertion</td>\n",
       "      <td>None</td>\n",
       "      <td>13019.0</td>\n",
       "      <td>None</td>\n",
       "      <td>1.0</td>\n",
       "      <td>sequence for accession not supported by vrs-py...</td>\n",
       "      <td>137</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>Duplication</td>\n",
       "      <td>None</td>\n",
       "      <td>73474.0</td>\n",
       "      <td>None</td>\n",
       "      <td>357.0</td>\n",
       "      <td>No viable variation members identified.</td>\n",
       "      <td>116</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>Indel</td>\n",
       "      <td>None</td>\n",
       "      <td>17038.0</td>\n",
       "      <td>None</td>\n",
       "      <td>2.0</td>\n",
       "      <td>No viable variation members identified.</td>\n",
       "      <td>102</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>protein only</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>No viable variation members identified.</td>\n",
       "      <td>93</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>Complex</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>No viable variation members identified.</td>\n",
       "      <td>87</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>copy number gain</td>\n",
       "      <td>None</td>\n",
       "      <td>21484.0</td>\n",
       "      <td>None</td>\n",
       "      <td>438.0</td>\n",
       "      <td>range copies are not supported.</td>\n",
       "      <td>76</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>Insertion</td>\n",
       "      <td>None</td>\n",
       "      <td>13019.0</td>\n",
       "      <td>None</td>\n",
       "      <td>1.0</td>\n",
       "      <td>unsupported hgvs expression.</td>\n",
       "      <td>69</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>Inversion</td>\n",
       "      <td>None</td>\n",
       "      <td>1401.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>No viable variation members identified.</td>\n",
       "      <td>69</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>Deletion</td>\n",
       "      <td>None</td>\n",
       "      <td>160232.0</td>\n",
       "      <td>None</td>\n",
       "      <td>711.0</td>\n",
       "      <td>Variant not described on GRCh37/GRCh38 assembly</td>\n",
       "      <td>68</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>Microsatellite</td>\n",
       "      <td>None</td>\n",
       "      <td>36095.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>unsupported hgvs expression.</td>\n",
       "      <td>67</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>Duplication</td>\n",
       "      <td>None</td>\n",
       "      <td>73474.0</td>\n",
       "      <td>None</td>\n",
       "      <td>357.0</td>\n",
       "      <td>Variant not described on GRCh37/GRCh38 assembly</td>\n",
       "      <td>50</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>Indel</td>\n",
       "      <td>None</td>\n",
       "      <td>17038.0</td>\n",
       "      <td>None</td>\n",
       "      <td>2.0</td>\n",
       "      <td>unsupported hgvs expression.</td>\n",
       "      <td>43</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>Duplication</td>\n",
       "      <td>None</td>\n",
       "      <td>73474.0</td>\n",
       "      <td>None</td>\n",
       "      <td>357.0</td>\n",
       "      <td>unsupported hgvs expression.</td>\n",
       "      <td>42</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>Microsatellite</td>\n",
       "      <td>None</td>\n",
       "      <td>36095.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>No viable variation members identified.</td>\n",
       "      <td>33</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>Duplication</td>\n",
       "      <td>None</td>\n",
       "      <td>73474.0</td>\n",
       "      <td>None</td>\n",
       "      <td>357.0</td>\n",
       "      <td>intronic positions are not resolvable in seque...</td>\n",
       "      <td>33</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>Inversion</td>\n",
       "      <td>None</td>\n",
       "      <td>1401.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>unsupported hgvs expression.</td>\n",
       "      <td>32</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>Indel</td>\n",
       "      <td>None</td>\n",
       "      <td>17038.0</td>\n",
       "      <td>None</td>\n",
       "      <td>2.0</td>\n",
       "      <td>sequence for accession not supported by vrs-py...</td>\n",
       "      <td>22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>Haplotype, single variant</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>haplotype and genotype variations are not supp...</td>\n",
       "      <td>21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>copy number loss</td>\n",
       "      <td>None</td>\n",
       "      <td>20367.0</td>\n",
       "      <td>None</td>\n",
       "      <td>538.0</td>\n",
       "      <td>range copies are not supported.</td>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>Microsatellite</td>\n",
       "      <td>None</td>\n",
       "      <td>36095.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>sequence for accession not supported by vrs-py...</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>Phase unknown</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>haplotype and genotype variations are not supp...</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>Insertion</td>\n",
       "      <td>None</td>\n",
       "      <td>13019.0</td>\n",
       "      <td>None</td>\n",
       "      <td>1.0</td>\n",
       "      <td>intronic positions are not resolvable in seque...</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>Indel</td>\n",
       "      <td>None</td>\n",
       "      <td>17038.0</td>\n",
       "      <td>None</td>\n",
       "      <td>2.0</td>\n",
       "      <td>intronic positions are not resolvable in seque...</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>Variation</td>\n",
       "      <td>None</td>\n",
       "      <td>379.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>unsupported hgvs expression.</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>fusion</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>No viable variation members identified.</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>Insertion</td>\n",
       "      <td>None</td>\n",
       "      <td>13019.0</td>\n",
       "      <td>None</td>\n",
       "      <td>1.0</td>\n",
       "      <td>repeat expressions are not supported.</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>Deletion</td>\n",
       "      <td>None</td>\n",
       "      <td>160232.0</td>\n",
       "      <td>None</td>\n",
       "      <td>711.0</td>\n",
       "      <td>repeat expressions are not supported.</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>Variation</td>\n",
       "      <td>None</td>\n",
       "      <td>379.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>repeat expressions are not supported.</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>Variation</td>\n",
       "      <td>None</td>\n",
       "      <td>379.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>sequence for accession not supported by vrs-py...</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>copy number gain</td>\n",
       "      <td>None</td>\n",
       "      <td>21484.0</td>\n",
       "      <td>None</td>\n",
       "      <td>438.0</td>\n",
       "      <td>No viable variation members identified.</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>Deletion</td>\n",
       "      <td>None</td>\n",
       "      <td>160232.0</td>\n",
       "      <td>None</td>\n",
       "      <td>711.0</td>\n",
       "      <td>expression contains unbalaned paretheses.</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>Complex</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>unsupported hgvs expression.</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>None</td>\n",
       "      <td>3401625.0</td>\n",
       "      <td>None</td>\n",
       "      <td>71.0</td>\n",
       "      <td>unsupported hgvs expression.</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>Variation</td>\n",
       "      <td>None</td>\n",
       "      <td>379.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>intronic positions are not resolvable in seque...</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>Duplication</td>\n",
       "      <td>None</td>\n",
       "      <td>73474.0</td>\n",
       "      <td>None</td>\n",
       "      <td>357.0</td>\n",
       "      <td>repeat expressions are not supported.</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50</th>\n",
       "      <td>Distinct chromosomes</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>haplotype and genotype variations are not supp...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51</th>\n",
       "      <td>copy number gain</td>\n",
       "      <td>None</td>\n",
       "      <td>21484.0</td>\n",
       "      <td>None</td>\n",
       "      <td>438.0</td>\n",
       "      <td>repeat expressions are not supported.</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52</th>\n",
       "      <td>Translocation</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>unsupported hgvs expression.</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53</th>\n",
       "      <td>Microsatellite</td>\n",
       "      <td>None</td>\n",
       "      <td>36095.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>intronic positions are not resolvable in seque...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54</th>\n",
       "      <td>Complex</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>sequence for accession not supported by vrs-py...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55</th>\n",
       "      <td>Indel</td>\n",
       "      <td>None</td>\n",
       "      <td>17038.0</td>\n",
       "      <td>None</td>\n",
       "      <td>2.0</td>\n",
       "      <td>repeat expressions are not supported.</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56</th>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>None</td>\n",
       "      <td>3401625.0</td>\n",
       "      <td>None</td>\n",
       "      <td>71.0</td>\n",
       "      <td>intronic positions are not resolvable in seque...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>57</th>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>None</td>\n",
       "      <td>3401625.0</td>\n",
       "      <td>None</td>\n",
       "      <td>71.0</td>\n",
       "      <td>sequence for accession not supported by vrs-py...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>58</th>\n",
       "      <td>Tandem duplication</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>No viable variation members identified.</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            in.variation_type in.issue_x    count_x in.issue_y  count_y  \\\n",
       "0            copy number gain       None    21484.0       None    438.0   \n",
       "1            copy number loss       None    20367.0       None    538.0   \n",
       "2                   Haplotype          0        0.0          0      0.0   \n",
       "3                   Diplotype          0        0.0          0      0.0   \n",
       "4                    Deletion       None   160232.0       None    711.0   \n",
       "5              Microsatellite       None    36095.0          0      0.0   \n",
       "6                    Deletion       None   160232.0       None    711.0   \n",
       "7        CompoundHeterozygote          0        0.0          0      0.0   \n",
       "8   single nucleotide variant       None  3401625.0       None     71.0   \n",
       "9               Translocation          0        0.0          0      0.0   \n",
       "10                  Insertion       None    13019.0       None      1.0   \n",
       "11                   Deletion       None   160232.0       None    711.0   \n",
       "12                   Deletion       None   160232.0       None    711.0   \n",
       "13                Duplication       None    73474.0       None    357.0   \n",
       "14                  Variation       None      379.0          0      0.0   \n",
       "15                  Insertion       None    13019.0       None      1.0   \n",
       "16                Duplication       None    73474.0       None    357.0   \n",
       "17                      Indel       None    17038.0       None      2.0   \n",
       "18               protein only          0        0.0          0      0.0   \n",
       "19                    Complex          0        0.0          0      0.0   \n",
       "20           copy number gain       None    21484.0       None    438.0   \n",
       "21                  Insertion       None    13019.0       None      1.0   \n",
       "22                  Inversion       None     1401.0          0      0.0   \n",
       "23                   Deletion       None   160232.0       None    711.0   \n",
       "24             Microsatellite       None    36095.0          0      0.0   \n",
       "25                Duplication       None    73474.0       None    357.0   \n",
       "26                      Indel       None    17038.0       None      2.0   \n",
       "27                Duplication       None    73474.0       None    357.0   \n",
       "28             Microsatellite       None    36095.0          0      0.0   \n",
       "29                Duplication       None    73474.0       None    357.0   \n",
       "30                  Inversion       None     1401.0          0      0.0   \n",
       "31                      Indel       None    17038.0       None      2.0   \n",
       "32  Haplotype, single variant          0        0.0          0      0.0   \n",
       "33           copy number loss       None    20367.0       None    538.0   \n",
       "34             Microsatellite       None    36095.0          0      0.0   \n",
       "35              Phase unknown          0        0.0          0      0.0   \n",
       "36                  Insertion       None    13019.0       None      1.0   \n",
       "37                      Indel       None    17038.0       None      2.0   \n",
       "38                  Variation       None      379.0          0      0.0   \n",
       "39                     fusion          0        0.0          0      0.0   \n",
       "40                  Insertion       None    13019.0       None      1.0   \n",
       "41                   Deletion       None   160232.0       None    711.0   \n",
       "42                  Variation       None      379.0          0      0.0   \n",
       "43                  Variation       None      379.0          0      0.0   \n",
       "44           copy number gain       None    21484.0       None    438.0   \n",
       "45                   Deletion       None   160232.0       None    711.0   \n",
       "46                    Complex          0        0.0          0      0.0   \n",
       "47  single nucleotide variant       None  3401625.0       None     71.0   \n",
       "48                  Variation       None      379.0          0      0.0   \n",
       "49                Duplication       None    73474.0       None    357.0   \n",
       "50       Distinct chromosomes          0        0.0          0      0.0   \n",
       "51           copy number gain       None    21484.0       None    438.0   \n",
       "52              Translocation          0        0.0          0      0.0   \n",
       "53             Microsatellite       None    36095.0          0      0.0   \n",
       "54                    Complex          0        0.0          0      0.0   \n",
       "55                      Indel       None    17038.0       None      2.0   \n",
       "56  single nucleotide variant       None  3401625.0       None     71.0   \n",
       "57  single nucleotide variant       None  3401625.0       None     71.0   \n",
       "58         Tandem duplication          0        0.0          0      0.0   \n",
       "\n",
       "                                             in.issue  count  \n",
       "0     Variant not described on GRCh37/GRCh38 assembly   2807  \n",
       "1     Variant not described on GRCh37/GRCh38 assembly   1706  \n",
       "2   haplotype and genotype variations are not supp...    617  \n",
       "3   haplotype and genotype variations are not supp...    596  \n",
       "4             No viable variation members identified.    593  \n",
       "5               repeat expressions are not supported.    456  \n",
       "6   sequence for accession not supported by vrs-py...    336  \n",
       "7   haplotype and genotype variations are not supp...    297  \n",
       "8             No viable variation members identified.    290  \n",
       "9             No viable variation members identified.    282  \n",
       "10            No viable variation members identified.    264  \n",
       "11  intronic positions are not resolvable in seque...    220  \n",
       "12                       unsupported hgvs expression.    193  \n",
       "13  sequence for accession not supported by vrs-py...    157  \n",
       "14            No viable variation members identified.    157  \n",
       "15  sequence for accession not supported by vrs-py...    137  \n",
       "16            No viable variation members identified.    116  \n",
       "17            No viable variation members identified.    102  \n",
       "18            No viable variation members identified.     93  \n",
       "19            No viable variation members identified.     87  \n",
       "20                    range copies are not supported.     76  \n",
       "21                       unsupported hgvs expression.     69  \n",
       "22            No viable variation members identified.     69  \n",
       "23    Variant not described on GRCh37/GRCh38 assembly     68  \n",
       "24                       unsupported hgvs expression.     67  \n",
       "25    Variant not described on GRCh37/GRCh38 assembly     50  \n",
       "26                       unsupported hgvs expression.     43  \n",
       "27                       unsupported hgvs expression.     42  \n",
       "28            No viable variation members identified.     33  \n",
       "29  intronic positions are not resolvable in seque...     33  \n",
       "30                       unsupported hgvs expression.     32  \n",
       "31  sequence for accession not supported by vrs-py...     22  \n",
       "32  haplotype and genotype variations are not supp...     21  \n",
       "33                    range copies are not supported.     19  \n",
       "34  sequence for accession not supported by vrs-py...     11  \n",
       "35  haplotype and genotype variations are not supp...      8  \n",
       "36  intronic positions are not resolvable in seque...      7  \n",
       "37  intronic positions are not resolvable in seque...      7  \n",
       "38                       unsupported hgvs expression.      6  \n",
       "39            No viable variation members identified.      5  \n",
       "40              repeat expressions are not supported.      5  \n",
       "41              repeat expressions are not supported.      5  \n",
       "42              repeat expressions are not supported.      4  \n",
       "43  sequence for accession not supported by vrs-py...      4  \n",
       "44            No viable variation members identified.      3  \n",
       "45          expression contains unbalaned paretheses.      3  \n",
       "46                       unsupported hgvs expression.      2  \n",
       "47                       unsupported hgvs expression.      2  \n",
       "48  intronic positions are not resolvable in seque...      2  \n",
       "49              repeat expressions are not supported.      2  \n",
       "50  haplotype and genotype variations are not supp...      1  \n",
       "51              repeat expressions are not supported.      1  \n",
       "52                       unsupported hgvs expression.      1  \n",
       "53  intronic positions are not resolvable in seque...      1  \n",
       "54  sequence for accession not supported by vrs-py...      1  \n",
       "55              repeat expressions are not supported.      1  \n",
       "56  intronic positions are not resolvable in seque...      1  \n",
       "57  sequence for accession not supported by vrs-py...      1  \n",
       "58            No viable variation members identified.      1  "
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "variation_type_count_summary_df.to_csv(\n",
    "    \"variation_analysis_output/variation_type_count_summary_df.csv\"\n",
    ")\n",
    "variation_type_count_summary_df"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "variation-normalizer-manuscript",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.13.5"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
