{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# <a id='toc1_'></a>[ClinVar Analysis](#toc0_)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Table of contents**<a id='toc0_'></a>    \n",
    "- [ClinVar Analysis](#toc1_)    \n",
    "  - [Initialize](#toc1_1_)    \n",
    "    - [Import necessary libraries](#toc1_1_1_)    \n",
    "    - [Import variant file from downloads- change to personal username](#toc1_1_2_)    \n",
    "  - [Add Supported Status of Variant based on in.vrs_xform_plan.policy](#toc1_2_)    \n",
    "  - [Add Normalization Status of Variant based on out.errors](#toc1_3_)    \n",
    "    - [Set Normalize Status of Variant as T/F](#toc1_3_1_)    \n",
    "      - [Summary Table](#toc1_3_1_1_)    \n",
    "  - [Create subgroups based on Variant Status](#toc1_4_)    \n",
    "    - [Supported and Normalized Variants](#toc1_4_1_)    \n",
    "    - [Supported and Not Normalized Variants](#toc1_4_2_)    \n",
    "    - [Not Supported Variants](#toc1_4_3_)    \n",
    "  - [Counting variants from each group](#toc1_5_)    \n",
    "\n",
    "<!-- vscode-jupyter-toc-config\n",
    "\tnumbering=false\n",
    "\tanchor=true\n",
    "\tflat=false\n",
    "\tminLevel=1\n",
    "\tmaxLevel=6\n",
    "\t/vscode-jupyter-toc-config -->\n",
    "<!-- THIS CELL WILL BE REPLACED ON TOC UPDATE. DO NOT WRITE YOUR TEXT IN THIS CELL -->"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## <a id='toc1_1_'></a>[Initialize](#toc0_)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### <a id='toc1_1_1_'></a>[Import necessary libraries](#toc0_)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [],
   "source": [
    "import ndjson\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import re\n",
    "from dotenv import load_dotenv"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### <a id='toc1_1_2_'></a>[Import variant file from downloads](#toc0_)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "change to personal username below"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [],
   "source": [
    "file = open(\"/Users/rsaxs014/Downloads/output-variation_identity-all-liftover.ndjson\")\n",
    "records = ndjson.load(file)\n",
    "\n",
    "df0 = pd.json_normalize(records)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = df0.copy()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## <a id='toc1_2_'></a>[Add Supported Status of Variant based on in.vrs_xform_plan.policy](#toc0_)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Checking for blanks"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [],
   "source": [
    "df[\"in.vrs_xform_plan.policy\"] = df[\"in.vrs_xform_plan.policy\"].fillna(\"None\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "in.vrs_xform_plan.policy\n",
       "Canonical SPDI                                      2118669\n",
       "Absolute copy count                                   53263\n",
       "Copy number change (cn loss|del and cn gain|dup)      27104\n",
       "NCBI36 genomic only                                    4771\n",
       "No hgvs or location info                               3089\n",
       "Genotype/Haplotype                                     1440\n",
       "Invalid/unsupported hgvs                               1336\n",
       "Remaining valid hgvs alleles                            941\n",
       "Min/max copy count range not supported                   14\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 58,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df[\"in.vrs_xform_plan.policy\"].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {},
   "outputs": [],
   "source": [
    "df[\"support_status\"] = df[\"in.vrs_xform_plan.policy\"].copy()\n",
    "\n",
    "df.loc[df[\"support_status\"] == \"Canonical SPDI\", \"support_status\"] = True\n",
    "df.loc[df[\"support_status\"] == \"Absolute copy count\", \"support_status\"] = True\n",
    "df.loc[df[\"support_status\"] == \"Copy number change (cn loss|del and cn gain|dup)\",\n",
    "    \"support_status\"] = True\n",
    "df.loc[df[\"support_status\"] == \"NCBI36 genomic only\", \"support_status\"] = False\n",
    "df.loc[df[\"support_status\"] == \"No hgvs or location info\", \"support_status\"] = False\n",
    "df.loc[df[\"support_status\"] == \"Genotype/Haplotype\", \"support_status\"] = False\n",
    "df.loc[df[\"support_status\"] == \"Invalid/unsupported hgvs\", \"support_status\"] = False\n",
    "df.loc[df[\"support_status\"] == \"Remaining valid hgvs alleles\", \"support_status\"] = True\n",
    "df.loc[df[\"support_status\"] == \"Min/max copy count range not supported\", \n",
    "    \"support_status\"] = False"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "support_status\n",
       "True     2199977\n",
       "False      10650\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 60,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df[\"support_status\"].value_counts()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## <a id='toc1_3_'></a>[Add Normalization Status of Variant based on out.errors](#toc0_)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The errors are stored as a list of values, some of which are strings and other of which are dictionaries (determined by whether error was handled at the level of Variation Normalizer or after the normalizer)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The \"get_errors\" function extracts the text error responses for better readability and ease string processing"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_errors(errors):\n",
    "    errors_out = []\n",
    "    for e in errors:\n",
    "        if type(e) == str:\n",
    "            errors_out.append(e)\n",
    "        elif type(e) == dict:\n",
    "            for k, v in e.items():\n",
    "                if k not in [\n",
    "                    \"msg\",\n",
    "                    \"response-errors\",]:\n",
    "                ## only get these keys from normalizer response\n",
    "                    continue\n",
    "                if type(v) == str:\n",
    "                    errors_out.append(v)\n",
    "                elif type(v) == list:\n",
    "                    errors_out.append(\";\".join(v))\n",
    "    return \";\".join(errors_out)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {},
   "outputs": [],
   "source": [
    "df[\"error_string\"] = df[\"out.errors\"].fillna(\"\").apply(get_errors)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This is the number of unique error strings"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "There are many different strings because many of the errors contain specific genomic coordinates, which are unlikely to occur more than once"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "532"
      ]
     },
     "execution_count": 63,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df[\"error_string\"].nunique()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "To get the core error message, the numeric values are replaced with \"#\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {},
   "outputs": [],
   "source": [
    "def reduce_errors(error_string):\n",
    "    out = error_string.lower()\n",
    "    out = re.sub(\"\\d+\", \"#\", out)\n",
    "    return out"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "metadata": {},
   "outputs": [],
   "source": [
    "def reduce_errors_more(error_string):\n",
    "    errs = error_string.split(\";\")\n",
    "    new_errs = [re.sub(\"\\:[ ]?[^\\s]+[\\s]?\", \"\", err) for err in errs]\n",
    "    return \";\".join(new_errs)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {},
   "outputs": [],
   "source": [
    "df[\"error_string_reduce\"] = df[\"error_string\"].apply(reduce_errors)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "error_string_reduce\n",
       "                                                                                                                                                         2198936\n",
       "Not Supported                                                                                                                                              10200\n",
       "error returned from variation normalizer;unable to find a grch# accession for: nc_#.#                                                                        539\n",
       "unrecognized variation record                                                                                                                                431\n",
       "error returned from variation normalizer;unable to find classification for: nc_#.#:g.(#_?)_(?_#)del                                                          233\n",
       "error returned from variation normalizer;unable to find classification for: nc_#.#:g.(#_?)_(?_#)dup                                                          151\n",
       "error returned from variation normalizer;unable to find classification for: cm#.#:g.#_#dup;unable to tokenize: cm#.#:g.#_#dup                                 21\n",
       "error returned from variation normalizer;unable to tokenize: nt_#.#:g.(?_#)_(#_?)del;unable to find classification for: nt_#.#:g.(?_#)_(#_?)del               16\n",
       "error returned from variation normalizer;unable to tokenize: cm#.#:g.#_#dup;unable to find classification for: cm#.#:g.#_#dup                                 15\n",
       "error returned from variation normalizer;unable to find classification for: nt_#.#:g.(?_#)_(#_?)del;unable to tokenize: nt_#.#:g.(?_#)_(#_?)del               13\n",
       "error returned from variation normalizer;unable to tokenize: nc_#.#:m.#_#del;unable to find classification for: nc_#.#:m.#_#del                               12\n",
       "error returned from variation normalizer;unable to translate variation                                                                                         8\n",
       "error returned from variation normalizer;unable to find classification for: nc_#.#:g.(#_#)_(#_?)dup                                                            6\n",
       "error returned from variation normalizer;unable to find classification for: nc_#.#:g.(?_#)_(#_#)dup                                                            6\n",
       "error returned from variation normalizer;unable to find valid result for classification: genomic_deletion_ambiguous                                            5\n",
       "error returned from variation normalizer;unable to find classification for: nc_#.#:m.#_#del;unable to tokenize: nc_#.#:m.#_#del                                4\n",
       "error returned from variation normalizer;unable to tokenize: nt_#.#:g.(?_#)_(#_?)dup;unable to find classification for: nt_#.#:g.(?_#)_(#_?)dup                4\n",
       "error returned from variation normalizer;unable to find classification for: nt_#.#:g.(?_#)_(#_?)dup;unable to tokenize: nt_#.#:g.(?_#)_(#_?)dup                3\n",
       "error returned from variation normalizer;unable to find classification for: nc_#.#:g.(#_#)_(#_?)del                                                            3\n",
       "error returned from variation normalizer;unable to tokenize: nw_#.#:g.(?_#)_(#_?)del;unable to find classification for: nw_#.#:g.(?_#)_(#_?)del                2\n",
       "error returned from variation normalizer;unable to find valid result for classification: genomic_duplication_ambiguous                                         2\n",
       "error returned from variation normalizer;unable to find classification for: nt_#.#:g.#_#dup;unable to tokenize: nt_#.#:g.#_#dup                                1\n",
       "error returned from variation normalizer;unable to find classification for: nc_#.#:g.#_(?_#)dup                                                                1\n",
       "error returned from variation normalizer;unable to translate nc_#.#:g.#t>a                                                                                     1\n",
       "error returned from variation normalizer;nc_#.#:g.#_#insgccgtgggccccgccaccgccgccgtgccaccgcc is not a supported hgvs genomic duplication or deletion            1\n",
       "error returned from variation normalizer;unable to find classification for: nc_#.#:g.(?_#)_(#_#)del                                                            1\n",
       "error returned from variation normalizer;unable to find classification for: nc_#.#:g.(#_?)_(#_?)del                                                            1\n",
       "error returned from variation normalizer;unable to translate nc_#.#:g.#g>a                                                                                     1\n",
       "error returned from variation normalizer;unable to tokenize: nw_#.#:g.#g>a                                                                                     1\n",
       "error returned from variation normalizer;unable to find classification for: nw_#.#:g.(?_#)_(#_?)del;unable to tokenize: nw_#.#:g.(?_#)_(#_?)del                1\n",
       "error returned from variation normalizer;unable to translate nc_#.#:g.#_#instattt                                                                              1\n",
       "error returned from variation normalizer;unable to tokenize: nt_#.#:g.#_#del;unable to find classification for: nt_#.#:g.#_#del                                1\n",
       "error returned from variation normalizer;unable to tokenize: nw_#.#:g.(?_#)_(#_?)dup;unable to find classification for: nw_#.#:g.(?_#)_(#_?)dup                1\n",
       "error returned from variation normalizer;unable to tokenize: nw_#.#:g.#c>a                                                                                     1\n",
       "error returned from variation normalizer;unable to find classification for: nc_#.#:g.#_(#_#)del                                                                1\n",
       "error returned from variation normalizer;unable to find classification for: nw_#.#:g.#del;unable to tokenize: nw_#.#:g.#del                                    1\n",
       "error returned from variation normalizer;nc_#.#:g.#_#delinsatgaaaagctttacagaagttggaattaaaaaaa is not a supported hgvs genomic duplication or deletion          1\n",
       "error returned from variation normalizer;unable to find valid result for classification: genomic_duplication                                                   1\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 82,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df[\"error_string_reduce\"].value_counts()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "There are Not Supported variants that have no error because they were labeled \"Not Supported\" manually."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "An error (\"Not Supported\") is entered manually for those variants so that they are not categorized as normalized"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {},
   "outputs": [],
   "source": [
    "df.loc[\n",
    "    (df[\"support_status\"] == False) & (df[\"error_string_reduce\"] == \"\"),\n",
    "    \"error_string_reduce\",\n",
    "] = \"Not Supported\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The error strings had to be reduced further"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "metadata": {},
   "outputs": [],
   "source": [
    "df[\"error_string_reduce_2\"] = df[\"error_string_reduce\"].apply(reduce_errors_more)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "error_string_reduce_2\n",
       "                                                                                                          2198936\n",
       "Not Supported                                                                                               10200\n",
       "error returned from variation normalizer;unable to find a grch# accession for                                 539\n",
       "unrecognized variation record                                                                                 431\n",
       "error returned from variation normalizer;unable to find classification for                                    403\n",
       "error returned from variation normalizer;unable to tokenize;unable to find classification for                  51\n",
       "error returned from variation normalizer;unable to find classification for;unable to tokenize                  44\n",
       "error returned from variation normalizer;unable to find valid result for classification                         8\n",
       "error returned from variation normalizer;unable to translate variation                                          8\n",
       "error returned from variation normalizer;unable to translate nc_#.#                                             3\n",
       "error returned from variation normalizer;nc_#.#is not a supported hgvs genomic duplication or deletion          2\n",
       "error returned from variation normalizer;unable to tokenize                                                     2\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 69,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df[\"error_string_reduce_2\"].value_counts()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### <a id='toc1_3_1_'></a>[Set Normalize Status of Variant as T/F](#toc0_)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "If an error is present, the variant was not normalized and therefore has a False Normalize Status"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>in.id</th>\n",
       "      <th>in.name</th>\n",
       "      <th>in.subclass_type</th>\n",
       "      <th>in.variation_type</th>\n",
       "      <th>in.xrefs</th>\n",
       "      <th>in.vrs_xform_plan.type</th>\n",
       "      <th>in.vrs_xform_plan.inputs</th>\n",
       "      <th>in.vrs_xform_plan.policy</th>\n",
       "      <th>out.id</th>\n",
       "      <th>out.type</th>\n",
       "      <th>...</th>\n",
       "      <th>out.seq.ref_allele_vcf</th>\n",
       "      <th>out.seq.position_vcf</th>\n",
       "      <th>out.seq.alt_allele_vcf</th>\n",
       "      <th>in.max_copies</th>\n",
       "      <th>in.min_copies</th>\n",
       "      <th>support_status</th>\n",
       "      <th>error_string</th>\n",
       "      <th>error_string_reduce</th>\n",
       "      <th>error_string_reduce_2</th>\n",
       "      <th>normalize_status</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>16098</td>\n",
       "      <td>SLC2A2, 1-BP INS, 793C</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>Insertion</td>\n",
       "      <td>OMIM:138160.0009</td>\n",
       "      <td>Text</td>\n",
       "      <td>[id]</td>\n",
       "      <td>No hgvs or location info</td>\n",
       "      <td>Text:clinvar:16098</td>\n",
       "      <td>Text</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td></td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>425693</td>\n",
       "      <td>NM_001204.6(BMPR2):c.77-?_247+?dup</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>Duplication</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Text</td>\n",
       "      <td>[id]</td>\n",
       "      <td>No hgvs or location info</td>\n",
       "      <td>Text:clinvar:425693</td>\n",
       "      <td>Text</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td></td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2446408</td>\n",
       "      <td>CDHR1, 783G-A (rs147346345)</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>OMIM:609502.0005</td>\n",
       "      <td>Text</td>\n",
       "      <td>[id]</td>\n",
       "      <td>No hgvs or location info</td>\n",
       "      <td>Text:clinvar:2446408</td>\n",
       "      <td>Text</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td></td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>90650</td>\n",
       "      <td>NM_000251.2(MSH2):c.1387-?_1510+?del</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Text</td>\n",
       "      <td>[id]</td>\n",
       "      <td>No hgvs or location info</td>\n",
       "      <td>Text:clinvar:90650</td>\n",
       "      <td>Text</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td></td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>14905</td>\n",
       "      <td>HLA-DRB1, HLA-DRB1*1101</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>Variation</td>\n",
       "      <td>OMIM:142857.0001</td>\n",
       "      <td>Text</td>\n",
       "      <td>[id]</td>\n",
       "      <td>No hgvs or location info</td>\n",
       "      <td>Text:clinvar:14905</td>\n",
       "      <td>Text</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td></td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2210622</th>\n",
       "      <td>1496502</td>\n",
       "      <td>NM_007255.3(B4GALT7):c.881_882insTGAGGTGGATTAA...</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>Insertion</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Allele</td>\n",
       "      <td>[canonical_spdi]</td>\n",
       "      <td>Canonical SPDI</td>\n",
       "      <td>ga4gh:VA.P-rdDkBZMl4kwW-ZHmsMOy0ik5MKa4LI</td>\n",
       "      <td>Allele</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2210623</th>\n",
       "      <td>1801464</td>\n",
       "      <td>NM_176787.5(PIGN):c.2673-514_2673-513insGCAAAC...</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>Insertion</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Allele</td>\n",
       "      <td>[hgvs]</td>\n",
       "      <td>Remaining valid hgvs alleles</td>\n",
       "      <td>ga4gh:VA.3BXZ9oWeEqz4VM16yV8mgjvQ8ZTElB2J</td>\n",
       "      <td>Allele</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2210624</th>\n",
       "      <td>1513408</td>\n",
       "      <td>NM_024928.5(STN1):c.340_352AAG[2]CTACAAGGCCGGG...</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>Insertion</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Text</td>\n",
       "      <td>[id]</td>\n",
       "      <td>Invalid/unsupported hgvs</td>\n",
       "      <td>Text:clinvar:1513408</td>\n",
       "      <td>Text</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td></td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2210625</th>\n",
       "      <td>1464440</td>\n",
       "      <td>NM_001267550.2(TTN):c.29512_29513insGGCCGGGCGC...</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>Insertion</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Text</td>\n",
       "      <td>[id]</td>\n",
       "      <td>Invalid/unsupported hgvs</td>\n",
       "      <td>Text:clinvar:1464440</td>\n",
       "      <td>Text</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td></td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2210626</th>\n",
       "      <td>1453116</td>\n",
       "      <td>NM_000051.4(ATM):c.3376_3377insGGCCGGGCGCGGTGG...</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>Insertion</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Text</td>\n",
       "      <td>[id]</td>\n",
       "      <td>Invalid/unsupported hgvs</td>\n",
       "      <td>Text:clinvar:1453116</td>\n",
       "      <td>Text</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td></td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2210627 rows × 92 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           in.id                                            in.name  \\\n",
       "0          16098                             SLC2A2, 1-BP INS, 793C   \n",
       "1         425693                 NM_001204.6(BMPR2):c.77-?_247+?dup   \n",
       "2        2446408                        CDHR1, 783G-A (rs147346345)   \n",
       "3          90650               NM_000251.2(MSH2):c.1387-?_1510+?del   \n",
       "4          14905                            HLA-DRB1, HLA-DRB1*1101   \n",
       "...          ...                                                ...   \n",
       "2210622  1496502  NM_007255.3(B4GALT7):c.881_882insTGAGGTGGATTAA...   \n",
       "2210623  1801464  NM_176787.5(PIGN):c.2673-514_2673-513insGCAAAC...   \n",
       "2210624  1513408  NM_024928.5(STN1):c.340_352AAG[2]CTACAAGGCCGGG...   \n",
       "2210625  1464440  NM_001267550.2(TTN):c.29512_29513insGGCCGGGCGC...   \n",
       "2210626  1453116  NM_000051.4(ATM):c.3376_3377insGGCCGGGCGCGGTGG...   \n",
       "\n",
       "        in.subclass_type          in.variation_type          in.xrefs  \\\n",
       "0           SimpleAllele                  Insertion  OMIM:138160.0009   \n",
       "1           SimpleAllele                Duplication               NaN   \n",
       "2           SimpleAllele  single nucleotide variant  OMIM:609502.0005   \n",
       "3           SimpleAllele                   Deletion               NaN   \n",
       "4           SimpleAllele                  Variation  OMIM:142857.0001   \n",
       "...                  ...                        ...               ...   \n",
       "2210622     SimpleAllele                  Insertion               NaN   \n",
       "2210623     SimpleAllele                  Insertion               NaN   \n",
       "2210624     SimpleAllele                  Insertion               NaN   \n",
       "2210625     SimpleAllele                  Insertion               NaN   \n",
       "2210626     SimpleAllele                  Insertion               NaN   \n",
       "\n",
       "        in.vrs_xform_plan.type in.vrs_xform_plan.inputs  \\\n",
       "0                         Text                     [id]   \n",
       "1                         Text                     [id]   \n",
       "2                         Text                     [id]   \n",
       "3                         Text                     [id]   \n",
       "4                         Text                     [id]   \n",
       "...                        ...                      ...   \n",
       "2210622                 Allele         [canonical_spdi]   \n",
       "2210623                 Allele                   [hgvs]   \n",
       "2210624                   Text                     [id]   \n",
       "2210625                   Text                     [id]   \n",
       "2210626                   Text                     [id]   \n",
       "\n",
       "             in.vrs_xform_plan.policy  \\\n",
       "0            No hgvs or location info   \n",
       "1            No hgvs or location info   \n",
       "2            No hgvs or location info   \n",
       "3            No hgvs or location info   \n",
       "4            No hgvs or location info   \n",
       "...                               ...   \n",
       "2210622                Canonical SPDI   \n",
       "2210623  Remaining valid hgvs alleles   \n",
       "2210624      Invalid/unsupported hgvs   \n",
       "2210625      Invalid/unsupported hgvs   \n",
       "2210626      Invalid/unsupported hgvs   \n",
       "\n",
       "                                            out.id out.type  ...  \\\n",
       "0                               Text:clinvar:16098     Text  ...   \n",
       "1                              Text:clinvar:425693     Text  ...   \n",
       "2                             Text:clinvar:2446408     Text  ...   \n",
       "3                               Text:clinvar:90650     Text  ...   \n",
       "4                               Text:clinvar:14905     Text  ...   \n",
       "...                                            ...      ...  ...   \n",
       "2210622  ga4gh:VA.P-rdDkBZMl4kwW-ZHmsMOy0ik5MKa4LI   Allele  ...   \n",
       "2210623  ga4gh:VA.3BXZ9oWeEqz4VM16yV8mgjvQ8ZTElB2J   Allele  ...   \n",
       "2210624                       Text:clinvar:1513408     Text  ...   \n",
       "2210625                       Text:clinvar:1464440     Text  ...   \n",
       "2210626                       Text:clinvar:1453116     Text  ...   \n",
       "\n",
       "        out.seq.ref_allele_vcf out.seq.position_vcf out.seq.alt_allele_vcf  \\\n",
       "0                          NaN                  NaN                    NaN   \n",
       "1                          NaN                  NaN                    NaN   \n",
       "2                          NaN                  NaN                    NaN   \n",
       "3                          NaN                  NaN                    NaN   \n",
       "4                          NaN                  NaN                    NaN   \n",
       "...                        ...                  ...                    ...   \n",
       "2210622                    NaN                  NaN                    NaN   \n",
       "2210623                    NaN                  NaN                    NaN   \n",
       "2210624                    NaN                  NaN                    NaN   \n",
       "2210625                    NaN                  NaN                    NaN   \n",
       "2210626                    NaN                  NaN                    NaN   \n",
       "\n",
       "        in.max_copies in.min_copies support_status error_string  \\\n",
       "0                 NaN           NaN          False                \n",
       "1                 NaN           NaN          False                \n",
       "2                 NaN           NaN          False                \n",
       "3                 NaN           NaN          False                \n",
       "4                 NaN           NaN          False                \n",
       "...               ...           ...            ...          ...   \n",
       "2210622           NaN           NaN           True                \n",
       "2210623           NaN           NaN           True                \n",
       "2210624           NaN           NaN          False                \n",
       "2210625           NaN           NaN          False                \n",
       "2210626           NaN           NaN          False                \n",
       "\n",
       "         error_string_reduce error_string_reduce_2  normalize_status  \n",
       "0              Not Supported         Not Supported             False  \n",
       "1              Not Supported         Not Supported             False  \n",
       "2              Not Supported         Not Supported             False  \n",
       "3              Not Supported         Not Supported             False  \n",
       "4              Not Supported         Not Supported             False  \n",
       "...                      ...                   ...               ...  \n",
       "2210622                                                         True  \n",
       "2210623                                                         True  \n",
       "2210624        Not Supported         Not Supported             False  \n",
       "2210625        Not Supported         Not Supported             False  \n",
       "2210626        Not Supported         Not Supported             False  \n",
       "\n",
       "[2210627 rows x 92 columns]"
      ]
     },
     "execution_count": 70,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df[\"normalize_status\"] = df[\"error_string_reduce_2\"] == \"\"\n",
    "df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### <a id='toc1_3_1_1_'></a>[Summary Table](#toc0_)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "In the table below, the cells show the number of variants with each expected behavior and how they actually ended up performing."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "If a variant was in an \"expected to pass\" category and ends up as text, that is an instance of a normalizer failure on a supported variant"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead tr th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe thead tr:last-of-type th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th colspan=\"5\" halign=\"left\">in.id</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th></th>\n",
       "      <th>out.type</th>\n",
       "      <th>Allele</th>\n",
       "      <th>CopyNumberChange</th>\n",
       "      <th>CopyNumberCount</th>\n",
       "      <th>NONE</th>\n",
       "      <th>Text</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>support_status</th>\n",
       "      <th>in.vrs_xform_plan.policy</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th rowspan=\"5\" valign=\"top\">False</th>\n",
       "      <th>Genotype/Haplotype</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1440</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Invalid/unsupported hgvs</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>19</td>\n",
       "      <td>1317</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Min/max copy count range not supported</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NCBI36 genomic only</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4771</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>No hgvs or location info</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3089</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"4\" valign=\"top\">True</th>\n",
       "      <th>Absolute copy count</th>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>52440</td>\n",
       "      <td>819</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Canonical SPDI</th>\n",
       "      <td>2118669</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Copy number change (cn loss|del and cn gain|dup)</th>\n",
       "      <td>0</td>\n",
       "      <td>26889</td>\n",
       "      <td>0</td>\n",
       "      <td>209</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Remaining valid hgvs alleles</th>\n",
       "      <td>927</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>13</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                   in.id  \\\n",
       "out.type                                                          Allele   \n",
       "support_status in.vrs_xform_plan.policy                                    \n",
       "False          Genotype/Haplotype                                      0   \n",
       "               Invalid/unsupported hgvs                                0   \n",
       "               Min/max copy count range not supported                  0   \n",
       "               NCBI36 genomic only                                     0   \n",
       "               No hgvs or location info                                0   \n",
       "True           Absolute copy count                                     0   \n",
       "               Canonical SPDI                                    2118669   \n",
       "               Copy number change (cn loss|del and cn gain|dup)        0   \n",
       "               Remaining valid hgvs alleles                          927   \n",
       "\n",
       "                                                                                  \\\n",
       "out.type                                                        CopyNumberChange   \n",
       "support_status in.vrs_xform_plan.policy                                            \n",
       "False          Genotype/Haplotype                                              0   \n",
       "               Invalid/unsupported hgvs                                        0   \n",
       "               Min/max copy count range not supported                          0   \n",
       "               NCBI36 genomic only                                             0   \n",
       "               No hgvs or location info                                        0   \n",
       "True           Absolute copy count                                             1   \n",
       "               Canonical SPDI                                                  0   \n",
       "               Copy number change (cn loss|del and cn gain|dup)            26889   \n",
       "               Remaining valid hgvs alleles                                    0   \n",
       "\n",
       "                                                                                 \\\n",
       "out.type                                                        CopyNumberCount   \n",
       "support_status in.vrs_xform_plan.policy                                           \n",
       "False          Genotype/Haplotype                                             0   \n",
       "               Invalid/unsupported hgvs                                       0   \n",
       "               Min/max copy count range not supported                         0   \n",
       "               NCBI36 genomic only                                            0   \n",
       "               No hgvs or location info                                       0   \n",
       "True           Absolute copy count                                        52440   \n",
       "               Canonical SPDI                                                 0   \n",
       "               Copy number change (cn loss|del and cn gain|dup)               0   \n",
       "               Remaining valid hgvs alleles                                   0   \n",
       "\n",
       "                                                                            \n",
       "out.type                                                        NONE  Text  \n",
       "support_status in.vrs_xform_plan.policy                                     \n",
       "False          Genotype/Haplotype                                  0  1440  \n",
       "               Invalid/unsupported hgvs                           19  1317  \n",
       "               Min/max copy count range not supported              0    14  \n",
       "               NCBI36 genomic only                                 0  4771  \n",
       "               No hgvs or location info                            0  3089  \n",
       "True           Absolute copy count                               819     3  \n",
       "               Canonical SPDI                                      0     0  \n",
       "               Copy number change (cn loss|del and cn gain|dup)  209     6  \n",
       "               Remaining valid hgvs alleles                       13     1  "
      ]
     },
     "execution_count": 71,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df[[\"in.id\", \"support_status\", \"in.vrs_xform_plan.policy\", \"out.type\"]].fillna(\n",
    "    \"NONE\"\n",
    ").groupby([\"support_status\", \"in.vrs_xform_plan.policy\", \"out.type\"]).count().unstack(\n",
    "    level=2\n",
    ").fillna(\n",
    "    0\n",
    ").astype(\n",
    "    int\n",
    ")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## <a id='toc1_4_'></a>[Create subgroups based on Variant Status](#toc0_)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### <a id='toc1_4_1_'></a>[Supported and Normalized Variants](#toc0_)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {},
   "outputs": [],
   "source": [
    "supported_df = df.copy()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>in.id</th>\n",
       "      <th>in.name</th>\n",
       "      <th>in.subclass_type</th>\n",
       "      <th>in.variation_type</th>\n",
       "      <th>in.xrefs</th>\n",
       "      <th>in.vrs_xform_plan.type</th>\n",
       "      <th>in.vrs_xform_plan.inputs</th>\n",
       "      <th>in.vrs_xform_plan.policy</th>\n",
       "      <th>out.id</th>\n",
       "      <th>out.type</th>\n",
       "      <th>...</th>\n",
       "      <th>out.seq.ref_allele_vcf</th>\n",
       "      <th>out.seq.position_vcf</th>\n",
       "      <th>out.seq.alt_allele_vcf</th>\n",
       "      <th>in.max_copies</th>\n",
       "      <th>in.min_copies</th>\n",
       "      <th>support_status</th>\n",
       "      <th>error_string</th>\n",
       "      <th>error_string_reduce</th>\n",
       "      <th>error_string_reduce_2</th>\n",
       "      <th>normalize_status</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>149</th>\n",
       "      <td>1676638</td>\n",
       "      <td>NM_000094.4(COL7A1):c.8729G&gt;T (p.Gly2910Val)</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Allele</td>\n",
       "      <td>[hgvs]</td>\n",
       "      <td>Remaining valid hgvs alleles</td>\n",
       "      <td>ga4gh:VA.aUAN4LCHVZ_CPwAQfkVuX42g_Gi6WlBu</td>\n",
       "      <td>Allele</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>150</th>\n",
       "      <td>1676635</td>\n",
       "      <td>NM_000094.4(COL7A1):c.80A&gt;T (p.Glu27Val)</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Allele</td>\n",
       "      <td>[hgvs]</td>\n",
       "      <td>Remaining valid hgvs alleles</td>\n",
       "      <td>ga4gh:VA.gQ9qhjkJfB8GnGCeVxyDDLZRkl2W2HK8</td>\n",
       "      <td>Allele</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>151</th>\n",
       "      <td>1676330</td>\n",
       "      <td>NM_054027.6(ANKH):c.259G&gt;A (p.Val87Ile)</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Allele</td>\n",
       "      <td>[hgvs]</td>\n",
       "      <td>Remaining valid hgvs alleles</td>\n",
       "      <td>ga4gh:VA.T3hLVZajyx7AGQKV7la2RgE9CO0Wiv2y</td>\n",
       "      <td>Allele</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>152</th>\n",
       "      <td>1676394</td>\n",
       "      <td>NM_000786.4(CYP51A1):c.1291C&gt;T (p.Arg431Cys)</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Allele</td>\n",
       "      <td>[hgvs]</td>\n",
       "      <td>Remaining valid hgvs alleles</td>\n",
       "      <td>ga4gh:VA.fQvrcvvCsScmvBvq4d6nxm09Q7avAmbX</td>\n",
       "      <td>Allele</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>153</th>\n",
       "      <td>1676476</td>\n",
       "      <td>NM_012309.5(SHANK2):c.460C&gt;T (p.Gln154Ter)</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Allele</td>\n",
       "      <td>[hgvs]</td>\n",
       "      <td>Remaining valid hgvs alleles</td>\n",
       "      <td>ga4gh:VA.dZAOK2Sy6aJlXan36nZYUjEJpk5WX8pF</td>\n",
       "      <td>Allele</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2210619</th>\n",
       "      <td>2202105</td>\n",
       "      <td>NM_153676.4(USH1C):c.496+14_496+15insGTACTCCAT...</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>Microsatellite</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Allele</td>\n",
       "      <td>[canonical_spdi]</td>\n",
       "      <td>Canonical SPDI</td>\n",
       "      <td>ga4gh:VA.5jQEJZIbf1ttjbIGJVhfrHR9XcB7vqWD</td>\n",
       "      <td>Allele</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2210620</th>\n",
       "      <td>1972383</td>\n",
       "      <td>NM_003470.3(USP7):c.383+10_383+11insGTTTAAATGA...</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>Insertion</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Allele</td>\n",
       "      <td>[canonical_spdi]</td>\n",
       "      <td>Canonical SPDI</td>\n",
       "      <td>ga4gh:VA.MODdP1SJ77g3APQ51wa_-2O3y6IZaKNZ</td>\n",
       "      <td>Allele</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2210621</th>\n",
       "      <td>1801441</td>\n",
       "      <td>NM_015189.3(EXOC6B):c.2197-66917_2197-66916ins...</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>Insertion</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Allele</td>\n",
       "      <td>[canonical_spdi]</td>\n",
       "      <td>Canonical SPDI</td>\n",
       "      <td>ga4gh:VA.Vvl6S6es_pV762RXhsgjZHEPHL5PpRU7</td>\n",
       "      <td>Allele</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2210622</th>\n",
       "      <td>1496502</td>\n",
       "      <td>NM_007255.3(B4GALT7):c.881_882insTGAGGTGGATTAA...</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>Insertion</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Allele</td>\n",
       "      <td>[canonical_spdi]</td>\n",
       "      <td>Canonical SPDI</td>\n",
       "      <td>ga4gh:VA.P-rdDkBZMl4kwW-ZHmsMOy0ik5MKa4LI</td>\n",
       "      <td>Allele</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2210623</th>\n",
       "      <td>1801464</td>\n",
       "      <td>NM_176787.5(PIGN):c.2673-514_2673-513insGCAAAC...</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>Insertion</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Allele</td>\n",
       "      <td>[hgvs]</td>\n",
       "      <td>Remaining valid hgvs alleles</td>\n",
       "      <td>ga4gh:VA.3BXZ9oWeEqz4VM16yV8mgjvQ8ZTElB2J</td>\n",
       "      <td>Allele</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2198936 rows × 92 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           in.id                                            in.name  \\\n",
       "149      1676638       NM_000094.4(COL7A1):c.8729G>T (p.Gly2910Val)   \n",
       "150      1676635           NM_000094.4(COL7A1):c.80A>T (p.Glu27Val)   \n",
       "151      1676330            NM_054027.6(ANKH):c.259G>A (p.Val87Ile)   \n",
       "152      1676394       NM_000786.4(CYP51A1):c.1291C>T (p.Arg431Cys)   \n",
       "153      1676476         NM_012309.5(SHANK2):c.460C>T (p.Gln154Ter)   \n",
       "...          ...                                                ...   \n",
       "2210619  2202105  NM_153676.4(USH1C):c.496+14_496+15insGTACTCCAT...   \n",
       "2210620  1972383  NM_003470.3(USP7):c.383+10_383+11insGTTTAAATGA...   \n",
       "2210621  1801441  NM_015189.3(EXOC6B):c.2197-66917_2197-66916ins...   \n",
       "2210622  1496502  NM_007255.3(B4GALT7):c.881_882insTGAGGTGGATTAA...   \n",
       "2210623  1801464  NM_176787.5(PIGN):c.2673-514_2673-513insGCAAAC...   \n",
       "\n",
       "        in.subclass_type          in.variation_type in.xrefs  \\\n",
       "149         SimpleAllele  single nucleotide variant      NaN   \n",
       "150         SimpleAllele  single nucleotide variant      NaN   \n",
       "151         SimpleAllele  single nucleotide variant      NaN   \n",
       "152         SimpleAllele  single nucleotide variant      NaN   \n",
       "153         SimpleAllele  single nucleotide variant      NaN   \n",
       "...                  ...                        ...      ...   \n",
       "2210619     SimpleAllele             Microsatellite      NaN   \n",
       "2210620     SimpleAllele                  Insertion      NaN   \n",
       "2210621     SimpleAllele                  Insertion      NaN   \n",
       "2210622     SimpleAllele                  Insertion      NaN   \n",
       "2210623     SimpleAllele                  Insertion      NaN   \n",
       "\n",
       "        in.vrs_xform_plan.type in.vrs_xform_plan.inputs  \\\n",
       "149                     Allele                   [hgvs]   \n",
       "150                     Allele                   [hgvs]   \n",
       "151                     Allele                   [hgvs]   \n",
       "152                     Allele                   [hgvs]   \n",
       "153                     Allele                   [hgvs]   \n",
       "...                        ...                      ...   \n",
       "2210619                 Allele         [canonical_spdi]   \n",
       "2210620                 Allele         [canonical_spdi]   \n",
       "2210621                 Allele         [canonical_spdi]   \n",
       "2210622                 Allele         [canonical_spdi]   \n",
       "2210623                 Allele                   [hgvs]   \n",
       "\n",
       "             in.vrs_xform_plan.policy  \\\n",
       "149      Remaining valid hgvs alleles   \n",
       "150      Remaining valid hgvs alleles   \n",
       "151      Remaining valid hgvs alleles   \n",
       "152      Remaining valid hgvs alleles   \n",
       "153      Remaining valid hgvs alleles   \n",
       "...                               ...   \n",
       "2210619                Canonical SPDI   \n",
       "2210620                Canonical SPDI   \n",
       "2210621                Canonical SPDI   \n",
       "2210622                Canonical SPDI   \n",
       "2210623  Remaining valid hgvs alleles   \n",
       "\n",
       "                                            out.id out.type  ...  \\\n",
       "149      ga4gh:VA.aUAN4LCHVZ_CPwAQfkVuX42g_Gi6WlBu   Allele  ...   \n",
       "150      ga4gh:VA.gQ9qhjkJfB8GnGCeVxyDDLZRkl2W2HK8   Allele  ...   \n",
       "151      ga4gh:VA.T3hLVZajyx7AGQKV7la2RgE9CO0Wiv2y   Allele  ...   \n",
       "152      ga4gh:VA.fQvrcvvCsScmvBvq4d6nxm09Q7avAmbX   Allele  ...   \n",
       "153      ga4gh:VA.dZAOK2Sy6aJlXan36nZYUjEJpk5WX8pF   Allele  ...   \n",
       "...                                            ...      ...  ...   \n",
       "2210619  ga4gh:VA.5jQEJZIbf1ttjbIGJVhfrHR9XcB7vqWD   Allele  ...   \n",
       "2210620  ga4gh:VA.MODdP1SJ77g3APQ51wa_-2O3y6IZaKNZ   Allele  ...   \n",
       "2210621  ga4gh:VA.Vvl6S6es_pV762RXhsgjZHEPHL5PpRU7   Allele  ...   \n",
       "2210622  ga4gh:VA.P-rdDkBZMl4kwW-ZHmsMOy0ik5MKa4LI   Allele  ...   \n",
       "2210623  ga4gh:VA.3BXZ9oWeEqz4VM16yV8mgjvQ8ZTElB2J   Allele  ...   \n",
       "\n",
       "        out.seq.ref_allele_vcf out.seq.position_vcf out.seq.alt_allele_vcf  \\\n",
       "149                        NaN                  NaN                    NaN   \n",
       "150                        NaN                  NaN                    NaN   \n",
       "151                        NaN                  NaN                    NaN   \n",
       "152                        NaN                  NaN                    NaN   \n",
       "153                        NaN                  NaN                    NaN   \n",
       "...                        ...                  ...                    ...   \n",
       "2210619                    NaN                  NaN                    NaN   \n",
       "2210620                    NaN                  NaN                    NaN   \n",
       "2210621                    NaN                  NaN                    NaN   \n",
       "2210622                    NaN                  NaN                    NaN   \n",
       "2210623                    NaN                  NaN                    NaN   \n",
       "\n",
       "        in.max_copies in.min_copies support_status error_string  \\\n",
       "149               NaN           NaN           True                \n",
       "150               NaN           NaN           True                \n",
       "151               NaN           NaN           True                \n",
       "152               NaN           NaN           True                \n",
       "153               NaN           NaN           True                \n",
       "...               ...           ...            ...          ...   \n",
       "2210619           NaN           NaN           True                \n",
       "2210620           NaN           NaN           True                \n",
       "2210621           NaN           NaN           True                \n",
       "2210622           NaN           NaN           True                \n",
       "2210623           NaN           NaN           True                \n",
       "\n",
       "         error_string_reduce error_string_reduce_2  normalize_status  \n",
       "149                                                             True  \n",
       "150                                                             True  \n",
       "151                                                             True  \n",
       "152                                                             True  \n",
       "153                                                             True  \n",
       "...                      ...                   ...               ...  \n",
       "2210619                                                         True  \n",
       "2210620                                                         True  \n",
       "2210621                                                         True  \n",
       "2210622                                                         True  \n",
       "2210623                                                         True  \n",
       "\n",
       "[2198936 rows x 92 columns]"
      ]
     },
     "execution_count": 73,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "supported_df = supported_df.loc[\n",
    "    (supported_df[\"support_status\"] == True)\n",
    "    & (supported_df[\"normalize_status\"] == True)\n",
    "]\n",
    "supported_df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### <a id='toc1_4_2_'></a>[Supported and Not Normalized Variants](#toc0_)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "metadata": {},
   "outputs": [],
   "source": [
    "supported_not_normalized_df = df.copy()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>in.id</th>\n",
       "      <th>in.name</th>\n",
       "      <th>in.subclass_type</th>\n",
       "      <th>in.variation_type</th>\n",
       "      <th>in.xrefs</th>\n",
       "      <th>in.vrs_xform_plan.type</th>\n",
       "      <th>in.vrs_xform_plan.inputs</th>\n",
       "      <th>in.vrs_xform_plan.policy</th>\n",
       "      <th>out.id</th>\n",
       "      <th>out.type</th>\n",
       "      <th>...</th>\n",
       "      <th>out.seq.ref_allele_vcf</th>\n",
       "      <th>out.seq.position_vcf</th>\n",
       "      <th>out.seq.alt_allele_vcf</th>\n",
       "      <th>in.max_copies</th>\n",
       "      <th>in.min_copies</th>\n",
       "      <th>support_status</th>\n",
       "      <th>error_string</th>\n",
       "      <th>error_string_reduce</th>\n",
       "      <th>error_string_reduce_2</th>\n",
       "      <th>normalize_status</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>170</th>\n",
       "      <td>989220</td>\n",
       "      <td>NC_000015.9:g.(44884528_44881613)_(44877833_44...</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>Duplication</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CopyNumberChange</td>\n",
       "      <td>[hgvs]</td>\n",
       "      <td>Copy number change (cn loss|del and cn gain|dup)</td>\n",
       "      <td>989220</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>Error returned from variation normalizer;Unabl...</td>\n",
       "      <td>error returned from variation normalizer;unabl...</td>\n",
       "      <td>error returned from variation normalizer;unabl...</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>171</th>\n",
       "      <td>10342</td>\n",
       "      <td>NG_011403.2:g.(80027_96047)_(131648_164496)del</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>OMIM:300841.0259</td>\n",
       "      <td>CopyNumberChange</td>\n",
       "      <td>[hgvs]</td>\n",
       "      <td>Copy number change (cn loss|del and cn gain|dup)</td>\n",
       "      <td>10342</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>Error returned from variation normalizer;Unabl...</td>\n",
       "      <td>error returned from variation normalizer;unabl...</td>\n",
       "      <td>error returned from variation normalizer;unabl...</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>212</th>\n",
       "      <td>602019</td>\n",
       "      <td>GRCh37/hg19 15q11.2(chr15:22750305-23140114)x3</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>copy number gain</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CopyNumberCount</td>\n",
       "      <td>[hgvs, absolute_copies]</td>\n",
       "      <td>Absolute copy count</td>\n",
       "      <td>602019</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>Error returned from variation normalizer;Unabl...</td>\n",
       "      <td>error returned from variation normalizer;unabl...</td>\n",
       "      <td>error returned from variation normalizer;unabl...</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>284</th>\n",
       "      <td>1706497</td>\n",
       "      <td>GRCh37/hg19 16p11.2(chr16:29432212-30177807)x1</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>copy number loss</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CopyNumberCount</td>\n",
       "      <td>[hgvs, absolute_copies]</td>\n",
       "      <td>Absolute copy count</td>\n",
       "      <td>1706497</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>Error returned from variation normalizer;Unabl...</td>\n",
       "      <td>error returned from variation normalizer;unabl...</td>\n",
       "      <td>error returned from variation normalizer;unabl...</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>315</th>\n",
       "      <td>1705934</td>\n",
       "      <td>GRCh37/hg19 Xp22.33(chrX:566009-1356042)x3</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>copy number gain</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CopyNumberCount</td>\n",
       "      <td>[hgvs, absolute_copies]</td>\n",
       "      <td>Absolute copy count</td>\n",
       "      <td>1705934</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>Error returned from variation normalizer;Unabl...</td>\n",
       "      <td>error returned from variation normalizer;unabl...</td>\n",
       "      <td>error returned from variation normalizer;unabl...</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2143310</th>\n",
       "      <td>146444</td>\n",
       "      <td>GRCh37/hg19 1q21.1-21.2(chr1:145425395-1488676...</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>copy number loss</td>\n",
       "      <td>dbVar:nssv1415305,dbVar:nsv531853</td>\n",
       "      <td>CopyNumberCount</td>\n",
       "      <td>[hgvs, absolute_copies]</td>\n",
       "      <td>Absolute copy count</td>\n",
       "      <td>146444</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>Error returned from variation normalizer;Unabl...</td>\n",
       "      <td>error returned from variation normalizer;unabl...</td>\n",
       "      <td>error returned from variation normalizer;unabl...</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2143311</th>\n",
       "      <td>830807</td>\n",
       "      <td>NC_000001.10:g.(?_145498103)_(145538307_?)dup</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>Duplication</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CopyNumberChange</td>\n",
       "      <td>[hgvs]</td>\n",
       "      <td>Copy number change (cn loss|del and cn gain|dup)</td>\n",
       "      <td>830807</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>Error returned from variation normalizer;Unabl...</td>\n",
       "      <td>error returned from variation normalizer;unabl...</td>\n",
       "      <td>error returned from variation normalizer;unabl...</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2143339</th>\n",
       "      <td>565216</td>\n",
       "      <td>GRCh37/hg19 1q32.1(chr1:206173911-206288157)x1</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>copy number loss</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CopyNumberCount</td>\n",
       "      <td>[hgvs, absolute_copies]</td>\n",
       "      <td>Absolute copy count</td>\n",
       "      <td>565216</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>Error returned from variation normalizer;Unabl...</td>\n",
       "      <td>error returned from variation normalizer;unabl...</td>\n",
       "      <td>error returned from variation normalizer;unabl...</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2143340</th>\n",
       "      <td>600207</td>\n",
       "      <td>GRCh37/hg19 1q32.1(chr1:206315933-206331193)x1</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>copy number loss</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CopyNumberCount</td>\n",
       "      <td>[hgvs, absolute_copies]</td>\n",
       "      <td>Absolute copy count</td>\n",
       "      <td>600207</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>Error returned from variation normalizer;Unabl...</td>\n",
       "      <td>error returned from variation normalizer;unabl...</td>\n",
       "      <td>error returned from variation normalizer;unabl...</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2143679</th>\n",
       "      <td>393540</td>\n",
       "      <td>GRCh37/hg19 12q23.1(chr12:100367541-100489446)x1</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>copy number loss</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CopyNumberCount</td>\n",
       "      <td>[hgvs, absolute_copies]</td>\n",
       "      <td>Absolute copy count</td>\n",
       "      <td>393540</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>Error returned from variation normalizer;Unabl...</td>\n",
       "      <td>error returned from variation normalizer;unabl...</td>\n",
       "      <td>error returned from variation normalizer;unabl...</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1041 rows × 92 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           in.id                                            in.name  \\\n",
       "170       989220  NC_000015.9:g.(44884528_44881613)_(44877833_44...   \n",
       "171        10342     NG_011403.2:g.(80027_96047)_(131648_164496)del   \n",
       "212       602019     GRCh37/hg19 15q11.2(chr15:22750305-23140114)x3   \n",
       "284      1706497     GRCh37/hg19 16p11.2(chr16:29432212-30177807)x1   \n",
       "315      1705934         GRCh37/hg19 Xp22.33(chrX:566009-1356042)x3   \n",
       "...          ...                                                ...   \n",
       "2143310   146444  GRCh37/hg19 1q21.1-21.2(chr1:145425395-1488676...   \n",
       "2143311   830807      NC_000001.10:g.(?_145498103)_(145538307_?)dup   \n",
       "2143339   565216     GRCh37/hg19 1q32.1(chr1:206173911-206288157)x1   \n",
       "2143340   600207     GRCh37/hg19 1q32.1(chr1:206315933-206331193)x1   \n",
       "2143679   393540   GRCh37/hg19 12q23.1(chr12:100367541-100489446)x1   \n",
       "\n",
       "        in.subclass_type in.variation_type                           in.xrefs  \\\n",
       "170         SimpleAllele       Duplication                                NaN   \n",
       "171         SimpleAllele          Deletion                   OMIM:300841.0259   \n",
       "212         SimpleAllele  copy number gain                                NaN   \n",
       "284         SimpleAllele  copy number loss                                NaN   \n",
       "315         SimpleAllele  copy number gain                                NaN   \n",
       "...                  ...               ...                                ...   \n",
       "2143310     SimpleAllele  copy number loss  dbVar:nssv1415305,dbVar:nsv531853   \n",
       "2143311     SimpleAllele       Duplication                                NaN   \n",
       "2143339     SimpleAllele  copy number loss                                NaN   \n",
       "2143340     SimpleAllele  copy number loss                                NaN   \n",
       "2143679     SimpleAllele  copy number loss                                NaN   \n",
       "\n",
       "        in.vrs_xform_plan.type in.vrs_xform_plan.inputs  \\\n",
       "170           CopyNumberChange                   [hgvs]   \n",
       "171           CopyNumberChange                   [hgvs]   \n",
       "212            CopyNumberCount  [hgvs, absolute_copies]   \n",
       "284            CopyNumberCount  [hgvs, absolute_copies]   \n",
       "315            CopyNumberCount  [hgvs, absolute_copies]   \n",
       "...                        ...                      ...   \n",
       "2143310        CopyNumberCount  [hgvs, absolute_copies]   \n",
       "2143311       CopyNumberChange                   [hgvs]   \n",
       "2143339        CopyNumberCount  [hgvs, absolute_copies]   \n",
       "2143340        CopyNumberCount  [hgvs, absolute_copies]   \n",
       "2143679        CopyNumberCount  [hgvs, absolute_copies]   \n",
       "\n",
       "                                 in.vrs_xform_plan.policy   out.id out.type  \\\n",
       "170      Copy number change (cn loss|del and cn gain|dup)   989220      NaN   \n",
       "171      Copy number change (cn loss|del and cn gain|dup)    10342      NaN   \n",
       "212                                   Absolute copy count   602019      NaN   \n",
       "284                                   Absolute copy count  1706497      NaN   \n",
       "315                                   Absolute copy count  1705934      NaN   \n",
       "...                                                   ...      ...      ...   \n",
       "2143310                               Absolute copy count   146444      NaN   \n",
       "2143311  Copy number change (cn loss|del and cn gain|dup)   830807      NaN   \n",
       "2143339                               Absolute copy count   565216      NaN   \n",
       "2143340                               Absolute copy count   600207      NaN   \n",
       "2143679                               Absolute copy count   393540      NaN   \n",
       "\n",
       "         ... out.seq.ref_allele_vcf out.seq.position_vcf  \\\n",
       "170      ...                    NaN                  NaN   \n",
       "171      ...                    NaN                  NaN   \n",
       "212      ...                    NaN                  NaN   \n",
       "284      ...                    NaN                  NaN   \n",
       "315      ...                    NaN                  NaN   \n",
       "...      ...                    ...                  ...   \n",
       "2143310  ...                    NaN                  NaN   \n",
       "2143311  ...                    NaN                  NaN   \n",
       "2143339  ...                    NaN                  NaN   \n",
       "2143340  ...                    NaN                  NaN   \n",
       "2143679  ...                    NaN                  NaN   \n",
       "\n",
       "        out.seq.alt_allele_vcf in.max_copies in.min_copies support_status  \\\n",
       "170                        NaN           NaN           NaN           True   \n",
       "171                        NaN           NaN           NaN           True   \n",
       "212                        NaN           NaN           NaN           True   \n",
       "284                        NaN           NaN           NaN           True   \n",
       "315                        NaN           NaN           NaN           True   \n",
       "...                        ...           ...           ...            ...   \n",
       "2143310                    NaN           NaN           NaN           True   \n",
       "2143311                    NaN           NaN           NaN           True   \n",
       "2143339                    NaN           NaN           NaN           True   \n",
       "2143340                    NaN           NaN           NaN           True   \n",
       "2143679                    NaN           NaN           NaN           True   \n",
       "\n",
       "                                              error_string  \\\n",
       "170      Error returned from variation normalizer;Unabl...   \n",
       "171      Error returned from variation normalizer;Unabl...   \n",
       "212      Error returned from variation normalizer;Unabl...   \n",
       "284      Error returned from variation normalizer;Unabl...   \n",
       "315      Error returned from variation normalizer;Unabl...   \n",
       "...                                                    ...   \n",
       "2143310  Error returned from variation normalizer;Unabl...   \n",
       "2143311  Error returned from variation normalizer;Unabl...   \n",
       "2143339  Error returned from variation normalizer;Unabl...   \n",
       "2143340  Error returned from variation normalizer;Unabl...   \n",
       "2143679  Error returned from variation normalizer;Unabl...   \n",
       "\n",
       "                                       error_string_reduce  \\\n",
       "170      error returned from variation normalizer;unabl...   \n",
       "171      error returned from variation normalizer;unabl...   \n",
       "212      error returned from variation normalizer;unabl...   \n",
       "284      error returned from variation normalizer;unabl...   \n",
       "315      error returned from variation normalizer;unabl...   \n",
       "...                                                    ...   \n",
       "2143310  error returned from variation normalizer;unabl...   \n",
       "2143311  error returned from variation normalizer;unabl...   \n",
       "2143339  error returned from variation normalizer;unabl...   \n",
       "2143340  error returned from variation normalizer;unabl...   \n",
       "2143679  error returned from variation normalizer;unabl...   \n",
       "\n",
       "                                     error_string_reduce_2  normalize_status  \n",
       "170      error returned from variation normalizer;unabl...             False  \n",
       "171      error returned from variation normalizer;unabl...             False  \n",
       "212      error returned from variation normalizer;unabl...             False  \n",
       "284      error returned from variation normalizer;unabl...             False  \n",
       "315      error returned from variation normalizer;unabl...             False  \n",
       "...                                                    ...               ...  \n",
       "2143310  error returned from variation normalizer;unabl...             False  \n",
       "2143311  error returned from variation normalizer;unabl...             False  \n",
       "2143339  error returned from variation normalizer;unabl...             False  \n",
       "2143340  error returned from variation normalizer;unabl...             False  \n",
       "2143679  error returned from variation normalizer;unabl...             False  \n",
       "\n",
       "[1041 rows x 92 columns]"
      ]
     },
     "execution_count": 75,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "supported_not_normalized_df = supported_not_normalized_df.loc[\n",
    "    (supported_not_normalized_df[\"support_status\"] == True)\n",
    "    & (supported_not_normalized_df[\"normalize_status\"] == False)\n",
    "]\n",
    "supported_not_normalized_df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### <a id='toc1_4_3_'></a>[Not Supported Variants](#toc0_)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "metadata": {},
   "outputs": [],
   "source": [
    "not_supported_df = df.copy()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>in.id</th>\n",
       "      <th>in.name</th>\n",
       "      <th>in.subclass_type</th>\n",
       "      <th>in.variation_type</th>\n",
       "      <th>in.xrefs</th>\n",
       "      <th>in.vrs_xform_plan.type</th>\n",
       "      <th>in.vrs_xform_plan.inputs</th>\n",
       "      <th>in.vrs_xform_plan.policy</th>\n",
       "      <th>out.id</th>\n",
       "      <th>out.type</th>\n",
       "      <th>...</th>\n",
       "      <th>out.seq.ref_allele_vcf</th>\n",
       "      <th>out.seq.position_vcf</th>\n",
       "      <th>out.seq.alt_allele_vcf</th>\n",
       "      <th>in.max_copies</th>\n",
       "      <th>in.min_copies</th>\n",
       "      <th>support_status</th>\n",
       "      <th>error_string</th>\n",
       "      <th>error_string_reduce</th>\n",
       "      <th>error_string_reduce_2</th>\n",
       "      <th>normalize_status</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>16098</td>\n",
       "      <td>SLC2A2, 1-BP INS, 793C</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>Insertion</td>\n",
       "      <td>OMIM:138160.0009</td>\n",
       "      <td>Text</td>\n",
       "      <td>[id]</td>\n",
       "      <td>No hgvs or location info</td>\n",
       "      <td>Text:clinvar:16098</td>\n",
       "      <td>Text</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td></td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>425693</td>\n",
       "      <td>NM_001204.6(BMPR2):c.77-?_247+?dup</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>Duplication</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Text</td>\n",
       "      <td>[id]</td>\n",
       "      <td>No hgvs or location info</td>\n",
       "      <td>Text:clinvar:425693</td>\n",
       "      <td>Text</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td></td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2446408</td>\n",
       "      <td>CDHR1, 783G-A (rs147346345)</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>OMIM:609502.0005</td>\n",
       "      <td>Text</td>\n",
       "      <td>[id]</td>\n",
       "      <td>No hgvs or location info</td>\n",
       "      <td>Text:clinvar:2446408</td>\n",
       "      <td>Text</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td></td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>90650</td>\n",
       "      <td>NM_000251.2(MSH2):c.1387-?_1510+?del</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Text</td>\n",
       "      <td>[id]</td>\n",
       "      <td>No hgvs or location info</td>\n",
       "      <td>Text:clinvar:90650</td>\n",
       "      <td>Text</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td></td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>14905</td>\n",
       "      <td>HLA-DRB1, HLA-DRB1*1101</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>Variation</td>\n",
       "      <td>OMIM:142857.0001</td>\n",
       "      <td>Text</td>\n",
       "      <td>[id]</td>\n",
       "      <td>No hgvs or location info</td>\n",
       "      <td>Text:clinvar:14905</td>\n",
       "      <td>Text</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td></td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2199354</th>\n",
       "      <td>1418992</td>\n",
       "      <td>NM_002076.4(GNS):c.841_842insTTTTTTTTTTTTTTTTT...</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>Insertion</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Text</td>\n",
       "      <td>[id]</td>\n",
       "      <td>Invalid/unsupported hgvs</td>\n",
       "      <td>Text:clinvar:1418992</td>\n",
       "      <td>Text</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td></td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2199355</th>\n",
       "      <td>2134754</td>\n",
       "      <td>NM_152564.5(VPS13B):c.5627_5628insTTTTTTTTTTTT...</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>Insertion</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Text</td>\n",
       "      <td>[id]</td>\n",
       "      <td>Invalid/unsupported hgvs</td>\n",
       "      <td>Text:clinvar:2134754</td>\n",
       "      <td>Text</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td></td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2210624</th>\n",
       "      <td>1513408</td>\n",
       "      <td>NM_024928.5(STN1):c.340_352AAG[2]CTACAAGGCCGGG...</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>Insertion</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Text</td>\n",
       "      <td>[id]</td>\n",
       "      <td>Invalid/unsupported hgvs</td>\n",
       "      <td>Text:clinvar:1513408</td>\n",
       "      <td>Text</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td></td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2210625</th>\n",
       "      <td>1464440</td>\n",
       "      <td>NM_001267550.2(TTN):c.29512_29513insGGCCGGGCGC...</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>Insertion</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Text</td>\n",
       "      <td>[id]</td>\n",
       "      <td>Invalid/unsupported hgvs</td>\n",
       "      <td>Text:clinvar:1464440</td>\n",
       "      <td>Text</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td></td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2210626</th>\n",
       "      <td>1453116</td>\n",
       "      <td>NM_000051.4(ATM):c.3376_3377insGGCCGGGCGCGGTGG...</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>Insertion</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Text</td>\n",
       "      <td>[id]</td>\n",
       "      <td>Invalid/unsupported hgvs</td>\n",
       "      <td>Text:clinvar:1453116</td>\n",
       "      <td>Text</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td></td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>10650 rows × 92 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           in.id                                            in.name  \\\n",
       "0          16098                             SLC2A2, 1-BP INS, 793C   \n",
       "1         425693                 NM_001204.6(BMPR2):c.77-?_247+?dup   \n",
       "2        2446408                        CDHR1, 783G-A (rs147346345)   \n",
       "3          90650               NM_000251.2(MSH2):c.1387-?_1510+?del   \n",
       "4          14905                            HLA-DRB1, HLA-DRB1*1101   \n",
       "...          ...                                                ...   \n",
       "2199354  1418992  NM_002076.4(GNS):c.841_842insTTTTTTTTTTTTTTTTT...   \n",
       "2199355  2134754  NM_152564.5(VPS13B):c.5627_5628insTTTTTTTTTTTT...   \n",
       "2210624  1513408  NM_024928.5(STN1):c.340_352AAG[2]CTACAAGGCCGGG...   \n",
       "2210625  1464440  NM_001267550.2(TTN):c.29512_29513insGGCCGGGCGC...   \n",
       "2210626  1453116  NM_000051.4(ATM):c.3376_3377insGGCCGGGCGCGGTGG...   \n",
       "\n",
       "        in.subclass_type          in.variation_type          in.xrefs  \\\n",
       "0           SimpleAllele                  Insertion  OMIM:138160.0009   \n",
       "1           SimpleAllele                Duplication               NaN   \n",
       "2           SimpleAllele  single nucleotide variant  OMIM:609502.0005   \n",
       "3           SimpleAllele                   Deletion               NaN   \n",
       "4           SimpleAllele                  Variation  OMIM:142857.0001   \n",
       "...                  ...                        ...               ...   \n",
       "2199354     SimpleAllele                  Insertion               NaN   \n",
       "2199355     SimpleAllele                  Insertion               NaN   \n",
       "2210624     SimpleAllele                  Insertion               NaN   \n",
       "2210625     SimpleAllele                  Insertion               NaN   \n",
       "2210626     SimpleAllele                  Insertion               NaN   \n",
       "\n",
       "        in.vrs_xform_plan.type in.vrs_xform_plan.inputs  \\\n",
       "0                         Text                     [id]   \n",
       "1                         Text                     [id]   \n",
       "2                         Text                     [id]   \n",
       "3                         Text                     [id]   \n",
       "4                         Text                     [id]   \n",
       "...                        ...                      ...   \n",
       "2199354                   Text                     [id]   \n",
       "2199355                   Text                     [id]   \n",
       "2210624                   Text                     [id]   \n",
       "2210625                   Text                     [id]   \n",
       "2210626                   Text                     [id]   \n",
       "\n",
       "         in.vrs_xform_plan.policy                out.id out.type  ...  \\\n",
       "0        No hgvs or location info    Text:clinvar:16098     Text  ...   \n",
       "1        No hgvs or location info   Text:clinvar:425693     Text  ...   \n",
       "2        No hgvs or location info  Text:clinvar:2446408     Text  ...   \n",
       "3        No hgvs or location info    Text:clinvar:90650     Text  ...   \n",
       "4        No hgvs or location info    Text:clinvar:14905     Text  ...   \n",
       "...                           ...                   ...      ...  ...   \n",
       "2199354  Invalid/unsupported hgvs  Text:clinvar:1418992     Text  ...   \n",
       "2199355  Invalid/unsupported hgvs  Text:clinvar:2134754     Text  ...   \n",
       "2210624  Invalid/unsupported hgvs  Text:clinvar:1513408     Text  ...   \n",
       "2210625  Invalid/unsupported hgvs  Text:clinvar:1464440     Text  ...   \n",
       "2210626  Invalid/unsupported hgvs  Text:clinvar:1453116     Text  ...   \n",
       "\n",
       "        out.seq.ref_allele_vcf out.seq.position_vcf out.seq.alt_allele_vcf  \\\n",
       "0                          NaN                  NaN                    NaN   \n",
       "1                          NaN                  NaN                    NaN   \n",
       "2                          NaN                  NaN                    NaN   \n",
       "3                          NaN                  NaN                    NaN   \n",
       "4                          NaN                  NaN                    NaN   \n",
       "...                        ...                  ...                    ...   \n",
       "2199354                    NaN                  NaN                    NaN   \n",
       "2199355                    NaN                  NaN                    NaN   \n",
       "2210624                    NaN                  NaN                    NaN   \n",
       "2210625                    NaN                  NaN                    NaN   \n",
       "2210626                    NaN                  NaN                    NaN   \n",
       "\n",
       "        in.max_copies in.min_copies support_status error_string  \\\n",
       "0                 NaN           NaN          False                \n",
       "1                 NaN           NaN          False                \n",
       "2                 NaN           NaN          False                \n",
       "3                 NaN           NaN          False                \n",
       "4                 NaN           NaN          False                \n",
       "...               ...           ...            ...          ...   \n",
       "2199354           NaN           NaN          False                \n",
       "2199355           NaN           NaN          False                \n",
       "2210624           NaN           NaN          False                \n",
       "2210625           NaN           NaN          False                \n",
       "2210626           NaN           NaN          False                \n",
       "\n",
       "         error_string_reduce error_string_reduce_2  normalize_status  \n",
       "0              Not Supported         Not Supported             False  \n",
       "1              Not Supported         Not Supported             False  \n",
       "2              Not Supported         Not Supported             False  \n",
       "3              Not Supported         Not Supported             False  \n",
       "4              Not Supported         Not Supported             False  \n",
       "...                      ...                   ...               ...  \n",
       "2199354        Not Supported         Not Supported             False  \n",
       "2199355        Not Supported         Not Supported             False  \n",
       "2210624        Not Supported         Not Supported             False  \n",
       "2210625        Not Supported         Not Supported             False  \n",
       "2210626        Not Supported         Not Supported             False  \n",
       "\n",
       "[10650 rows x 92 columns]"
      ]
     },
     "execution_count": 77,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "not_supported_df = not_supported_df.loc[\n",
    "    (not_supported_df[\"support_status\"] == False)\n",
    "    & (not_supported_df[\"normalize_status\"] == False)\n",
    "]\n",
    "not_supported_df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Sanity check: making sure there are no supported variants that have been marked as normalized"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "metadata": {},
   "outputs": [],
   "source": [
    "not_supported_but_normalized_df = df.copy()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>in.id</th>\n",
       "      <th>in.name</th>\n",
       "      <th>in.subclass_type</th>\n",
       "      <th>in.variation_type</th>\n",
       "      <th>in.xrefs</th>\n",
       "      <th>in.vrs_xform_plan.type</th>\n",
       "      <th>in.vrs_xform_plan.inputs</th>\n",
       "      <th>in.vrs_xform_plan.policy</th>\n",
       "      <th>out.id</th>\n",
       "      <th>out.type</th>\n",
       "      <th>...</th>\n",
       "      <th>out.seq.ref_allele_vcf</th>\n",
       "      <th>out.seq.position_vcf</th>\n",
       "      <th>out.seq.alt_allele_vcf</th>\n",
       "      <th>in.max_copies</th>\n",
       "      <th>in.min_copies</th>\n",
       "      <th>support_status</th>\n",
       "      <th>error_string</th>\n",
       "      <th>error_string_reduce</th>\n",
       "      <th>error_string_reduce_2</th>\n",
       "      <th>normalize_status</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>0 rows × 92 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [in.id, in.name, in.subclass_type, in.variation_type, in.xrefs, in.vrs_xform_plan.type, in.vrs_xform_plan.inputs, in.vrs_xform_plan.policy, out.id, out.type, out.definition, in.hgvs.assembly, in.hgvs.nucleotide, out.location.id, out.location.type, out.location.sequence_id, out.location.start.type, out.location.start.value, out.location.end.type, out.location.end.value, out.state.type, out.state.sequence, out.subject.id, out.subject.type, out.subject.sequence_id, out.subject.start.type, out.subject.start.value, out.subject.end.type, out.subject.end.value, out.copy_change, out.subject.start.min, out.subject.start.max, out.subject.end.min, out.subject.end.max, out.name, out.subclass_type, out.variation_type, out.hgvs.assembly, out.hgvs.nucleotide, out.vrs_xform_plan.type, out.vrs_xform_plan.inputs, out.vrs_xform_plan.policy, out.errors, out.xrefs, in.seq.assembly, in.seq.accession, in.seq.chr, in.seq.disp_start, in.seq.disp_stop, in.seq.inner_start, in.seq.inner_stop, in.absolute_copies, in.derived_hgvs, in.cyto, out.subject.start.comparator, out.subject.end.comparator, out.copies.type, out.copies.value, in.seq.outer_stop, in.seq.outer_start, in.seq.variant_length, out.seq.assembly, out.seq.accession, out.seq.chr, out.seq.disp_start, out.seq.disp_stop, out.seq.inner_start, out.seq.inner_stop, out.absolute_copies, out.derived_hgvs, out.cyto, in.seq.start, in.seq.stop, out.seq.outer_start, out.seq.outer_stop, out.seq.variant_length, out.seq.start, out.seq.stop, in.seq.ref_allele_vcf, in.seq.position_vcf, in.seq.alt_allele_vcf, in.canonical_spdi, out.seq.ref_allele_vcf, out.seq.position_vcf, out.seq.alt_allele_vcf, in.max_copies, in.min_copies, support_status, error_string, error_string_reduce, error_string_reduce_2, normalize_status]\n",
       "Index: []\n",
       "\n",
       "[0 rows x 92 columns]"
      ]
     },
     "execution_count": 79,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "not_supported_but_normalized_df = not_supported_but_normalized_df.loc[\n",
    "    (not_supported_but_normalized_df[\"support_status\"] == False)\n",
    "    & (not_supported_but_normalized_df[\"normalize_status\"] == True)\n",
    "]\n",
    "not_supported_but_normalized_df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## <a id='toc1_5_'></a>[Counting variants from each group](#toc0_)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "metadata": {},
   "outputs": [],
   "source": [
    "num_supported = len(supported_df)\n",
    "num_supported_not_normalized = len(supported_not_normalized_df)\n",
    "num_not_supported_but_normalized = len(not_supported_but_normalized_df)\n",
    "num_not_supported = len(not_supported_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2198936\n",
      "1041\n",
      "0\n",
      "10650\n"
     ]
    }
   ],
   "source": [
    "print(num_supported)\n",
    "print(num_supported_not_normalized)\n",
    "print(num_not_supported_but_normalized)\n",
    "print(num_not_supported)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "| |Supported | Not Supported\n",
    "| --- | --- | --- |\n",
    "| Normalized | 2198936 | 0 |\n",
    "| Not Normalized | 1041 | 10650|"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3.10.9 ('.venv': venv)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.9"
  },
  "orig_nbformat": 4,
  "vscode": {
   "interpreter": {
    "hash": "60bedfc6dd95a1d756c285a8080496495ec396023cb257f004626f94ae504b38"
   }
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
