{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# <a id='toc1_'></a>[ClinVar Variant Analysis](#toc0_)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Table of contents**<a id='toc0_'></a>    \n",
    "- [ClinVar Variant Analysis](#toc1_)    \n",
    "  - [Initialize](#toc1_1_)    \n",
    "    - [Import necessary libraries](#toc1_1_1_)    \n",
    "    - [Create output directory](#toc1_1_2_)    \n",
    "    - [Import variant information file](#toc1_1_3_)    \n",
    "  - [Add Supported Status of Variant based on in.vrs_xform_plan.policy](#toc1_2_)    \n",
    "  - [Add Normalization Status of Variant based on out.errors](#toc1_3_)    \n",
    "    - [Set Normalize Status of Variant as T/F](#toc1_3_1_)    \n",
    "      - [Summary Table](#toc1_3_1_1_)    \n",
    "  - [Create subgroups based on Variant Status](#toc1_4_)    \n",
    "    - [Supported and Normalized Variants](#toc1_4_1_)    \n",
    "    - [Supported and Not Normalized Variants](#toc1_4_2_)    \n",
    "    - [Not Supported Variants](#toc1_4_3_)    \n",
    "  - [Counting variants from each group](#toc1_5_)    \n",
    "\n",
    "<!-- vscode-jupyter-toc-config\n",
    "\tnumbering=false\n",
    "\tanchor=true\n",
    "\tflat=false\n",
    "\tminLevel=1\n",
    "\tmaxLevel=6\n",
    "\t/vscode-jupyter-toc-config -->\n",
    "<!-- THIS CELL WILL BE REPLACED ON TOC UPDATE. DO NOT WRITE YOUR TEXT IN THIS CELL -->"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## <a id='toc1_1_'></a>[Initialize](#toc0_)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### <a id='toc1_1_1_'></a>[Import necessary libraries](#toc0_)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import ndjson\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import re\n",
    "from dotenv import load_dotenv\n",
    "from pathlib import Path\n",
    "from boto3.exceptions import ResourceLoadException\n",
    "from botocore.config import Config\n",
    "import boto3\n",
    "import gzip"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### <a id='toc1_1_2_'></a>[Create output directory](#toc0_)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "path = Path(\"clinvar_variation_analysis_output\")\n",
    "path.mkdir(exist_ok = True)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### <a id='toc1_1_3_'></a>[Import variant information file](#toc0_)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "#to refresh SSO session, run aws sso login\n",
    "\n",
    "# s3 = boto3.client('s3')\n",
    "\n",
    "# with open('../clinvar/output-variation_identity-vrs-1.3.ndjson.gz', 'wb') as data:\n",
    "#     s3.download_fileobj('nch-igm-wagner-lab-public', 'variation-normalizer-manuscript/output-variation_identity-vrs-1.3.ndjson.gz', data)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "with gzip.open('output-variation_identity-vrs-1.3.ndjson.gz', 'rb') as f:\n",
    "    file_content = ndjson.load(f)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "df0 = pd.json_normalize(file_content)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = df0.copy()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## <a id='toc1_2_'></a>[Add Supported Status of Variant based on in.vrs_xform_plan.policy](#toc0_)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Checking for blanks"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "df[\"in.vrs_xform_plan.policy\"] = df[\"in.vrs_xform_plan.policy\"].fillna(\"None\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "in.vrs_xform_plan.policy\n",
       "Canonical SPDI                                      2118669\n",
       "Absolute copy count                                   53263\n",
       "Copy number change (cn loss|del and cn gain|dup)      27104\n",
       "NCBI36 genomic only                                    4771\n",
       "No hgvs or location info                               3089\n",
       "Genotype/Haplotype                                     1440\n",
       "Invalid/unsupported hgvs                               1336\n",
       "Remaining valid hgvs alleles                            941\n",
       "Min/max copy count range not supported                   14\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df[\"in.vrs_xform_plan.policy\"].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "df[\"support_status\"] = df[\"in.vrs_xform_plan.policy\"].copy()\n",
    "\n",
    "df.loc[df[\"support_status\"] == \"Canonical SPDI\", \"support_status\"] = True\n",
    "df.loc[df[\"support_status\"] == \"Absolute copy count\", \"support_status\"] = True\n",
    "df.loc[df[\"support_status\"] == \"Copy number change (cn loss|del and cn gain|dup)\",\n",
    "    \"support_status\"] = True\n",
    "df.loc[df[\"support_status\"] == \"NCBI36 genomic only\", \"support_status\"] = False\n",
    "df.loc[df[\"support_status\"] == \"No hgvs or location info\", \"support_status\"] = False\n",
    "df.loc[df[\"support_status\"] == \"Genotype/Haplotype\", \"support_status\"] = False\n",
    "df.loc[df[\"support_status\"] == \"Invalid/unsupported hgvs\", \"support_status\"] = False\n",
    "df.loc[df[\"support_status\"] == \"Remaining valid hgvs alleles\", \"support_status\"] = True\n",
    "df.loc[df[\"support_status\"] == \"Min/max copy count range not supported\", \n",
    "    \"support_status\"] = False"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "support_status\n",
       "True     2199977\n",
       "False      10650\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df[\"support_status\"].value_counts()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## <a id='toc1_3_'></a>[Add Normalization Status of Variant based on out.errors](#toc0_)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The errors are stored as a list of values, some of which are strings and other of which are dictionaries (determined by whether error was handled at the level of Variation Normalizer or after the normalizer)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The \"get_errors\" function extracts the text error responses for better readability and ease string processing"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_errors(errors):\n",
    "    errors_out = []\n",
    "    for e in errors:\n",
    "        if type(e) == str:\n",
    "            errors_out.append(e)\n",
    "        elif type(e) == dict:\n",
    "            for k, v in e.items():\n",
    "                if k not in [\n",
    "                    \"msg\",\n",
    "                    \"response-errors\",]:\n",
    "                ## only get these keys from normalizer response\n",
    "                    continue\n",
    "                if type(v) == str:\n",
    "                    errors_out.append(v)\n",
    "                elif type(v) == list:\n",
    "                    errors_out.append(\";\".join(v))\n",
    "    return \";\".join(errors_out)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "df[\"error_string\"] = df[\"out.errors\"].fillna(\"\").apply(get_errors)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This is the number of unique error strings"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "There are many different strings because many of the errors contain specific genomic coordinates, which are unlikely to occur more than once"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "532"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df[\"error_string\"].nunique()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "To get the core error message, the numeric values are replaced with \"#\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "def reduce_errors(error_string):\n",
    "    out = error_string.lower()\n",
    "    out = re.sub(\"\\d+\", \"#\", out)\n",
    "    return out"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "def reduce_errors_more(error_string):\n",
    "    errs = error_string.split(\";\")\n",
    "    new_errs = [re.sub(\"\\:[ ]?[^\\s]+[\\s]?\", \"\", err) for err in errs]\n",
    "    return \";\".join(new_errs)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "df[\"error_string_reduce\"] = df[\"error_string\"].apply(reduce_errors)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "df[\"error_string_reduce\"] = df[\"error_string_reduce\"].replace(\"\", \"Success\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "error_string_reduce\n",
       "Success                                                                                                                                                  2209136\n",
       "error returned from variation normalizer;unable to find a grch# accession for: nc_#.#                                                                        539\n",
       "unrecognized variation record                                                                                                                                431\n",
       "error returned from variation normalizer;unable to find classification for: nc_#.#:g.(#_?)_(?_#)del                                                          233\n",
       "error returned from variation normalizer;unable to find classification for: nc_#.#:g.(#_?)_(?_#)dup                                                          151\n",
       "error returned from variation normalizer;unable to tokenize: cm#.#:g.#_#dup;unable to find classification for: cm#.#:g.#_#dup                                 23\n",
       "error returned from variation normalizer;unable to find classification for: nt_#.#:g.(?_#)_(#_?)del;unable to tokenize: nt_#.#:g.(?_#)_(#_?)del               19\n",
       "error returned from variation normalizer;unable to find classification for: cm#.#:g.#_#dup;unable to tokenize: cm#.#:g.#_#dup                                 13\n",
       "error returned from variation normalizer;unable to tokenize: nt_#.#:g.(?_#)_(#_?)del;unable to find classification for: nt_#.#:g.(?_#)_(#_?)del               10\n",
       "error returned from variation normalizer;unable to tokenize: nc_#.#:m.#_#del;unable to find classification for: nc_#.#:m.#_#del                                9\n",
       "error returned from variation normalizer;unable to translate variation                                                                                         8\n",
       "error returned from variation normalizer;unable to find classification for: nc_#.#:m.#_#del;unable to tokenize: nc_#.#:m.#_#del                                7\n",
       "error returned from variation normalizer;unable to find classification for: nc_#.#:g.(?_#)_(#_#)dup                                                            6\n",
       "error returned from variation normalizer;unable to find classification for: nc_#.#:g.(#_#)_(#_?)dup                                                            6\n",
       "error returned from variation normalizer;unable to find valid result for classification: genomic_deletion_ambiguous                                            5\n",
       "error returned from variation normalizer;unable to tokenize: nt_#.#:g.(?_#)_(#_?)dup;unable to find classification for: nt_#.#:g.(?_#)_(#_?)dup                5\n",
       "error returned from variation normalizer;unable to find classification for: nc_#.#:g.(#_#)_(#_?)del                                                            3\n",
       "error returned from variation normalizer;unable to tokenize: nw_#.#:g.(?_#)_(#_?)del;unable to find classification for: nw_#.#:g.(?_#)_(#_?)del                2\n",
       "error returned from variation normalizer;unable to find valid result for classification: genomic_duplication_ambiguous                                         2\n",
       "error returned from variation normalizer;unable to find classification for: nt_#.#:g.(?_#)_(#_?)dup;unable to tokenize: nt_#.#:g.(?_#)_(#_?)dup                2\n",
       "error returned from variation normalizer;unable to find classification for: nc_#.#:g.#_(?_#)dup                                                                1\n",
       "error returned from variation normalizer;unable to translate nc_#.#:g.#t>a                                                                                     1\n",
       "error returned from variation normalizer;nc_#.#:g.#_#insgccgtgggccccgccaccgccgccgtgccaccgcc is not a supported hgvs genomic duplication or deletion            1\n",
       "error returned from variation normalizer;unable to find classification for: nc_#.#:g.(?_#)_(#_#)del                                                            1\n",
       "error returned from variation normalizer;unable to find classification for: nc_#.#:g.(#_?)_(#_?)del                                                            1\n",
       "error returned from variation normalizer;unable to translate nc_#.#:g.#g>a                                                                                     1\n",
       "error returned from variation normalizer;unable to find classification for: nt_#.#:g.#_#dup;unable to tokenize: nt_#.#:g.#_#dup                                1\n",
       "error returned from variation normalizer;unable to tokenize: nw_#.#:g.#g>a                                                                                     1\n",
       "error returned from variation normalizer;unable to find classification for: nw_#.#:g.(?_#)_(#_?)del;unable to tokenize: nw_#.#:g.(?_#)_(#_?)del                1\n",
       "error returned from variation normalizer;unable to translate nc_#.#:g.#_#instattt                                                                              1\n",
       "error returned from variation normalizer;unable to find classification for: nt_#.#:g.#_#del;unable to tokenize: nt_#.#:g.#_#del                                1\n",
       "error returned from variation normalizer;unable to find classification for: nw_#.#:g.(?_#)_(#_?)dup;unable to tokenize: nw_#.#:g.(?_#)_(#_?)dup                1\n",
       "error returned from variation normalizer;unable to tokenize: nw_#.#:g.#c>a                                                                                     1\n",
       "error returned from variation normalizer;unable to find classification for: nc_#.#:g.#_(#_#)del                                                                1\n",
       "error returned from variation normalizer;unable to find classification for: nw_#.#:g.#del;unable to tokenize: nw_#.#:g.#del                                    1\n",
       "error returned from variation normalizer;nc_#.#:g.#_#delinsatgaaaagctttacagaagttggaattaaaaaaa is not a supported hgvs genomic duplication or deletion          1\n",
       "error returned from variation normalizer;unable to find valid result for classification: genomic_duplication                                                   1\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df[\"error_string_reduce\"].value_counts()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "There are Not Supported variants that have no error (marked as success inaccurately) because they were labeled \"Not Supported\" manually."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "An error (\"Not Supported\") is entered manually for those variants so that they are not categorized as normalized"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "df.loc[\n",
    "    (df[\"support_status\"] == False) & (df[\"error_string_reduce\"] == \"Success\"),\n",
    "    \"error_string_reduce\",\n",
    "] = \"Not Supported\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The error strings had to be reduced further"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "df[\"error_string_reduce_2\"] = df[\"error_string_reduce\"].apply(reduce_errors_more)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "error_string_reduce_2\n",
       "Success                                                                                                   2198936\n",
       "Not Supported                                                                                               10200\n",
       "error returned from variation normalizer;unable to find a grch# accession for                                 539\n",
       "unrecognized variation record                                                                                 431\n",
       "error returned from variation normalizer;unable to find classification for                                    403\n",
       "error returned from variation normalizer;unable to tokenize;unable to find classification for                  49\n",
       "error returned from variation normalizer;unable to find classification for;unable to tokenize                  46\n",
       "error returned from variation normalizer;unable to find valid result for classification                         8\n",
       "error returned from variation normalizer;unable to translate variation                                          8\n",
       "error returned from variation normalizer;unable to translate nc_#.#                                             3\n",
       "error returned from variation normalizer;nc_#.#is not a supported hgvs genomic duplication or deletion          2\n",
       "error returned from variation normalizer;unable to tokenize                                                     2\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df[\"error_string_reduce_2\"].value_counts()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### <a id='toc1_3_1_'></a>[Set Normalize Status of Variant as T/F](#toc0_)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "If an error is present, the variant was not normalized and therefore has a False Normalize Status"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>in.id</th>\n",
       "      <th>in.name</th>\n",
       "      <th>in.subclass_type</th>\n",
       "      <th>in.variation_type</th>\n",
       "      <th>in.vrs_xform_plan.type</th>\n",
       "      <th>in.vrs_xform_plan.inputs</th>\n",
       "      <th>in.vrs_xform_plan.policy</th>\n",
       "      <th>out.id</th>\n",
       "      <th>out.type</th>\n",
       "      <th>out.definition</th>\n",
       "      <th>...</th>\n",
       "      <th>out.seq.ref_allele_vcf</th>\n",
       "      <th>out.seq.position_vcf</th>\n",
       "      <th>out.seq.alt_allele_vcf</th>\n",
       "      <th>in.max_copies</th>\n",
       "      <th>in.min_copies</th>\n",
       "      <th>support_status</th>\n",
       "      <th>error_string</th>\n",
       "      <th>error_string_reduce</th>\n",
       "      <th>error_string_reduce_2</th>\n",
       "      <th>normalize_status</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>425693</td>\n",
       "      <td>NM_001204.6(BMPR2):c.77-?_247+?dup</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>Duplication</td>\n",
       "      <td>Text</td>\n",
       "      <td>[id]</td>\n",
       "      <td>No hgvs or location info</td>\n",
       "      <td>Text:clinvar:425693</td>\n",
       "      <td>Text</td>\n",
       "      <td>clinvar:425693</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td></td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>90650</td>\n",
       "      <td>NM_000251.2(MSH2):c.1387-?_1510+?del</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>Text</td>\n",
       "      <td>[id]</td>\n",
       "      <td>No hgvs or location info</td>\n",
       "      <td>Text:clinvar:90650</td>\n",
       "      <td>Text</td>\n",
       "      <td>clinvar:90650</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td></td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>16098</td>\n",
       "      <td>SLC2A2, 1-BP INS, 793C</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>Insertion</td>\n",
       "      <td>Text</td>\n",
       "      <td>[id]</td>\n",
       "      <td>No hgvs or location info</td>\n",
       "      <td>Text:clinvar:16098</td>\n",
       "      <td>Text</td>\n",
       "      <td>clinvar:16098</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td></td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>14905</td>\n",
       "      <td>HLA-DRB1, HLA-DRB1*1101</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>Variation</td>\n",
       "      <td>Text</td>\n",
       "      <td>[id]</td>\n",
       "      <td>No hgvs or location info</td>\n",
       "      <td>Text:clinvar:14905</td>\n",
       "      <td>Text</td>\n",
       "      <td>clinvar:14905</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td></td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1048409</td>\n",
       "      <td>NM_000512.5:c.567_1002dup</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>Duplication</td>\n",
       "      <td>Text</td>\n",
       "      <td>[id]</td>\n",
       "      <td>No hgvs or location info</td>\n",
       "      <td>Text:clinvar:1048409</td>\n",
       "      <td>Text</td>\n",
       "      <td>clinvar:1048409</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td></td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2210622</th>\n",
       "      <td>1801441</td>\n",
       "      <td>NM_015189.3(EXOC6B):c.2197-66917_2197-66916ins...</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>Insertion</td>\n",
       "      <td>Allele</td>\n",
       "      <td>[canonical_spdi]</td>\n",
       "      <td>Canonical SPDI</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Allele</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td></td>\n",
       "      <td>Success</td>\n",
       "      <td>Success</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2210623</th>\n",
       "      <td>2202105</td>\n",
       "      <td>NM_153676.4(USH1C):c.496+14_496+15insGTACTCCAT...</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>Microsatellite</td>\n",
       "      <td>Allele</td>\n",
       "      <td>[canonical_spdi]</td>\n",
       "      <td>Canonical SPDI</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Allele</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td></td>\n",
       "      <td>Success</td>\n",
       "      <td>Success</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2210624</th>\n",
       "      <td>1972383</td>\n",
       "      <td>NM_003470.3(USP7):c.383+10_383+11insGTTTAAATGA...</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>Insertion</td>\n",
       "      <td>Allele</td>\n",
       "      <td>[canonical_spdi]</td>\n",
       "      <td>Canonical SPDI</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Allele</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td></td>\n",
       "      <td>Success</td>\n",
       "      <td>Success</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2210625</th>\n",
       "      <td>1496502</td>\n",
       "      <td>NM_007255.3(B4GALT7):c.881_882insTGAGGTGGATTAA...</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>Insertion</td>\n",
       "      <td>Allele</td>\n",
       "      <td>[canonical_spdi]</td>\n",
       "      <td>Canonical SPDI</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Allele</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td></td>\n",
       "      <td>Success</td>\n",
       "      <td>Success</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2210626</th>\n",
       "      <td>1801464</td>\n",
       "      <td>NM_176787.5(PIGN):c.2673-514_2673-513insGCAAAC...</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>Insertion</td>\n",
       "      <td>Allele</td>\n",
       "      <td>[hgvs]</td>\n",
       "      <td>Remaining valid hgvs alleles</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Allele</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td></td>\n",
       "      <td>Success</td>\n",
       "      <td>Success</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2210627 rows × 95 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           in.id                                            in.name  \\\n",
       "0         425693                 NM_001204.6(BMPR2):c.77-?_247+?dup   \n",
       "1          90650               NM_000251.2(MSH2):c.1387-?_1510+?del   \n",
       "2          16098                             SLC2A2, 1-BP INS, 793C   \n",
       "3          14905                            HLA-DRB1, HLA-DRB1*1101   \n",
       "4        1048409                          NM_000512.5:c.567_1002dup   \n",
       "...          ...                                                ...   \n",
       "2210622  1801441  NM_015189.3(EXOC6B):c.2197-66917_2197-66916ins...   \n",
       "2210623  2202105  NM_153676.4(USH1C):c.496+14_496+15insGTACTCCAT...   \n",
       "2210624  1972383  NM_003470.3(USP7):c.383+10_383+11insGTTTAAATGA...   \n",
       "2210625  1496502  NM_007255.3(B4GALT7):c.881_882insTGAGGTGGATTAA...   \n",
       "2210626  1801464  NM_176787.5(PIGN):c.2673-514_2673-513insGCAAAC...   \n",
       "\n",
       "        in.subclass_type in.variation_type in.vrs_xform_plan.type  \\\n",
       "0           SimpleAllele       Duplication                   Text   \n",
       "1           SimpleAllele          Deletion                   Text   \n",
       "2           SimpleAllele         Insertion                   Text   \n",
       "3           SimpleAllele         Variation                   Text   \n",
       "4           SimpleAllele       Duplication                   Text   \n",
       "...                  ...               ...                    ...   \n",
       "2210622     SimpleAllele         Insertion                 Allele   \n",
       "2210623     SimpleAllele    Microsatellite                 Allele   \n",
       "2210624     SimpleAllele         Insertion                 Allele   \n",
       "2210625     SimpleAllele         Insertion                 Allele   \n",
       "2210626     SimpleAllele         Insertion                 Allele   \n",
       "\n",
       "        in.vrs_xform_plan.inputs      in.vrs_xform_plan.policy  \\\n",
       "0                           [id]      No hgvs or location info   \n",
       "1                           [id]      No hgvs or location info   \n",
       "2                           [id]      No hgvs or location info   \n",
       "3                           [id]      No hgvs or location info   \n",
       "4                           [id]      No hgvs or location info   \n",
       "...                          ...                           ...   \n",
       "2210622         [canonical_spdi]                Canonical SPDI   \n",
       "2210623         [canonical_spdi]                Canonical SPDI   \n",
       "2210624         [canonical_spdi]                Canonical SPDI   \n",
       "2210625         [canonical_spdi]                Canonical SPDI   \n",
       "2210626                   [hgvs]  Remaining valid hgvs alleles   \n",
       "\n",
       "                       out.id out.type   out.definition  ...  \\\n",
       "0         Text:clinvar:425693     Text   clinvar:425693  ...   \n",
       "1          Text:clinvar:90650     Text    clinvar:90650  ...   \n",
       "2          Text:clinvar:16098     Text    clinvar:16098  ...   \n",
       "3          Text:clinvar:14905     Text    clinvar:14905  ...   \n",
       "4        Text:clinvar:1048409     Text  clinvar:1048409  ...   \n",
       "...                       ...      ...              ...  ...   \n",
       "2210622                   NaN   Allele              NaN  ...   \n",
       "2210623                   NaN   Allele              NaN  ...   \n",
       "2210624                   NaN   Allele              NaN  ...   \n",
       "2210625                   NaN   Allele              NaN  ...   \n",
       "2210626                   NaN   Allele              NaN  ...   \n",
       "\n",
       "        out.seq.ref_allele_vcf out.seq.position_vcf out.seq.alt_allele_vcf  \\\n",
       "0                          NaN                  NaN                    NaN   \n",
       "1                          NaN                  NaN                    NaN   \n",
       "2                          NaN                  NaN                    NaN   \n",
       "3                          NaN                  NaN                    NaN   \n",
       "4                          NaN                  NaN                    NaN   \n",
       "...                        ...                  ...                    ...   \n",
       "2210622                    NaN                  NaN                    NaN   \n",
       "2210623                    NaN                  NaN                    NaN   \n",
       "2210624                    NaN                  NaN                    NaN   \n",
       "2210625                    NaN                  NaN                    NaN   \n",
       "2210626                    NaN                  NaN                    NaN   \n",
       "\n",
       "        in.max_copies in.min_copies support_status error_string  \\\n",
       "0                 NaN           NaN          False                \n",
       "1                 NaN           NaN          False                \n",
       "2                 NaN           NaN          False                \n",
       "3                 NaN           NaN          False                \n",
       "4                 NaN           NaN          False                \n",
       "...               ...           ...            ...          ...   \n",
       "2210622           NaN           NaN           True                \n",
       "2210623           NaN           NaN           True                \n",
       "2210624           NaN           NaN           True                \n",
       "2210625           NaN           NaN           True                \n",
       "2210626           NaN           NaN           True                \n",
       "\n",
       "        error_string_reduce error_string_reduce_2  normalize_status  \n",
       "0             Not Supported         Not Supported             False  \n",
       "1             Not Supported         Not Supported             False  \n",
       "2             Not Supported         Not Supported             False  \n",
       "3             Not Supported         Not Supported             False  \n",
       "4             Not Supported         Not Supported             False  \n",
       "...                     ...                   ...               ...  \n",
       "2210622             Success               Success              True  \n",
       "2210623             Success               Success              True  \n",
       "2210624             Success               Success              True  \n",
       "2210625             Success               Success              True  \n",
       "2210626             Success               Success              True  \n",
       "\n",
       "[2210627 rows x 95 columns]"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df[\"normalize_status\"] = df[\"error_string_reduce_2\"] == \"Success\"\n",
    "df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### <a id='toc1_3_1_1_'></a>[Summary Table](#toc0_)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "In the table below, the cells show the number of variants with each expected behavior and how they actually ended up performing."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "If a variant was in an \"expected to pass\" category and ends up as text, that is an instance of a normalizer failure on a supported variant"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [],
   "source": [
    "summary_df = df[[\"in.id\", \"support_status\", \"in.vrs_xform_plan.policy\", \"out.type\"]].fillna(\n",
    "    \"NONE\"\n",
    ").groupby([\"support_status\", \"in.vrs_xform_plan.policy\", \"out.type\"]).count().unstack(\n",
    "    level=2\n",
    ").fillna(\n",
    "    0\n",
    ").astype(\n",
    "    int\n",
    ")[\"in.id\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "summary_df[\"VariantSum\"] = summary_df.sum(axis = 1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [],
   "source": [
    "summary_df[\"NormalizedSum\"] = summary_df[[\"Allele\", \"CopyNumberChange\", \"CopyNumberCount\"]].sum(axis = 1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [],
   "source": [
    "summary_df[\"NormalizedPercent\"] = (summary_df[\"NormalizedSum\"] / summary_df[\"VariantSum\"]).apply(lambda x : f\"{round(x * 100, 2)}%\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>out.type</th>\n",
       "      <th>Allele</th>\n",
       "      <th>CopyNumberChange</th>\n",
       "      <th>CopyNumberCount</th>\n",
       "      <th>NONE</th>\n",
       "      <th>Text</th>\n",
       "      <th>NormalizedPercent</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>support_status</th>\n",
       "      <th>in.vrs_xform_plan.policy</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th rowspan=\"5\" valign=\"top\">False</th>\n",
       "      <th>Genotype/Haplotype</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1440</td>\n",
       "      <td>0.0%</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Invalid/unsupported hgvs</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>19</td>\n",
       "      <td>1317</td>\n",
       "      <td>0.0%</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Min/max copy count range not supported</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "      <td>0.0%</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NCBI36 genomic only</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4771</td>\n",
       "      <td>0.0%</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>No hgvs or location info</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3089</td>\n",
       "      <td>0.0%</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"4\" valign=\"top\">True</th>\n",
       "      <th>Absolute copy count</th>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>52440</td>\n",
       "      <td>819</td>\n",
       "      <td>3</td>\n",
       "      <td>98.46%</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Canonical SPDI</th>\n",
       "      <td>2118669</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>100.0%</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Copy number change (cn loss|del and cn gain|dup)</th>\n",
       "      <td>0</td>\n",
       "      <td>26889</td>\n",
       "      <td>0</td>\n",
       "      <td>209</td>\n",
       "      <td>6</td>\n",
       "      <td>99.21%</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Remaining valid hgvs alleles</th>\n",
       "      <td>927</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>13</td>\n",
       "      <td>1</td>\n",
       "      <td>98.51%</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "out.type                                                          Allele  \\\n",
       "support_status in.vrs_xform_plan.policy                                    \n",
       "False          Genotype/Haplotype                                      0   \n",
       "               Invalid/unsupported hgvs                                0   \n",
       "               Min/max copy count range not supported                  0   \n",
       "               NCBI36 genomic only                                     0   \n",
       "               No hgvs or location info                                0   \n",
       "True           Absolute copy count                                     0   \n",
       "               Canonical SPDI                                    2118669   \n",
       "               Copy number change (cn loss|del and cn gain|dup)        0   \n",
       "               Remaining valid hgvs alleles                          927   \n",
       "\n",
       "out.type                                                         CopyNumberChange  \\\n",
       "support_status in.vrs_xform_plan.policy                                             \n",
       "False          Genotype/Haplotype                                               0   \n",
       "               Invalid/unsupported hgvs                                         0   \n",
       "               Min/max copy count range not supported                           0   \n",
       "               NCBI36 genomic only                                              0   \n",
       "               No hgvs or location info                                         0   \n",
       "True           Absolute copy count                                              1   \n",
       "               Canonical SPDI                                                   0   \n",
       "               Copy number change (cn loss|del and cn gain|dup)             26889   \n",
       "               Remaining valid hgvs alleles                                     0   \n",
       "\n",
       "out.type                                                         CopyNumberCount  \\\n",
       "support_status in.vrs_xform_plan.policy                                            \n",
       "False          Genotype/Haplotype                                              0   \n",
       "               Invalid/unsupported hgvs                                        0   \n",
       "               Min/max copy count range not supported                          0   \n",
       "               NCBI36 genomic only                                             0   \n",
       "               No hgvs or location info                                        0   \n",
       "True           Absolute copy count                                         52440   \n",
       "               Canonical SPDI                                                  0   \n",
       "               Copy number change (cn loss|del and cn gain|dup)                0   \n",
       "               Remaining valid hgvs alleles                                    0   \n",
       "\n",
       "out.type                                                         NONE  Text  \\\n",
       "support_status in.vrs_xform_plan.policy                                       \n",
       "False          Genotype/Haplotype                                   0  1440   \n",
       "               Invalid/unsupported hgvs                            19  1317   \n",
       "               Min/max copy count range not supported               0    14   \n",
       "               NCBI36 genomic only                                  0  4771   \n",
       "               No hgvs or location info                             0  3089   \n",
       "True           Absolute copy count                                819     3   \n",
       "               Canonical SPDI                                       0     0   \n",
       "               Copy number change (cn loss|del and cn gain|dup)   209     6   \n",
       "               Remaining valid hgvs alleles                        13     1   \n",
       "\n",
       "out.type                                                        NormalizedPercent  \n",
       "support_status in.vrs_xform_plan.policy                                            \n",
       "False          Genotype/Haplotype                                            0.0%  \n",
       "               Invalid/unsupported hgvs                                      0.0%  \n",
       "               Min/max copy count range not supported                        0.0%  \n",
       "               NCBI36 genomic only                                           0.0%  \n",
       "               No hgvs or location info                                      0.0%  \n",
       "True           Absolute copy count                                         98.46%  \n",
       "               Canonical SPDI                                              100.0%  \n",
       "               Copy number change (cn loss|del and cn gain|dup)            99.21%  \n",
       "               Remaining valid hgvs alleles                                98.51%  "
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "summary_df = summary_df.drop([\"VariantSum\", \"NormalizedSum\"], axis=1)\n",
    "summary_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [],
   "source": [
    "summary_df.to_csv(\n",
    "    \"clinvar_variation_analysis_output/variant_analysis_summary_df.csv\"\n",
    ")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## <a id='toc1_4_'></a>[Create subgroups based on Variant Status](#toc0_)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### <a id='toc1_4_1_'></a>[Supported and Normalized Variants](#toc0_)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [],
   "source": [
    "supported_df = df.copy()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>in.id</th>\n",
       "      <th>in.name</th>\n",
       "      <th>in.subclass_type</th>\n",
       "      <th>in.variation_type</th>\n",
       "      <th>in.vrs_xform_plan.type</th>\n",
       "      <th>in.vrs_xform_plan.inputs</th>\n",
       "      <th>in.vrs_xform_plan.policy</th>\n",
       "      <th>out.id</th>\n",
       "      <th>out.type</th>\n",
       "      <th>out.definition</th>\n",
       "      <th>...</th>\n",
       "      <th>out.seq.ref_allele_vcf</th>\n",
       "      <th>out.seq.position_vcf</th>\n",
       "      <th>out.seq.alt_allele_vcf</th>\n",
       "      <th>in.max_copies</th>\n",
       "      <th>in.min_copies</th>\n",
       "      <th>support_status</th>\n",
       "      <th>error_string</th>\n",
       "      <th>error_string_reduce</th>\n",
       "      <th>error_string_reduce_2</th>\n",
       "      <th>normalize_status</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>149</th>\n",
       "      <td>1676638</td>\n",
       "      <td>NM_000094.4(COL7A1):c.8729G&gt;T (p.Gly2910Val)</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>Allele</td>\n",
       "      <td>[hgvs]</td>\n",
       "      <td>Remaining valid hgvs alleles</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Allele</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td></td>\n",
       "      <td>Success</td>\n",
       "      <td>Success</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>150</th>\n",
       "      <td>1676377</td>\n",
       "      <td>NM_012064.4(MIP):c.20C&gt;T (p.Ala7Val)</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>Allele</td>\n",
       "      <td>[hgvs]</td>\n",
       "      <td>Remaining valid hgvs alleles</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Allele</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td></td>\n",
       "      <td>Success</td>\n",
       "      <td>Success</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>151</th>\n",
       "      <td>1676330</td>\n",
       "      <td>NM_054027.6(ANKH):c.259G&gt;A (p.Val87Ile)</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>Allele</td>\n",
       "      <td>[hgvs]</td>\n",
       "      <td>Remaining valid hgvs alleles</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Allele</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td></td>\n",
       "      <td>Success</td>\n",
       "      <td>Success</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>152</th>\n",
       "      <td>1325429</td>\n",
       "      <td>NM_000138.5(FBN1):c.5855G&gt;T (p.Gly1952Val)</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>Allele</td>\n",
       "      <td>[hgvs]</td>\n",
       "      <td>Remaining valid hgvs alleles</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Allele</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td></td>\n",
       "      <td>Success</td>\n",
       "      <td>Success</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>153</th>\n",
       "      <td>1676394</td>\n",
       "      <td>NM_000786.4(CYP51A1):c.1291C&gt;T (p.Arg431Cys)</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>Allele</td>\n",
       "      <td>[hgvs]</td>\n",
       "      <td>Remaining valid hgvs alleles</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Allele</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td></td>\n",
       "      <td>Success</td>\n",
       "      <td>Success</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2210622</th>\n",
       "      <td>1801441</td>\n",
       "      <td>NM_015189.3(EXOC6B):c.2197-66917_2197-66916ins...</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>Insertion</td>\n",
       "      <td>Allele</td>\n",
       "      <td>[canonical_spdi]</td>\n",
       "      <td>Canonical SPDI</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Allele</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td></td>\n",
       "      <td>Success</td>\n",
       "      <td>Success</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2210623</th>\n",
       "      <td>2202105</td>\n",
       "      <td>NM_153676.4(USH1C):c.496+14_496+15insGTACTCCAT...</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>Microsatellite</td>\n",
       "      <td>Allele</td>\n",
       "      <td>[canonical_spdi]</td>\n",
       "      <td>Canonical SPDI</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Allele</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td></td>\n",
       "      <td>Success</td>\n",
       "      <td>Success</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2210624</th>\n",
       "      <td>1972383</td>\n",
       "      <td>NM_003470.3(USP7):c.383+10_383+11insGTTTAAATGA...</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>Insertion</td>\n",
       "      <td>Allele</td>\n",
       "      <td>[canonical_spdi]</td>\n",
       "      <td>Canonical SPDI</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Allele</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td></td>\n",
       "      <td>Success</td>\n",
       "      <td>Success</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2210625</th>\n",
       "      <td>1496502</td>\n",
       "      <td>NM_007255.3(B4GALT7):c.881_882insTGAGGTGGATTAA...</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>Insertion</td>\n",
       "      <td>Allele</td>\n",
       "      <td>[canonical_spdi]</td>\n",
       "      <td>Canonical SPDI</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Allele</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td></td>\n",
       "      <td>Success</td>\n",
       "      <td>Success</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2210626</th>\n",
       "      <td>1801464</td>\n",
       "      <td>NM_176787.5(PIGN):c.2673-514_2673-513insGCAAAC...</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>Insertion</td>\n",
       "      <td>Allele</td>\n",
       "      <td>[hgvs]</td>\n",
       "      <td>Remaining valid hgvs alleles</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Allele</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td></td>\n",
       "      <td>Success</td>\n",
       "      <td>Success</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2198936 rows × 95 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           in.id                                            in.name  \\\n",
       "149      1676638       NM_000094.4(COL7A1):c.8729G>T (p.Gly2910Val)   \n",
       "150      1676377               NM_012064.4(MIP):c.20C>T (p.Ala7Val)   \n",
       "151      1676330            NM_054027.6(ANKH):c.259G>A (p.Val87Ile)   \n",
       "152      1325429         NM_000138.5(FBN1):c.5855G>T (p.Gly1952Val)   \n",
       "153      1676394       NM_000786.4(CYP51A1):c.1291C>T (p.Arg431Cys)   \n",
       "...          ...                                                ...   \n",
       "2210622  1801441  NM_015189.3(EXOC6B):c.2197-66917_2197-66916ins...   \n",
       "2210623  2202105  NM_153676.4(USH1C):c.496+14_496+15insGTACTCCAT...   \n",
       "2210624  1972383  NM_003470.3(USP7):c.383+10_383+11insGTTTAAATGA...   \n",
       "2210625  1496502  NM_007255.3(B4GALT7):c.881_882insTGAGGTGGATTAA...   \n",
       "2210626  1801464  NM_176787.5(PIGN):c.2673-514_2673-513insGCAAAC...   \n",
       "\n",
       "        in.subclass_type          in.variation_type in.vrs_xform_plan.type  \\\n",
       "149         SimpleAllele  single nucleotide variant                 Allele   \n",
       "150         SimpleAllele  single nucleotide variant                 Allele   \n",
       "151         SimpleAllele  single nucleotide variant                 Allele   \n",
       "152         SimpleAllele  single nucleotide variant                 Allele   \n",
       "153         SimpleAllele  single nucleotide variant                 Allele   \n",
       "...                  ...                        ...                    ...   \n",
       "2210622     SimpleAllele                  Insertion                 Allele   \n",
       "2210623     SimpleAllele             Microsatellite                 Allele   \n",
       "2210624     SimpleAllele                  Insertion                 Allele   \n",
       "2210625     SimpleAllele                  Insertion                 Allele   \n",
       "2210626     SimpleAllele                  Insertion                 Allele   \n",
       "\n",
       "        in.vrs_xform_plan.inputs      in.vrs_xform_plan.policy out.id  \\\n",
       "149                       [hgvs]  Remaining valid hgvs alleles    NaN   \n",
       "150                       [hgvs]  Remaining valid hgvs alleles    NaN   \n",
       "151                       [hgvs]  Remaining valid hgvs alleles    NaN   \n",
       "152                       [hgvs]  Remaining valid hgvs alleles    NaN   \n",
       "153                       [hgvs]  Remaining valid hgvs alleles    NaN   \n",
       "...                          ...                           ...    ...   \n",
       "2210622         [canonical_spdi]                Canonical SPDI    NaN   \n",
       "2210623         [canonical_spdi]                Canonical SPDI    NaN   \n",
       "2210624         [canonical_spdi]                Canonical SPDI    NaN   \n",
       "2210625         [canonical_spdi]                Canonical SPDI    NaN   \n",
       "2210626                   [hgvs]  Remaining valid hgvs alleles    NaN   \n",
       "\n",
       "        out.type out.definition  ... out.seq.ref_allele_vcf  \\\n",
       "149       Allele            NaN  ...                    NaN   \n",
       "150       Allele            NaN  ...                    NaN   \n",
       "151       Allele            NaN  ...                    NaN   \n",
       "152       Allele            NaN  ...                    NaN   \n",
       "153       Allele            NaN  ...                    NaN   \n",
       "...          ...            ...  ...                    ...   \n",
       "2210622   Allele            NaN  ...                    NaN   \n",
       "2210623   Allele            NaN  ...                    NaN   \n",
       "2210624   Allele            NaN  ...                    NaN   \n",
       "2210625   Allele            NaN  ...                    NaN   \n",
       "2210626   Allele            NaN  ...                    NaN   \n",
       "\n",
       "        out.seq.position_vcf out.seq.alt_allele_vcf in.max_copies  \\\n",
       "149                      NaN                    NaN           NaN   \n",
       "150                      NaN                    NaN           NaN   \n",
       "151                      NaN                    NaN           NaN   \n",
       "152                      NaN                    NaN           NaN   \n",
       "153                      NaN                    NaN           NaN   \n",
       "...                      ...                    ...           ...   \n",
       "2210622                  NaN                    NaN           NaN   \n",
       "2210623                  NaN                    NaN           NaN   \n",
       "2210624                  NaN                    NaN           NaN   \n",
       "2210625                  NaN                    NaN           NaN   \n",
       "2210626                  NaN                    NaN           NaN   \n",
       "\n",
       "        in.min_copies support_status error_string error_string_reduce  \\\n",
       "149               NaN           True                          Success   \n",
       "150               NaN           True                          Success   \n",
       "151               NaN           True                          Success   \n",
       "152               NaN           True                          Success   \n",
       "153               NaN           True                          Success   \n",
       "...               ...            ...          ...                 ...   \n",
       "2210622           NaN           True                          Success   \n",
       "2210623           NaN           True                          Success   \n",
       "2210624           NaN           True                          Success   \n",
       "2210625           NaN           True                          Success   \n",
       "2210626           NaN           True                          Success   \n",
       "\n",
       "        error_string_reduce_2  normalize_status  \n",
       "149                   Success              True  \n",
       "150                   Success              True  \n",
       "151                   Success              True  \n",
       "152                   Success              True  \n",
       "153                   Success              True  \n",
       "...                       ...               ...  \n",
       "2210622               Success              True  \n",
       "2210623               Success              True  \n",
       "2210624               Success              True  \n",
       "2210625               Success              True  \n",
       "2210626               Success              True  \n",
       "\n",
       "[2198936 rows x 95 columns]"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "supported_df = supported_df.loc[\n",
    "    (supported_df[\"support_status\"] == True)\n",
    "    & (supported_df[\"normalize_status\"] == True)\n",
    "]\n",
    "supported_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>in.variation_type</th>\n",
       "      <th>in.vrs_xform_plan.policy</th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>Canonical SPDI</td>\n",
       "      <td>1934692</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Deletion</td>\n",
       "      <td>Canonical SPDI</td>\n",
       "      <td>92875</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Duplication</td>\n",
       "      <td>Canonical SPDI</td>\n",
       "      <td>43185</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>copy number loss</td>\n",
       "      <td>Absolute copy count</td>\n",
       "      <td>26857</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Microsatellite</td>\n",
       "      <td>Canonical SPDI</td>\n",
       "      <td>26678</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>copy number gain</td>\n",
       "      <td>Absolute copy count</td>\n",
       "      <td>25583</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>Deletion</td>\n",
       "      <td>Copy number change (cn loss|del and cn gain|dup)</td>\n",
       "      <td>16249</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>Indel</td>\n",
       "      <td>Canonical SPDI</td>\n",
       "      <td>10986</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>Insertion</td>\n",
       "      <td>Canonical SPDI</td>\n",
       "      <td>8865</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>Duplication</td>\n",
       "      <td>Copy number change (cn loss|del and cn gain|dup)</td>\n",
       "      <td>8119</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>copy number loss</td>\n",
       "      <td>Copy number change (cn loss|del and cn gain|dup)</td>\n",
       "      <td>1524</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>Inversion</td>\n",
       "      <td>Canonical SPDI</td>\n",
       "      <td>1051</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>copy number gain</td>\n",
       "      <td>Copy number change (cn loss|del and cn gain|dup)</td>\n",
       "      <td>1003</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>Remaining valid hgvs alleles</td>\n",
       "      <td>396</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>Indel</td>\n",
       "      <td>Remaining valid hgvs alleles</td>\n",
       "      <td>340</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>Variation</td>\n",
       "      <td>Canonical SPDI</td>\n",
       "      <td>337</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>Insertion</td>\n",
       "      <td>Remaining valid hgvs alleles</td>\n",
       "      <td>191</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>Deletion</td>\n",
       "      <td>Absolute copy count</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>Variation</td>\n",
       "      <td>Remaining valid hgvs alleles</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>Duplication</td>\n",
       "      <td>Absolute copy count</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            in.variation_type  \\\n",
       "0   single nucleotide variant   \n",
       "1                    Deletion   \n",
       "2                 Duplication   \n",
       "3            copy number loss   \n",
       "4              Microsatellite   \n",
       "5            copy number gain   \n",
       "6                    Deletion   \n",
       "7                       Indel   \n",
       "8                   Insertion   \n",
       "9                 Duplication   \n",
       "10           copy number loss   \n",
       "11                  Inversion   \n",
       "12           copy number gain   \n",
       "13  single nucleotide variant   \n",
       "14                      Indel   \n",
       "15                  Variation   \n",
       "16                  Insertion   \n",
       "17                   Deletion   \n",
       "18                  Variation   \n",
       "19                Duplication   \n",
       "\n",
       "                            in.vrs_xform_plan.policy    count  \n",
       "0                                     Canonical SPDI  1934692  \n",
       "1                                     Canonical SPDI    92875  \n",
       "2                                     Canonical SPDI    43185  \n",
       "3                                Absolute copy count    26857  \n",
       "4                                     Canonical SPDI    26678  \n",
       "5                                Absolute copy count    25583  \n",
       "6   Copy number change (cn loss|del and cn gain|dup)    16249  \n",
       "7                                     Canonical SPDI    10986  \n",
       "8                                     Canonical SPDI     8865  \n",
       "9   Copy number change (cn loss|del and cn gain|dup)     8119  \n",
       "10  Copy number change (cn loss|del and cn gain|dup)     1524  \n",
       "11                                    Canonical SPDI     1051  \n",
       "12  Copy number change (cn loss|del and cn gain|dup)     1003  \n",
       "13                      Remaining valid hgvs alleles      396  \n",
       "14                      Remaining valid hgvs alleles      340  \n",
       "15                                    Canonical SPDI      337  \n",
       "16                      Remaining valid hgvs alleles      191  \n",
       "17                               Absolute copy count        3  \n",
       "18                      Remaining valid hgvs alleles        1  \n",
       "19                               Absolute copy count        1  "
      ]
     },
     "execution_count": 39,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "variation_type_count_supported_df = supported_df.value_counts([\"in.variation_type\", \"in.vrs_xform_plan.policy\"]).reset_index()\n",
    "variation_type_count_supported_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [],
   "source": [
    "variation_type_count_supported_df.to_csv(\n",
    "    \"clinvar_variation_analysis_output/variation_type_count_supported_df.csv\"\n",
    ")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### <a id='toc1_4_2_'></a>[Supported and Not Normalized Variants](#toc0_)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [],
   "source": [
    "supported_not_normalized_df = df.copy()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>in.id</th>\n",
       "      <th>in.name</th>\n",
       "      <th>in.subclass_type</th>\n",
       "      <th>in.variation_type</th>\n",
       "      <th>in.vrs_xform_plan.type</th>\n",
       "      <th>in.vrs_xform_plan.inputs</th>\n",
       "      <th>in.vrs_xform_plan.policy</th>\n",
       "      <th>out.id</th>\n",
       "      <th>out.type</th>\n",
       "      <th>out.definition</th>\n",
       "      <th>...</th>\n",
       "      <th>out.seq.ref_allele_vcf</th>\n",
       "      <th>out.seq.position_vcf</th>\n",
       "      <th>out.seq.alt_allele_vcf</th>\n",
       "      <th>in.max_copies</th>\n",
       "      <th>in.min_copies</th>\n",
       "      <th>support_status</th>\n",
       "      <th>error_string</th>\n",
       "      <th>error_string_reduce</th>\n",
       "      <th>error_string_reduce_2</th>\n",
       "      <th>normalize_status</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>164</th>\n",
       "      <td>989220</td>\n",
       "      <td>NC_000015.9:g.(44884528_44881613)_(44877833_44...</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>Duplication</td>\n",
       "      <td>CopyNumberChange</td>\n",
       "      <td>[hgvs]</td>\n",
       "      <td>Copy number change (cn loss|del and cn gain|dup)</td>\n",
       "      <td>989220</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>Error returned from variation normalizer;Unabl...</td>\n",
       "      <td>error returned from variation normalizer;unabl...</td>\n",
       "      <td>error returned from variation normalizer;unabl...</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>166</th>\n",
       "      <td>10342</td>\n",
       "      <td>NG_011403.2:g.(80027_96047)_(131648_164496)del</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>CopyNumberChange</td>\n",
       "      <td>[hgvs]</td>\n",
       "      <td>Copy number change (cn loss|del and cn gain|dup)</td>\n",
       "      <td>10342</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>Error returned from variation normalizer;Unabl...</td>\n",
       "      <td>error returned from variation normalizer;unabl...</td>\n",
       "      <td>error returned from variation normalizer;unabl...</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>210</th>\n",
       "      <td>602019</td>\n",
       "      <td>GRCh37/hg19 15q11.2(chr15:22750305-23140114)x3</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>copy number gain</td>\n",
       "      <td>CopyNumberCount</td>\n",
       "      <td>[hgvs, absolute_copies]</td>\n",
       "      <td>Absolute copy count</td>\n",
       "      <td>602019</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>Error returned from variation normalizer;Unabl...</td>\n",
       "      <td>error returned from variation normalizer;unabl...</td>\n",
       "      <td>error returned from variation normalizer;unabl...</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>277</th>\n",
       "      <td>1706497</td>\n",
       "      <td>GRCh37/hg19 16p11.2(chr16:29432212-30177807)x1</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>copy number loss</td>\n",
       "      <td>CopyNumberCount</td>\n",
       "      <td>[hgvs, absolute_copies]</td>\n",
       "      <td>Absolute copy count</td>\n",
       "      <td>1706497</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>Error returned from variation normalizer;Unabl...</td>\n",
       "      <td>error returned from variation normalizer;unabl...</td>\n",
       "      <td>error returned from variation normalizer;unabl...</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>308</th>\n",
       "      <td>1705934</td>\n",
       "      <td>GRCh37/hg19 Xp22.33(chrX:566009-1356042)x3</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>copy number gain</td>\n",
       "      <td>CopyNumberCount</td>\n",
       "      <td>[hgvs, absolute_copies]</td>\n",
       "      <td>Absolute copy count</td>\n",
       "      <td>1705934</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>Error returned from variation normalizer;Unabl...</td>\n",
       "      <td>error returned from variation normalizer;unabl...</td>\n",
       "      <td>error returned from variation normalizer;unabl...</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2143307</th>\n",
       "      <td>830807</td>\n",
       "      <td>NC_000001.10:g.(?_145498103)_(145538307_?)dup</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>Duplication</td>\n",
       "      <td>CopyNumberChange</td>\n",
       "      <td>[hgvs]</td>\n",
       "      <td>Copy number change (cn loss|del and cn gain|dup)</td>\n",
       "      <td>830807</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>Error returned from variation normalizer;Unabl...</td>\n",
       "      <td>error returned from variation normalizer;unabl...</td>\n",
       "      <td>error returned from variation normalizer;unabl...</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2143312</th>\n",
       "      <td>625541</td>\n",
       "      <td>GRCh37/hg19 1q21.1(chr1:145395604-145704146)</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>copy number loss</td>\n",
       "      <td>CopyNumberChange</td>\n",
       "      <td>[hgvs]</td>\n",
       "      <td>Copy number change (cn loss|del and cn gain|dup)</td>\n",
       "      <td>625541</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>Error returned from variation normalizer;Unabl...</td>\n",
       "      <td>error returned from variation normalizer;unabl...</td>\n",
       "      <td>error returned from variation normalizer;unabl...</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2143342</th>\n",
       "      <td>600207</td>\n",
       "      <td>GRCh37/hg19 1q32.1(chr1:206315933-206331193)x1</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>copy number loss</td>\n",
       "      <td>CopyNumberCount</td>\n",
       "      <td>[hgvs, absolute_copies]</td>\n",
       "      <td>Absolute copy count</td>\n",
       "      <td>600207</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>Error returned from variation normalizer;Unabl...</td>\n",
       "      <td>error returned from variation normalizer;unabl...</td>\n",
       "      <td>error returned from variation normalizer;unabl...</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2143345</th>\n",
       "      <td>565216</td>\n",
       "      <td>GRCh37/hg19 1q32.1(chr1:206173911-206288157)x1</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>copy number loss</td>\n",
       "      <td>CopyNumberCount</td>\n",
       "      <td>[hgvs, absolute_copies]</td>\n",
       "      <td>Absolute copy count</td>\n",
       "      <td>565216</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>Error returned from variation normalizer;Unabl...</td>\n",
       "      <td>error returned from variation normalizer;unabl...</td>\n",
       "      <td>error returned from variation normalizer;unabl...</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2143675</th>\n",
       "      <td>393540</td>\n",
       "      <td>GRCh37/hg19 12q23.1(chr12:100367541-100489446)x1</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>copy number loss</td>\n",
       "      <td>CopyNumberCount</td>\n",
       "      <td>[hgvs, absolute_copies]</td>\n",
       "      <td>Absolute copy count</td>\n",
       "      <td>393540</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>Error returned from variation normalizer;Unabl...</td>\n",
       "      <td>error returned from variation normalizer;unabl...</td>\n",
       "      <td>error returned from variation normalizer;unabl...</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1041 rows × 95 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           in.id                                            in.name  \\\n",
       "164       989220  NC_000015.9:g.(44884528_44881613)_(44877833_44...   \n",
       "166        10342     NG_011403.2:g.(80027_96047)_(131648_164496)del   \n",
       "210       602019     GRCh37/hg19 15q11.2(chr15:22750305-23140114)x3   \n",
       "277      1706497     GRCh37/hg19 16p11.2(chr16:29432212-30177807)x1   \n",
       "308      1705934         GRCh37/hg19 Xp22.33(chrX:566009-1356042)x3   \n",
       "...          ...                                                ...   \n",
       "2143307   830807      NC_000001.10:g.(?_145498103)_(145538307_?)dup   \n",
       "2143312   625541       GRCh37/hg19 1q21.1(chr1:145395604-145704146)   \n",
       "2143342   600207     GRCh37/hg19 1q32.1(chr1:206315933-206331193)x1   \n",
       "2143345   565216     GRCh37/hg19 1q32.1(chr1:206173911-206288157)x1   \n",
       "2143675   393540   GRCh37/hg19 12q23.1(chr12:100367541-100489446)x1   \n",
       "\n",
       "        in.subclass_type in.variation_type in.vrs_xform_plan.type  \\\n",
       "164         SimpleAllele       Duplication       CopyNumberChange   \n",
       "166         SimpleAllele          Deletion       CopyNumberChange   \n",
       "210         SimpleAllele  copy number gain        CopyNumberCount   \n",
       "277         SimpleAllele  copy number loss        CopyNumberCount   \n",
       "308         SimpleAllele  copy number gain        CopyNumberCount   \n",
       "...                  ...               ...                    ...   \n",
       "2143307     SimpleAllele       Duplication       CopyNumberChange   \n",
       "2143312     SimpleAllele  copy number loss       CopyNumberChange   \n",
       "2143342     SimpleAllele  copy number loss        CopyNumberCount   \n",
       "2143345     SimpleAllele  copy number loss        CopyNumberCount   \n",
       "2143675     SimpleAllele  copy number loss        CopyNumberCount   \n",
       "\n",
       "        in.vrs_xform_plan.inputs  \\\n",
       "164                       [hgvs]   \n",
       "166                       [hgvs]   \n",
       "210      [hgvs, absolute_copies]   \n",
       "277      [hgvs, absolute_copies]   \n",
       "308      [hgvs, absolute_copies]   \n",
       "...                          ...   \n",
       "2143307                   [hgvs]   \n",
       "2143312                   [hgvs]   \n",
       "2143342  [hgvs, absolute_copies]   \n",
       "2143345  [hgvs, absolute_copies]   \n",
       "2143675  [hgvs, absolute_copies]   \n",
       "\n",
       "                                 in.vrs_xform_plan.policy   out.id out.type  \\\n",
       "164      Copy number change (cn loss|del and cn gain|dup)   989220      NaN   \n",
       "166      Copy number change (cn loss|del and cn gain|dup)    10342      NaN   \n",
       "210                                   Absolute copy count   602019      NaN   \n",
       "277                                   Absolute copy count  1706497      NaN   \n",
       "308                                   Absolute copy count  1705934      NaN   \n",
       "...                                                   ...      ...      ...   \n",
       "2143307  Copy number change (cn loss|del and cn gain|dup)   830807      NaN   \n",
       "2143312  Copy number change (cn loss|del and cn gain|dup)   625541      NaN   \n",
       "2143342                               Absolute copy count   600207      NaN   \n",
       "2143345                               Absolute copy count   565216      NaN   \n",
       "2143675                               Absolute copy count   393540      NaN   \n",
       "\n",
       "        out.definition  ... out.seq.ref_allele_vcf out.seq.position_vcf  \\\n",
       "164                NaN  ...                    NaN                  NaN   \n",
       "166                NaN  ...                    NaN                  NaN   \n",
       "210                NaN  ...                    NaN                  NaN   \n",
       "277                NaN  ...                    NaN                  NaN   \n",
       "308                NaN  ...                    NaN                  NaN   \n",
       "...                ...  ...                    ...                  ...   \n",
       "2143307            NaN  ...                    NaN                  NaN   \n",
       "2143312            NaN  ...                    NaN                  NaN   \n",
       "2143342            NaN  ...                    NaN                  NaN   \n",
       "2143345            NaN  ...                    NaN                  NaN   \n",
       "2143675            NaN  ...                    NaN                  NaN   \n",
       "\n",
       "        out.seq.alt_allele_vcf in.max_copies in.min_copies support_status  \\\n",
       "164                        NaN           NaN           NaN           True   \n",
       "166                        NaN           NaN           NaN           True   \n",
       "210                        NaN           NaN           NaN           True   \n",
       "277                        NaN           NaN           NaN           True   \n",
       "308                        NaN           NaN           NaN           True   \n",
       "...                        ...           ...           ...            ...   \n",
       "2143307                    NaN           NaN           NaN           True   \n",
       "2143312                    NaN           NaN           NaN           True   \n",
       "2143342                    NaN           NaN           NaN           True   \n",
       "2143345                    NaN           NaN           NaN           True   \n",
       "2143675                    NaN           NaN           NaN           True   \n",
       "\n",
       "                                              error_string  \\\n",
       "164      Error returned from variation normalizer;Unabl...   \n",
       "166      Error returned from variation normalizer;Unabl...   \n",
       "210      Error returned from variation normalizer;Unabl...   \n",
       "277      Error returned from variation normalizer;Unabl...   \n",
       "308      Error returned from variation normalizer;Unabl...   \n",
       "...                                                    ...   \n",
       "2143307  Error returned from variation normalizer;Unabl...   \n",
       "2143312  Error returned from variation normalizer;Unabl...   \n",
       "2143342  Error returned from variation normalizer;Unabl...   \n",
       "2143345  Error returned from variation normalizer;Unabl...   \n",
       "2143675  Error returned from variation normalizer;Unabl...   \n",
       "\n",
       "                                       error_string_reduce  \\\n",
       "164      error returned from variation normalizer;unabl...   \n",
       "166      error returned from variation normalizer;unabl...   \n",
       "210      error returned from variation normalizer;unabl...   \n",
       "277      error returned from variation normalizer;unabl...   \n",
       "308      error returned from variation normalizer;unabl...   \n",
       "...                                                    ...   \n",
       "2143307  error returned from variation normalizer;unabl...   \n",
       "2143312  error returned from variation normalizer;unabl...   \n",
       "2143342  error returned from variation normalizer;unabl...   \n",
       "2143345  error returned from variation normalizer;unabl...   \n",
       "2143675  error returned from variation normalizer;unabl...   \n",
       "\n",
       "                                     error_string_reduce_2  normalize_status  \n",
       "164      error returned from variation normalizer;unabl...             False  \n",
       "166      error returned from variation normalizer;unabl...             False  \n",
       "210      error returned from variation normalizer;unabl...             False  \n",
       "277      error returned from variation normalizer;unabl...             False  \n",
       "308      error returned from variation normalizer;unabl...             False  \n",
       "...                                                    ...               ...  \n",
       "2143307  error returned from variation normalizer;unabl...             False  \n",
       "2143312  error returned from variation normalizer;unabl...             False  \n",
       "2143342  error returned from variation normalizer;unabl...             False  \n",
       "2143345  error returned from variation normalizer;unabl...             False  \n",
       "2143675  error returned from variation normalizer;unabl...             False  \n",
       "\n",
       "[1041 rows x 95 columns]"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "supported_not_normalized_df = supported_not_normalized_df.loc[\n",
    "    (supported_not_normalized_df[\"support_status\"] == True)\n",
    "    & (supported_not_normalized_df[\"normalize_status\"] == False)\n",
    "]\n",
    "supported_not_normalized_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>in.variation_type</th>\n",
       "      <th>in.vrs_xform_plan.policy</th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>copy number loss</td>\n",
       "      <td>Absolute copy count</td>\n",
       "      <td>475</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>copy number gain</td>\n",
       "      <td>Absolute copy count</td>\n",
       "      <td>344</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Deletion</td>\n",
       "      <td>Copy number change (cn loss|del and cn gain|dup)</td>\n",
       "      <td>89</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Duplication</td>\n",
       "      <td>Copy number change (cn loss|del and cn gain|dup)</td>\n",
       "      <td>72</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>copy number loss</td>\n",
       "      <td>Copy number change (cn loss|del and cn gain|dup)</td>\n",
       "      <td>31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>copy number gain</td>\n",
       "      <td>Copy number change (cn loss|del and cn gain|dup)</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>Remaining valid hgvs alleles</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>Insertion</td>\n",
       "      <td>Remaining valid hgvs alleles</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>Variation</td>\n",
       "      <td>Remaining valid hgvs alleles</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "           in.variation_type  \\\n",
       "0           copy number loss   \n",
       "1           copy number gain   \n",
       "2                   Deletion   \n",
       "3                Duplication   \n",
       "4           copy number loss   \n",
       "5           copy number gain   \n",
       "6  single nucleotide variant   \n",
       "7                  Insertion   \n",
       "8                  Variation   \n",
       "\n",
       "                           in.vrs_xform_plan.policy  count  \n",
       "0                               Absolute copy count    475  \n",
       "1                               Absolute copy count    344  \n",
       "2  Copy number change (cn loss|del and cn gain|dup)     89  \n",
       "3  Copy number change (cn loss|del and cn gain|dup)     72  \n",
       "4  Copy number change (cn loss|del and cn gain|dup)     31  \n",
       "5  Copy number change (cn loss|del and cn gain|dup)     17  \n",
       "6                      Remaining valid hgvs alleles     11  \n",
       "7                      Remaining valid hgvs alleles      1  \n",
       "8                      Remaining valid hgvs alleles      1  "
      ]
     },
     "execution_count": 42,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "variation_type_count_supported_not_normalized_df = supported_not_normalized_df.value_counts([\"in.variation_type\", \"in.vrs_xform_plan.policy\"]).reset_index()\n",
    "variation_type_count_supported_not_normalized_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [],
   "source": [
    "variation_type_count_supported_not_normalized_df.to_csv(\n",
    "    \"clinvar_variation_analysis_output/variation_type_count_supported_not_normalized_df.csv\"\n",
    ")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### <a id='toc1_4_3_'></a>[Not Supported Variants](#toc0_)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [],
   "source": [
    "not_supported_df = df.copy()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>in.id</th>\n",
       "      <th>in.name</th>\n",
       "      <th>in.subclass_type</th>\n",
       "      <th>in.variation_type</th>\n",
       "      <th>in.vrs_xform_plan.type</th>\n",
       "      <th>in.vrs_xform_plan.inputs</th>\n",
       "      <th>in.vrs_xform_plan.policy</th>\n",
       "      <th>out.id</th>\n",
       "      <th>out.type</th>\n",
       "      <th>out.definition</th>\n",
       "      <th>...</th>\n",
       "      <th>out.seq.ref_allele_vcf</th>\n",
       "      <th>out.seq.position_vcf</th>\n",
       "      <th>out.seq.alt_allele_vcf</th>\n",
       "      <th>in.max_copies</th>\n",
       "      <th>in.min_copies</th>\n",
       "      <th>support_status</th>\n",
       "      <th>error_string</th>\n",
       "      <th>error_string_reduce</th>\n",
       "      <th>error_string_reduce_2</th>\n",
       "      <th>normalize_status</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>425693</td>\n",
       "      <td>NM_001204.6(BMPR2):c.77-?_247+?dup</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>Duplication</td>\n",
       "      <td>Text</td>\n",
       "      <td>[id]</td>\n",
       "      <td>No hgvs or location info</td>\n",
       "      <td>Text:clinvar:425693</td>\n",
       "      <td>Text</td>\n",
       "      <td>clinvar:425693</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td></td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>90650</td>\n",
       "      <td>NM_000251.2(MSH2):c.1387-?_1510+?del</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>Text</td>\n",
       "      <td>[id]</td>\n",
       "      <td>No hgvs or location info</td>\n",
       "      <td>Text:clinvar:90650</td>\n",
       "      <td>Text</td>\n",
       "      <td>clinvar:90650</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td></td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>16098</td>\n",
       "      <td>SLC2A2, 1-BP INS, 793C</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>Insertion</td>\n",
       "      <td>Text</td>\n",
       "      <td>[id]</td>\n",
       "      <td>No hgvs or location info</td>\n",
       "      <td>Text:clinvar:16098</td>\n",
       "      <td>Text</td>\n",
       "      <td>clinvar:16098</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td></td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>14905</td>\n",
       "      <td>HLA-DRB1, HLA-DRB1*1101</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>Variation</td>\n",
       "      <td>Text</td>\n",
       "      <td>[id]</td>\n",
       "      <td>No hgvs or location info</td>\n",
       "      <td>Text:clinvar:14905</td>\n",
       "      <td>Text</td>\n",
       "      <td>clinvar:14905</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td></td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1048409</td>\n",
       "      <td>NM_000512.5:c.567_1002dup</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>Duplication</td>\n",
       "      <td>Text</td>\n",
       "      <td>[id]</td>\n",
       "      <td>No hgvs or location info</td>\n",
       "      <td>Text:clinvar:1048409</td>\n",
       "      <td>Text</td>\n",
       "      <td>clinvar:1048409</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td></td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2199349</th>\n",
       "      <td>1418992</td>\n",
       "      <td>NM_002076.4(GNS):c.841_842insTTTTTTTTTTTTTTTTT...</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>Insertion</td>\n",
       "      <td>Text</td>\n",
       "      <td>[id]</td>\n",
       "      <td>Invalid/unsupported hgvs</td>\n",
       "      <td>Text:clinvar:1418992</td>\n",
       "      <td>Text</td>\n",
       "      <td>clinvar:1418992</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td></td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2199350</th>\n",
       "      <td>2134754</td>\n",
       "      <td>NM_152564.5(VPS13B):c.5627_5628insTTTTTTTTTTTT...</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>Insertion</td>\n",
       "      <td>Text</td>\n",
       "      <td>[id]</td>\n",
       "      <td>Invalid/unsupported hgvs</td>\n",
       "      <td>Text:clinvar:2134754</td>\n",
       "      <td>Text</td>\n",
       "      <td>clinvar:2134754</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td></td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2210617</th>\n",
       "      <td>1513408</td>\n",
       "      <td>NM_024928.5(STN1):c.340_352AAG[2]CTACAAGGCCGGG...</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>Insertion</td>\n",
       "      <td>Text</td>\n",
       "      <td>[id]</td>\n",
       "      <td>Invalid/unsupported hgvs</td>\n",
       "      <td>Text:clinvar:1513408</td>\n",
       "      <td>Text</td>\n",
       "      <td>clinvar:1513408</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td></td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2210618</th>\n",
       "      <td>1464440</td>\n",
       "      <td>NM_001267550.2(TTN):c.29512_29513insGGCCGGGCGC...</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>Insertion</td>\n",
       "      <td>Text</td>\n",
       "      <td>[id]</td>\n",
       "      <td>Invalid/unsupported hgvs</td>\n",
       "      <td>Text:clinvar:1464440</td>\n",
       "      <td>Text</td>\n",
       "      <td>clinvar:1464440</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td></td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2210619</th>\n",
       "      <td>1453116</td>\n",
       "      <td>NM_000051.4(ATM):c.3376_3377insGGCCGGGCGCGGTGG...</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>Insertion</td>\n",
       "      <td>Text</td>\n",
       "      <td>[id]</td>\n",
       "      <td>Invalid/unsupported hgvs</td>\n",
       "      <td>Text:clinvar:1453116</td>\n",
       "      <td>Text</td>\n",
       "      <td>clinvar:1453116</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td></td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>10650 rows × 95 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           in.id                                            in.name  \\\n",
       "0         425693                 NM_001204.6(BMPR2):c.77-?_247+?dup   \n",
       "1          90650               NM_000251.2(MSH2):c.1387-?_1510+?del   \n",
       "2          16098                             SLC2A2, 1-BP INS, 793C   \n",
       "3          14905                            HLA-DRB1, HLA-DRB1*1101   \n",
       "4        1048409                          NM_000512.5:c.567_1002dup   \n",
       "...          ...                                                ...   \n",
       "2199349  1418992  NM_002076.4(GNS):c.841_842insTTTTTTTTTTTTTTTTT...   \n",
       "2199350  2134754  NM_152564.5(VPS13B):c.5627_5628insTTTTTTTTTTTT...   \n",
       "2210617  1513408  NM_024928.5(STN1):c.340_352AAG[2]CTACAAGGCCGGG...   \n",
       "2210618  1464440  NM_001267550.2(TTN):c.29512_29513insGGCCGGGCGC...   \n",
       "2210619  1453116  NM_000051.4(ATM):c.3376_3377insGGCCGGGCGCGGTGG...   \n",
       "\n",
       "        in.subclass_type in.variation_type in.vrs_xform_plan.type  \\\n",
       "0           SimpleAllele       Duplication                   Text   \n",
       "1           SimpleAllele          Deletion                   Text   \n",
       "2           SimpleAllele         Insertion                   Text   \n",
       "3           SimpleAllele         Variation                   Text   \n",
       "4           SimpleAllele       Duplication                   Text   \n",
       "...                  ...               ...                    ...   \n",
       "2199349     SimpleAllele         Insertion                   Text   \n",
       "2199350     SimpleAllele         Insertion                   Text   \n",
       "2210617     SimpleAllele         Insertion                   Text   \n",
       "2210618     SimpleAllele         Insertion                   Text   \n",
       "2210619     SimpleAllele         Insertion                   Text   \n",
       "\n",
       "        in.vrs_xform_plan.inputs  in.vrs_xform_plan.policy  \\\n",
       "0                           [id]  No hgvs or location info   \n",
       "1                           [id]  No hgvs or location info   \n",
       "2                           [id]  No hgvs or location info   \n",
       "3                           [id]  No hgvs or location info   \n",
       "4                           [id]  No hgvs or location info   \n",
       "...                          ...                       ...   \n",
       "2199349                     [id]  Invalid/unsupported hgvs   \n",
       "2199350                     [id]  Invalid/unsupported hgvs   \n",
       "2210617                     [id]  Invalid/unsupported hgvs   \n",
       "2210618                     [id]  Invalid/unsupported hgvs   \n",
       "2210619                     [id]  Invalid/unsupported hgvs   \n",
       "\n",
       "                       out.id out.type   out.definition  ...  \\\n",
       "0         Text:clinvar:425693     Text   clinvar:425693  ...   \n",
       "1          Text:clinvar:90650     Text    clinvar:90650  ...   \n",
       "2          Text:clinvar:16098     Text    clinvar:16098  ...   \n",
       "3          Text:clinvar:14905     Text    clinvar:14905  ...   \n",
       "4        Text:clinvar:1048409     Text  clinvar:1048409  ...   \n",
       "...                       ...      ...              ...  ...   \n",
       "2199349  Text:clinvar:1418992     Text  clinvar:1418992  ...   \n",
       "2199350  Text:clinvar:2134754     Text  clinvar:2134754  ...   \n",
       "2210617  Text:clinvar:1513408     Text  clinvar:1513408  ...   \n",
       "2210618  Text:clinvar:1464440     Text  clinvar:1464440  ...   \n",
       "2210619  Text:clinvar:1453116     Text  clinvar:1453116  ...   \n",
       "\n",
       "        out.seq.ref_allele_vcf out.seq.position_vcf out.seq.alt_allele_vcf  \\\n",
       "0                          NaN                  NaN                    NaN   \n",
       "1                          NaN                  NaN                    NaN   \n",
       "2                          NaN                  NaN                    NaN   \n",
       "3                          NaN                  NaN                    NaN   \n",
       "4                          NaN                  NaN                    NaN   \n",
       "...                        ...                  ...                    ...   \n",
       "2199349                    NaN                  NaN                    NaN   \n",
       "2199350                    NaN                  NaN                    NaN   \n",
       "2210617                    NaN                  NaN                    NaN   \n",
       "2210618                    NaN                  NaN                    NaN   \n",
       "2210619                    NaN                  NaN                    NaN   \n",
       "\n",
       "        in.max_copies in.min_copies support_status error_string  \\\n",
       "0                 NaN           NaN          False                \n",
       "1                 NaN           NaN          False                \n",
       "2                 NaN           NaN          False                \n",
       "3                 NaN           NaN          False                \n",
       "4                 NaN           NaN          False                \n",
       "...               ...           ...            ...          ...   \n",
       "2199349           NaN           NaN          False                \n",
       "2199350           NaN           NaN          False                \n",
       "2210617           NaN           NaN          False                \n",
       "2210618           NaN           NaN          False                \n",
       "2210619           NaN           NaN          False                \n",
       "\n",
       "        error_string_reduce error_string_reduce_2  normalize_status  \n",
       "0             Not Supported         Not Supported             False  \n",
       "1             Not Supported         Not Supported             False  \n",
       "2             Not Supported         Not Supported             False  \n",
       "3             Not Supported         Not Supported             False  \n",
       "4             Not Supported         Not Supported             False  \n",
       "...                     ...                   ...               ...  \n",
       "2199349       Not Supported         Not Supported             False  \n",
       "2199350       Not Supported         Not Supported             False  \n",
       "2210617       Not Supported         Not Supported             False  \n",
       "2210618       Not Supported         Not Supported             False  \n",
       "2210619       Not Supported         Not Supported             False  \n",
       "\n",
       "[10650 rows x 95 columns]"
      ]
     },
     "execution_count": 47,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "not_supported_df = not_supported_df.loc[\n",
    "    (not_supported_df[\"support_status\"] == False)\n",
    "    & (not_supported_df[\"normalize_status\"] == False)\n",
    "]\n",
    "not_supported_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>in.variation_type</th>\n",
       "      <th>in.vrs_xform_plan.policy</th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>copy number gain</td>\n",
       "      <td>NCBI36 genomic only</td>\n",
       "      <td>2897</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>copy number loss</td>\n",
       "      <td>NCBI36 genomic only</td>\n",
       "      <td>1749</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Deletion</td>\n",
       "      <td>No hgvs or location info</td>\n",
       "      <td>1236</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Diplotype</td>\n",
       "      <td>Genotype/Haplotype</td>\n",
       "      <td>596</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Haplotype</td>\n",
       "      <td>Genotype/Haplotype</td>\n",
       "      <td>565</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>Microsatellite</td>\n",
       "      <td>Invalid/unsupported hgvs</td>\n",
       "      <td>415</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>Deletion</td>\n",
       "      <td>Invalid/unsupported hgvs</td>\n",
       "      <td>413</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>Insertion</td>\n",
       "      <td>No hgvs or location info</td>\n",
       "      <td>382</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>Insertion</td>\n",
       "      <td>Invalid/unsupported hgvs</td>\n",
       "      <td>306</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>No hgvs or location info</td>\n",
       "      <td>287</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>Translocation</td>\n",
       "      <td>No hgvs or location info</td>\n",
       "      <td>271</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>Duplication</td>\n",
       "      <td>No hgvs or location info</td>\n",
       "      <td>265</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>CompoundHeterozygote</td>\n",
       "      <td>Genotype/Haplotype</td>\n",
       "      <td>249</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>Variation</td>\n",
       "      <td>No hgvs or location info</td>\n",
       "      <td>206</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>Indel</td>\n",
       "      <td>No hgvs or location info</td>\n",
       "      <td>138</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>Duplication</td>\n",
       "      <td>Invalid/unsupported hgvs</td>\n",
       "      <td>116</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>protein only</td>\n",
       "      <td>No hgvs or location info</td>\n",
       "      <td>95</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>Complex</td>\n",
       "      <td>No hgvs or location info</td>\n",
       "      <td>72</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>Deletion</td>\n",
       "      <td>NCBI36 genomic only</td>\n",
       "      <td>71</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>Microsatellite</td>\n",
       "      <td>No hgvs or location info</td>\n",
       "      <td>67</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>Inversion</td>\n",
       "      <td>No hgvs or location info</td>\n",
       "      <td>63</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>Duplication</td>\n",
       "      <td>NCBI36 genomic only</td>\n",
       "      <td>50</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>Indel</td>\n",
       "      <td>Invalid/unsupported hgvs</td>\n",
       "      <td>40</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>Inversion</td>\n",
       "      <td>Invalid/unsupported hgvs</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>Haplotype, single variant</td>\n",
       "      <td>Genotype/Haplotype</td>\n",
       "      <td>21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>copy number gain</td>\n",
       "      <td>Min/max copy count range not supported</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>Phase unknown</td>\n",
       "      <td>Genotype/Haplotype</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>copy number loss</td>\n",
       "      <td>Invalid/unsupported hgvs</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>fusion</td>\n",
       "      <td>No hgvs or location info</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>copy number gain</td>\n",
       "      <td>Invalid/unsupported hgvs</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>Complex</td>\n",
       "      <td>NCBI36 genomic only</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>Variation</td>\n",
       "      <td>Invalid/unsupported hgvs</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>Complex</td>\n",
       "      <td>Invalid/unsupported hgvs</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>Translocation</td>\n",
       "      <td>Invalid/unsupported hgvs</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>Translocation</td>\n",
       "      <td>NCBI36 genomic only</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>Tandem duplication</td>\n",
       "      <td>No hgvs or location info</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>Distinct chromosomes</td>\n",
       "      <td>Genotype/Haplotype</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            in.variation_type                in.vrs_xform_plan.policy  count\n",
       "0            copy number gain                     NCBI36 genomic only   2897\n",
       "1            copy number loss                     NCBI36 genomic only   1749\n",
       "2                    Deletion                No hgvs or location info   1236\n",
       "3                   Diplotype                      Genotype/Haplotype    596\n",
       "4                   Haplotype                      Genotype/Haplotype    565\n",
       "5              Microsatellite                Invalid/unsupported hgvs    415\n",
       "6                    Deletion                Invalid/unsupported hgvs    413\n",
       "7                   Insertion                No hgvs or location info    382\n",
       "8                   Insertion                Invalid/unsupported hgvs    306\n",
       "9   single nucleotide variant                No hgvs or location info    287\n",
       "10              Translocation                No hgvs or location info    271\n",
       "11                Duplication                No hgvs or location info    265\n",
       "12       CompoundHeterozygote                      Genotype/Haplotype    249\n",
       "13                  Variation                No hgvs or location info    206\n",
       "14                      Indel                No hgvs or location info    138\n",
       "15                Duplication                Invalid/unsupported hgvs    116\n",
       "16               protein only                No hgvs or location info     95\n",
       "17                    Complex                No hgvs or location info     72\n",
       "18                   Deletion                     NCBI36 genomic only     71\n",
       "19             Microsatellite                No hgvs or location info     67\n",
       "20                  Inversion                No hgvs or location info     63\n",
       "21                Duplication                     NCBI36 genomic only     50\n",
       "22                      Indel                Invalid/unsupported hgvs     40\n",
       "23                  Inversion                Invalid/unsupported hgvs     29\n",
       "24  Haplotype, single variant                      Genotype/Haplotype     21\n",
       "25           copy number gain  Min/max copy count range not supported     14\n",
       "26              Phase unknown                      Genotype/Haplotype      8\n",
       "27           copy number loss                Invalid/unsupported hgvs      7\n",
       "28                     fusion                No hgvs or location info      6\n",
       "29           copy number gain                Invalid/unsupported hgvs      5\n",
       "30                    Complex                     NCBI36 genomic only      3\n",
       "31                  Variation                Invalid/unsupported hgvs      2\n",
       "32                    Complex                Invalid/unsupported hgvs      2\n",
       "33              Translocation                Invalid/unsupported hgvs      1\n",
       "34              Translocation                     NCBI36 genomic only      1\n",
       "35         Tandem duplication                No hgvs or location info      1\n",
       "36       Distinct chromosomes                      Genotype/Haplotype      1"
      ]
     },
     "execution_count": 49,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "variation_type_count_not_supported_df = not_supported_df.value_counts([\"in.variation_type\", \"in.vrs_xform_plan.policy\"]).reset_index()\n",
    "variation_type_count_not_supported_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [],
   "source": [
    "variation_type_count_not_supported_df.to_csv(\n",
    "    \"clinvar_variation_analysis_output/variation_type_count_not_supported_df.csv\"\n",
    ")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Sanity check: making sure there are no supported variants that have been marked as normalized"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [],
   "source": [
    "not_supported_but_normalized_df = df.copy()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>in.id</th>\n",
       "      <th>in.name</th>\n",
       "      <th>in.subclass_type</th>\n",
       "      <th>in.variation_type</th>\n",
       "      <th>in.vrs_xform_plan.type</th>\n",
       "      <th>in.vrs_xform_plan.inputs</th>\n",
       "      <th>in.vrs_xform_plan.policy</th>\n",
       "      <th>out.id</th>\n",
       "      <th>out.type</th>\n",
       "      <th>out.definition</th>\n",
       "      <th>...</th>\n",
       "      <th>out.seq.ref_allele_vcf</th>\n",
       "      <th>out.seq.position_vcf</th>\n",
       "      <th>out.seq.alt_allele_vcf</th>\n",
       "      <th>in.max_copies</th>\n",
       "      <th>in.min_copies</th>\n",
       "      <th>support_status</th>\n",
       "      <th>error_string</th>\n",
       "      <th>error_string_reduce</th>\n",
       "      <th>error_string_reduce_2</th>\n",
       "      <th>normalize_status</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>0 rows × 95 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [in.id, in.name, in.subclass_type, in.variation_type, in.vrs_xform_plan.type, in.vrs_xform_plan.inputs, in.vrs_xform_plan.policy, out.id, out.type, out.definition, in.xrefs, in.hgvs.assembly, in.hgvs.nucleotide, out._id, out.location._id, out.location.type, out.location.sequence_id, out.location.interval.type, out.location.interval.start.type, out.location.interval.start.value, out.location.interval.end.type, out.location.interval.end.value, out.state.type, out.state.sequence, out.name, out.subclass_type, out.variation_type, out.hgvs.assembly, out.hgvs.nucleotide, out.vrs_xform_plan.type, out.vrs_xform_plan.inputs, out.vrs_xform_plan.policy, out.errors, out.subject._id, out.subject.type, out.subject.sequence_id, out.subject.interval.type, out.subject.interval.start.type, out.subject.interval.start.value, out.subject.interval.end.type, out.subject.interval.end.value, out.copy_change, out.xrefs, in.seq.disp_start, in.seq.inner_start, in.seq.chr, in.seq.inner_stop, in.seq.accession, in.seq.assembly, in.seq.disp_stop, in.seq.outer_stop, in.seq.outer_start, in.absolute_copies, in.derived_hgvs, in.cyto, out.subject.interval.start.min, out.subject.interval.start.max, out.subject.interval.end.min, out.subject.interval.end.max, out.subject.interval.start.comparator, out.subject.interval.end.comparator, out.copies.type, out.copies.value, in.seq.variant_length, out.seq.assembly, out.seq.accession, out.seq.chr, out.seq.disp_start, out.seq.disp_stop, out.seq.inner_start, out.seq.inner_stop, out.absolute_copies, out.derived_hgvs, out.cyto, in.seq.start, in.seq.stop, out.seq.outer_start, out.seq.outer_stop, out.seq.variant_length, out.seq.start, out.seq.stop, in.seq.ref_allele_vcf, in.seq.position_vcf, in.seq.alt_allele_vcf, in.canonical_spdi, out.seq.ref_allele_vcf, out.seq.position_vcf, out.seq.alt_allele_vcf, in.max_copies, in.min_copies, support_status, error_string, error_string_reduce, error_string_reduce_2, normalize_status]\n",
       "Index: []\n",
       "\n",
       "[0 rows x 95 columns]"
      ]
     },
     "execution_count": 52,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "not_supported_but_normalized_df = not_supported_but_normalized_df.loc[\n",
    "    (not_supported_but_normalized_df[\"support_status\"] == False)\n",
    "    & (not_supported_but_normalized_df[\"normalize_status\"] == True)\n",
    "]\n",
    "not_supported_but_normalized_df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## <a id='toc1_5_'></a>[Counting variants from each group](#toc0_)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "num_supported = len(supported_df)\n",
    "num_supported_not_normalized = len(supported_not_normalized_df)\n",
    "num_not_supported_but_normalized = len(not_supported_but_normalized_df)\n",
    "num_not_supported = len(not_supported_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "summary_df2 = pd.DataFrame({\"Supported\":[num_supported, num_supported_not_normalized],\n",
    "                \"Not Supported\":[num_not_supported_but_normalized, num_not_supported]})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "summary_df2.index = ['Normalized', 'Not Normalized']\n",
    "summary_df2"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Counting variant types for each group"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "variation_type_count_summary_df = pd.merge(pd.merge(variation_type_count_supported_df,variation_type_count_supported_not_normalized_df, on='in.variation_type', how = \"left\"), variation_type_count_not_supported_df, on='in.variation_type', how = \"right\")\n",
    "variation_type_count_summary_df = variation_type_count_summary_df.replace(np.nan,'',regex=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "variation_type_count_summary_df = variation_type_count_summary_df.rename(columns={\"in.id_x\": \"supported\", \"in.id_y\": \"supported_not_normalized\", \"in.id\": \"not_supported\"})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "variation_type_count_summary_df.to_csv(\n",
    "    \"clinvar_variation_analysis_output/variation_type_count_summary_df.csv\"\n",
    ")"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3.10.9 ('.venv': venv)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.4"
  },
  "orig_nbformat": 4,
  "vscode": {
   "interpreter": {
    "hash": "60bedfc6dd95a1d756c285a8080496495ec396023cb257f004626f94ae504b38"
   }
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
