{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# <a id='toc1_'></a>[ClinVar Variant Analysis](#toc0_)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The clinvar_variation_analysis notebook contains an analysis on ClinVar variant data"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Table of contents**<a id='toc0_'></a>    \n",
    "- [ClinVar Variant Analysis](#toc1_)    \n",
    "  - [Initialize](#toc1_1_)    \n",
    "    - [Import necessary libraries](#toc1_1_1_)    \n",
    "    - [Create output directory](#toc1_1_2_)    \n",
    "    - [Import variant information file](#toc1_1_3_)    \n",
    "  - [Add Supported Status of Variant based on in.vrs_xform_plan.policy](#toc1_2_)    \n",
    "  - [Add Normalization Status of Variant based on out.errors](#toc1_3_)    \n",
    "    - [Set Normalize Status of Variant as T/F](#toc1_3_1_)    \n",
    "      - [Summary Table](#toc1_3_1_1_)    \n",
    "  - [Create subgroups based on Variant Status](#toc1_4_)    \n",
    "    - [Supported and Normalized Variants](#toc1_4_1_)    \n",
    "    - [Supported and Not Normalized Variants](#toc1_4_2_)    \n",
    "    - [Not Supported Variants](#toc1_4_3_)    \n",
    "  - [Counting variants from each group](#toc1_5_)    \n",
    "  - [Counting variant types for each group](#toc1_6_)    \n",
    "\n",
    "<!-- vscode-jupyter-toc-config\n",
    "\tnumbering=false\n",
    "\tanchor=true\n",
    "\tflat=false\n",
    "\tminLevel=1\n",
    "\tmaxLevel=6\n",
    "\t/vscode-jupyter-toc-config -->\n",
    "<!-- THIS CELL WILL BE REPLACED ON TOC UPDATE. DO NOT WRITE YOUR TEXT IN THIS CELL -->"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## <a id='toc1_1_'></a>[Initialize](#toc0_)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### <a id='toc1_1_1_'></a>[Import necessary libraries](#toc0_)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import ndjson\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "from pathlib import Path\n",
    "import gzip"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### <a id='toc1_1_2_'></a>[Create output directory](#toc0_)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "path = Path(\"variation_analysis_output\")\n",
    "path.mkdir(exist_ok=True)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### <a id='toc1_1_3_'></a>[Import variant information file](#toc0_)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "with gzip.open(\"normalized-vi.json.gz\", \"rb\") as f:\n",
    "    file_content = ndjson.load(f)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.json_normalize(file_content)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## <a id='toc1_2_'></a>[Add Supported Status of Variant based on in.issue](#toc0_)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Checking for blanks"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "in.issue\n",
       "No viable variation members identified.                       2093\n",
       "haplotype and genotype variations are not supported.          1540\n",
       "sequence for accession not supported by vrs-python release     668\n",
       "repeat expressions are not supported.                          474\n",
       "unsupported hgvs expression.                                   457\n",
       "intronic positions are not resolvable in sequence.             272\n",
       "range copies are not supported.                                 95\n",
       "expression contains unbalaned paretheses.                        3\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df[\"in.issue\"].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "df[\"in.issue\"] = df[\"in.issue\"].fillna(\"None\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "df[\"support_status\"] = df[\"in.issue\"].copy()\n",
    "\n",
    "df.loc[df[\"support_status\"] == \"No viable variation members identified.\", \"support_status\"] = False\n",
    "df.loc[\n",
    "    df[\"support_status\"] == \"haplotype and genotype variations are not supported.\",\n",
    "    \"support_status\",\n",
    "] = False\n",
    "df.loc[df[\"support_status\"] == \"sequence for accession not supported by vrs-python release\", \"support_status\"] = False\n",
    "df.loc[df[\"support_status\"] == \"repeat expressions are not supported.\", \"support_status\"] = False\n",
    "df.loc[df[\"support_status\"] == \"unsupported hgvs expression.\", \"support_status\"] = False\n",
    "df.loc[df[\"support_status\"] == \"intronic positions are not resolvable in sequence.\", \"support_status\"] = False\n",
    "df.loc[df[\"support_status\"] == \"range copies are not supported.\", \"support_status\"] = False\n",
    "df.loc[\n",
    "    df[\"support_status\"] == \"expression contains unbalaned paretheses.\", \"support_status\"\n",
    "] = False\n",
    "df[\"support_status\"] = df[\"support_status\"].fillna(False).astype(bool)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "support_status\n",
       "True     3738092\n",
       "False       5602\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df[\"support_status\"].value_counts()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## <a id='toc1_3_'></a>[Add Normalization Status of Variant based on out.errors](#toc0_)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The errors are stored as a list of values, some of which are strings and other of which are dictionaries (determined by whether error was handled at the level of Variation Normalizer or after the normalizer)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The \"get_errors\" function extracts the text error responses for better readability and ease string processing"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "import ast\n",
    "\n",
    "def get_errors(errors: list) -> str:\n",
    "    \"\"\"Takes the values for the errors and represents them as a string\n",
    "    :param errors: list of errors\n",
    "    :return: string representing error\n",
    "    \"\"\"\n",
    "    if pd.isna(errors):\n",
    "        return \"Success\"\n",
    "\n",
    "    # Parse if it's a stringified list/dict\n",
    "    if isinstance(errors, str):\n",
    "        try:\n",
    "            errors = ast.literal_eval(errors)\n",
    "        except Exception:\n",
    "            return errors  # return raw string if not a valid list or dict\n",
    "\n",
    "    errors_out = []\n",
    "\n",
    "    # Normalize to list\n",
    "    if not isinstance(errors, list):\n",
    "        errors = [errors]\n",
    "\n",
    "    for e in errors:\n",
    "        if isinstance(e, str):\n",
    "            errors_out.append(e)\n",
    "        elif isinstance(e, dict):\n",
    "            for k, v in e.items():\n",
    "                if k not in [\"msg\", \"response-errors\"]:\n",
    "                    continue\n",
    "                if isinstance(v, str):\n",
    "                    errors_out.append(v)\n",
    "                elif isinstance(v, list):\n",
    "                    errors_out.extend(v)  # multiple error messages\n",
    "    return \"; \".join(errors_out)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "df[\"error_string\"] = df[\"out.errors\"].apply(get_errors)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "error_string\n",
       "Success                                                                                  3736937\n",
       "Unable to get GRCh37/GRCh38 assembly for: NC_000023.9                                        422\n",
       "Unable to get GRCh37/GRCh38 assembly for: NC_000024.8                                        392\n",
       "Unable to get GRCh37/GRCh38 assembly for: NC_000002.10                                       305\n",
       "Unable to get GRCh37/GRCh38 assembly for: NC_000009.10                                       279\n",
       "                                                                                          ...   \n",
       "Unable to find classification for: NC_000014.8:g.(?_88399357)_(88417093_88429727)del           1\n",
       "Unable to find classification for: NC_000013.11:g.(32380146_32394688)_(32398771_?)del          1\n",
       "Unable to find classification for: NC_000006.11:g.(?_32006191)_(32007026_32007132)del          1\n",
       "Unable to find classification for: NC_000001.10:g.(?_1447541)_(1454371_1455520)del             1\n",
       "Unable to tokenize: NG_070230.1:g.18747C>T                                                     1\n",
       "Name: count, Length: 1664, dtype: int64"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df[\"error_string\"].value_counts()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "There are Not Supported variants that have no error (marked as success inaccurately) because they were labeled \"Not Supported\" manually."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "An error (\"Not Supported\") is entered manually for those variants so that they are not categorized as normalized"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "df.loc[\n",
    "    (~df[\"support_status\"]) & (df[\"error_string\"] == \"Success\"),\n",
    "    \"error_string\",\n",
    "] = \"Not Supported\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### <a id='toc1_3_1_'></a>[Set Normalize Status of Variant as T/F](#toc0_)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "If an error is present, the variant was not normalized and therefore has a False Normalize Status"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>out</th>\n",
       "      <th>in.variation_id</th>\n",
       "      <th>in.name</th>\n",
       "      <th>in.vrs_class</th>\n",
       "      <th>in.range_copies</th>\n",
       "      <th>in.issue</th>\n",
       "      <th>in.variation_type</th>\n",
       "      <th>in.subclass_type</th>\n",
       "      <th>in.mappings</th>\n",
       "      <th>in.cytogenetic</th>\n",
       "      <th>...</th>\n",
       "      <th>out.state.type</th>\n",
       "      <th>out.state.length</th>\n",
       "      <th>out.state.sequence</th>\n",
       "      <th>out.state.repeatSubunitLength</th>\n",
       "      <th>out.extensions</th>\n",
       "      <th>in.absolute_copies</th>\n",
       "      <th>out.copies</th>\n",
       "      <th>support_status</th>\n",
       "      <th>error_string</th>\n",
       "      <th>normalize_status</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NaN</td>\n",
       "      <td>427832</td>\n",
       "      <td>NM_000785.3(CYP27B1):c.[1319_1325dupCCCACCC];[...</td>\n",
       "      <td>Not Available</td>\n",
       "      <td>[]</td>\n",
       "      <td>haplotype and genotype variations are not supp...</td>\n",
       "      <td>CompoundHeterozygote</td>\n",
       "      <td>Genotype</td>\n",
       "      <td>[{'system': 'ClinVar', 'code': '427832', 'rela...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NaN</td>\n",
       "      <td>424704</td>\n",
       "      <td>NM_003977.3(AIP):c.[-125-145_-125-144delCGinsA...</td>\n",
       "      <td>Not Available</td>\n",
       "      <td>[]</td>\n",
       "      <td>haplotype and genotype variations are not supp...</td>\n",
       "      <td>CompoundHeterozygote</td>\n",
       "      <td>Genotype</td>\n",
       "      <td>[{'system': 'ClinVar', 'code': '424704', 'rela...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NaN</td>\n",
       "      <td>982544</td>\n",
       "      <td>NM_000329.3(RPE65):c.[1067dup];[1543C&gt;T]</td>\n",
       "      <td>Not Available</td>\n",
       "      <td>[]</td>\n",
       "      <td>haplotype and genotype variations are not supp...</td>\n",
       "      <td>CompoundHeterozygote</td>\n",
       "      <td>Genotype</td>\n",
       "      <td>[{'system': 'ClinVar', 'code': '982544', 'rela...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NaN</td>\n",
       "      <td>424730</td>\n",
       "      <td>NM_000372.4(TYR):c.[1276_1282delATGGTTC];[139G&gt;A]</td>\n",
       "      <td>Not Available</td>\n",
       "      <td>[]</td>\n",
       "      <td>haplotype and genotype variations are not supp...</td>\n",
       "      <td>CompoundHeterozygote</td>\n",
       "      <td>Genotype</td>\n",
       "      <td>[{'system': 'ClinVar', 'code': '424730', 'rela...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NaN</td>\n",
       "      <td>424736</td>\n",
       "      <td>NM_005045.3(RELN):c.[2213G&gt;A];[9427T&gt;G]</td>\n",
       "      <td>Not Available</td>\n",
       "      <td>[]</td>\n",
       "      <td>haplotype and genotype variations are not supp...</td>\n",
       "      <td>CompoundHeterozygote</td>\n",
       "      <td>Genotype</td>\n",
       "      <td>[{'system': 'ClinVar', 'code': '424736', 'rela...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3743689</th>\n",
       "      <td>NaN</td>\n",
       "      <td>3911672</td>\n",
       "      <td>NC_012920.1(MT-CO3):m.9698T&gt;C</td>\n",
       "      <td>Allele</td>\n",
       "      <td>[]</td>\n",
       "      <td>None</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>[]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>LiteralSequenceExpression</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>Success</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3743690</th>\n",
       "      <td>NaN</td>\n",
       "      <td>693215</td>\n",
       "      <td>NC_012920.1(MT-CO3):m.9751T&gt;C</td>\n",
       "      <td>Allele</td>\n",
       "      <td>[]</td>\n",
       "      <td>None</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>[]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>LiteralSequenceExpression</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>Success</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3743691</th>\n",
       "      <td>NaN</td>\n",
       "      <td>693235</td>\n",
       "      <td>NC_012920.1(MT-CO3):m.9828G&gt;A</td>\n",
       "      <td>Allele</td>\n",
       "      <td>[]</td>\n",
       "      <td>None</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>[]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>LiteralSequenceExpression</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>Success</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3743692</th>\n",
       "      <td>NaN</td>\n",
       "      <td>3049652</td>\n",
       "      <td>NM_001372044.2(SHANK3):c.1111-8C&gt;T</td>\n",
       "      <td>Allele</td>\n",
       "      <td>[]</td>\n",
       "      <td>None</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>[]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>Unable to tokenize: NG_070230.1:g.18747C&gt;T</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3743693</th>\n",
       "      <td>NaN</td>\n",
       "      <td>2223234</td>\n",
       "      <td>NM_080615.1:c.1096G&gt;A</td>\n",
       "      <td>Allele</td>\n",
       "      <td>[]</td>\n",
       "      <td>None</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>[]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>LiteralSequenceExpression</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>Success</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3743694 rows × 40 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         out in.variation_id  \\\n",
       "0        NaN          427832   \n",
       "1        NaN          424704   \n",
       "2        NaN          982544   \n",
       "3        NaN          424730   \n",
       "4        NaN          424736   \n",
       "...      ...             ...   \n",
       "3743689  NaN         3911672   \n",
       "3743690  NaN          693215   \n",
       "3743691  NaN          693235   \n",
       "3743692  NaN         3049652   \n",
       "3743693  NaN         2223234   \n",
       "\n",
       "                                                   in.name   in.vrs_class  \\\n",
       "0        NM_000785.3(CYP27B1):c.[1319_1325dupCCCACCC];[...  Not Available   \n",
       "1        NM_003977.3(AIP):c.[-125-145_-125-144delCGinsA...  Not Available   \n",
       "2                 NM_000329.3(RPE65):c.[1067dup];[1543C>T]  Not Available   \n",
       "3        NM_000372.4(TYR):c.[1276_1282delATGGTTC];[139G>A]  Not Available   \n",
       "4                  NM_005045.3(RELN):c.[2213G>A];[9427T>G]  Not Available   \n",
       "...                                                    ...            ...   \n",
       "3743689                      NC_012920.1(MT-CO3):m.9698T>C         Allele   \n",
       "3743690                      NC_012920.1(MT-CO3):m.9751T>C         Allele   \n",
       "3743691                      NC_012920.1(MT-CO3):m.9828G>A         Allele   \n",
       "3743692                 NM_001372044.2(SHANK3):c.1111-8C>T         Allele   \n",
       "3743693                              NM_080615.1:c.1096G>A         Allele   \n",
       "\n",
       "        in.range_copies                                           in.issue  \\\n",
       "0                    []  haplotype and genotype variations are not supp...   \n",
       "1                    []  haplotype and genotype variations are not supp...   \n",
       "2                    []  haplotype and genotype variations are not supp...   \n",
       "3                    []  haplotype and genotype variations are not supp...   \n",
       "4                    []  haplotype and genotype variations are not supp...   \n",
       "...                 ...                                                ...   \n",
       "3743689              []                                               None   \n",
       "3743690              []                                               None   \n",
       "3743691              []                                               None   \n",
       "3743692              []                                               None   \n",
       "3743693              []                                               None   \n",
       "\n",
       "                 in.variation_type in.subclass_type  \\\n",
       "0             CompoundHeterozygote         Genotype   \n",
       "1             CompoundHeterozygote         Genotype   \n",
       "2             CompoundHeterozygote         Genotype   \n",
       "3             CompoundHeterozygote         Genotype   \n",
       "4             CompoundHeterozygote         Genotype   \n",
       "...                            ...              ...   \n",
       "3743689  single nucleotide variant     SimpleAllele   \n",
       "3743690  single nucleotide variant     SimpleAllele   \n",
       "3743691  single nucleotide variant     SimpleAllele   \n",
       "3743692  single nucleotide variant     SimpleAllele   \n",
       "3743693  single nucleotide variant     SimpleAllele   \n",
       "\n",
       "                                               in.mappings in.cytogenetic  \\\n",
       "0        [{'system': 'ClinVar', 'code': '427832', 'rela...            NaN   \n",
       "1        [{'system': 'ClinVar', 'code': '424704', 'rela...            NaN   \n",
       "2        [{'system': 'ClinVar', 'code': '982544', 'rela...            NaN   \n",
       "3        [{'system': 'ClinVar', 'code': '424730', 'rela...            NaN   \n",
       "4        [{'system': 'ClinVar', 'code': '424736', 'rela...            NaN   \n",
       "...                                                    ...            ...   \n",
       "3743689                                                 []            NaN   \n",
       "3743690                                                 []            NaN   \n",
       "3743691                                                 []            NaN   \n",
       "3743692                                                 []            NaN   \n",
       "3743693                                                 []            NaN   \n",
       "\n",
       "         ...             out.state.type out.state.length out.state.sequence  \\\n",
       "0        ...                        NaN              NaN                NaN   \n",
       "1        ...                        NaN              NaN                NaN   \n",
       "2        ...                        NaN              NaN                NaN   \n",
       "3        ...                        NaN              NaN                NaN   \n",
       "4        ...                        NaN              NaN                NaN   \n",
       "...      ...                        ...              ...                ...   \n",
       "3743689  ...  LiteralSequenceExpression              NaN                  C   \n",
       "3743690  ...  LiteralSequenceExpression              NaN                  C   \n",
       "3743691  ...  LiteralSequenceExpression              NaN                  A   \n",
       "3743692  ...                        NaN              NaN                NaN   \n",
       "3743693  ...  LiteralSequenceExpression              NaN                  A   \n",
       "\n",
       "        out.state.repeatSubunitLength out.extensions in.absolute_copies  \\\n",
       "0                                 NaN            NaN                NaN   \n",
       "1                                 NaN            NaN                NaN   \n",
       "2                                 NaN            NaN                NaN   \n",
       "3                                 NaN            NaN                NaN   \n",
       "4                                 NaN            NaN                NaN   \n",
       "...                               ...            ...                ...   \n",
       "3743689                           NaN            NaN                NaN   \n",
       "3743690                           NaN            NaN                NaN   \n",
       "3743691                           NaN            NaN                NaN   \n",
       "3743692                           NaN            NaN                NaN   \n",
       "3743693                           NaN            NaN                NaN   \n",
       "\n",
       "        out.copies support_status                                error_string  \\\n",
       "0              NaN          False                               Not Supported   \n",
       "1              NaN          False                               Not Supported   \n",
       "2              NaN          False                               Not Supported   \n",
       "3              NaN          False                               Not Supported   \n",
       "4              NaN          False                               Not Supported   \n",
       "...            ...            ...                                         ...   \n",
       "3743689        NaN           True                                     Success   \n",
       "3743690        NaN           True                                     Success   \n",
       "3743691        NaN           True                                     Success   \n",
       "3743692        NaN           True  Unable to tokenize: NG_070230.1:g.18747C>T   \n",
       "3743693        NaN           True                                     Success   \n",
       "\n",
       "        normalize_status  \n",
       "0                  False  \n",
       "1                  False  \n",
       "2                  False  \n",
       "3                  False  \n",
       "4                  False  \n",
       "...                  ...  \n",
       "3743689             True  \n",
       "3743690             True  \n",
       "3743691             True  \n",
       "3743692            False  \n",
       "3743693             True  \n",
       "\n",
       "[3743694 rows x 40 columns]"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df[\"normalize_status\"] = df[\"error_string\"] == \"Success\"\n",
    "df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### <a id='toc1_3_1_1_'></a>[Summary Table](#toc0_)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "In the table below, the cells show the number of variants with each expected behavior and how they actually ended up performing."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "If a variant was in an \"expected to pass\" category and ends up as text, that is an instance of a normalizer failure on a supported variant"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "summary_df = (\n",
    "    df[[\"in.variation_id\", \"support_status\", \"in.issue\", \"out.type\"]]\n",
    "    .fillna(\"NONE\")\n",
    "    .groupby([\"support_status\", \"in.issue\", \"out.type\"])\n",
    "    .count()\n",
    "    .unstack(level=2)\n",
    "    .fillna(0)\n",
    "    .astype(int)[\"in.variation_id\"]\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "summary_df[\"VariantSum\"] = summary_df.sum(axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "summary_df[\"NormalizedSum\"] = summary_df[\n",
    "    [\"Allele\", \"CopyNumberChange\", \"CopyNumberCount\"]\n",
    "].sum(axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "summary_df[\"NormalizedPercent\"] = (\n",
    "    summary_df[\"NormalizedSum\"] / summary_df[\"VariantSum\"]\n",
    ").apply(lambda x: f\"{round(x * 100, 2)}%\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>out.type</th>\n",
       "      <th>Allele</th>\n",
       "      <th>CopyNumberChange</th>\n",
       "      <th>CopyNumberCount</th>\n",
       "      <th>NONE</th>\n",
       "      <th>NormalizedPercent</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>support_status</th>\n",
       "      <th>in.issue</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th rowspan=\"8\" valign=\"top\">False</th>\n",
       "      <th>No viable variation members identified.</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2093</td>\n",
       "      <td>0.0%</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>expression contains unbalaned paretheses.</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>0.0%</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>haplotype and genotype variations are not supported.</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1540</td>\n",
       "      <td>0.0%</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>intronic positions are not resolvable in sequence.</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>272</td>\n",
       "      <td>0.0%</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>range copies are not supported.</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>95</td>\n",
       "      <td>0.0%</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>repeat expressions are not supported.</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>474</td>\n",
       "      <td>0.0%</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>sequence for accession not supported by vrs-python release</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>668</td>\n",
       "      <td>0.0%</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>unsupported hgvs expression.</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>457</td>\n",
       "      <td>0.0%</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>True</th>\n",
       "      <th>None</th>\n",
       "      <td>3659725</td>\n",
       "      <td>32230</td>\n",
       "      <td>39379</td>\n",
       "      <td>6758</td>\n",
       "      <td>99.82%</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "out.type                                                            Allele  \\\n",
       "support_status in.issue                                                      \n",
       "False          No viable variation members identified.                   0   \n",
       "               expression contains unbalaned paretheses.                 0   \n",
       "               haplotype and genotype variations are not suppo...        0   \n",
       "               intronic positions are not resolvable in sequence.        0   \n",
       "               range copies are not supported.                           0   \n",
       "               repeat expressions are not supported.                     0   \n",
       "               sequence for accession not supported by vrs-pyt...        0   \n",
       "               unsupported hgvs expression.                              0   \n",
       "True           None                                                3659725   \n",
       "\n",
       "out.type                                                           CopyNumberChange  \\\n",
       "support_status in.issue                                                               \n",
       "False          No viable variation members identified.                            0   \n",
       "               expression contains unbalaned paretheses.                          0   \n",
       "               haplotype and genotype variations are not suppo...                 0   \n",
       "               intronic positions are not resolvable in sequence.                 0   \n",
       "               range copies are not supported.                                    0   \n",
       "               repeat expressions are not supported.                              0   \n",
       "               sequence for accession not supported by vrs-pyt...                 0   \n",
       "               unsupported hgvs expression.                                       0   \n",
       "True           None                                                           32230   \n",
       "\n",
       "out.type                                                           CopyNumberCount  \\\n",
       "support_status in.issue                                                              \n",
       "False          No viable variation members identified.                           0   \n",
       "               expression contains unbalaned paretheses.                         0   \n",
       "               haplotype and genotype variations are not suppo...                0   \n",
       "               intronic positions are not resolvable in sequence.                0   \n",
       "               range copies are not supported.                                   0   \n",
       "               repeat expressions are not supported.                             0   \n",
       "               sequence for accession not supported by vrs-pyt...                0   \n",
       "               unsupported hgvs expression.                                      0   \n",
       "True           None                                                          39379   \n",
       "\n",
       "out.type                                                           NONE  \\\n",
       "support_status in.issue                                                   \n",
       "False          No viable variation members identified.             2093   \n",
       "               expression contains unbalaned paretheses.              3   \n",
       "               haplotype and genotype variations are not suppo...  1540   \n",
       "               intronic positions are not resolvable in sequence.   272   \n",
       "               range copies are not supported.                       95   \n",
       "               repeat expressions are not supported.                474   \n",
       "               sequence for accession not supported by vrs-pyt...   668   \n",
       "               unsupported hgvs expression.                         457   \n",
       "True           None                                                6758   \n",
       "\n",
       "out.type                                                          NormalizedPercent  \n",
       "support_status in.issue                                                              \n",
       "False          No viable variation members identified.                         0.0%  \n",
       "               expression contains unbalaned paretheses.                       0.0%  \n",
       "               haplotype and genotype variations are not suppo...              0.0%  \n",
       "               intronic positions are not resolvable in sequence.              0.0%  \n",
       "               range copies are not supported.                                 0.0%  \n",
       "               repeat expressions are not supported.                           0.0%  \n",
       "               sequence for accession not supported by vrs-pyt...              0.0%  \n",
       "               unsupported hgvs expression.                                    0.0%  \n",
       "True           None                                                          99.82%  "
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "summary_df = summary_df.drop([\"VariantSum\", \"NormalizedSum\"], axis=1)\n",
    "summary_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "summary_df.to_csv(\"variation_analysis_output/variant_analysis_summary_df.csv\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## <a id='toc1_4_'></a>[Create subgroups based on Variant Status](#toc0_)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### <a id='toc1_4_1_'></a>[Supported and Normalized Variants](#toc0_)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "supported_df = df.copy()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>out</th>\n",
       "      <th>in.variation_id</th>\n",
       "      <th>in.name</th>\n",
       "      <th>in.vrs_class</th>\n",
       "      <th>in.range_copies</th>\n",
       "      <th>in.issue</th>\n",
       "      <th>in.variation_type</th>\n",
       "      <th>in.subclass_type</th>\n",
       "      <th>in.mappings</th>\n",
       "      <th>in.cytogenetic</th>\n",
       "      <th>...</th>\n",
       "      <th>out.state.type</th>\n",
       "      <th>out.state.length</th>\n",
       "      <th>out.state.sequence</th>\n",
       "      <th>out.state.repeatSubunitLength</th>\n",
       "      <th>out.extensions</th>\n",
       "      <th>in.absolute_copies</th>\n",
       "      <th>out.copies</th>\n",
       "      <th>support_status</th>\n",
       "      <th>error_string</th>\n",
       "      <th>normalize_status</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>NaN</td>\n",
       "      <td>3247925</td>\n",
       "      <td>NC_000001.10:g.(?_103385847)_(103496820_?)del</td>\n",
       "      <td>CopyNumberChange</td>\n",
       "      <td>[]</td>\n",
       "      <td>None</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>[]</td>\n",
       "      <td>1p21.1</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>Success</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>NaN</td>\n",
       "      <td>832762</td>\n",
       "      <td>NC_000001.11:g.(?_119415410)_(119422630_?)del</td>\n",
       "      <td>CopyNumberChange</td>\n",
       "      <td>[]</td>\n",
       "      <td>None</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>[]</td>\n",
       "      <td>1p12</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>Success</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>NaN</td>\n",
       "      <td>3247694</td>\n",
       "      <td>NC_000001.10:g.(?_154141771)_(154148734_?)del</td>\n",
       "      <td>CopyNumberChange</td>\n",
       "      <td>[]</td>\n",
       "      <td>None</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>[]</td>\n",
       "      <td>1q21.3</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>Success</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>NaN</td>\n",
       "      <td>1457138</td>\n",
       "      <td>NC_000001.10:g.(?_156084710)_(156085085_?)del</td>\n",
       "      <td>CopyNumberChange</td>\n",
       "      <td>[]</td>\n",
       "      <td>None</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>[]</td>\n",
       "      <td>1q22</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>Success</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>NaN</td>\n",
       "      <td>3247637</td>\n",
       "      <td>NC_000001.10:g.(?_156830727)_(156836790_?)del</td>\n",
       "      <td>CopyNumberChange</td>\n",
       "      <td>[]</td>\n",
       "      <td>None</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>[]</td>\n",
       "      <td>1q23.1</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>Success</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3743688</th>\n",
       "      <td>NaN</td>\n",
       "      <td>693181</td>\n",
       "      <td>NC_012920.1(MT-CO3):m.9549C&gt;T</td>\n",
       "      <td>Allele</td>\n",
       "      <td>[]</td>\n",
       "      <td>None</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>[{'system': 'dbSNP', 'code': '693181', 'relati...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>LiteralSequenceExpression</td>\n",
       "      <td>NaN</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>Success</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3743689</th>\n",
       "      <td>NaN</td>\n",
       "      <td>3911672</td>\n",
       "      <td>NC_012920.1(MT-CO3):m.9698T&gt;C</td>\n",
       "      <td>Allele</td>\n",
       "      <td>[]</td>\n",
       "      <td>None</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>[]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>LiteralSequenceExpression</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>Success</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3743690</th>\n",
       "      <td>NaN</td>\n",
       "      <td>693215</td>\n",
       "      <td>NC_012920.1(MT-CO3):m.9751T&gt;C</td>\n",
       "      <td>Allele</td>\n",
       "      <td>[]</td>\n",
       "      <td>None</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>[]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>LiteralSequenceExpression</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>Success</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3743691</th>\n",
       "      <td>NaN</td>\n",
       "      <td>693235</td>\n",
       "      <td>NC_012920.1(MT-CO3):m.9828G&gt;A</td>\n",
       "      <td>Allele</td>\n",
       "      <td>[]</td>\n",
       "      <td>None</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>[]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>LiteralSequenceExpression</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>Success</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3743693</th>\n",
       "      <td>NaN</td>\n",
       "      <td>2223234</td>\n",
       "      <td>NM_080615.1:c.1096G&gt;A</td>\n",
       "      <td>Allele</td>\n",
       "      <td>[]</td>\n",
       "      <td>None</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>[]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>LiteralSequenceExpression</td>\n",
       "      <td>NaN</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>Success</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3731335 rows × 40 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         out in.variation_id                                        in.name  \\\n",
       "25       NaN         3247925  NC_000001.10:g.(?_103385847)_(103496820_?)del   \n",
       "26       NaN          832762  NC_000001.11:g.(?_119415410)_(119422630_?)del   \n",
       "27       NaN         3247694  NC_000001.10:g.(?_154141771)_(154148734_?)del   \n",
       "28       NaN         1457138  NC_000001.10:g.(?_156084710)_(156085085_?)del   \n",
       "29       NaN         3247637  NC_000001.10:g.(?_156830727)_(156836790_?)del   \n",
       "...      ...             ...                                            ...   \n",
       "3743688  NaN          693181                  NC_012920.1(MT-CO3):m.9549C>T   \n",
       "3743689  NaN         3911672                  NC_012920.1(MT-CO3):m.9698T>C   \n",
       "3743690  NaN          693215                  NC_012920.1(MT-CO3):m.9751T>C   \n",
       "3743691  NaN          693235                  NC_012920.1(MT-CO3):m.9828G>A   \n",
       "3743693  NaN         2223234                          NM_080615.1:c.1096G>A   \n",
       "\n",
       "             in.vrs_class in.range_copies in.issue          in.variation_type  \\\n",
       "25       CopyNumberChange              []     None                   Deletion   \n",
       "26       CopyNumberChange              []     None                   Deletion   \n",
       "27       CopyNumberChange              []     None                   Deletion   \n",
       "28       CopyNumberChange              []     None                   Deletion   \n",
       "29       CopyNumberChange              []     None                   Deletion   \n",
       "...                   ...             ...      ...                        ...   \n",
       "3743688            Allele              []     None  single nucleotide variant   \n",
       "3743689            Allele              []     None  single nucleotide variant   \n",
       "3743690            Allele              []     None  single nucleotide variant   \n",
       "3743691            Allele              []     None  single nucleotide variant   \n",
       "3743693            Allele              []     None  single nucleotide variant   \n",
       "\n",
       "        in.subclass_type                                        in.mappings  \\\n",
       "25          SimpleAllele                                                 []   \n",
       "26          SimpleAllele                                                 []   \n",
       "27          SimpleAllele                                                 []   \n",
       "28          SimpleAllele                                                 []   \n",
       "29          SimpleAllele                                                 []   \n",
       "...                  ...                                                ...   \n",
       "3743688     SimpleAllele  [{'system': 'dbSNP', 'code': '693181', 'relati...   \n",
       "3743689     SimpleAllele                                                 []   \n",
       "3743690     SimpleAllele                                                 []   \n",
       "3743691     SimpleAllele                                                 []   \n",
       "3743693     SimpleAllele                                                 []   \n",
       "\n",
       "        in.cytogenetic  ...             out.state.type out.state.length  \\\n",
       "25              1p21.1  ...                        NaN              NaN   \n",
       "26                1p12  ...                        NaN              NaN   \n",
       "27              1q21.3  ...                        NaN              NaN   \n",
       "28                1q22  ...                        NaN              NaN   \n",
       "29              1q23.1  ...                        NaN              NaN   \n",
       "...                ...  ...                        ...              ...   \n",
       "3743688            NaN  ...  LiteralSequenceExpression              NaN   \n",
       "3743689            NaN  ...  LiteralSequenceExpression              NaN   \n",
       "3743690            NaN  ...  LiteralSequenceExpression              NaN   \n",
       "3743691            NaN  ...  LiteralSequenceExpression              NaN   \n",
       "3743693            NaN  ...  LiteralSequenceExpression              NaN   \n",
       "\n",
       "        out.state.sequence out.state.repeatSubunitLength out.extensions  \\\n",
       "25                     NaN                           NaN            NaN   \n",
       "26                     NaN                           NaN            NaN   \n",
       "27                     NaN                           NaN            NaN   \n",
       "28                     NaN                           NaN            NaN   \n",
       "29                     NaN                           NaN            NaN   \n",
       "...                    ...                           ...            ...   \n",
       "3743688                  T                           NaN            NaN   \n",
       "3743689                  C                           NaN            NaN   \n",
       "3743690                  C                           NaN            NaN   \n",
       "3743691                  A                           NaN            NaN   \n",
       "3743693                  A                           NaN            NaN   \n",
       "\n",
       "        in.absolute_copies out.copies support_status error_string  \\\n",
       "25                     NaN        NaN           True      Success   \n",
       "26                     NaN        NaN           True      Success   \n",
       "27                     NaN        NaN           True      Success   \n",
       "28                     NaN        NaN           True      Success   \n",
       "29                     NaN        NaN           True      Success   \n",
       "...                    ...        ...            ...          ...   \n",
       "3743688                NaN        NaN           True      Success   \n",
       "3743689                NaN        NaN           True      Success   \n",
       "3743690                NaN        NaN           True      Success   \n",
       "3743691                NaN        NaN           True      Success   \n",
       "3743693                NaN        NaN           True      Success   \n",
       "\n",
       "        normalize_status  \n",
       "25                  True  \n",
       "26                  True  \n",
       "27                  True  \n",
       "28                  True  \n",
       "29                  True  \n",
       "...                  ...  \n",
       "3743688             True  \n",
       "3743689             True  \n",
       "3743690             True  \n",
       "3743691             True  \n",
       "3743693             True  \n",
       "\n",
       "[3731335 rows x 40 columns]"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "supported_df = supported_df.loc[\n",
    "    (supported_df[\"support_status\"] & supported_df[\"normalize_status\"])\n",
    "]\n",
    "supported_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>in.variation_type</th>\n",
       "      <th>in.issue</th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>None</td>\n",
       "      <td>3388754</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Deletion</td>\n",
       "      <td>None</td>\n",
       "      <td>159681</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Duplication</td>\n",
       "      <td>None</td>\n",
       "      <td>73241</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Microsatellite</td>\n",
       "      <td>None</td>\n",
       "      <td>36028</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>copy number gain</td>\n",
       "      <td>None</td>\n",
       "      <td>21484</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>copy number loss</td>\n",
       "      <td>None</td>\n",
       "      <td>20367</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>Indel</td>\n",
       "      <td>None</td>\n",
       "      <td>17018</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>Insertion</td>\n",
       "      <td>None</td>\n",
       "      <td>12982</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>Inversion</td>\n",
       "      <td>None</td>\n",
       "      <td>1401</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>Variation</td>\n",
       "      <td>None</td>\n",
       "      <td>379</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "           in.variation_type in.issue    count\n",
       "0  single nucleotide variant     None  3388754\n",
       "1                   Deletion     None   159681\n",
       "2                Duplication     None    73241\n",
       "3             Microsatellite     None    36028\n",
       "4           copy number gain     None    21484\n",
       "5           copy number loss     None    20367\n",
       "6                      Indel     None    17018\n",
       "7                  Insertion     None    12982\n",
       "8                  Inversion     None     1401\n",
       "9                  Variation     None      379"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "variation_type_count_supported_df = supported_df.value_counts(\n",
    "    [\"in.variation_type\", \"in.issue\"]\n",
    ").reset_index()\n",
    "variation_type_count_supported_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [],
   "source": [
    "variation_type_count_supported_df.to_csv(\n",
    "    \"variation_analysis_output/variation_type_count_supported_df.csv\"\n",
    ")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### <a id='toc1_4_2_'></a>[Supported and Not Normalized Variants](#toc0_)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "supported_not_normalized_df = df.copy()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>out</th>\n",
       "      <th>in.variation_id</th>\n",
       "      <th>in.name</th>\n",
       "      <th>in.vrs_class</th>\n",
       "      <th>in.range_copies</th>\n",
       "      <th>in.issue</th>\n",
       "      <th>in.variation_type</th>\n",
       "      <th>in.subclass_type</th>\n",
       "      <th>in.mappings</th>\n",
       "      <th>in.cytogenetic</th>\n",
       "      <th>...</th>\n",
       "      <th>out.state.type</th>\n",
       "      <th>out.state.length</th>\n",
       "      <th>out.state.sequence</th>\n",
       "      <th>out.state.repeatSubunitLength</th>\n",
       "      <th>out.extensions</th>\n",
       "      <th>in.absolute_copies</th>\n",
       "      <th>out.copies</th>\n",
       "      <th>support_status</th>\n",
       "      <th>error_string</th>\n",
       "      <th>normalize_status</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>NaN</td>\n",
       "      <td>1174529</td>\n",
       "      <td>NC_000001.10:g.(103388956_103400026)_(10409439...</td>\n",
       "      <td>CopyNumberChange</td>\n",
       "      <td>[]</td>\n",
       "      <td>None</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>[]</td>\n",
       "      <td>1p21.1</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>Unable to find classification for: NC_000001.1...</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>558</th>\n",
       "      <td>NaN</td>\n",
       "      <td>254090</td>\n",
       "      <td>NM_000251.2(MSH2):c.-125_1076+?del</td>\n",
       "      <td>CopyNumberChange</td>\n",
       "      <td>[]</td>\n",
       "      <td>None</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>[]</td>\n",
       "      <td>2p21</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>Unable to find classification for: NC_000002.1...</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1147</th>\n",
       "      <td>NaN</td>\n",
       "      <td>987857</td>\n",
       "      <td>NC_000007.13:g.(?_6010555)_(6027252_6029430)del</td>\n",
       "      <td>CopyNumberChange</td>\n",
       "      <td>[]</td>\n",
       "      <td>None</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>[]</td>\n",
       "      <td>7p22.1</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>Unable to find classification for: NC_000007.1...</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2377</th>\n",
       "      <td>NaN</td>\n",
       "      <td>236334</td>\n",
       "      <td>Single allele</td>\n",
       "      <td>CopyNumberChange</td>\n",
       "      <td>[]</td>\n",
       "      <td>None</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>[]</td>\n",
       "      <td>16p13.3-13.2</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>Unable to get GRCh37/GRCh38 assembly for: NC_0...</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2379</th>\n",
       "      <td>NaN</td>\n",
       "      <td>974178</td>\n",
       "      <td>NC_000016.10:g.(89752223_89758576)_(89816657_?...</td>\n",
       "      <td>CopyNumberChange</td>\n",
       "      <td>[]</td>\n",
       "      <td>None</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>[]</td>\n",
       "      <td>16q24.3</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>Unable to find classification for: NC_000016.9...</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3676061</th>\n",
       "      <td>NaN</td>\n",
       "      <td>399136</td>\n",
       "      <td>NCBI36/hg18 Yq11.223(chrY:23282395-23906316)x1</td>\n",
       "      <td>CopyNumberCount</td>\n",
       "      <td>[]</td>\n",
       "      <td>None</td>\n",
       "      <td>copy number loss</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>[]</td>\n",
       "      <td>Yq11.223</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>Unable to get GRCh37/GRCh38 assembly for: NC_0...</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3676062</th>\n",
       "      <td>NaN</td>\n",
       "      <td>401800</td>\n",
       "      <td>NCBI36/hg18 Yq11.223(chrY:24283222-24691810)x1</td>\n",
       "      <td>CopyNumberCount</td>\n",
       "      <td>[]</td>\n",
       "      <td>None</td>\n",
       "      <td>copy number loss</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>[]</td>\n",
       "      <td>Yq11.223</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>Unable to get GRCh37/GRCh38 assembly for: NC_0...</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3676063</th>\n",
       "      <td>NaN</td>\n",
       "      <td>400073</td>\n",
       "      <td>NCBI36/hg18 Yq11.223-11.23(chrY:25252100-25583...</td>\n",
       "      <td>CopyNumberCount</td>\n",
       "      <td>[]</td>\n",
       "      <td>None</td>\n",
       "      <td>copy number loss</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>[]</td>\n",
       "      <td>Yq11.223-11.23</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>Unable to get GRCh37/GRCh38 assembly for: NC_0...</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3676064</th>\n",
       "      <td>NaN</td>\n",
       "      <td>398808</td>\n",
       "      <td>NCBI36/hg18 Yq11.23(chrY:26218479-26497939)x1</td>\n",
       "      <td>CopyNumberCount</td>\n",
       "      <td>[]</td>\n",
       "      <td>None</td>\n",
       "      <td>copy number loss</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>[]</td>\n",
       "      <td>Yq11.23</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>Unable to get GRCh37/GRCh38 assembly for: NC_0...</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3743692</th>\n",
       "      <td>NaN</td>\n",
       "      <td>3049652</td>\n",
       "      <td>NM_001372044.2(SHANK3):c.1111-8C&gt;T</td>\n",
       "      <td>Allele</td>\n",
       "      <td>[]</td>\n",
       "      <td>None</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>[]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>Unable to tokenize: NG_070230.1:g.18747C&gt;T</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>6757 rows × 40 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         out in.variation_id  \\\n",
       "24       NaN         1174529   \n",
       "558      NaN          254090   \n",
       "1147     NaN          987857   \n",
       "2377     NaN          236334   \n",
       "2379     NaN          974178   \n",
       "...      ...             ...   \n",
       "3676061  NaN          399136   \n",
       "3676062  NaN          401800   \n",
       "3676063  NaN          400073   \n",
       "3676064  NaN          398808   \n",
       "3743692  NaN         3049652   \n",
       "\n",
       "                                                   in.name      in.vrs_class  \\\n",
       "24       NC_000001.10:g.(103388956_103400026)_(10409439...  CopyNumberChange   \n",
       "558                     NM_000251.2(MSH2):c.-125_1076+?del  CopyNumberChange   \n",
       "1147       NC_000007.13:g.(?_6010555)_(6027252_6029430)del  CopyNumberChange   \n",
       "2377                                         Single allele  CopyNumberChange   \n",
       "2379     NC_000016.10:g.(89752223_89758576)_(89816657_?...  CopyNumberChange   \n",
       "...                                                    ...               ...   \n",
       "3676061     NCBI36/hg18 Yq11.223(chrY:23282395-23906316)x1   CopyNumberCount   \n",
       "3676062     NCBI36/hg18 Yq11.223(chrY:24283222-24691810)x1   CopyNumberCount   \n",
       "3676063  NCBI36/hg18 Yq11.223-11.23(chrY:25252100-25583...   CopyNumberCount   \n",
       "3676064      NCBI36/hg18 Yq11.23(chrY:26218479-26497939)x1   CopyNumberCount   \n",
       "3743692                 NM_001372044.2(SHANK3):c.1111-8C>T            Allele   \n",
       "\n",
       "        in.range_copies in.issue          in.variation_type in.subclass_type  \\\n",
       "24                   []     None                   Deletion     SimpleAllele   \n",
       "558                  []     None                   Deletion     SimpleAllele   \n",
       "1147                 []     None                   Deletion     SimpleAllele   \n",
       "2377                 []     None                   Deletion     SimpleAllele   \n",
       "2379                 []     None                   Deletion     SimpleAllele   \n",
       "...                 ...      ...                        ...              ...   \n",
       "3676061              []     None           copy number loss     SimpleAllele   \n",
       "3676062              []     None           copy number loss     SimpleAllele   \n",
       "3676063              []     None           copy number loss     SimpleAllele   \n",
       "3676064              []     None           copy number loss     SimpleAllele   \n",
       "3743692              []     None  single nucleotide variant     SimpleAllele   \n",
       "\n",
       "        in.mappings  in.cytogenetic  ... out.state.type out.state.length  \\\n",
       "24               []          1p21.1  ...            NaN              NaN   \n",
       "558              []            2p21  ...            NaN              NaN   \n",
       "1147             []          7p22.1  ...            NaN              NaN   \n",
       "2377             []    16p13.3-13.2  ...            NaN              NaN   \n",
       "2379             []         16q24.3  ...            NaN              NaN   \n",
       "...             ...             ...  ...            ...              ...   \n",
       "3676061          []        Yq11.223  ...            NaN              NaN   \n",
       "3676062          []        Yq11.223  ...            NaN              NaN   \n",
       "3676063          []  Yq11.223-11.23  ...            NaN              NaN   \n",
       "3676064          []         Yq11.23  ...            NaN              NaN   \n",
       "3743692          []             NaN  ...            NaN              NaN   \n",
       "\n",
       "        out.state.sequence out.state.repeatSubunitLength out.extensions  \\\n",
       "24                     NaN                           NaN            NaN   \n",
       "558                    NaN                           NaN            NaN   \n",
       "1147                   NaN                           NaN            NaN   \n",
       "2377                   NaN                           NaN            NaN   \n",
       "2379                   NaN                           NaN            NaN   \n",
       "...                    ...                           ...            ...   \n",
       "3676061                NaN                           NaN            NaN   \n",
       "3676062                NaN                           NaN            NaN   \n",
       "3676063                NaN                           NaN            NaN   \n",
       "3676064                NaN                           NaN            NaN   \n",
       "3743692                NaN                           NaN            NaN   \n",
       "\n",
       "        in.absolute_copies out.copies support_status  \\\n",
       "24                     NaN        NaN           True   \n",
       "558                    NaN        NaN           True   \n",
       "1147                   NaN        NaN           True   \n",
       "2377                   NaN        NaN           True   \n",
       "2379                   NaN        NaN           True   \n",
       "...                    ...        ...            ...   \n",
       "3676061                  1        NaN           True   \n",
       "3676062                  1        NaN           True   \n",
       "3676063                  1        NaN           True   \n",
       "3676064                  1        NaN           True   \n",
       "3743692                NaN        NaN           True   \n",
       "\n",
       "                                              error_string normalize_status  \n",
       "24       Unable to find classification for: NC_000001.1...            False  \n",
       "558      Unable to find classification for: NC_000002.1...            False  \n",
       "1147     Unable to find classification for: NC_000007.1...            False  \n",
       "2377     Unable to get GRCh37/GRCh38 assembly for: NC_0...            False  \n",
       "2379     Unable to find classification for: NC_000016.9...            False  \n",
       "...                                                    ...              ...  \n",
       "3676061  Unable to get GRCh37/GRCh38 assembly for: NC_0...            False  \n",
       "3676062  Unable to get GRCh37/GRCh38 assembly for: NC_0...            False  \n",
       "3676063  Unable to get GRCh37/GRCh38 assembly for: NC_0...            False  \n",
       "3676064  Unable to get GRCh37/GRCh38 assembly for: NC_0...            False  \n",
       "3743692         Unable to tokenize: NG_070230.1:g.18747C>T            False  \n",
       "\n",
       "[6757 rows x 40 columns]"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "supported_not_normalized_df = supported_not_normalized_df.loc[\n",
    "    (\n",
    "        supported_not_normalized_df[\"support_status\"]\n",
    "        & ~supported_not_normalized_df[\"normalize_status\"]\n",
    "    )\n",
    "]\n",
    "supported_not_normalized_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>in.variation_type</th>\n",
       "      <th>in.issue</th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>copy number gain</td>\n",
       "      <td>None</td>\n",
       "      <td>3245</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>copy number loss</td>\n",
       "      <td>None</td>\n",
       "      <td>2243</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Deletion</td>\n",
       "      <td>None</td>\n",
       "      <td>779</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Duplication</td>\n",
       "      <td>None</td>\n",
       "      <td>407</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>None</td>\n",
       "      <td>80</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>Indel</td>\n",
       "      <td>None</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>Insertion</td>\n",
       "      <td>None</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "           in.variation_type in.issue  count\n",
       "0           copy number gain     None   3245\n",
       "1           copy number loss     None   2243\n",
       "2                   Deletion     None    779\n",
       "3                Duplication     None    407\n",
       "4  single nucleotide variant     None     80\n",
       "5                      Indel     None      2\n",
       "6                  Insertion     None      1"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "variation_type_count_supported_not_normalized_df = (\n",
    "    supported_not_normalized_df.value_counts(\n",
    "        [\"in.variation_type\", \"in.issue\"]\n",
    "    ).reset_index()\n",
    ")\n",
    "variation_type_count_supported_not_normalized_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [],
   "source": [
    "variation_type_count_supported_not_normalized_df.to_csv(\n",
    "    \"variation_analysis_output/variation_type_count_supported_not_normalized_df.csv\"\n",
    ")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### <a id='toc1_4_3_'></a>[Not Supported Variants](#toc0_)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [],
   "source": [
    "not_supported_df = df.copy()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>out</th>\n",
       "      <th>in.variation_id</th>\n",
       "      <th>in.name</th>\n",
       "      <th>in.vrs_class</th>\n",
       "      <th>in.range_copies</th>\n",
       "      <th>in.issue</th>\n",
       "      <th>in.variation_type</th>\n",
       "      <th>in.subclass_type</th>\n",
       "      <th>in.mappings</th>\n",
       "      <th>in.cytogenetic</th>\n",
       "      <th>...</th>\n",
       "      <th>out.state.type</th>\n",
       "      <th>out.state.length</th>\n",
       "      <th>out.state.sequence</th>\n",
       "      <th>out.state.repeatSubunitLength</th>\n",
       "      <th>out.extensions</th>\n",
       "      <th>in.absolute_copies</th>\n",
       "      <th>out.copies</th>\n",
       "      <th>support_status</th>\n",
       "      <th>error_string</th>\n",
       "      <th>normalize_status</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NaN</td>\n",
       "      <td>427832</td>\n",
       "      <td>NM_000785.3(CYP27B1):c.[1319_1325dupCCCACCC];[...</td>\n",
       "      <td>Not Available</td>\n",
       "      <td>[]</td>\n",
       "      <td>haplotype and genotype variations are not supp...</td>\n",
       "      <td>CompoundHeterozygote</td>\n",
       "      <td>Genotype</td>\n",
       "      <td>[{'system': 'ClinVar', 'code': '427832', 'rela...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NaN</td>\n",
       "      <td>424704</td>\n",
       "      <td>NM_003977.3(AIP):c.[-125-145_-125-144delCGinsA...</td>\n",
       "      <td>Not Available</td>\n",
       "      <td>[]</td>\n",
       "      <td>haplotype and genotype variations are not supp...</td>\n",
       "      <td>CompoundHeterozygote</td>\n",
       "      <td>Genotype</td>\n",
       "      <td>[{'system': 'ClinVar', 'code': '424704', 'rela...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NaN</td>\n",
       "      <td>982544</td>\n",
       "      <td>NM_000329.3(RPE65):c.[1067dup];[1543C&gt;T]</td>\n",
       "      <td>Not Available</td>\n",
       "      <td>[]</td>\n",
       "      <td>haplotype and genotype variations are not supp...</td>\n",
       "      <td>CompoundHeterozygote</td>\n",
       "      <td>Genotype</td>\n",
       "      <td>[{'system': 'ClinVar', 'code': '982544', 'rela...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NaN</td>\n",
       "      <td>424730</td>\n",
       "      <td>NM_000372.4(TYR):c.[1276_1282delATGGTTC];[139G&gt;A]</td>\n",
       "      <td>Not Available</td>\n",
       "      <td>[]</td>\n",
       "      <td>haplotype and genotype variations are not supp...</td>\n",
       "      <td>CompoundHeterozygote</td>\n",
       "      <td>Genotype</td>\n",
       "      <td>[{'system': 'ClinVar', 'code': '424730', 'rela...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NaN</td>\n",
       "      <td>424736</td>\n",
       "      <td>NM_005045.3(RELN):c.[2213G&gt;A];[9427T&gt;G]</td>\n",
       "      <td>Not Available</td>\n",
       "      <td>[]</td>\n",
       "      <td>haplotype and genotype variations are not supp...</td>\n",
       "      <td>CompoundHeterozygote</td>\n",
       "      <td>Genotype</td>\n",
       "      <td>[{'system': 'ClinVar', 'code': '424736', 'rela...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3676071</th>\n",
       "      <td>NaN</td>\n",
       "      <td>3778638</td>\n",
       "      <td>MON1A, ARG249CYS (rs555274206)</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>[]</td>\n",
       "      <td>No viable variation members identified.</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>[]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3676072</th>\n",
       "      <td>NaN</td>\n",
       "      <td>10563</td>\n",
       "      <td>F9, IVS1, 192A-G</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>[]</td>\n",
       "      <td>No viable variation members identified.</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>[]</td>\n",
       "      <td>Xq27.1-q27.2</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3676073</th>\n",
       "      <td>NaN</td>\n",
       "      <td>3901867</td>\n",
       "      <td>ATP2A2, GLU373LYS</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>[]</td>\n",
       "      <td>No viable variation members identified.</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>[]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3676074</th>\n",
       "      <td>NaN</td>\n",
       "      <td>3901246</td>\n",
       "      <td>CST3, IVS2DS, G-T, +1</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>[]</td>\n",
       "      <td>No viable variation members identified.</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>[]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3676075</th>\n",
       "      <td>NaN</td>\n",
       "      <td>3381198</td>\n",
       "      <td>ACVRL1, ARG67TRP</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>[]</td>\n",
       "      <td>No viable variation members identified.</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>[]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5602 rows × 40 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         out in.variation_id  \\\n",
       "0        NaN          427832   \n",
       "1        NaN          424704   \n",
       "2        NaN          982544   \n",
       "3        NaN          424730   \n",
       "4        NaN          424736   \n",
       "...      ...             ...   \n",
       "3676071  NaN         3778638   \n",
       "3676072  NaN           10563   \n",
       "3676073  NaN         3901867   \n",
       "3676074  NaN         3901246   \n",
       "3676075  NaN         3381198   \n",
       "\n",
       "                                                   in.name   in.vrs_class  \\\n",
       "0        NM_000785.3(CYP27B1):c.[1319_1325dupCCCACCC];[...  Not Available   \n",
       "1        NM_003977.3(AIP):c.[-125-145_-125-144delCGinsA...  Not Available   \n",
       "2                 NM_000329.3(RPE65):c.[1067dup];[1543C>T]  Not Available   \n",
       "3        NM_000372.4(TYR):c.[1276_1282delATGGTTC];[139G>A]  Not Available   \n",
       "4                  NM_005045.3(RELN):c.[2213G>A];[9427T>G]  Not Available   \n",
       "...                                                    ...            ...   \n",
       "3676071                     MON1A, ARG249CYS (rs555274206)        Unknown   \n",
       "3676072                                   F9, IVS1, 192A-G        Unknown   \n",
       "3676073                                  ATP2A2, GLU373LYS        Unknown   \n",
       "3676074                              CST3, IVS2DS, G-T, +1        Unknown   \n",
       "3676075                                   ACVRL1, ARG67TRP        Unknown   \n",
       "\n",
       "        in.range_copies                                           in.issue  \\\n",
       "0                    []  haplotype and genotype variations are not supp...   \n",
       "1                    []  haplotype and genotype variations are not supp...   \n",
       "2                    []  haplotype and genotype variations are not supp...   \n",
       "3                    []  haplotype and genotype variations are not supp...   \n",
       "4                    []  haplotype and genotype variations are not supp...   \n",
       "...                 ...                                                ...   \n",
       "3676071              []            No viable variation members identified.   \n",
       "3676072              []            No viable variation members identified.   \n",
       "3676073              []            No viable variation members identified.   \n",
       "3676074              []            No viable variation members identified.   \n",
       "3676075              []            No viable variation members identified.   \n",
       "\n",
       "                 in.variation_type in.subclass_type  \\\n",
       "0             CompoundHeterozygote         Genotype   \n",
       "1             CompoundHeterozygote         Genotype   \n",
       "2             CompoundHeterozygote         Genotype   \n",
       "3             CompoundHeterozygote         Genotype   \n",
       "4             CompoundHeterozygote         Genotype   \n",
       "...                            ...              ...   \n",
       "3676071  single nucleotide variant     SimpleAllele   \n",
       "3676072  single nucleotide variant     SimpleAllele   \n",
       "3676073  single nucleotide variant     SimpleAllele   \n",
       "3676074  single nucleotide variant     SimpleAllele   \n",
       "3676075  single nucleotide variant     SimpleAllele   \n",
       "\n",
       "                                               in.mappings in.cytogenetic  \\\n",
       "0        [{'system': 'ClinVar', 'code': '427832', 'rela...            NaN   \n",
       "1        [{'system': 'ClinVar', 'code': '424704', 'rela...            NaN   \n",
       "2        [{'system': 'ClinVar', 'code': '982544', 'rela...            NaN   \n",
       "3        [{'system': 'ClinVar', 'code': '424730', 'rela...            NaN   \n",
       "4        [{'system': 'ClinVar', 'code': '424736', 'rela...            NaN   \n",
       "...                                                    ...            ...   \n",
       "3676071                                                 []            NaN   \n",
       "3676072                                                 []   Xq27.1-q27.2   \n",
       "3676073                                                 []            NaN   \n",
       "3676074                                                 []            NaN   \n",
       "3676075                                                 []            NaN   \n",
       "\n",
       "         ... out.state.type out.state.length out.state.sequence  \\\n",
       "0        ...            NaN              NaN                NaN   \n",
       "1        ...            NaN              NaN                NaN   \n",
       "2        ...            NaN              NaN                NaN   \n",
       "3        ...            NaN              NaN                NaN   \n",
       "4        ...            NaN              NaN                NaN   \n",
       "...      ...            ...              ...                ...   \n",
       "3676071  ...            NaN              NaN                NaN   \n",
       "3676072  ...            NaN              NaN                NaN   \n",
       "3676073  ...            NaN              NaN                NaN   \n",
       "3676074  ...            NaN              NaN                NaN   \n",
       "3676075  ...            NaN              NaN                NaN   \n",
       "\n",
       "        out.state.repeatSubunitLength out.extensions in.absolute_copies  \\\n",
       "0                                 NaN            NaN                NaN   \n",
       "1                                 NaN            NaN                NaN   \n",
       "2                                 NaN            NaN                NaN   \n",
       "3                                 NaN            NaN                NaN   \n",
       "4                                 NaN            NaN                NaN   \n",
       "...                               ...            ...                ...   \n",
       "3676071                           NaN            NaN                NaN   \n",
       "3676072                           NaN            NaN                NaN   \n",
       "3676073                           NaN            NaN                NaN   \n",
       "3676074                           NaN            NaN                NaN   \n",
       "3676075                           NaN            NaN                NaN   \n",
       "\n",
       "        out.copies support_status   error_string normalize_status  \n",
       "0              NaN          False  Not Supported            False  \n",
       "1              NaN          False  Not Supported            False  \n",
       "2              NaN          False  Not Supported            False  \n",
       "3              NaN          False  Not Supported            False  \n",
       "4              NaN          False  Not Supported            False  \n",
       "...            ...            ...            ...              ...  \n",
       "3676071        NaN          False  Not Supported            False  \n",
       "3676072        NaN          False  Not Supported            False  \n",
       "3676073        NaN          False  Not Supported            False  \n",
       "3676074        NaN          False  Not Supported            False  \n",
       "3676075        NaN          False  Not Supported            False  \n",
       "\n",
       "[5602 rows x 40 columns]"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "not_supported_df = not_supported_df.loc[\n",
    "    ~not_supported_df[\"support_status\"] & ~not_supported_df[\"normalize_status\"]\n",
    "]\n",
    "not_supported_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>in.variation_type</th>\n",
       "      <th>in.issue</th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Haplotype</td>\n",
       "      <td>haplotype and genotype variations are not supp...</td>\n",
       "      <td>617</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Diplotype</td>\n",
       "      <td>haplotype and genotype variations are not supp...</td>\n",
       "      <td>596</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Deletion</td>\n",
       "      <td>No viable variation members identified.</td>\n",
       "      <td>593</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Microsatellite</td>\n",
       "      <td>repeat expressions are not supported.</td>\n",
       "      <td>456</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Deletion</td>\n",
       "      <td>sequence for accession not supported by vrs-py...</td>\n",
       "      <td>336</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>CompoundHeterozygote</td>\n",
       "      <td>haplotype and genotype variations are not supp...</td>\n",
       "      <td>297</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>No viable variation members identified.</td>\n",
       "      <td>290</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>Translocation</td>\n",
       "      <td>No viable variation members identified.</td>\n",
       "      <td>282</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>Insertion</td>\n",
       "      <td>No viable variation members identified.</td>\n",
       "      <td>264</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>Deletion</td>\n",
       "      <td>intronic positions are not resolvable in seque...</td>\n",
       "      <td>220</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>Deletion</td>\n",
       "      <td>unsupported hgvs expression.</td>\n",
       "      <td>193</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>Duplication</td>\n",
       "      <td>sequence for accession not supported by vrs-py...</td>\n",
       "      <td>157</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>Variation</td>\n",
       "      <td>No viable variation members identified.</td>\n",
       "      <td>156</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>Insertion</td>\n",
       "      <td>sequence for accession not supported by vrs-py...</td>\n",
       "      <td>137</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>Duplication</td>\n",
       "      <td>No viable variation members identified.</td>\n",
       "      <td>116</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>Indel</td>\n",
       "      <td>No viable variation members identified.</td>\n",
       "      <td>102</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>protein only</td>\n",
       "      <td>No viable variation members identified.</td>\n",
       "      <td>93</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>Complex</td>\n",
       "      <td>No viable variation members identified.</td>\n",
       "      <td>86</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>copy number gain</td>\n",
       "      <td>range copies are not supported.</td>\n",
       "      <td>76</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>Insertion</td>\n",
       "      <td>unsupported hgvs expression.</td>\n",
       "      <td>69</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>Inversion</td>\n",
       "      <td>No viable variation members identified.</td>\n",
       "      <td>69</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>Microsatellite</td>\n",
       "      <td>unsupported hgvs expression.</td>\n",
       "      <td>67</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>Indel</td>\n",
       "      <td>unsupported hgvs expression.</td>\n",
       "      <td>43</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>Duplication</td>\n",
       "      <td>unsupported hgvs expression.</td>\n",
       "      <td>42</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>Microsatellite</td>\n",
       "      <td>No viable variation members identified.</td>\n",
       "      <td>33</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>Duplication</td>\n",
       "      <td>intronic positions are not resolvable in seque...</td>\n",
       "      <td>33</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>Inversion</td>\n",
       "      <td>unsupported hgvs expression.</td>\n",
       "      <td>32</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>Indel</td>\n",
       "      <td>sequence for accession not supported by vrs-py...</td>\n",
       "      <td>22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>Haplotype, single variant</td>\n",
       "      <td>haplotype and genotype variations are not supp...</td>\n",
       "      <td>21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>copy number loss</td>\n",
       "      <td>range copies are not supported.</td>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>Microsatellite</td>\n",
       "      <td>sequence for accession not supported by vrs-py...</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>Phase unknown</td>\n",
       "      <td>haplotype and genotype variations are not supp...</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>Insertion</td>\n",
       "      <td>intronic positions are not resolvable in seque...</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>Indel</td>\n",
       "      <td>intronic positions are not resolvable in seque...</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>Variation</td>\n",
       "      <td>unsupported hgvs expression.</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>Insertion</td>\n",
       "      <td>repeat expressions are not supported.</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>fusion</td>\n",
       "      <td>No viable variation members identified.</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>Deletion</td>\n",
       "      <td>repeat expressions are not supported.</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>Variation</td>\n",
       "      <td>sequence for accession not supported by vrs-py...</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>Variation</td>\n",
       "      <td>repeat expressions are not supported.</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>Deletion</td>\n",
       "      <td>expression contains unbalaned paretheses.</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>copy number gain</td>\n",
       "      <td>No viable variation members identified.</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>intronic positions are not resolvable in seque...</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>Complex</td>\n",
       "      <td>unsupported hgvs expression.</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>unsupported hgvs expression.</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>Variation</td>\n",
       "      <td>intronic positions are not resolvable in seque...</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>Duplication</td>\n",
       "      <td>repeat expressions are not supported.</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>Complex</td>\n",
       "      <td>sequence for accession not supported by vrs-py...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>Indel</td>\n",
       "      <td>repeat expressions are not supported.</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>copy number gain</td>\n",
       "      <td>repeat expressions are not supported.</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50</th>\n",
       "      <td>Translocation</td>\n",
       "      <td>unsupported hgvs expression.</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51</th>\n",
       "      <td>Distinct chromosomes</td>\n",
       "      <td>haplotype and genotype variations are not supp...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52</th>\n",
       "      <td>Tandem duplication</td>\n",
       "      <td>No viable variation members identified.</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53</th>\n",
       "      <td>Microsatellite</td>\n",
       "      <td>intronic positions are not resolvable in seque...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            in.variation_type  \\\n",
       "0                   Haplotype   \n",
       "1                   Diplotype   \n",
       "2                    Deletion   \n",
       "3              Microsatellite   \n",
       "4                    Deletion   \n",
       "5        CompoundHeterozygote   \n",
       "6   single nucleotide variant   \n",
       "7               Translocation   \n",
       "8                   Insertion   \n",
       "9                    Deletion   \n",
       "10                   Deletion   \n",
       "11                Duplication   \n",
       "12                  Variation   \n",
       "13                  Insertion   \n",
       "14                Duplication   \n",
       "15                      Indel   \n",
       "16               protein only   \n",
       "17                    Complex   \n",
       "18           copy number gain   \n",
       "19                  Insertion   \n",
       "20                  Inversion   \n",
       "21             Microsatellite   \n",
       "22                      Indel   \n",
       "23                Duplication   \n",
       "24             Microsatellite   \n",
       "25                Duplication   \n",
       "26                  Inversion   \n",
       "27                      Indel   \n",
       "28  Haplotype, single variant   \n",
       "29           copy number loss   \n",
       "30             Microsatellite   \n",
       "31              Phase unknown   \n",
       "32                  Insertion   \n",
       "33                      Indel   \n",
       "34                  Variation   \n",
       "35                  Insertion   \n",
       "36                     fusion   \n",
       "37                   Deletion   \n",
       "38                  Variation   \n",
       "39                  Variation   \n",
       "40                   Deletion   \n",
       "41           copy number gain   \n",
       "42  single nucleotide variant   \n",
       "43                    Complex   \n",
       "44  single nucleotide variant   \n",
       "45                  Variation   \n",
       "46                Duplication   \n",
       "47                    Complex   \n",
       "48                      Indel   \n",
       "49           copy number gain   \n",
       "50              Translocation   \n",
       "51       Distinct chromosomes   \n",
       "52         Tandem duplication   \n",
       "53             Microsatellite   \n",
       "\n",
       "                                             in.issue  count  \n",
       "0   haplotype and genotype variations are not supp...    617  \n",
       "1   haplotype and genotype variations are not supp...    596  \n",
       "2             No viable variation members identified.    593  \n",
       "3               repeat expressions are not supported.    456  \n",
       "4   sequence for accession not supported by vrs-py...    336  \n",
       "5   haplotype and genotype variations are not supp...    297  \n",
       "6             No viable variation members identified.    290  \n",
       "7             No viable variation members identified.    282  \n",
       "8             No viable variation members identified.    264  \n",
       "9   intronic positions are not resolvable in seque...    220  \n",
       "10                       unsupported hgvs expression.    193  \n",
       "11  sequence for accession not supported by vrs-py...    157  \n",
       "12            No viable variation members identified.    156  \n",
       "13  sequence for accession not supported by vrs-py...    137  \n",
       "14            No viable variation members identified.    116  \n",
       "15            No viable variation members identified.    102  \n",
       "16            No viable variation members identified.     93  \n",
       "17            No viable variation members identified.     86  \n",
       "18                    range copies are not supported.     76  \n",
       "19                       unsupported hgvs expression.     69  \n",
       "20            No viable variation members identified.     69  \n",
       "21                       unsupported hgvs expression.     67  \n",
       "22                       unsupported hgvs expression.     43  \n",
       "23                       unsupported hgvs expression.     42  \n",
       "24            No viable variation members identified.     33  \n",
       "25  intronic positions are not resolvable in seque...     33  \n",
       "26                       unsupported hgvs expression.     32  \n",
       "27  sequence for accession not supported by vrs-py...     22  \n",
       "28  haplotype and genotype variations are not supp...     21  \n",
       "29                    range copies are not supported.     19  \n",
       "30  sequence for accession not supported by vrs-py...     11  \n",
       "31  haplotype and genotype variations are not supp...      8  \n",
       "32  intronic positions are not resolvable in seque...      7  \n",
       "33  intronic positions are not resolvable in seque...      7  \n",
       "34                       unsupported hgvs expression.      6  \n",
       "35              repeat expressions are not supported.      5  \n",
       "36            No viable variation members identified.      5  \n",
       "37              repeat expressions are not supported.      5  \n",
       "38  sequence for accession not supported by vrs-py...      4  \n",
       "39              repeat expressions are not supported.      4  \n",
       "40          expression contains unbalaned paretheses.      3  \n",
       "41            No viable variation members identified.      3  \n",
       "42  intronic positions are not resolvable in seque...      2  \n",
       "43                       unsupported hgvs expression.      2  \n",
       "44                       unsupported hgvs expression.      2  \n",
       "45  intronic positions are not resolvable in seque...      2  \n",
       "46              repeat expressions are not supported.      2  \n",
       "47  sequence for accession not supported by vrs-py...      1  \n",
       "48              repeat expressions are not supported.      1  \n",
       "49              repeat expressions are not supported.      1  \n",
       "50                       unsupported hgvs expression.      1  \n",
       "51  haplotype and genotype variations are not supp...      1  \n",
       "52            No viable variation members identified.      1  \n",
       "53  intronic positions are not resolvable in seque...      1  "
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "variation_type_count_not_supported_df = not_supported_df.value_counts(\n",
    "    [\"in.variation_type\", \"in.issue\"]\n",
    ").reset_index()\n",
    "variation_type_count_not_supported_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [],
   "source": [
    "variation_type_count_not_supported_df.to_csv(\n",
    "    \"variation_analysis_output/variation_type_count_not_supported_df.csv\"\n",
    ")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Sanity check: making sure there are no supported variants that have been marked as normalized"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [],
   "source": [
    "not_supported_but_normalized_df = df.copy()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>out</th>\n",
       "      <th>in.variation_id</th>\n",
       "      <th>in.name</th>\n",
       "      <th>in.vrs_class</th>\n",
       "      <th>in.range_copies</th>\n",
       "      <th>in.issue</th>\n",
       "      <th>in.variation_type</th>\n",
       "      <th>in.subclass_type</th>\n",
       "      <th>in.mappings</th>\n",
       "      <th>in.cytogenetic</th>\n",
       "      <th>...</th>\n",
       "      <th>out.state.type</th>\n",
       "      <th>out.state.length</th>\n",
       "      <th>out.state.sequence</th>\n",
       "      <th>out.state.repeatSubunitLength</th>\n",
       "      <th>out.extensions</th>\n",
       "      <th>in.absolute_copies</th>\n",
       "      <th>out.copies</th>\n",
       "      <th>support_status</th>\n",
       "      <th>error_string</th>\n",
       "      <th>normalize_status</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>0 rows × 40 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [out, in.variation_id, in.name, in.vrs_class, in.range_copies, in.issue, in.variation_type, in.subclass_type, in.mappings, in.cytogenetic, in.accession, in.fmt, in.source, in.precedence, in.copy_change_type, in.assembly_version, in.chr, in.variant_length, out.errors, out.id, out.type, out.digest, out.location.id, out.location.type, out.location.digest, out.location.sequenceReference.type, out.location.sequenceReference.refgetAccession, out.location.start, out.location.end, out.copyChange, out.state.type, out.state.length, out.state.sequence, out.state.repeatSubunitLength, out.extensions, in.absolute_copies, out.copies, support_status, error_string, normalize_status]\n",
       "Index: []\n",
       "\n",
       "[0 rows x 40 columns]"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "not_supported_but_normalized_df = not_supported_but_normalized_df.loc[\n",
    "    (\n",
    "        ~not_supported_but_normalized_df[\"support_status\"]\n",
    "        & not_supported_but_normalized_df[\"normalize_status\"]\n",
    "    )\n",
    "]\n",
    "not_supported_but_normalized_df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## <a id='toc1_5_'></a>[Counting variants from each group](#toc0_)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [],
   "source": [
    "num_supported = len(supported_df)\n",
    "num_supported_not_normalized = len(supported_not_normalized_df)\n",
    "num_not_supported_but_normalized = len(not_supported_but_normalized_df)\n",
    "num_not_supported = len(not_supported_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [],
   "source": [
    "summary_df2 = pd.DataFrame(\n",
    "    {\n",
    "        \"Supported\": [num_supported, num_supported_not_normalized],\n",
    "        \"Not Supported\": [num_not_supported_but_normalized, num_not_supported],\n",
    "    }\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Supported</th>\n",
       "      <th>Not Supported</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Normalized</th>\n",
       "      <td>3731335</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Not Normalized</th>\n",
       "      <td>6757</td>\n",
       "      <td>5602</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                Supported  Not Supported\n",
       "Normalized        3731335              0\n",
       "Not Normalized       6757           5602"
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "summary_df2.index = [\"Normalized\", \"Not Normalized\"]\n",
    "summary_df2"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## <a id='toc1_6_'></a>[Counting variant types for each group](#toc0_)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [],
   "source": [
    "variation_type_count_summary_df = pd.merge(\n",
    "    pd.merge(\n",
    "        variation_type_count_supported_df,\n",
    "        variation_type_count_supported_not_normalized_df,\n",
    "        on=\"in.variation_type\",\n",
    "        how=\"left\",\n",
    "    ),\n",
    "    variation_type_count_not_supported_df,\n",
    "    on=\"in.variation_type\",\n",
    "    how=\"right\",\n",
    ")\n",
    "variation_type_count_summary_df = variation_type_count_summary_df.replace(\n",
    "    np.nan, 0, regex=True\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [],
   "source": [
    "variation_type_count_summary_df = variation_type_count_summary_df.rename(\n",
    "    columns={\n",
    "        \"in.id_x\": \"supported\",\n",
    "        \"in.id_y\": \"supported_not_normalized\",\n",
    "        \"in.id\": \"not_supported\",\n",
    "    }\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>in.variation_type</th>\n",
       "      <th>in.issue_x</th>\n",
       "      <th>count_x</th>\n",
       "      <th>in.issue_y</th>\n",
       "      <th>count_y</th>\n",
       "      <th>in.issue</th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Haplotype</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>haplotype and genotype variations are not supp...</td>\n",
       "      <td>617</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Diplotype</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>haplotype and genotype variations are not supp...</td>\n",
       "      <td>596</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Deletion</td>\n",
       "      <td>None</td>\n",
       "      <td>159681.0</td>\n",
       "      <td>None</td>\n",
       "      <td>779.0</td>\n",
       "      <td>No viable variation members identified.</td>\n",
       "      <td>593</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Microsatellite</td>\n",
       "      <td>None</td>\n",
       "      <td>36028.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>repeat expressions are not supported.</td>\n",
       "      <td>456</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Deletion</td>\n",
       "      <td>None</td>\n",
       "      <td>159681.0</td>\n",
       "      <td>None</td>\n",
       "      <td>779.0</td>\n",
       "      <td>sequence for accession not supported by vrs-py...</td>\n",
       "      <td>336</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>CompoundHeterozygote</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>haplotype and genotype variations are not supp...</td>\n",
       "      <td>297</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>None</td>\n",
       "      <td>3388754.0</td>\n",
       "      <td>None</td>\n",
       "      <td>80.0</td>\n",
       "      <td>No viable variation members identified.</td>\n",
       "      <td>290</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>Translocation</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>No viable variation members identified.</td>\n",
       "      <td>282</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>Insertion</td>\n",
       "      <td>None</td>\n",
       "      <td>12982.0</td>\n",
       "      <td>None</td>\n",
       "      <td>1.0</td>\n",
       "      <td>No viable variation members identified.</td>\n",
       "      <td>264</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>Deletion</td>\n",
       "      <td>None</td>\n",
       "      <td>159681.0</td>\n",
       "      <td>None</td>\n",
       "      <td>779.0</td>\n",
       "      <td>intronic positions are not resolvable in seque...</td>\n",
       "      <td>220</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>Deletion</td>\n",
       "      <td>None</td>\n",
       "      <td>159681.0</td>\n",
       "      <td>None</td>\n",
       "      <td>779.0</td>\n",
       "      <td>unsupported hgvs expression.</td>\n",
       "      <td>193</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>Duplication</td>\n",
       "      <td>None</td>\n",
       "      <td>73241.0</td>\n",
       "      <td>None</td>\n",
       "      <td>407.0</td>\n",
       "      <td>sequence for accession not supported by vrs-py...</td>\n",
       "      <td>157</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>Variation</td>\n",
       "      <td>None</td>\n",
       "      <td>379.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>No viable variation members identified.</td>\n",
       "      <td>156</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>Insertion</td>\n",
       "      <td>None</td>\n",
       "      <td>12982.0</td>\n",
       "      <td>None</td>\n",
       "      <td>1.0</td>\n",
       "      <td>sequence for accession not supported by vrs-py...</td>\n",
       "      <td>137</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>Duplication</td>\n",
       "      <td>None</td>\n",
       "      <td>73241.0</td>\n",
       "      <td>None</td>\n",
       "      <td>407.0</td>\n",
       "      <td>No viable variation members identified.</td>\n",
       "      <td>116</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>Indel</td>\n",
       "      <td>None</td>\n",
       "      <td>17018.0</td>\n",
       "      <td>None</td>\n",
       "      <td>2.0</td>\n",
       "      <td>No viable variation members identified.</td>\n",
       "      <td>102</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>protein only</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>No viable variation members identified.</td>\n",
       "      <td>93</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>Complex</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>No viable variation members identified.</td>\n",
       "      <td>86</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>copy number gain</td>\n",
       "      <td>None</td>\n",
       "      <td>21484.0</td>\n",
       "      <td>None</td>\n",
       "      <td>3245.0</td>\n",
       "      <td>range copies are not supported.</td>\n",
       "      <td>76</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>Insertion</td>\n",
       "      <td>None</td>\n",
       "      <td>12982.0</td>\n",
       "      <td>None</td>\n",
       "      <td>1.0</td>\n",
       "      <td>unsupported hgvs expression.</td>\n",
       "      <td>69</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>Inversion</td>\n",
       "      <td>None</td>\n",
       "      <td>1401.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>No viable variation members identified.</td>\n",
       "      <td>69</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>Microsatellite</td>\n",
       "      <td>None</td>\n",
       "      <td>36028.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>unsupported hgvs expression.</td>\n",
       "      <td>67</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>Indel</td>\n",
       "      <td>None</td>\n",
       "      <td>17018.0</td>\n",
       "      <td>None</td>\n",
       "      <td>2.0</td>\n",
       "      <td>unsupported hgvs expression.</td>\n",
       "      <td>43</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>Duplication</td>\n",
       "      <td>None</td>\n",
       "      <td>73241.0</td>\n",
       "      <td>None</td>\n",
       "      <td>407.0</td>\n",
       "      <td>unsupported hgvs expression.</td>\n",
       "      <td>42</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>Microsatellite</td>\n",
       "      <td>None</td>\n",
       "      <td>36028.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>No viable variation members identified.</td>\n",
       "      <td>33</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>Duplication</td>\n",
       "      <td>None</td>\n",
       "      <td>73241.0</td>\n",
       "      <td>None</td>\n",
       "      <td>407.0</td>\n",
       "      <td>intronic positions are not resolvable in seque...</td>\n",
       "      <td>33</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>Inversion</td>\n",
       "      <td>None</td>\n",
       "      <td>1401.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>unsupported hgvs expression.</td>\n",
       "      <td>32</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>Indel</td>\n",
       "      <td>None</td>\n",
       "      <td>17018.0</td>\n",
       "      <td>None</td>\n",
       "      <td>2.0</td>\n",
       "      <td>sequence for accession not supported by vrs-py...</td>\n",
       "      <td>22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>Haplotype, single variant</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>haplotype and genotype variations are not supp...</td>\n",
       "      <td>21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>copy number loss</td>\n",
       "      <td>None</td>\n",
       "      <td>20367.0</td>\n",
       "      <td>None</td>\n",
       "      <td>2243.0</td>\n",
       "      <td>range copies are not supported.</td>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>Microsatellite</td>\n",
       "      <td>None</td>\n",
       "      <td>36028.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>sequence for accession not supported by vrs-py...</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>Phase unknown</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>haplotype and genotype variations are not supp...</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>Insertion</td>\n",
       "      <td>None</td>\n",
       "      <td>12982.0</td>\n",
       "      <td>None</td>\n",
       "      <td>1.0</td>\n",
       "      <td>intronic positions are not resolvable in seque...</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>Indel</td>\n",
       "      <td>None</td>\n",
       "      <td>17018.0</td>\n",
       "      <td>None</td>\n",
       "      <td>2.0</td>\n",
       "      <td>intronic positions are not resolvable in seque...</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>Variation</td>\n",
       "      <td>None</td>\n",
       "      <td>379.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>unsupported hgvs expression.</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>Insertion</td>\n",
       "      <td>None</td>\n",
       "      <td>12982.0</td>\n",
       "      <td>None</td>\n",
       "      <td>1.0</td>\n",
       "      <td>repeat expressions are not supported.</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>fusion</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>No viable variation members identified.</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>Deletion</td>\n",
       "      <td>None</td>\n",
       "      <td>159681.0</td>\n",
       "      <td>None</td>\n",
       "      <td>779.0</td>\n",
       "      <td>repeat expressions are not supported.</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>Variation</td>\n",
       "      <td>None</td>\n",
       "      <td>379.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>sequence for accession not supported by vrs-py...</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>Variation</td>\n",
       "      <td>None</td>\n",
       "      <td>379.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>repeat expressions are not supported.</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>Deletion</td>\n",
       "      <td>None</td>\n",
       "      <td>159681.0</td>\n",
       "      <td>None</td>\n",
       "      <td>779.0</td>\n",
       "      <td>expression contains unbalaned paretheses.</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>copy number gain</td>\n",
       "      <td>None</td>\n",
       "      <td>21484.0</td>\n",
       "      <td>None</td>\n",
       "      <td>3245.0</td>\n",
       "      <td>No viable variation members identified.</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>None</td>\n",
       "      <td>3388754.0</td>\n",
       "      <td>None</td>\n",
       "      <td>80.0</td>\n",
       "      <td>intronic positions are not resolvable in seque...</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>Complex</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>unsupported hgvs expression.</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>None</td>\n",
       "      <td>3388754.0</td>\n",
       "      <td>None</td>\n",
       "      <td>80.0</td>\n",
       "      <td>unsupported hgvs expression.</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>Variation</td>\n",
       "      <td>None</td>\n",
       "      <td>379.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>intronic positions are not resolvable in seque...</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>Duplication</td>\n",
       "      <td>None</td>\n",
       "      <td>73241.0</td>\n",
       "      <td>None</td>\n",
       "      <td>407.0</td>\n",
       "      <td>repeat expressions are not supported.</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>Complex</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>sequence for accession not supported by vrs-py...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>Indel</td>\n",
       "      <td>None</td>\n",
       "      <td>17018.0</td>\n",
       "      <td>None</td>\n",
       "      <td>2.0</td>\n",
       "      <td>repeat expressions are not supported.</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>copy number gain</td>\n",
       "      <td>None</td>\n",
       "      <td>21484.0</td>\n",
       "      <td>None</td>\n",
       "      <td>3245.0</td>\n",
       "      <td>repeat expressions are not supported.</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50</th>\n",
       "      <td>Translocation</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>unsupported hgvs expression.</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51</th>\n",
       "      <td>Distinct chromosomes</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>haplotype and genotype variations are not supp...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52</th>\n",
       "      <td>Tandem duplication</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>No viable variation members identified.</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53</th>\n",
       "      <td>Microsatellite</td>\n",
       "      <td>None</td>\n",
       "      <td>36028.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>intronic positions are not resolvable in seque...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            in.variation_type in.issue_x    count_x in.issue_y  count_y  \\\n",
       "0                   Haplotype          0        0.0          0      0.0   \n",
       "1                   Diplotype          0        0.0          0      0.0   \n",
       "2                    Deletion       None   159681.0       None    779.0   \n",
       "3              Microsatellite       None    36028.0          0      0.0   \n",
       "4                    Deletion       None   159681.0       None    779.0   \n",
       "5        CompoundHeterozygote          0        0.0          0      0.0   \n",
       "6   single nucleotide variant       None  3388754.0       None     80.0   \n",
       "7               Translocation          0        0.0          0      0.0   \n",
       "8                   Insertion       None    12982.0       None      1.0   \n",
       "9                    Deletion       None   159681.0       None    779.0   \n",
       "10                   Deletion       None   159681.0       None    779.0   \n",
       "11                Duplication       None    73241.0       None    407.0   \n",
       "12                  Variation       None      379.0          0      0.0   \n",
       "13                  Insertion       None    12982.0       None      1.0   \n",
       "14                Duplication       None    73241.0       None    407.0   \n",
       "15                      Indel       None    17018.0       None      2.0   \n",
       "16               protein only          0        0.0          0      0.0   \n",
       "17                    Complex          0        0.0          0      0.0   \n",
       "18           copy number gain       None    21484.0       None   3245.0   \n",
       "19                  Insertion       None    12982.0       None      1.0   \n",
       "20                  Inversion       None     1401.0          0      0.0   \n",
       "21             Microsatellite       None    36028.0          0      0.0   \n",
       "22                      Indel       None    17018.0       None      2.0   \n",
       "23                Duplication       None    73241.0       None    407.0   \n",
       "24             Microsatellite       None    36028.0          0      0.0   \n",
       "25                Duplication       None    73241.0       None    407.0   \n",
       "26                  Inversion       None     1401.0          0      0.0   \n",
       "27                      Indel       None    17018.0       None      2.0   \n",
       "28  Haplotype, single variant          0        0.0          0      0.0   \n",
       "29           copy number loss       None    20367.0       None   2243.0   \n",
       "30             Microsatellite       None    36028.0          0      0.0   \n",
       "31              Phase unknown          0        0.0          0      0.0   \n",
       "32                  Insertion       None    12982.0       None      1.0   \n",
       "33                      Indel       None    17018.0       None      2.0   \n",
       "34                  Variation       None      379.0          0      0.0   \n",
       "35                  Insertion       None    12982.0       None      1.0   \n",
       "36                     fusion          0        0.0          0      0.0   \n",
       "37                   Deletion       None   159681.0       None    779.0   \n",
       "38                  Variation       None      379.0          0      0.0   \n",
       "39                  Variation       None      379.0          0      0.0   \n",
       "40                   Deletion       None   159681.0       None    779.0   \n",
       "41           copy number gain       None    21484.0       None   3245.0   \n",
       "42  single nucleotide variant       None  3388754.0       None     80.0   \n",
       "43                    Complex          0        0.0          0      0.0   \n",
       "44  single nucleotide variant       None  3388754.0       None     80.0   \n",
       "45                  Variation       None      379.0          0      0.0   \n",
       "46                Duplication       None    73241.0       None    407.0   \n",
       "47                    Complex          0        0.0          0      0.0   \n",
       "48                      Indel       None    17018.0       None      2.0   \n",
       "49           copy number gain       None    21484.0       None   3245.0   \n",
       "50              Translocation          0        0.0          0      0.0   \n",
       "51       Distinct chromosomes          0        0.0          0      0.0   \n",
       "52         Tandem duplication          0        0.0          0      0.0   \n",
       "53             Microsatellite       None    36028.0          0      0.0   \n",
       "\n",
       "                                             in.issue  count  \n",
       "0   haplotype and genotype variations are not supp...    617  \n",
       "1   haplotype and genotype variations are not supp...    596  \n",
       "2             No viable variation members identified.    593  \n",
       "3               repeat expressions are not supported.    456  \n",
       "4   sequence for accession not supported by vrs-py...    336  \n",
       "5   haplotype and genotype variations are not supp...    297  \n",
       "6             No viable variation members identified.    290  \n",
       "7             No viable variation members identified.    282  \n",
       "8             No viable variation members identified.    264  \n",
       "9   intronic positions are not resolvable in seque...    220  \n",
       "10                       unsupported hgvs expression.    193  \n",
       "11  sequence for accession not supported by vrs-py...    157  \n",
       "12            No viable variation members identified.    156  \n",
       "13  sequence for accession not supported by vrs-py...    137  \n",
       "14            No viable variation members identified.    116  \n",
       "15            No viable variation members identified.    102  \n",
       "16            No viable variation members identified.     93  \n",
       "17            No viable variation members identified.     86  \n",
       "18                    range copies are not supported.     76  \n",
       "19                       unsupported hgvs expression.     69  \n",
       "20            No viable variation members identified.     69  \n",
       "21                       unsupported hgvs expression.     67  \n",
       "22                       unsupported hgvs expression.     43  \n",
       "23                       unsupported hgvs expression.     42  \n",
       "24            No viable variation members identified.     33  \n",
       "25  intronic positions are not resolvable in seque...     33  \n",
       "26                       unsupported hgvs expression.     32  \n",
       "27  sequence for accession not supported by vrs-py...     22  \n",
       "28  haplotype and genotype variations are not supp...     21  \n",
       "29                    range copies are not supported.     19  \n",
       "30  sequence for accession not supported by vrs-py...     11  \n",
       "31  haplotype and genotype variations are not supp...      8  \n",
       "32  intronic positions are not resolvable in seque...      7  \n",
       "33  intronic positions are not resolvable in seque...      7  \n",
       "34                       unsupported hgvs expression.      6  \n",
       "35              repeat expressions are not supported.      5  \n",
       "36            No viable variation members identified.      5  \n",
       "37              repeat expressions are not supported.      5  \n",
       "38  sequence for accession not supported by vrs-py...      4  \n",
       "39              repeat expressions are not supported.      4  \n",
       "40          expression contains unbalaned paretheses.      3  \n",
       "41            No viable variation members identified.      3  \n",
       "42  intronic positions are not resolvable in seque...      2  \n",
       "43                       unsupported hgvs expression.      2  \n",
       "44                       unsupported hgvs expression.      2  \n",
       "45  intronic positions are not resolvable in seque...      2  \n",
       "46              repeat expressions are not supported.      2  \n",
       "47  sequence for accession not supported by vrs-py...      1  \n",
       "48              repeat expressions are not supported.      1  \n",
       "49              repeat expressions are not supported.      1  \n",
       "50                       unsupported hgvs expression.      1  \n",
       "51  haplotype and genotype variations are not supp...      1  \n",
       "52            No viable variation members identified.      1  \n",
       "53  intronic positions are not resolvable in seque...      1  "
      ]
     },
     "execution_count": 39,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "variation_type_count_summary_df.to_csv(\n",
    "    \"variation_analysis_output/variation_type_count_summary_df.csv\"\n",
    ")\n",
    "variation_type_count_summary_df"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "variation-normalizer-manuscript",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.13.5"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
