{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "True"
      ]
     },
     "execution_count": 1,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Load environment variables. This MUST be the first cell.\n",
    "from dotenv import load_dotenv\n",
    "\n",
    "load_dotenv(\"../../.env.shared\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/rsjxa001/variation-normalizer-manuscript/.venv/lib/python3.13/site-packages/variation/main.py:12: UserWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html. The pkg_resources package is slated for removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.\n",
      "  import pkg_resources\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "***Using Gene Database Endpoint: http://localhost:8000***\n"
     ]
    }
   ],
   "source": [
    "import json  # noqa: E402\n",
    "import re  # noqa: E402\n",
    "import logging\n",
    "\n",
    "import pandas as pd  # noqa: E402\n",
    "from tqdm.notebook import tqdm  # noqa: E402\n",
    "from variation.main import parsed_to_cn_var  # noqa: E402\n",
    "from variation.schemas.copy_number_schema import ParsedToCnVarQuery  # noqa: E402\n",
    "\n",
    "log_filename = \"prep_normalize_cnv.log\"\n",
    "logging.basicConfig(\n",
    "    filename=log_filename,\n",
    "    format=\"[%(asctime)s] - %(name)s - %(levelname)s : %(message)s\",\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Download cool-seq-tool files. This must be done before initializing QueryHandler\n",
    "import os \n",
    "import sys\n",
    "\n",
    "module_path = os.path.abspath(os.path.join(\"../..\"))\n",
    "if module_path not in sys.path:\n",
    "    sys.path.append(module_path)\n",
    "from analysis.download_cool_seq_tool_files import download_cool_seq_tool_files  # noqa: E402\n",
    "\n",
    "download_cool_seq_tool_files(is_docker_env=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "pd.set_option(\"display.max_columns\", 100)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "MANUSCRIPT_S3_URL = \"https://nch-igm-wagner-lab-public.s3.us-east-2.amazonaws.com/variation-normalizer-manuscript/2025\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Download NCH microarray CNV data from s3 bucket\n",
    "\n",
    "from pathlib import Path\n",
    "import requests\n",
    "\n",
    "def download_s3(url: str, outfile_path: Path) -> None:\n",
    "    \"\"\"Download objects from public s3 bucket\n",
    "\n",
    "    :param url: URL for file in s3 bucket\n",
    "    :param outfile_path: Path where file should be saved\n",
    "    \"\"\"\n",
    "    with requests.get(url, stream=True) as r:\n",
    "        r.raise_for_status()\n",
    "        with open(outfile_path, \"wb\") as h:\n",
    "            for chunk in r.iter_content(chunk_size=8192):\n",
    "                if chunk:\n",
    "                    h.write(chunk)\n",
    "path = \"cnv_data\"\n",
    "Path(path).mkdir(exist_ok=True)\n",
    "\n",
    "\n",
    "url = f\"{MANUSCRIPT_S3_URL}/cnv_data/NCH-microarray-CNVs.csv\"\n",
    "outfile_path = Path(f\"{path}/NCH-microarray-CNVs.csv\")\n",
    "download_s3(url, outfile_path)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "### Load NCH microarray CNV data\n",
    "da = pd.read_csv(\"cnv_data/NCH-microarray-CNVs.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>variant</th>\n",
       "      <th>build</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>7q35(146581450_146661741)x3</td>\n",
       "      <td>GRCh37</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>22q11.21q11.23(21798705_23751189)x3</td>\n",
       "      <td>GRCh37</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>18p11.32(146484_408044)x1</td>\n",
       "      <td>GRCh37</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3q25.1(150353937_150944378)x1</td>\n",
       "      <td>GRCh37</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>16p12.1(27864452_27995317)x3</td>\n",
       "      <td>GRCh37</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                               variant   build\n",
       "0          7q35(146581450_146661741)x3  GRCh37\n",
       "1  22q11.21q11.23(21798705_23751189)x3  GRCh37\n",
       "2            18p11.32(146484_408044)x1  GRCh37\n",
       "3        3q25.1(150353937_150944378)x1  GRCh37\n",
       "4         16p12.1(27864452_27995317)x3  GRCh37"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "da.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "19106"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(da)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Remove duplicates"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "14710"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "da = da.drop_duplicates(\"variant\")\n",
    "len(da)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Compute and check chromosome values"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "da[\"chromosome\"] = da[\"variant\"].apply(lambda v: re.findall(r\"([\\dXY]+)[pq]\", v)[0])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['7', '22', '18', '3', '16', 'X', '2', '17', '1', '20', '5', '6',\n",
       "       '9', '14', '11', '10', '13', '4', '8', '12', '21', '15', '19', 'Y',\n",
       "       '1919'], dtype=object)"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "da[\"chromosome\"].unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "# clean incorrect values\n",
    "da[\"chromosome\"] = da[\"chromosome\"].replace(\"1919\", \"19\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Compute start/stop values"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "da[\"start\"] = (\n",
    "    da[\"variant\"]\n",
    "    .apply(lambda v: re.findall(r\"\\(([\\d,]+)\", v)[0].replace(\",\", \"\"))\n",
    "    .astype(int)\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "da[\"stop\"] = (\n",
    "    da[\"variant\"]\n",
    "    .apply(lambda v: re.findall(r\"([\\d,]+)\\)\", v)[-1].replace(\",\", \"\"))\n",
    "    .astype(int)\n",
    ")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Identify errors in start/stop values"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "da[\"var_len\"] = da[\"stop\"] - da[\"start\"] + 1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "def count_positions(v: str) -> int:\n",
    "    \"\"\"given an ISCN nomenclature string, returns the number of numerical positions detected (expected is 2 per the nomenclature)\"\"\"\n",
    "    numranges = re.findall(r\"\\([\\d,\\-\\_]+\\)\", v)[0]\n",
    "    numranges = re.sub(\",\", \"\", numranges)\n",
    "    return len(re.findall(r\"\\d+\", numranges))\n",
    "\n",
    "\n",
    "da[\"#positions\"] = da[\"variant\"].apply(count_positions)\n",
    "if list(da[\"#positions\"].unique()) == [1]:\n",
    "    del da[\"#positions\"]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Remove malformed variants: incorrect number of positions\n",
    "(these all appear to data entry errors; not all have a clear resolution, so we will drop them from this analysis)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>variant</th>\n",
       "      <th>build</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>start</th>\n",
       "      <th>stop</th>\n",
       "      <th>var_len</th>\n",
       "      <th>#positions</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>5251</th>\n",
       "      <td>13q31.3(90,653,47091,471,740)x3</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>13</td>\n",
       "      <td>9065347091471740</td>\n",
       "      <td>9065347091471740</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5802</th>\n",
       "      <td>15q21.3(55,499,672,55,593,588)x1</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>15</td>\n",
       "      <td>5549967255593588</td>\n",
       "      <td>5549967255593588</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5940</th>\n",
       "      <td>1p36.33p34.3(1,341,185-35,888,147-35,888,147)x...</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>1</td>\n",
       "      <td>1341185</td>\n",
       "      <td>35888147</td>\n",
       "      <td>34546963</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12101</th>\n",
       "      <td>1p33(47,070,474-47,282-269)x3</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>1</td>\n",
       "      <td>47070474</td>\n",
       "      <td>269</td>\n",
       "      <td>-47070204</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13757</th>\n",
       "      <td>1q21.1(144,998,070-146,179-215)x1</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>1</td>\n",
       "      <td>144998070</td>\n",
       "      <td>215</td>\n",
       "      <td>-144997854</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15978</th>\n",
       "      <td>12q13.12(49,370,149-49-678,707)x3</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>12</td>\n",
       "      <td>49370149</td>\n",
       "      <td>678707</td>\n",
       "      <td>-48691441</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16842</th>\n",
       "      <td>20q13.2q13.32(52,096,110-57-389,480)x2 hmz</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>20</td>\n",
       "      <td>52096110</td>\n",
       "      <td>389480</td>\n",
       "      <td>-51706629</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17398</th>\n",
       "      <td>8q24.21q24.22(129,951,769-133-621,197)x2 hmz</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>8</td>\n",
       "      <td>129951769</td>\n",
       "      <td>621197</td>\n",
       "      <td>-129330571</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                 variant   build chromosome  \\\n",
       "5251                     13q31.3(90,653,47091,471,740)x3  GRCh38         13   \n",
       "5802                    15q21.3(55,499,672,55,593,588)x1  GRCh38         15   \n",
       "5940   1p36.33p34.3(1,341,185-35,888,147-35,888,147)x...  GRCh38          1   \n",
       "12101                      1p33(47,070,474-47,282-269)x3  GRCh37          1   \n",
       "13757                  1q21.1(144,998,070-146,179-215)x1  GRCh37          1   \n",
       "15978                  12q13.12(49,370,149-49-678,707)x3  GRCh38         12   \n",
       "16842         20q13.2q13.32(52,096,110-57-389,480)x2 hmz  GRCh38         20   \n",
       "17398       8q24.21q24.22(129,951,769-133-621,197)x2 hmz  GRCh38          8   \n",
       "\n",
       "                  start              stop    var_len  #positions  \n",
       "5251   9065347091471740  9065347091471740          1           1  \n",
       "5802   5549967255593588  5549967255593588          1           1  \n",
       "5940            1341185          35888147   34546963           3  \n",
       "12101          47070474               269  -47070204           3  \n",
       "13757         144998070               215 -144997854           3  \n",
       "15978          49370149            678707  -48691441           3  \n",
       "16842          52096110            389480  -51706629           3  \n",
       "17398         129951769            621197 -129330571           3  "
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "da[(da[\"#positions\"] != 2)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "14702"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "da = da[(da[\"#positions\"] == 2)].copy()\n",
    "len(da)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Remove Malformed variants: start > stop\n",
    "(these all appear to be data entry errors; not all have a clear resolution, so we will drop them from this analysis)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>variant</th>\n",
       "      <th>build</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>start</th>\n",
       "      <th>stop</th>\n",
       "      <th>var_len</th>\n",
       "      <th>#positions</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>4295</th>\n",
       "      <td>6q22.1q23.2(1117,662,682-134,504,204)x2 hmz</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>6</td>\n",
       "      <td>1117662682</td>\n",
       "      <td>134504204</td>\n",
       "      <td>-983158477</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4963</th>\n",
       "      <td>6q16.1(95,032,6225-95,297,694)x3</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>6</td>\n",
       "      <td>950326225</td>\n",
       "      <td>95297694</td>\n",
       "      <td>-855028530</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6081</th>\n",
       "      <td>10q22.3(793155275_79370971)x1</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>10</td>\n",
       "      <td>793155275</td>\n",
       "      <td>79370971</td>\n",
       "      <td>-713784303</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6192</th>\n",
       "      <td>22q13.31q13.3(345611578_51178150)x1</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>22</td>\n",
       "      <td>345611578</td>\n",
       "      <td>51178150</td>\n",
       "      <td>-294433427</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8304</th>\n",
       "      <td>9p24.1(9014741_5165768)x1</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>9</td>\n",
       "      <td>9014741</td>\n",
       "      <td>5165768</td>\n",
       "      <td>-3848972</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10562</th>\n",
       "      <td>9q22.2(914,974,185-92,471,302)x1</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>9</td>\n",
       "      <td>914974185</td>\n",
       "      <td>92471302</td>\n",
       "      <td>-822502882</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13673</th>\n",
       "      <td>15q26.3(98,111,496-97,413,981)x3</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>15</td>\n",
       "      <td>98111496</td>\n",
       "      <td>97413981</td>\n",
       "      <td>-697514</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14905</th>\n",
       "      <td>2q34(212,681,263-21,889,508)x1</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>2</td>\n",
       "      <td>212681263</td>\n",
       "      <td>21889508</td>\n",
       "      <td>-190791754</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14938</th>\n",
       "      <td>8q11.1q11.21(47,042,437-48,401,60)x3</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>8</td>\n",
       "      <td>47042437</td>\n",
       "      <td>4840160</td>\n",
       "      <td>-42202276</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15141</th>\n",
       "      <td>5q35.3(180,380,290-100,448,046)x3</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>5</td>\n",
       "      <td>180380290</td>\n",
       "      <td>100448046</td>\n",
       "      <td>-79932243</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15354</th>\n",
       "      <td>11p15.4(9,660,8406-10,126,909)x3</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>11</td>\n",
       "      <td>96608406</td>\n",
       "      <td>10126909</td>\n",
       "      <td>-86481496</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15503</th>\n",
       "      <td>15q14q21.3(34197524-5899549)x2 hmz</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>15</td>\n",
       "      <td>34197524</td>\n",
       "      <td>5899549</td>\n",
       "      <td>-28297974</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15909</th>\n",
       "      <td>4q21.1q21.3(76,558,545-66,960,086)x1</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>4</td>\n",
       "      <td>76558545</td>\n",
       "      <td>66960086</td>\n",
       "      <td>-9598458</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17053</th>\n",
       "      <td>7q31.32(122,149,117-112,178,851)x1</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>7</td>\n",
       "      <td>122149117</td>\n",
       "      <td>112178851</td>\n",
       "      <td>-9970265</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17422</th>\n",
       "      <td>2q31.1(175,173,118-117,322,981)x1</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>2</td>\n",
       "      <td>175173118</td>\n",
       "      <td>117322981</td>\n",
       "      <td>-57850136</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18014</th>\n",
       "      <td>9p21.3(22,2736,313-22,768,766)x1</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>9</td>\n",
       "      <td>222736313</td>\n",
       "      <td>22768766</td>\n",
       "      <td>-199967546</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18078</th>\n",
       "      <td>14q32.33(150,422,204-105,845,718)x1</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>14</td>\n",
       "      <td>150422204</td>\n",
       "      <td>105845718</td>\n",
       "      <td>-44576485</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18725</th>\n",
       "      <td>14q32.33(140,233,331-104,294,832)x3</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>14</td>\n",
       "      <td>140233331</td>\n",
       "      <td>104294832</td>\n",
       "      <td>-35938498</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19032</th>\n",
       "      <td>2q33.1(230,250,520-203,387,006)x3</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>2</td>\n",
       "      <td>230250520</td>\n",
       "      <td>203387006</td>\n",
       "      <td>-26863513</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                           variant   build chromosome  \\\n",
       "4295   6q22.1q23.2(1117,662,682-134,504,204)x2 hmz  GRCh38          6   \n",
       "4963              6q16.1(95,032,6225-95,297,694)x3  GRCh38          6   \n",
       "6081                 10q22.3(793155275_79370971)x1  GRCh37         10   \n",
       "6192           22q13.31q13.3(345611578_51178150)x1  GRCh37         22   \n",
       "8304                     9p24.1(9014741_5165768)x1  GRCh37          9   \n",
       "10562             9q22.2(914,974,185-92,471,302)x1  GRCh37          9   \n",
       "13673             15q26.3(98,111,496-97,413,981)x3  GRCh37         15   \n",
       "14905               2q34(212,681,263-21,889,508)x1  GRCh37          2   \n",
       "14938         8q11.1q11.21(47,042,437-48,401,60)x3  GRCh37          8   \n",
       "15141            5q35.3(180,380,290-100,448,046)x3  GRCh37          5   \n",
       "15354             11p15.4(9,660,8406-10,126,909)x3  GRCh38         11   \n",
       "15503           15q14q21.3(34197524-5899549)x2 hmz  GRCh38         15   \n",
       "15909         4q21.1q21.3(76,558,545-66,960,086)x1  GRCh38          4   \n",
       "17053           7q31.32(122,149,117-112,178,851)x1  GRCh38          7   \n",
       "17422            2q31.1(175,173,118-117,322,981)x1  GRCh38          2   \n",
       "18014             9p21.3(22,2736,313-22,768,766)x1  GRCh37          9   \n",
       "18078          14q32.33(150,422,204-105,845,718)x1  GRCh37         14   \n",
       "18725          14q32.33(140,233,331-104,294,832)x3  GRCh37         14   \n",
       "19032            2q33.1(230,250,520-203,387,006)x3  GRCh37          2   \n",
       "\n",
       "            start       stop    var_len  #positions  \n",
       "4295   1117662682  134504204 -983158477           2  \n",
       "4963    950326225   95297694 -855028530           2  \n",
       "6081    793155275   79370971 -713784303           2  \n",
       "6192    345611578   51178150 -294433427           2  \n",
       "8304      9014741    5165768   -3848972           2  \n",
       "10562   914974185   92471302 -822502882           2  \n",
       "13673    98111496   97413981    -697514           2  \n",
       "14905   212681263   21889508 -190791754           2  \n",
       "14938    47042437    4840160  -42202276           2  \n",
       "15141   180380290  100448046  -79932243           2  \n",
       "15354    96608406   10126909  -86481496           2  \n",
       "15503    34197524    5899549  -28297974           2  \n",
       "15909    76558545   66960086   -9598458           2  \n",
       "17053   122149117  112178851   -9970265           2  \n",
       "17422   175173118  117322981  -57850136           2  \n",
       "18014   222736313   22768766 -199967546           2  \n",
       "18078   150422204  105845718  -44576485           2  \n",
       "18725   140233331  104294832  -35938498           2  \n",
       "19032   230250520  203387006  -26863513           2  "
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "da[(da[\"var_len\"] < 0)].drop_duplicates(\"variant\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "14683"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "da = da[da[\"var_len\"] >= 0].copy()\n",
    "len(da)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Compute copy number/ranges for each variant"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [],
   "source": [
    "da[\"copy_number\"] = da[\"variant\"].apply(\n",
    "    lambda v: re.findall(r\"\\) ?x?[ ]?([\\d~\\-]*)\", v)[0]\n",
    ")\n",
    "da[\"copy_number_min\"] = da[\"copy_number\"].apply(lambda x: int(x[0]))\n",
    "da[\"copy_number_max\"] = da[\"copy_number\"].apply(lambda x: int(x[-1]))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Check copy number values"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['0',\n",
       " '0-2',\n",
       " '0~1',\n",
       " '1',\n",
       " '1-2',\n",
       " '1~2',\n",
       " '1~3',\n",
       " '2',\n",
       " '2-3',\n",
       " '2~3',\n",
       " '3',\n",
       " '3~4',\n",
       " '4',\n",
       " '4~5',\n",
       " '5~6']"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sorted(set(da[\"copy_number\"]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[0, 1, 2, 3, 4, 5]"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sorted(set(da[\"copy_number_min\"]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[0, 1, 2, 3, 4, 5, 6]"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sorted(set(da[\"copy_number_max\"]))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Regions of Homozygosity (\"hmz\") variants not to be considered in this analysis\n",
    "#### How many variants in our data with CN=2 are autosomal and not regions of homozygosity?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [],
   "source": [
    "da[\"region_of_homozygosity\"] = da[\"variant\"].str.contains(\"hmz\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "region_of_homozygosity\n",
       "False    9017\n",
       "True     5666\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "da[\"region_of_homozygosity\"].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "9017"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "da = da[~da[\"region_of_homozygosity\"]].copy()\n",
    "len(da)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>variant</th>\n",
       "      <th>build</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>start</th>\n",
       "      <th>stop</th>\n",
       "      <th>var_len</th>\n",
       "      <th>#positions</th>\n",
       "      <th>copy_number</th>\n",
       "      <th>copy_number_min</th>\n",
       "      <th>copy_number_max</th>\n",
       "      <th>region_of_homozygosity</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>151</th>\n",
       "      <td>2p13.2p12(73227186_79042754)x2</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>2</td>\n",
       "      <td>73227186</td>\n",
       "      <td>79042754</td>\n",
       "      <td>5815569</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4144</th>\n",
       "      <td>7p22.2p12.3(3,264,728-46,071,829)x2</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>7</td>\n",
       "      <td>3264728</td>\n",
       "      <td>46071829</td>\n",
       "      <td>42807102</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4399</th>\n",
       "      <td>3p25.3p24.1(10,514,008-29,532,824)x2</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>3</td>\n",
       "      <td>10514008</td>\n",
       "      <td>29532824</td>\n",
       "      <td>19018817</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5317</th>\n",
       "      <td>8q12.3q21.13(64,242,384-83,258,323)x2</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>8</td>\n",
       "      <td>64242384</td>\n",
       "      <td>83258323</td>\n",
       "      <td>19015940</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5338</th>\n",
       "      <td>20q11.21q13.2(29,991,219-50,728,172)x2</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>20</td>\n",
       "      <td>29991219</td>\n",
       "      <td>50728172</td>\n",
       "      <td>20736954</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5340</th>\n",
       "      <td>10p15.3p15.1(307,807-5,146,667)x2</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>10</td>\n",
       "      <td>307807</td>\n",
       "      <td>5146667</td>\n",
       "      <td>4838861</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7198</th>\n",
       "      <td>2q11.2q12.2(98703659_106160498)x2</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>2</td>\n",
       "      <td>98703659</td>\n",
       "      <td>106160498</td>\n",
       "      <td>7456840</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7596</th>\n",
       "      <td>12q23.3q24.31(108904090_124118489)x2</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>12</td>\n",
       "      <td>108904090</td>\n",
       "      <td>124118489</td>\n",
       "      <td>15214400</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15220</th>\n",
       "      <td>11q24.1(122,984,093-123,106,251)x2</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>11</td>\n",
       "      <td>122984093</td>\n",
       "      <td>123106251</td>\n",
       "      <td>122159</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15416</th>\n",
       "      <td>7p12.3p11.2(46,591,048-57,560,040)x2</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>7</td>\n",
       "      <td>46591048</td>\n",
       "      <td>57560040</td>\n",
       "      <td>10968993</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15418</th>\n",
       "      <td>6q21q22.3(114,543,815-129,022,028)x2</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>6</td>\n",
       "      <td>114543815</td>\n",
       "      <td>129022028</td>\n",
       "      <td>14478214</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15419</th>\n",
       "      <td>6q21q22.3(114,543,815-129,059,691)x2</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>6</td>\n",
       "      <td>114543815</td>\n",
       "      <td>129059691</td>\n",
       "      <td>14515877</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15421</th>\n",
       "      <td>7p12.3p11.2(46,591,048-57,487,634)x2</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>7</td>\n",
       "      <td>46591048</td>\n",
       "      <td>57487634</td>\n",
       "      <td>10896587</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15961</th>\n",
       "      <td>13q12.2q13.3(28,249,648-37,113,666)x2</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>13</td>\n",
       "      <td>28249648</td>\n",
       "      <td>37113666</td>\n",
       "      <td>8864019</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15971</th>\n",
       "      <td>13q12.2q13.3(28,249,648-37,209,815)x2</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>13</td>\n",
       "      <td>28249648</td>\n",
       "      <td>37209815</td>\n",
       "      <td>8960168</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16680</th>\n",
       "      <td>13q13.2q21.31(34,898,501-65,301,754)x2</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>13</td>\n",
       "      <td>34898501</td>\n",
       "      <td>65301754</td>\n",
       "      <td>30403254</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16695</th>\n",
       "      <td>13q13.2q21.31(34,898,501-65,337,816)x2</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>13</td>\n",
       "      <td>34898501</td>\n",
       "      <td>65337816</td>\n",
       "      <td>30439316</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                      variant   build chromosome      start  \\\n",
       "151            2p13.2p12(73227186_79042754)x2  GRCh37          2   73227186   \n",
       "4144      7p22.2p12.3(3,264,728-46,071,829)x2  GRCh38          7    3264728   \n",
       "4399     3p25.3p24.1(10,514,008-29,532,824)x2  GRCh38          3   10514008   \n",
       "5317    8q12.3q21.13(64,242,384-83,258,323)x2  GRCh38          8   64242384   \n",
       "5338   20q11.21q13.2(29,991,219-50,728,172)x2  GRCh38         20   29991219   \n",
       "5340        10p15.3p15.1(307,807-5,146,667)x2  GRCh38         10     307807   \n",
       "7198        2q11.2q12.2(98703659_106160498)x2  GRCh37          2   98703659   \n",
       "7596     12q23.3q24.31(108904090_124118489)x2  GRCh37         12  108904090   \n",
       "15220      11q24.1(122,984,093-123,106,251)x2  GRCh37         11  122984093   \n",
       "15416    7p12.3p11.2(46,591,048-57,560,040)x2  GRCh38          7   46591048   \n",
       "15418    6q21q22.3(114,543,815-129,022,028)x2  GRCh38          6  114543815   \n",
       "15419    6q21q22.3(114,543,815-129,059,691)x2  GRCh38          6  114543815   \n",
       "15421    7p12.3p11.2(46,591,048-57,487,634)x2  GRCh38          7   46591048   \n",
       "15961   13q12.2q13.3(28,249,648-37,113,666)x2  GRCh38         13   28249648   \n",
       "15971   13q12.2q13.3(28,249,648-37,209,815)x2  GRCh38         13   28249648   \n",
       "16680  13q13.2q21.31(34,898,501-65,301,754)x2  GRCh38         13   34898501   \n",
       "16695  13q13.2q21.31(34,898,501-65,337,816)x2  GRCh38         13   34898501   \n",
       "\n",
       "            stop   var_len  #positions copy_number  copy_number_min  \\\n",
       "151     79042754   5815569           2           2                2   \n",
       "4144    46071829  42807102           2           2                2   \n",
       "4399    29532824  19018817           2           2                2   \n",
       "5317    83258323  19015940           2           2                2   \n",
       "5338    50728172  20736954           2           2                2   \n",
       "5340     5146667   4838861           2           2                2   \n",
       "7198   106160498   7456840           2           2                2   \n",
       "7596   124118489  15214400           2           2                2   \n",
       "15220  123106251    122159           2           2                2   \n",
       "15416   57560040  10968993           2           2                2   \n",
       "15418  129022028  14478214           2           2                2   \n",
       "15419  129059691  14515877           2           2                2   \n",
       "15421   57487634  10896587           2           2                2   \n",
       "15961   37113666   8864019           2           2                2   \n",
       "15971   37209815   8960168           2           2                2   \n",
       "16680   65301754  30403254           2           2                2   \n",
       "16695   65337816  30439316           2           2                2   \n",
       "\n",
       "       copy_number_max  region_of_homozygosity  \n",
       "151                  2                   False  \n",
       "4144                 2                   False  \n",
       "4399                 2                   False  \n",
       "5317                 2                   False  \n",
       "5338                 2                   False  \n",
       "5340                 2                   False  \n",
       "7198                 2                   False  \n",
       "7596                 2                   False  \n",
       "15220                2                   False  \n",
       "15416                2                   False  \n",
       "15418                2                   False  \n",
       "15419                2                   False  \n",
       "15421                2                   False  \n",
       "15961                2                   False  \n",
       "15971                2                   False  \n",
       "16680                2                   False  \n",
       "16695                2                   False  "
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "da[(da[\"copy_number\"] == \"2\") & (~da[\"chromosome\"].isin([\"X\", \"Y\"]))]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### After inspection of orginal reports, it appears that these are all actually regions of homozygosity and minor typos/defects in regex matching strategy caused them not to be recognized as such\n",
    "* \"hmm\" instead of \"hmz\"\n",
    "* \",\" between \"x2\" and rest of variant, but \",\" was used as a tokenizing character between variant strings"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "9000"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "da = da[(da[\"copy_number\"] != \"2\") | (da[\"chromosome\"].isin([\"X\", \"Y\"]))].copy()\n",
    "len(da)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Remove any variants in build GRCh36"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "8989"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "da = da[da[\"build\"].isin([\"GRCh37\", \"GRCh38\"])]\n",
    "len(da)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [],
   "source": [
    "da = da.reset_index()\n",
    "del da[\"index\"]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Normalize NCH CNVs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "9bdc3d1075384c39b1c9436eeabdf450",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "  0%|          | 0/8989 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "### this cell runs the normalizer and saves the output\n",
    "### runs in ~40s\n",
    "### skip re-running by commenting out this cell\n",
    "\n",
    "norm_responses = []\n",
    "for ix, cnv in tqdm(da.iterrows(), total=len(da)):\n",
    "    args = {\n",
    "        \"assembly\": cnv[\"build\"],\n",
    "        \"chromosome\": f\"chr{cnv['chromosome']}\",\n",
    "        \"start0\": int(cnv[\"start\"]),\n",
    "        \"start_pos_type\": \"number\",\n",
    "        \"end0\": int(cnv[\"stop\"]),\n",
    "        \"end_pos_type\": \"number\",\n",
    "    }\n",
    "    if (\n",
    "        cnv[\"copy_number_min\"] == cnv[\"copy_number_max\"]\n",
    "    ):  ##if copy number is unambiguous, not a range\n",
    "        args[\"copies0\"] = int(cnv[\"copy_number_min\"])\n",
    "        args[\"copies_type\"] = \"number\"\n",
    "    else:  ##if copy number is a range e.g. 3~4\n",
    "        args[\"copies0\"] = int(cnv[\"copy_number_min\"])\n",
    "        args[\"copies1\"] = int(cnv[\"copy_number_max\"])\n",
    "        args[\"copies_type\"] = \"definite_range\"\n",
    "\n",
    "    args[\"do_liftover\"] = True\n",
    "    args[\"untranslatable_returns_text\"] = False\n",
    "\n",
    "    rec = dict(cnv)\n",
    "    resp = json.loads(parsed_to_cn_var(ParsedToCnVarQuery(**args)).model_dump_json(exclude_none=True))\n",
    "    rec.update(resp)\n",
    "\n",
    "    for arg, val in args.items():\n",
    "        rec[f\"query_args.{arg}\"] = val\n",
    "\n",
    "    norm_responses.append(rec)\n",
    "\n",
    "\n",
    "## save raw normalizer output\n",
    "with open(\"cnv_data/NCH-normalizer-results.json\", \"w\") as f:\n",
    "    json.dump(norm_responses, f)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [],
   "source": [
    "with open(\"cnv_data/NCH-normalizer-results.json\", \"r\") as f:\n",
    "    norm_responses = json.load(f)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>variant</th>\n",
       "      <th>build</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>start</th>\n",
       "      <th>stop</th>\n",
       "      <th>var_len</th>\n",
       "      <th>#positions</th>\n",
       "      <th>copy_number</th>\n",
       "      <th>copy_number_min</th>\n",
       "      <th>copy_number_max</th>\n",
       "      <th>region_of_homozygosity</th>\n",
       "      <th>warnings</th>\n",
       "      <th>query_args.assembly</th>\n",
       "      <th>query_args.chromosome</th>\n",
       "      <th>query_args.start0</th>\n",
       "      <th>query_args.start_pos_type</th>\n",
       "      <th>query_args.end0</th>\n",
       "      <th>query_args.end_pos_type</th>\n",
       "      <th>query_args.copies0</th>\n",
       "      <th>query_args.copies_type</th>\n",
       "      <th>query_args.do_liftover</th>\n",
       "      <th>query_args.untranslatable_returns_text</th>\n",
       "      <th>service_meta_.name</th>\n",
       "      <th>service_meta_.version</th>\n",
       "      <th>service_meta_.response_datetime</th>\n",
       "      <th>service_meta_.url</th>\n",
       "      <th>copy_number_count.id</th>\n",
       "      <th>copy_number_count.type</th>\n",
       "      <th>copy_number_count.digest</th>\n",
       "      <th>copy_number_count.location.id</th>\n",
       "      <th>copy_number_count.location.type</th>\n",
       "      <th>copy_number_count.location.digest</th>\n",
       "      <th>copy_number_count.location.sequenceReference.type</th>\n",
       "      <th>copy_number_count.location.sequenceReference.refgetAccession</th>\n",
       "      <th>copy_number_count.location.start</th>\n",
       "      <th>copy_number_count.location.end</th>\n",
       "      <th>copy_number_count.copies</th>\n",
       "      <th>query_args.copies1</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>7q35(146581450_146661741)x3</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>7</td>\n",
       "      <td>146581450</td>\n",
       "      <td>146661741</td>\n",
       "      <td>80292</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>False</td>\n",
       "      <td>[]</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>chr7</td>\n",
       "      <td>146581450</td>\n",
       "      <td>number</td>\n",
       "      <td>146661741</td>\n",
       "      <td>number</td>\n",
       "      <td>3</td>\n",
       "      <td>number</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>variation-normalizer</td>\n",
       "      <td>0.15.0</td>\n",
       "      <td>2025-07-28T17:53:11.575012Z</td>\n",
       "      <td>https://github.com/cancervariants/variation-no...</td>\n",
       "      <td>ga4gh:CN.MJuX6gNGt0BJ1iHpVj_UVxBBya0Sp8n_</td>\n",
       "      <td>CopyNumberCount</td>\n",
       "      <td>MJuX6gNGt0BJ1iHpVj_UVxBBya0Sp8n_</td>\n",
       "      <td>ga4gh:SL.qmNJl0mIgQ5mtMf-9CLxIelFedebRN4-</td>\n",
       "      <td>SequenceLocation</td>\n",
       "      <td>qmNJl0mIgQ5mtMf-9CLxIelFedebRN4-</td>\n",
       "      <td>SequenceReference</td>\n",
       "      <td>SQ.F-LrLMe1SRpfUZHkQmvkVKFEGaoDeHul</td>\n",
       "      <td>146884357.0</td>\n",
       "      <td>146964649.0</td>\n",
       "      <td>3</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>22q11.21q11.23(21798705_23751189)x3</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>22</td>\n",
       "      <td>21798705</td>\n",
       "      <td>23751189</td>\n",
       "      <td>1952485</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>False</td>\n",
       "      <td>[]</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>chr22</td>\n",
       "      <td>21798705</td>\n",
       "      <td>number</td>\n",
       "      <td>23751189</td>\n",
       "      <td>number</td>\n",
       "      <td>3</td>\n",
       "      <td>number</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>variation-normalizer</td>\n",
       "      <td>0.15.0</td>\n",
       "      <td>2025-07-28T17:53:11.576953Z</td>\n",
       "      <td>https://github.com/cancervariants/variation-no...</td>\n",
       "      <td>ga4gh:CN.FluTAMBMb2-R0oBoWMz8R40Qn2Dj6KXP</td>\n",
       "      <td>CopyNumberCount</td>\n",
       "      <td>FluTAMBMb2-R0oBoWMz8R40Qn2Dj6KXP</td>\n",
       "      <td>ga4gh:SL.JiiE178XNyrPA_QIdD0_O8fllMElI4Fj</td>\n",
       "      <td>SequenceLocation</td>\n",
       "      <td>JiiE178XNyrPA_QIdD0_O8fllMElI4Fj</td>\n",
       "      <td>SequenceReference</td>\n",
       "      <td>SQ.7B7SHsmchAR0dFcDCuSFjJAo7tX87krQ</td>\n",
       "      <td>21444415.0</td>\n",
       "      <td>23409002.0</td>\n",
       "      <td>3</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>18p11.32(146484_408044)x1</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>18</td>\n",
       "      <td>146484</td>\n",
       "      <td>408044</td>\n",
       "      <td>261561</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>False</td>\n",
       "      <td>[]</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>chr18</td>\n",
       "      <td>146484</td>\n",
       "      <td>number</td>\n",
       "      <td>408044</td>\n",
       "      <td>number</td>\n",
       "      <td>1</td>\n",
       "      <td>number</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>variation-normalizer</td>\n",
       "      <td>0.15.0</td>\n",
       "      <td>2025-07-28T17:53:11.579229Z</td>\n",
       "      <td>https://github.com/cancervariants/variation-no...</td>\n",
       "      <td>ga4gh:CN.B85IimabYh-WJ-r_tN6nmPkSCl9_rgWc</td>\n",
       "      <td>CopyNumberCount</td>\n",
       "      <td>B85IimabYh-WJ-r_tN6nmPkSCl9_rgWc</td>\n",
       "      <td>ga4gh:SL.Cu3iBhfqtXsd_cVsHtCdq325C6oAd0NH</td>\n",
       "      <td>SequenceLocation</td>\n",
       "      <td>Cu3iBhfqtXsd_cVsHtCdq325C6oAd0NH</td>\n",
       "      <td>SequenceReference</td>\n",
       "      <td>SQ.vWwFhJ5lQDMhh-czg06YtlWqu0lvFAZV</td>\n",
       "      <td>146483.0</td>\n",
       "      <td>408044.0</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3q25.1(150353937_150944378)x1</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>3</td>\n",
       "      <td>150353937</td>\n",
       "      <td>150944378</td>\n",
       "      <td>590442</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>False</td>\n",
       "      <td>[]</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>chr3</td>\n",
       "      <td>150353937</td>\n",
       "      <td>number</td>\n",
       "      <td>150944378</td>\n",
       "      <td>number</td>\n",
       "      <td>1</td>\n",
       "      <td>number</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>variation-normalizer</td>\n",
       "      <td>0.15.0</td>\n",
       "      <td>2025-07-28T17:53:11.580966Z</td>\n",
       "      <td>https://github.com/cancervariants/variation-no...</td>\n",
       "      <td>ga4gh:CN.zS1jV37WemE7wBrf4Bkdt4VSxyNHUp_N</td>\n",
       "      <td>CopyNumberCount</td>\n",
       "      <td>zS1jV37WemE7wBrf4Bkdt4VSxyNHUp_N</td>\n",
       "      <td>ga4gh:SL.jQiTZDtAM_PWPlTA3l-BnS9UKIuLPiOs</td>\n",
       "      <td>SequenceLocation</td>\n",
       "      <td>jQiTZDtAM_PWPlTA3l-BnS9UKIuLPiOs</td>\n",
       "      <td>SequenceReference</td>\n",
       "      <td>SQ.Zu7h9AggXxhTaGVsy7h_EZSChSZGcmgX</td>\n",
       "      <td>150636149.0</td>\n",
       "      <td>151226590.0</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>16p12.1(27864452_27995317)x3</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>16</td>\n",
       "      <td>27864452</td>\n",
       "      <td>27995317</td>\n",
       "      <td>130866</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>False</td>\n",
       "      <td>[]</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>chr16</td>\n",
       "      <td>27864452</td>\n",
       "      <td>number</td>\n",
       "      <td>27995317</td>\n",
       "      <td>number</td>\n",
       "      <td>3</td>\n",
       "      <td>number</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>variation-normalizer</td>\n",
       "      <td>0.15.0</td>\n",
       "      <td>2025-07-28T17:53:11.582848Z</td>\n",
       "      <td>https://github.com/cancervariants/variation-no...</td>\n",
       "      <td>ga4gh:CN.Wqsbt4qGYELdnzSkqx0AOr_tpa8zJ8N1</td>\n",
       "      <td>CopyNumberCount</td>\n",
       "      <td>Wqsbt4qGYELdnzSkqx0AOr_tpa8zJ8N1</td>\n",
       "      <td>ga4gh:SL.MNRSCufvIZcW7fm8nOkL2aujzX1MmOFH</td>\n",
       "      <td>SequenceLocation</td>\n",
       "      <td>MNRSCufvIZcW7fm8nOkL2aujzX1MmOFH</td>\n",
       "      <td>SequenceReference</td>\n",
       "      <td>SQ.yC_0RBj3fgBlvgyAuycbzdubtLxq-rE0</td>\n",
       "      <td>27853130.0</td>\n",
       "      <td>27983996.0</td>\n",
       "      <td>3</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                               variant   build chromosome      start  \\\n",
       "0          7q35(146581450_146661741)x3  GRCh37          7  146581450   \n",
       "1  22q11.21q11.23(21798705_23751189)x3  GRCh37         22   21798705   \n",
       "2            18p11.32(146484_408044)x1  GRCh37         18     146484   \n",
       "3        3q25.1(150353937_150944378)x1  GRCh37          3  150353937   \n",
       "4         16p12.1(27864452_27995317)x3  GRCh37         16   27864452   \n",
       "\n",
       "        stop  var_len  #positions copy_number  copy_number_min  \\\n",
       "0  146661741    80292           2           3                3   \n",
       "1   23751189  1952485           2           3                3   \n",
       "2     408044   261561           2           1                1   \n",
       "3  150944378   590442           2           1                1   \n",
       "4   27995317   130866           2           3                3   \n",
       "\n",
       "   copy_number_max  region_of_homozygosity warnings query_args.assembly  \\\n",
       "0                3                   False       []              GRCh37   \n",
       "1                3                   False       []              GRCh37   \n",
       "2                1                   False       []              GRCh37   \n",
       "3                1                   False       []              GRCh37   \n",
       "4                3                   False       []              GRCh37   \n",
       "\n",
       "  query_args.chromosome  query_args.start0 query_args.start_pos_type  \\\n",
       "0                  chr7          146581450                    number   \n",
       "1                 chr22           21798705                    number   \n",
       "2                 chr18             146484                    number   \n",
       "3                  chr3          150353937                    number   \n",
       "4                 chr16           27864452                    number   \n",
       "\n",
       "   query_args.end0 query_args.end_pos_type  query_args.copies0  \\\n",
       "0        146661741                  number                   3   \n",
       "1         23751189                  number                   3   \n",
       "2           408044                  number                   1   \n",
       "3        150944378                  number                   1   \n",
       "4         27995317                  number                   3   \n",
       "\n",
       "  query_args.copies_type  query_args.do_liftover  \\\n",
       "0                 number                    True   \n",
       "1                 number                    True   \n",
       "2                 number                    True   \n",
       "3                 number                    True   \n",
       "4                 number                    True   \n",
       "\n",
       "   query_args.untranslatable_returns_text    service_meta_.name  \\\n",
       "0                                   False  variation-normalizer   \n",
       "1                                   False  variation-normalizer   \n",
       "2                                   False  variation-normalizer   \n",
       "3                                   False  variation-normalizer   \n",
       "4                                   False  variation-normalizer   \n",
       "\n",
       "  service_meta_.version service_meta_.response_datetime  \\\n",
       "0                0.15.0     2025-07-28T17:53:11.575012Z   \n",
       "1                0.15.0     2025-07-28T17:53:11.576953Z   \n",
       "2                0.15.0     2025-07-28T17:53:11.579229Z   \n",
       "3                0.15.0     2025-07-28T17:53:11.580966Z   \n",
       "4                0.15.0     2025-07-28T17:53:11.582848Z   \n",
       "\n",
       "                                   service_meta_.url  \\\n",
       "0  https://github.com/cancervariants/variation-no...   \n",
       "1  https://github.com/cancervariants/variation-no...   \n",
       "2  https://github.com/cancervariants/variation-no...   \n",
       "3  https://github.com/cancervariants/variation-no...   \n",
       "4  https://github.com/cancervariants/variation-no...   \n",
       "\n",
       "                        copy_number_count.id copy_number_count.type  \\\n",
       "0  ga4gh:CN.MJuX6gNGt0BJ1iHpVj_UVxBBya0Sp8n_        CopyNumberCount   \n",
       "1  ga4gh:CN.FluTAMBMb2-R0oBoWMz8R40Qn2Dj6KXP        CopyNumberCount   \n",
       "2  ga4gh:CN.B85IimabYh-WJ-r_tN6nmPkSCl9_rgWc        CopyNumberCount   \n",
       "3  ga4gh:CN.zS1jV37WemE7wBrf4Bkdt4VSxyNHUp_N        CopyNumberCount   \n",
       "4  ga4gh:CN.Wqsbt4qGYELdnzSkqx0AOr_tpa8zJ8N1        CopyNumberCount   \n",
       "\n",
       "           copy_number_count.digest  \\\n",
       "0  MJuX6gNGt0BJ1iHpVj_UVxBBya0Sp8n_   \n",
       "1  FluTAMBMb2-R0oBoWMz8R40Qn2Dj6KXP   \n",
       "2  B85IimabYh-WJ-r_tN6nmPkSCl9_rgWc   \n",
       "3  zS1jV37WemE7wBrf4Bkdt4VSxyNHUp_N   \n",
       "4  Wqsbt4qGYELdnzSkqx0AOr_tpa8zJ8N1   \n",
       "\n",
       "               copy_number_count.location.id copy_number_count.location.type  \\\n",
       "0  ga4gh:SL.qmNJl0mIgQ5mtMf-9CLxIelFedebRN4-                SequenceLocation   \n",
       "1  ga4gh:SL.JiiE178XNyrPA_QIdD0_O8fllMElI4Fj                SequenceLocation   \n",
       "2  ga4gh:SL.Cu3iBhfqtXsd_cVsHtCdq325C6oAd0NH                SequenceLocation   \n",
       "3  ga4gh:SL.jQiTZDtAM_PWPlTA3l-BnS9UKIuLPiOs                SequenceLocation   \n",
       "4  ga4gh:SL.MNRSCufvIZcW7fm8nOkL2aujzX1MmOFH                SequenceLocation   \n",
       "\n",
       "  copy_number_count.location.digest  \\\n",
       "0  qmNJl0mIgQ5mtMf-9CLxIelFedebRN4-   \n",
       "1  JiiE178XNyrPA_QIdD0_O8fllMElI4Fj   \n",
       "2  Cu3iBhfqtXsd_cVsHtCdq325C6oAd0NH   \n",
       "3  jQiTZDtAM_PWPlTA3l-BnS9UKIuLPiOs   \n",
       "4  MNRSCufvIZcW7fm8nOkL2aujzX1MmOFH   \n",
       "\n",
       "  copy_number_count.location.sequenceReference.type  \\\n",
       "0                                 SequenceReference   \n",
       "1                                 SequenceReference   \n",
       "2                                 SequenceReference   \n",
       "3                                 SequenceReference   \n",
       "4                                 SequenceReference   \n",
       "\n",
       "  copy_number_count.location.sequenceReference.refgetAccession  \\\n",
       "0                SQ.F-LrLMe1SRpfUZHkQmvkVKFEGaoDeHul             \n",
       "1                SQ.7B7SHsmchAR0dFcDCuSFjJAo7tX87krQ             \n",
       "2                SQ.vWwFhJ5lQDMhh-czg06YtlWqu0lvFAZV             \n",
       "3                SQ.Zu7h9AggXxhTaGVsy7h_EZSChSZGcmgX             \n",
       "4                SQ.yC_0RBj3fgBlvgyAuycbzdubtLxq-rE0             \n",
       "\n",
       "   copy_number_count.location.start  copy_number_count.location.end  \\\n",
       "0                       146884357.0                     146964649.0   \n",
       "1                        21444415.0                      23409002.0   \n",
       "2                          146483.0                        408044.0   \n",
       "3                       150636149.0                     151226590.0   \n",
       "4                        27853130.0                      27983996.0   \n",
       "\n",
       "  copy_number_count.copies  query_args.copies1  \n",
       "0                        3                 NaN  \n",
       "1                        3                 NaN  \n",
       "2                        1                 NaN  \n",
       "3                        1                 NaN  \n",
       "4                        3                 NaN  "
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "### read normalizer responses into a dataframe for analysis\n",
    "dr = pd.json_normalize(norm_responses)\n",
    "dr.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [],
   "source": [
    "## warnings were stored in a list; convert to a string for readability/ease of processing\n",
    "dr[\"warning_string\"] = dr[\"warnings\"].apply(\";\".join)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "warning_string_reduce\n",
       "Success                                  8718\n",
       "Unable to liftover: chr# with pos #       208\n",
       "Position # is not valid on ga4gh:SQ.*      63\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "### remove numbers/identifiers of specific chromosome/positions/variants from warning strings to consolidate into a small number of error types\n",
    "\n",
    "\n",
    "def reduce_warning(warning: str) -> str:\n",
    "    \"\"\"removes specific numeric characters from error messages to reduce to broader category of error\"\"\"\n",
    "    warning = re.sub(r\"ga4gh\\:SQ\\.[A-Za-z0-9\\-\\_]+\", \"ga4gh:SQ.*\", warning)\n",
    "    warning = re.sub(r\"chr[XY\\d]+\", \"chr#\", warning)\n",
    "    warning = re.sub(r\"\\(\\d+\\)\", \"#\", warning)\n",
    "    warning = re.sub(r\"pos \\d+\", \"pos #\", warning)\n",
    "    if re.match(r\"\\d+ validation errors for\", warning):\n",
    "        warning = \"Pydantic Validation Error\"\n",
    "\n",
    "    if warning == \"\":\n",
    "        return \"Success\"\n",
    "    return warning\n",
    "\n",
    "\n",
    "dr[\"warning_string_reduce\"] = dr[\"warning_string\"].apply(reduce_warning)\n",
    "dr[\"warning_string_reduce\"].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [],
   "source": [
    "### get the normalized lifted over coordinates\n",
    "da[\"start_38\"] = dr[\"copy_number_count.location.start\"]\n",
    "da[\"stop_38\"] = dr[\"copy_number_count.location.end\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [],
   "source": [
    "### restrict to variants with valid normalized positions\n",
    "da = da[da[\"start_38\"].notna() & da[\"stop_38\"].notna()].copy()\n",
    "da[\"start_38\"] = da[\"start_38\"].astype(int)\n",
    "da[\"stop_38\"] = da[\"stop_38\"].astype(int)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "8718"
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(da)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [],
   "source": [
    "da[\n",
    "    [\n",
    "        \"variant\",\n",
    "        \"build\",\n",
    "        \"chromosome\",\n",
    "        \"start\",\n",
    "        \"stop\",\n",
    "        \"start_38\",\n",
    "        \"stop_38\",\n",
    "        \"copy_number\",\n",
    "        \"copy_number_min\",\n",
    "        \"copy_number_max\",\n",
    "    ]\n",
    "].dropna().to_csv(\"cnv_data/NCH-microarray-CNVs-cleaned.csv\", index=False)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "variation-normalizer-manuscript",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.13.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
