{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "True"
      ]
     },
     "execution_count": 1,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Load environment variables. This MUST be the first cell.\n",
    "from dotenv import load_dotenv\n",
    "\n",
    "load_dotenv(\"../../.env.shared\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "from tqdm.notebook import tqdm\n",
    "import ndjson\n",
    "import re\n",
    "import gzip\n",
    "import os\n",
    "import sys\n",
    "\n",
    "analysis_dir = os.path.abspath(os.path.join(os.getcwd(), '..'))\n",
    "sys.path.append(analysis_dir)\n",
    "\n",
    "from utils import get_errors  # noqa: E402"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "pd.set_option(\"display.max_columns\", 200)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "### Load normalizer output for ClinVar variants located in the \"clinvar\" directory of this repo\n",
    "\n",
    "with gzip.open(\"../clinvar/vi-normalized-with-liftover.jsonl.gz\", \"rb\") as f:\n",
    "    records = ndjson.load(f)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "3757467"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(records)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "8a06e937c1ad47f5a86296c92188cbb9",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "  0%|          | 0/38 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "## read ClinVar normalizer responses into a pandas dataframe for analysis\n",
    "\n",
    "batch_size = 100000\n",
    "n_batches = len(records) // batch_size + 1\n",
    "\n",
    "df0 = pd.concat(\n",
    "    [\n",
    "        pd.json_normalize(records[k * batch_size : (k + 1) * batch_size])\n",
    "        for k in tqdm(range(n_batches))\n",
    "    ]\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "### re-run to re-initialize the ClinVar dataframe without re-running the above cell to read in the full dataframe\n",
    "df = df0.copy()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#  Identify CNVs in ClinVar"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "in.vrs_class\n",
       "Allele              3675021\n",
       "CopyNumberCount       44901\n",
       "CopyNumberChange      34074\n",
       "Unknown                1713\n",
       "Not Available          1112\n",
       "Haplotype               646\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "### different predicted behavior of variants based on variant type/what information provided by ClinVar\n",
    "df[\"in.vrs_class\"].value_counts()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Restrict to copy number variants. \n",
    "ClinVar variant types were further specified and identified in the ```in.vrs_class``` field before running the normalizer on ClinVar. \n",
    "For our analysis, CNVs are variants with ```in.vrs_class``` equal to one of  \n",
    "* ```CopyNumberCount```\n",
    "* ```CopyNumberChange```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = df[df[\"in.vrs_class\"].str.lower().str.contains(\"copy\")].copy()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "78975"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(df)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Remove variants which failed to normalize"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "### errors stored as a list of values, some of which are strings and other of which are dictionaries (determined by whether error was handled at the level of Variation Normalizer or after the normalizer)\n",
    "### this function extracts the text error responses for better readability and ease string processing\n",
    "### to get the core error message, we'll simply strip all the numeric values out and replace them with \"#\"\n",
    "def reduce_errors(error_string: str) -> str:\n",
    "    out = error_string.lower()\n",
    "    out = re.sub(r\"\\d+\", \"#\", out)\n",
    "    return out"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "### apply the error extraction function to our dataframe's column containing the errors\n",
    "df[\"error_string\"] = df[\"out.errors\"].fillna(\"\").apply(get_errors)\n",
    "df[\"error_string_reduce\"] = df[\"error_string\"].apply(reduce_errors)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Remove variants which failed to normalize \n",
    "Due to\n",
    "* liftover error\n",
    "* classification error\n",
    "* tokenization error\n",
    "* not supported by normalizer"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = df[df[\"error_string_reduce\"] == \"\"].copy()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "71730"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(df)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Calculate start/stop positions to compare to NCH variants\n",
    "### What start/stop position fields are available\n",
    "* ```out.subject.start``` is the normalized start value when the start is specified (similarly for ```end```)\n",
    "* If the start is a range, we extract values from the range (similarly for ```end```)\n",
    "\n",
    "### Our policy for choosing start/stop positions: Prioritize the absolute start and stop position (if known), otherwise inner, then outer \n",
    "* choose the specified normalized start value when available; otherwise the inner start (i.e. the max start); otherwise the outer start (i.e. the min start)\n",
    "* similarly, choose the normalized end value when available; otherwise the inner stop (i.e. the max stop); otherwise the outer stop (i.e. the max stop)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_location_interval_endpoint(\n",
    "    loc_interval: int | list, \n",
    "    endpoint=max #max or min\n",
    ") -> int:\n",
    "    \"\"\"Extract the location interval endpoint from a provided location\n",
    "    \n",
    "    :param loc_interval: An integer or list representing the interval\n",
    "    :param endpoint: A callable describing if the max or min from the interval\n",
    "        should be used\n",
    "    :return: An integer representing the location interval endpoint\n",
    "    \"\"\"\n",
    "    if type(loc_interval) is int:\n",
    "        return loc_interval\n",
    "    if type(loc_interval) is list:\n",
    "        return endpoint(set(loc_interval) - {None})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "df[\"start_38\"] = df[\"out.location.start\"].apply(\n",
    "    lambda loc_interval: get_location_interval_endpoint(loc_interval, endpoint=max)\n",
    ")\n",
    "\n",
    "df[\"stop_38\"] = df['out.location.end'].apply(\n",
    "    lambda loc_interval: get_location_interval_endpoint(loc_interval, endpoint=min)\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>in.assembly_version</th>\n",
       "      <th>out.location.start</th>\n",
       "      <th>out.location.end</th>\n",
       "      <th>in.range_copies</th>\n",
       "      <th>in.absolute_copies</th>\n",
       "      <th>out.copies</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [in.assembly_version, out.location.start, out.location.end, in.range_copies, in.absolute_copies, out.copies]\n",
       "Index: []"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "### Sanity check: Confirm that all variants are able to populate the start/stop position fields using our chosen preference rule\n",
    "\n",
    "df[(df[\"start_38\"].isna()) | (df[\"stop_38\"].isna())][\n",
    "    [\n",
    "        \"in.assembly_version\",\n",
    "        \"out.location.start\",\n",
    "        \"out.location.end\",\n",
    "        \"in.range_copies\",\n",
    "        \"in.absolute_copies\",\n",
    "        \"out.copies\"\n",
    "    ]\n",
    "]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Restrict to CNVs with calculated location information"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>out</th>\n",
       "      <th>in.variation_id</th>\n",
       "      <th>in.name</th>\n",
       "      <th>in.vrs_class</th>\n",
       "      <th>in.range_copies</th>\n",
       "      <th>in.issue</th>\n",
       "      <th>in.variation_type</th>\n",
       "      <th>in.subclass_type</th>\n",
       "      <th>in.cytogenetic</th>\n",
       "      <th>in.mappings</th>\n",
       "      <th>in.accession</th>\n",
       "      <th>in.fmt</th>\n",
       "      <th>in.source</th>\n",
       "      <th>in.precedence</th>\n",
       "      <th>out.errors</th>\n",
       "      <th>in.copy_change_type</th>\n",
       "      <th>in.assembly_version</th>\n",
       "      <th>in.chr</th>\n",
       "      <th>in.variant_length</th>\n",
       "      <th>out.id</th>\n",
       "      <th>out.type</th>\n",
       "      <th>out.digest</th>\n",
       "      <th>out.location.id</th>\n",
       "      <th>out.location.type</th>\n",
       "      <th>out.location.digest</th>\n",
       "      <th>out.location.sequenceReference.type</th>\n",
       "      <th>out.location.sequenceReference.refgetAccession</th>\n",
       "      <th>out.location.start</th>\n",
       "      <th>out.location.end</th>\n",
       "      <th>out.copyChange</th>\n",
       "      <th>out.state.type</th>\n",
       "      <th>out.state.length</th>\n",
       "      <th>out.state.sequence</th>\n",
       "      <th>out.state.repeatSubunitLength</th>\n",
       "      <th>out.extensions</th>\n",
       "      <th>in.absolute_copies</th>\n",
       "      <th>out.copies</th>\n",
       "      <th>error_string</th>\n",
       "      <th>error_string_reduce</th>\n",
       "      <th>start_38</th>\n",
       "      <th>stop_38</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>NaN</td>\n",
       "      <td>3247636</td>\n",
       "      <td>NC_000001.10:g.(?_100381624)_(100382277_?)del</td>\n",
       "      <td>CopyNumberChange</td>\n",
       "      <td>[]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>1p21.2</td>\n",
       "      <td>[]</td>\n",
       "      <td>NC_000001.10</td>\n",
       "      <td>hgvs</td>\n",
       "      <td>NC_000001.10:g.(?_100381624)_(100382277_?)del</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>loss</td>\n",
       "      <td>37</td>\n",
       "      <td>1</td>\n",
       "      <td>654</td>\n",
       "      <td>ga4gh:CX.ZuUeYk-wKkSruFZ3e2dh_UXdF_Z26bS7</td>\n",
       "      <td>CopyNumberChange</td>\n",
       "      <td>ZuUeYk-wKkSruFZ3e2dh_UXdF_Z26bS7</td>\n",
       "      <td>ga4gh:SL.cy-yXgOqImqYmkZASlKTDG-UKBs0B2mW</td>\n",
       "      <td>SequenceLocation</td>\n",
       "      <td>cy-yXgOqImqYmkZASlKTDG-UKBs0B2mW</td>\n",
       "      <td>SequenceReference</td>\n",
       "      <td>SQ.Ya6Rs7DHhDeg7YaOSg1EoNi3U_nQ9SvO</td>\n",
       "      <td>[None, 99916067]</td>\n",
       "      <td>[99916721, None]</td>\n",
       "      <td>loss</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>99916067</td>\n",
       "      <td>99916721</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>NaN</td>\n",
       "      <td>2425513</td>\n",
       "      <td>NC_000001.10:g.(?_153963273)_(154580482_?)del</td>\n",
       "      <td>CopyNumberChange</td>\n",
       "      <td>[]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>1q21.3</td>\n",
       "      <td>[]</td>\n",
       "      <td>NC_000001.10</td>\n",
       "      <td>hgvs</td>\n",
       "      <td>NC_000001.10:g.(?_153963273)_(154580482_?)del</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>loss</td>\n",
       "      <td>37</td>\n",
       "      <td>1</td>\n",
       "      <td>617210</td>\n",
       "      <td>ga4gh:CX.mmX_J3BieQI7iYl1G3bfQpvZvx-AMusn</td>\n",
       "      <td>CopyNumberChange</td>\n",
       "      <td>mmX_J3BieQI7iYl1G3bfQpvZvx-AMusn</td>\n",
       "      <td>ga4gh:SL.PlrS9sDMxKG1Z9YVW9eRrWmT_cFXU7f_</td>\n",
       "      <td>SequenceLocation</td>\n",
       "      <td>PlrS9sDMxKG1Z9YVW9eRrWmT_cFXU7f_</td>\n",
       "      <td>SequenceReference</td>\n",
       "      <td>SQ.Ya6Rs7DHhDeg7YaOSg1EoNi3U_nQ9SvO</td>\n",
       "      <td>[None, 153990796]</td>\n",
       "      <td>[154608006, None]</td>\n",
       "      <td>loss</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>153990796</td>\n",
       "      <td>154608006</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>NaN</td>\n",
       "      <td>1067607</td>\n",
       "      <td>NC_000001.10:g.(?_156104586)_(156108907_?)del</td>\n",
       "      <td>CopyNumberChange</td>\n",
       "      <td>[]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>1q22</td>\n",
       "      <td>[]</td>\n",
       "      <td>NC_000001.10</td>\n",
       "      <td>hgvs</td>\n",
       "      <td>NC_000001.10:g.(?_156104586)_(156108907_?)del</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>loss</td>\n",
       "      <td>37</td>\n",
       "      <td>1</td>\n",
       "      <td>4322</td>\n",
       "      <td>ga4gh:CX.8F4hrGUG68YFpcJCW2Aj-iKpxwbKXbG2</td>\n",
       "      <td>CopyNumberChange</td>\n",
       "      <td>8F4hrGUG68YFpcJCW2Aj-iKpxwbKXbG2</td>\n",
       "      <td>ga4gh:SL.AAxRpLgK_6iqDk7i4_XNWRxh04YhVIij</td>\n",
       "      <td>SequenceLocation</td>\n",
       "      <td>AAxRpLgK_6iqDk7i4_XNWRxh04YhVIij</td>\n",
       "      <td>SequenceReference</td>\n",
       "      <td>SQ.Ya6Rs7DHhDeg7YaOSg1EoNi3U_nQ9SvO</td>\n",
       "      <td>[None, 156134794]</td>\n",
       "      <td>[156139116, None]</td>\n",
       "      <td>loss</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>156134794</td>\n",
       "      <td>156139116</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>NaN</td>\n",
       "      <td>3247983</td>\n",
       "      <td>NC_000001.10:g.(?_169446833)_(169454047_?)del</td>\n",
       "      <td>CopyNumberChange</td>\n",
       "      <td>[]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>1q24.2</td>\n",
       "      <td>[]</td>\n",
       "      <td>NC_000001.10</td>\n",
       "      <td>hgvs</td>\n",
       "      <td>NC_000001.10:g.(?_169446833)_(169454047_?)del</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>loss</td>\n",
       "      <td>37</td>\n",
       "      <td>1</td>\n",
       "      <td>7215</td>\n",
       "      <td>ga4gh:CX.soEykkuwN3ddRcRc2N4G7VvKCJqym8w9</td>\n",
       "      <td>CopyNumberChange</td>\n",
       "      <td>soEykkuwN3ddRcRc2N4G7VvKCJqym8w9</td>\n",
       "      <td>ga4gh:SL.E-GzdBhCBCXrzEOH2Jh9A5zhKdlr0hKa</td>\n",
       "      <td>SequenceLocation</td>\n",
       "      <td>E-GzdBhCBCXrzEOH2Jh9A5zhKdlr0hKa</td>\n",
       "      <td>SequenceReference</td>\n",
       "      <td>SQ.Ya6Rs7DHhDeg7YaOSg1EoNi3U_nQ9SvO</td>\n",
       "      <td>[None, 169477594]</td>\n",
       "      <td>[169484809, None]</td>\n",
       "      <td>loss</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>169477594</td>\n",
       "      <td>169484809</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>NaN</td>\n",
       "      <td>3247768</td>\n",
       "      <td>NC_000001.10:g.(?_183538257)_(183543776_?)del</td>\n",
       "      <td>CopyNumberChange</td>\n",
       "      <td>[]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>1q25.3</td>\n",
       "      <td>[]</td>\n",
       "      <td>NC_000001.10</td>\n",
       "      <td>hgvs</td>\n",
       "      <td>NC_000001.10:g.(?_183538257)_(183543776_?)del</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>loss</td>\n",
       "      <td>37</td>\n",
       "      <td>1</td>\n",
       "      <td>5520</td>\n",
       "      <td>ga4gh:CX.wgUo3HPmYiknojW5Hx6w_b-wZKquqOGB</td>\n",
       "      <td>CopyNumberChange</td>\n",
       "      <td>wgUo3HPmYiknojW5Hx6w_b-wZKquqOGB</td>\n",
       "      <td>ga4gh:SL.MAXLYdyIQHFsrwpEuP4pJOo7wwi0PDOu</td>\n",
       "      <td>SequenceLocation</td>\n",
       "      <td>MAXLYdyIQHFsrwpEuP4pJOo7wwi0PDOu</td>\n",
       "      <td>SequenceReference</td>\n",
       "      <td>SQ.Ya6Rs7DHhDeg7YaOSg1EoNi3U_nQ9SvO</td>\n",
       "      <td>[None, 183569121]</td>\n",
       "      <td>[183574641, None]</td>\n",
       "      <td>loss</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>183569121</td>\n",
       "      <td>183574641</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    out in.variation_id                                        in.name  \\\n",
       "30  NaN         3247636  NC_000001.10:g.(?_100381624)_(100382277_?)del   \n",
       "32  NaN         2425513  NC_000001.10:g.(?_153963273)_(154580482_?)del   \n",
       "33  NaN         1067607  NC_000001.10:g.(?_156104586)_(156108907_?)del   \n",
       "34  NaN         3247983  NC_000001.10:g.(?_169446833)_(169454047_?)del   \n",
       "35  NaN         3247768  NC_000001.10:g.(?_183538257)_(183543776_?)del   \n",
       "\n",
       "        in.vrs_class in.range_copies in.issue in.variation_type  \\\n",
       "30  CopyNumberChange              []      NaN          Deletion   \n",
       "32  CopyNumberChange              []      NaN          Deletion   \n",
       "33  CopyNumberChange              []      NaN          Deletion   \n",
       "34  CopyNumberChange              []      NaN          Deletion   \n",
       "35  CopyNumberChange              []      NaN          Deletion   \n",
       "\n",
       "   in.subclass_type in.cytogenetic in.mappings  in.accession in.fmt  \\\n",
       "30     SimpleAllele         1p21.2          []  NC_000001.10   hgvs   \n",
       "32     SimpleAllele         1q21.3          []  NC_000001.10   hgvs   \n",
       "33     SimpleAllele           1q22          []  NC_000001.10   hgvs   \n",
       "34     SimpleAllele         1q24.2          []  NC_000001.10   hgvs   \n",
       "35     SimpleAllele         1q25.3          []  NC_000001.10   hgvs   \n",
       "\n",
       "                                        in.source in.precedence out.errors  \\\n",
       "30  NC_000001.10:g.(?_100381624)_(100382277_?)del             2        NaN   \n",
       "32  NC_000001.10:g.(?_153963273)_(154580482_?)del             2        NaN   \n",
       "33  NC_000001.10:g.(?_156104586)_(156108907_?)del             2        NaN   \n",
       "34  NC_000001.10:g.(?_169446833)_(169454047_?)del             2        NaN   \n",
       "35  NC_000001.10:g.(?_183538257)_(183543776_?)del             2        NaN   \n",
       "\n",
       "   in.copy_change_type in.assembly_version in.chr in.variant_length  \\\n",
       "30                loss                  37      1               654   \n",
       "32                loss                  37      1            617210   \n",
       "33                loss                  37      1              4322   \n",
       "34                loss                  37      1              7215   \n",
       "35                loss                  37      1              5520   \n",
       "\n",
       "                                       out.id          out.type  \\\n",
       "30  ga4gh:CX.ZuUeYk-wKkSruFZ3e2dh_UXdF_Z26bS7  CopyNumberChange   \n",
       "32  ga4gh:CX.mmX_J3BieQI7iYl1G3bfQpvZvx-AMusn  CopyNumberChange   \n",
       "33  ga4gh:CX.8F4hrGUG68YFpcJCW2Aj-iKpxwbKXbG2  CopyNumberChange   \n",
       "34  ga4gh:CX.soEykkuwN3ddRcRc2N4G7VvKCJqym8w9  CopyNumberChange   \n",
       "35  ga4gh:CX.wgUo3HPmYiknojW5Hx6w_b-wZKquqOGB  CopyNumberChange   \n",
       "\n",
       "                          out.digest  \\\n",
       "30  ZuUeYk-wKkSruFZ3e2dh_UXdF_Z26bS7   \n",
       "32  mmX_J3BieQI7iYl1G3bfQpvZvx-AMusn   \n",
       "33  8F4hrGUG68YFpcJCW2Aj-iKpxwbKXbG2   \n",
       "34  soEykkuwN3ddRcRc2N4G7VvKCJqym8w9   \n",
       "35  wgUo3HPmYiknojW5Hx6w_b-wZKquqOGB   \n",
       "\n",
       "                              out.location.id out.location.type  \\\n",
       "30  ga4gh:SL.cy-yXgOqImqYmkZASlKTDG-UKBs0B2mW  SequenceLocation   \n",
       "32  ga4gh:SL.PlrS9sDMxKG1Z9YVW9eRrWmT_cFXU7f_  SequenceLocation   \n",
       "33  ga4gh:SL.AAxRpLgK_6iqDk7i4_XNWRxh04YhVIij  SequenceLocation   \n",
       "34  ga4gh:SL.E-GzdBhCBCXrzEOH2Jh9A5zhKdlr0hKa  SequenceLocation   \n",
       "35  ga4gh:SL.MAXLYdyIQHFsrwpEuP4pJOo7wwi0PDOu  SequenceLocation   \n",
       "\n",
       "                 out.location.digest out.location.sequenceReference.type  \\\n",
       "30  cy-yXgOqImqYmkZASlKTDG-UKBs0B2mW                   SequenceReference   \n",
       "32  PlrS9sDMxKG1Z9YVW9eRrWmT_cFXU7f_                   SequenceReference   \n",
       "33  AAxRpLgK_6iqDk7i4_XNWRxh04YhVIij                   SequenceReference   \n",
       "34  E-GzdBhCBCXrzEOH2Jh9A5zhKdlr0hKa                   SequenceReference   \n",
       "35  MAXLYdyIQHFsrwpEuP4pJOo7wwi0PDOu                   SequenceReference   \n",
       "\n",
       "   out.location.sequenceReference.refgetAccession out.location.start  \\\n",
       "30            SQ.Ya6Rs7DHhDeg7YaOSg1EoNi3U_nQ9SvO   [None, 99916067]   \n",
       "32            SQ.Ya6Rs7DHhDeg7YaOSg1EoNi3U_nQ9SvO  [None, 153990796]   \n",
       "33            SQ.Ya6Rs7DHhDeg7YaOSg1EoNi3U_nQ9SvO  [None, 156134794]   \n",
       "34            SQ.Ya6Rs7DHhDeg7YaOSg1EoNi3U_nQ9SvO  [None, 169477594]   \n",
       "35            SQ.Ya6Rs7DHhDeg7YaOSg1EoNi3U_nQ9SvO  [None, 183569121]   \n",
       "\n",
       "     out.location.end out.copyChange out.state.type  out.state.length  \\\n",
       "30   [99916721, None]           loss            NaN               NaN   \n",
       "32  [154608006, None]           loss            NaN               NaN   \n",
       "33  [156139116, None]           loss            NaN               NaN   \n",
       "34  [169484809, None]           loss            NaN               NaN   \n",
       "35  [183574641, None]           loss            NaN               NaN   \n",
       "\n",
       "   out.state.sequence  out.state.repeatSubunitLength out.extensions  \\\n",
       "30                NaN                            NaN            NaN   \n",
       "32                NaN                            NaN            NaN   \n",
       "33                NaN                            NaN            NaN   \n",
       "34                NaN                            NaN            NaN   \n",
       "35                NaN                            NaN            NaN   \n",
       "\n",
       "   in.absolute_copies  out.copies error_string error_string_reduce   start_38  \\\n",
       "30                NaN         NaN                                    99916067   \n",
       "32                NaN         NaN                                   153990796   \n",
       "33                NaN         NaN                                   156134794   \n",
       "34                NaN         NaN                                   169477594   \n",
       "35                NaN         NaN                                   183569121   \n",
       "\n",
       "      stop_38  \n",
       "30   99916721  \n",
       "32  154608006  \n",
       "33  156139116  \n",
       "34  169484809  \n",
       "35  183574641  "
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>variation_id</th>\n",
       "      <th>name</th>\n",
       "      <th>variation_type</th>\n",
       "      <th>assembly_version</th>\n",
       "      <th>chr</th>\n",
       "      <th>cytogenetic</th>\n",
       "      <th>start_38</th>\n",
       "      <th>stop_38</th>\n",
       "      <th>range_copies</th>\n",
       "      <th>absolute_copies</th>\n",
       "      <th>copies</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>3247636</td>\n",
       "      <td>NC_000001.10:g.(?_100381624)_(100382277_?)del</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>37</td>\n",
       "      <td>1</td>\n",
       "      <td>1p21.2</td>\n",
       "      <td>99916067</td>\n",
       "      <td>99916721</td>\n",
       "      <td>[]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>2425513</td>\n",
       "      <td>NC_000001.10:g.(?_153963273)_(154580482_?)del</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>37</td>\n",
       "      <td>1</td>\n",
       "      <td>1q21.3</td>\n",
       "      <td>153990796</td>\n",
       "      <td>154608006</td>\n",
       "      <td>[]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>1067607</td>\n",
       "      <td>NC_000001.10:g.(?_156104586)_(156108907_?)del</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>37</td>\n",
       "      <td>1</td>\n",
       "      <td>1q22</td>\n",
       "      <td>156134794</td>\n",
       "      <td>156139116</td>\n",
       "      <td>[]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>3247983</td>\n",
       "      <td>NC_000001.10:g.(?_169446833)_(169454047_?)del</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>37</td>\n",
       "      <td>1</td>\n",
       "      <td>1q24.2</td>\n",
       "      <td>169477594</td>\n",
       "      <td>169484809</td>\n",
       "      <td>[]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>3247768</td>\n",
       "      <td>NC_000001.10:g.(?_183538257)_(183543776_?)del</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>37</td>\n",
       "      <td>1</td>\n",
       "      <td>1q25.3</td>\n",
       "      <td>183569121</td>\n",
       "      <td>183574641</td>\n",
       "      <td>[]</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>89476</th>\n",
       "      <td>59269</td>\n",
       "      <td>GRCh38/hg38 Xp11.4-11.3(chrX:41823849-44240337)x1</td>\n",
       "      <td>copy number loss</td>\n",
       "      <td>38</td>\n",
       "      <td>X</td>\n",
       "      <td>Xp11.4-11.3</td>\n",
       "      <td>41823848</td>\n",
       "      <td>44240337</td>\n",
       "      <td>[]</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>89477</th>\n",
       "      <td>148019</td>\n",
       "      <td>GRCh38/hg38 Xq13.3-28(chrX:75086417-156022206)x1</td>\n",
       "      <td>copy number loss</td>\n",
       "      <td>38</td>\n",
       "      <td>X</td>\n",
       "      <td>Xq13.3-28</td>\n",
       "      <td>75086416</td>\n",
       "      <td>156022206</td>\n",
       "      <td>[]</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>89478</th>\n",
       "      <td>59236</td>\n",
       "      <td>GRCh38/hg38 Xp22.31-22.2(chrX:9540020-13128124)x1</td>\n",
       "      <td>copy number loss</td>\n",
       "      <td>38</td>\n",
       "      <td>X</td>\n",
       "      <td>Xp22.31-22.2</td>\n",
       "      <td>9540019</td>\n",
       "      <td>13128124</td>\n",
       "      <td>[]</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>89479</th>\n",
       "      <td>2579183</td>\n",
       "      <td>GRCh38/hg38 Xq24(chrX:119582581-119589158)x0</td>\n",
       "      <td>copy number loss</td>\n",
       "      <td>38</td>\n",
       "      <td>X</td>\n",
       "      <td>Xq24</td>\n",
       "      <td>119582580</td>\n",
       "      <td>119589158</td>\n",
       "      <td>[]</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>89482</th>\n",
       "      <td>149027</td>\n",
       "      <td>GRCh38/hg38 Yq11.223(chrY:21949807-22728213)x0</td>\n",
       "      <td>copy number loss</td>\n",
       "      <td>38</td>\n",
       "      <td>Y</td>\n",
       "      <td>Yq11.223</td>\n",
       "      <td>21949806</td>\n",
       "      <td>22728213</td>\n",
       "      <td>[]</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>71730 rows Ã— 11 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      variation_id                                               name  \\\n",
       "30         3247636      NC_000001.10:g.(?_100381624)_(100382277_?)del   \n",
       "32         2425513      NC_000001.10:g.(?_153963273)_(154580482_?)del   \n",
       "33         1067607      NC_000001.10:g.(?_156104586)_(156108907_?)del   \n",
       "34         3247983      NC_000001.10:g.(?_169446833)_(169454047_?)del   \n",
       "35         3247768      NC_000001.10:g.(?_183538257)_(183543776_?)del   \n",
       "...            ...                                                ...   \n",
       "89476        59269  GRCh38/hg38 Xp11.4-11.3(chrX:41823849-44240337)x1   \n",
       "89477       148019   GRCh38/hg38 Xq13.3-28(chrX:75086417-156022206)x1   \n",
       "89478        59236  GRCh38/hg38 Xp22.31-22.2(chrX:9540020-13128124)x1   \n",
       "89479      2579183       GRCh38/hg38 Xq24(chrX:119582581-119589158)x0   \n",
       "89482       149027     GRCh38/hg38 Yq11.223(chrY:21949807-22728213)x0   \n",
       "\n",
       "         variation_type assembly_version chr   cytogenetic   start_38  \\\n",
       "30             Deletion               37   1        1p21.2   99916067   \n",
       "32             Deletion               37   1        1q21.3  153990796   \n",
       "33             Deletion               37   1          1q22  156134794   \n",
       "34             Deletion               37   1        1q24.2  169477594   \n",
       "35             Deletion               37   1        1q25.3  183569121   \n",
       "...                 ...              ...  ..           ...        ...   \n",
       "89476  copy number loss               38   X   Xp11.4-11.3   41823848   \n",
       "89477  copy number loss               38   X     Xq13.3-28   75086416   \n",
       "89478  copy number loss               38   X  Xp22.31-22.2    9540019   \n",
       "89479  copy number loss               38   X          Xq24  119582580   \n",
       "89482  copy number loss               38   Y      Yq11.223   21949806   \n",
       "\n",
       "         stop_38 range_copies absolute_copies  copies  \n",
       "30      99916721           []             NaN     NaN  \n",
       "32     154608006           []             NaN     NaN  \n",
       "33     156139116           []             NaN     NaN  \n",
       "34     169484809           []             NaN     NaN  \n",
       "35     183574641           []             NaN     NaN  \n",
       "...          ...          ...             ...     ...  \n",
       "89476   44240337           []               1     1.0  \n",
       "89477  156022206           []               1     1.0  \n",
       "89478   13128124           []               1     1.0  \n",
       "89479  119589158           []               0     0.0  \n",
       "89482   22728213           []               0     0.0  \n",
       "\n",
       "[71730 rows x 11 columns]"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "## restrict dataframe to minimal necessary fields, clean up column names for export and downstream analyses\n",
    "\n",
    "df = df[\n",
    "    [\n",
    "        \"in.variation_id\",\n",
    "        \"in.name\",\n",
    "        \"in.variation_type\",\n",
    "        \"in.assembly_version\",\n",
    "        \"in.chr\",\n",
    "        \"in.cytogenetic\",\n",
    "        \"start_38\",\n",
    "        \"stop_38\",\n",
    "        \"in.range_copies\",\n",
    "        \"in.absolute_copies\",\n",
    "        \"out.copies\"\n",
    "    ]\n",
    "]\n",
    "df = df.rename(columns={c: c.split(\".\")[-1] for c in df.columns})\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "df.to_csv(\"cnv_data/ClinVar-CNVs-normalized.csv.gzip\", index=False, compression=\"gzip\")"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "variation-normalizer-manuscript",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.13.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
