{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "ClinVar!"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Initialize"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import ndjson\n",
    "import pandas as pd\n",
    "import numpy as np"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "#import logging\n",
    "#from enum import Enum\n",
    "import re\n",
    "#import csv\n",
    "#from pathlib import Path\n",
    "#import zipfile\n",
    "\n",
    "from dotenv import load_dotenv\n",
    "\n",
    "#from variation.query import QueryHandler\n",
    "\n",
    "#logging.getLogger(\"root\").setLevel(logging.WARNING)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 131,
   "metadata": {},
   "outputs": [],
   "source": [
    "file = open('/Users/rsaxs014/Downloads/output-variation_identity.ndjson')\n",
    "records = ndjson.load(file)\n",
    "\n",
    "df0 = pd.json_normalize(records)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 176,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = df0.copy()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Add Supported Status- based on in.vrs_xform_plan.policy"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 213,
   "metadata": {},
   "outputs": [],
   "source": [
    "df['in.vrs_xform_plan.policy'] = df['in.vrs_xform_plan.policy'].fillna(\"None\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 214,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "in.vrs_xform_plan.policy\n",
       "Canonical SPDI                                      2118669\n",
       "Absolute copy count                                   53263\n",
       "Copy number change (cn loss|del and cn gain|dup)      27104\n",
       "NCBI36 genomic only                                    4771\n",
       "No hgvs or location info                               3089\n",
       "Genotype/Haplotype                                     1440\n",
       "Invalid/unsupported hgvs                               1336\n",
       "Remaining valid hgvs alleles                            941\n",
       "Min/max copy count range not supported                   14\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 214,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df[\"in.vrs_xform_plan.policy\"].value_counts()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "TODO: look into using a dictionary to replace below"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 215,
   "metadata": {},
   "outputs": [],
   "source": [
    "df[\"support_status\"] = df[\"in.vrs_xform_plan.policy\"].copy()\n",
    "\n",
    "df.loc[df[\"support_status\"] == \"Canonical SPDI\", \"support_status\"] = True\n",
    "df.loc[df[\"support_status\"] == \"Absolute copy count\", \"support_status\"] = True\n",
    "df.loc[df[\"support_status\"] == \"Copy number change (cn loss|del and cn gain|dup)\", \"support_status\"] = True\n",
    "df.loc[df[\"support_status\"] == \"NCBI36 genomic only\", \"support_status\"] = False\n",
    "df.loc[df[\"support_status\"] == \"No hgvs or location info\", \"support_status\"] = False\n",
    "df.loc[df[\"support_status\"] == \"Genotype/Haplotype\", \"support_status\"] = False\n",
    "df.loc[df[\"support_status\"] == \"Invalid/unsupported hgvs\", \"support_status\"] = False\n",
    "df.loc[df[\"support_status\"] == \"Remaining valid hgvs alleles\", \"support_status\"] = True\n",
    "df.loc[df[\"support_status\"] == \"Min/max copy count range not supported\", \"support_status\"] = False\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 216,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "support_status\n",
       "True     2199977\n",
       "False      10650\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 216,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df['support_status'].value_counts()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Add Normalization Status- based on out.errors"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The errors are stored as a list of values, some of which are strings and other of which are dictionaries (determined by whether error was handled at the level of Variation Normalizer or after the normalizer)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The \"get_errors\" function extracts the text error responses for better readability and ease string processing"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 304,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_errors(errors):\n",
    "    errors_out = []\n",
    "    for e in errors:\n",
    "        if type(e)== str:\n",
    "            errors_out.append(e)\n",
    "        elif type(e)==dict:\n",
    "            for k,v in e.items():\n",
    "                if k not in ['msg','response-errors']: ## only get these keys from normalizer response\n",
    "                    continue\n",
    "                if type(v) == str:\n",
    "                    errors_out.append(v)\n",
    "                elif type(v) == list:\n",
    "                    errors_out.append(';'.join(v))\n",
    "                #errors_out.append('[NORMALIZER] '+';'.join(e.get('response-errors',[])))\n",
    "    return ';'.join(errors_out)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 305,
   "metadata": {},
   "outputs": [],
   "source": [
    "df['error_string'] = df['out.errors'].fillna('').apply(get_errors)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This is the number of unique error strings"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "There are many different strings because many of the errors contain specific genomic coordinates, which are unlikely to occur more than once"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 306,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "61185"
      ]
     },
     "execution_count": 306,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df['error_string'].nunique()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "To get the core error message, the numeric values are replaced with \"#\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 307,
   "metadata": {},
   "outputs": [],
   "source": [
    "def reduce_errors(error_string):\n",
    "    out = error_string.lower()\n",
    "    out = re.sub('\\d+','#',out)\n",
    "    return out"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 308,
   "metadata": {},
   "outputs": [],
   "source": [
    "df['error_string_reduce'] = df['error_string'].apply(reduce_errors)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 309,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "error_string_reduce\n",
       "                                                                                                                                                       2141287\n",
       "variation record was too long to normalize (#), treating as text                                                                                         61464\n",
       "error returned from variation normalizer;position, #, does not exist on nc_#.#                                                                            5832\n",
       "error returned from variation normalizer;copy_number_count mode requires `baseline_copies`                                                                1562\n",
       "unrecognized variation record                                                                                                                              431\n",
       "error returned from variation normalizer;unable to tokenize g.(#_?)_(?_#)del                                                                                16\n",
       "error returned from variation normalizer;unable to tokenize m.#_#del                                                                                        16\n",
       "error returned from variation normalizer;unable to tokenize g.(#_#)_(#_#)del                                                                                 3\n",
       "error returned from variation normalizer;unable to find valid result for classifications: {'genomic substitution'}                                           3\n",
       "error returned from variation normalizer;unable to tokenize g.(#_#)_(#_#)dup                                                                                 2\n",
       "error returned from variation normalizer;unable to find valid result for classifications: {'genomic duplication'}                                            2\n",
       "error returned from variation normalizer;unable to tokenize g.(?_#)_(#_?)del                                                                                 2\n",
       "error returned from variation normalizer;unable to tokenize g.(#_#)_(#_?)del                                                                                 2\n",
       "not status #                                                                                                                                                 1\n",
       "error returned from variation normalizer;nc_#.#:g.#_#insgccgtgggccccgccaccgccgccgtgccaccgcc is not a supported hgvs genomic duplication or deletion          1\n",
       "error returned from variation normalizer;unable to find valid result for classifications: {'genomic uncertain deletion'}                                     1\n",
       "error returned from variation normalizer;unable to tokenize g.#_#instattt                                                                                    1\n",
       "error returned from variation normalizer;unable to tokenize g.#_#dup                                                                                         1\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 309,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df['error_string_reduce'].value_counts()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "There are Not Supported variants that have no error because they were labeled \"Not Supported\" manually."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "An error (\"Not Supported\") is entered manually for those variants so that they are not categorized as normalized"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 310,
   "metadata": {},
   "outputs": [],
   "source": [
    "df.loc[(df[\"support_status\"] == False) & (df[\"error_string_reduce\"] == ''), \"error_string_reduce\"] = \"Not Supported\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The variants with no recorded error, have \"error_string_reduce\" string equal to the empty string and not NaN, so the empty string will be replaced with an NaN"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 311,
   "metadata": {},
   "outputs": [
    {
     "ename": "KeyboardInterrupt",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[311], line 1\u001b[0m\n\u001b[0;32m----> 1\u001b[0m df[\u001b[38;5;124m'\u001b[39m\u001b[38;5;124merror_string_reduce\u001b[39m\u001b[38;5;124m'\u001b[39m] \u001b[38;5;241m=\u001b[39m \u001b[43mdf\u001b[49m\u001b[43m[\u001b[49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[38;5;124;43merror_string_reduce\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[43m]\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mapply\u001b[49m\u001b[43m(\u001b[49m\u001b[43mreduce_errors\u001b[49m\u001b[43m)\u001b[49m\u001b[38;5;241m.\u001b[39mreplace(\u001b[38;5;124m'\u001b[39m\u001b[38;5;124m'\u001b[39m,np\u001b[38;5;241m.\u001b[39mnan)\n",
      "File \u001b[0;32m~/Documents/git/genomic_med_lab/variation-normalizer-manuscript/.venv/lib/python3.10/site-packages/pandas/core/series.py:4630\u001b[0m, in \u001b[0;36mSeries.apply\u001b[0;34m(self, func, convert_dtype, args, **kwargs)\u001b[0m\n\u001b[1;32m   4520\u001b[0m \u001b[39mdef\u001b[39;00m \u001b[39mapply\u001b[39m(\n\u001b[1;32m   4521\u001b[0m     \u001b[39mself\u001b[39m,\n\u001b[1;32m   4522\u001b[0m     func: AggFuncType,\n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m   4525\u001b[0m     \u001b[39m*\u001b[39m\u001b[39m*\u001b[39mkwargs,\n\u001b[1;32m   4526\u001b[0m ) \u001b[39m-\u001b[39m\u001b[39m>\u001b[39m DataFrame \u001b[39m|\u001b[39m Series:\n\u001b[1;32m   4527\u001b[0m \u001b[39m    \u001b[39m\u001b[39m\"\"\"\u001b[39;00m\n\u001b[1;32m   4528\u001b[0m \u001b[39m    Invoke function on values of Series.\u001b[39;00m\n\u001b[1;32m   4529\u001b[0m \n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m   4628\u001b[0m \u001b[39m    dtype: float64\u001b[39;00m\n\u001b[1;32m   4629\u001b[0m \u001b[39m    \"\"\"\u001b[39;00m\n\u001b[0;32m-> 4630\u001b[0m     \u001b[39mreturn\u001b[39;00m SeriesApply(\u001b[39mself\u001b[39;49m, func, convert_dtype, args, kwargs)\u001b[39m.\u001b[39;49mapply()\n",
      "File \u001b[0;32m~/Documents/git/genomic_med_lab/variation-normalizer-manuscript/.venv/lib/python3.10/site-packages/pandas/core/apply.py:1025\u001b[0m, in \u001b[0;36mSeriesApply.apply\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m   1022\u001b[0m     \u001b[39mreturn\u001b[39;00m \u001b[39mself\u001b[39m\u001b[39m.\u001b[39mapply_str()\n\u001b[1;32m   1024\u001b[0m \u001b[39m# self.f is Callable\u001b[39;00m\n\u001b[0;32m-> 1025\u001b[0m \u001b[39mreturn\u001b[39;00m \u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49mapply_standard()\n",
      "File \u001b[0;32m~/Documents/git/genomic_med_lab/variation-normalizer-manuscript/.venv/lib/python3.10/site-packages/pandas/core/apply.py:1076\u001b[0m, in \u001b[0;36mSeriesApply.apply_standard\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m   1074\u001b[0m     \u001b[39melse\u001b[39;00m:\n\u001b[1;32m   1075\u001b[0m         values \u001b[39m=\u001b[39m obj\u001b[39m.\u001b[39mastype(\u001b[39mobject\u001b[39m)\u001b[39m.\u001b[39m_values\n\u001b[0;32m-> 1076\u001b[0m         mapped \u001b[39m=\u001b[39m lib\u001b[39m.\u001b[39;49mmap_infer(\n\u001b[1;32m   1077\u001b[0m             values,\n\u001b[1;32m   1078\u001b[0m             f,\n\u001b[1;32m   1079\u001b[0m             convert\u001b[39m=\u001b[39;49m\u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49mconvert_dtype,\n\u001b[1;32m   1080\u001b[0m         )\n\u001b[1;32m   1082\u001b[0m \u001b[39mif\u001b[39;00m \u001b[39mlen\u001b[39m(mapped) \u001b[39mand\u001b[39;00m \u001b[39misinstance\u001b[39m(mapped[\u001b[39m0\u001b[39m], ABCSeries):\n\u001b[1;32m   1083\u001b[0m     \u001b[39m# GH#43986 Need to do list(mapped) in order to get treated as nested\u001b[39;00m\n\u001b[1;32m   1084\u001b[0m     \u001b[39m#  See also GH#25959 regarding EA support\u001b[39;00m\n\u001b[1;32m   1085\u001b[0m     \u001b[39mreturn\u001b[39;00m obj\u001b[39m.\u001b[39m_constructor_expanddim(\u001b[39mlist\u001b[39m(mapped), index\u001b[39m=\u001b[39mobj\u001b[39m.\u001b[39mindex)\n",
      "File \u001b[0;32m~/Documents/git/genomic_med_lab/variation-normalizer-manuscript/.venv/lib/python3.10/site-packages/pandas/_libs/lib.pyx:2834\u001b[0m, in \u001b[0;36mpandas._libs.lib.map_infer\u001b[0;34m()\u001b[0m\n",
      "Cell \u001b[0;32mIn[307], line 2\u001b[0m, in \u001b[0;36mreduce_errors\u001b[0;34m(error_string)\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21mreduce_errors\u001b[39m(error_string):\n\u001b[0;32m----> 2\u001b[0m     out \u001b[38;5;241m=\u001b[39m \u001b[43merror_string\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mlower\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m      3\u001b[0m     out \u001b[38;5;241m=\u001b[39m re\u001b[38;5;241m.\u001b[39msub(\u001b[38;5;124m'\u001b[39m\u001b[38;5;124m\\\u001b[39m\u001b[38;5;124md+\u001b[39m\u001b[38;5;124m'\u001b[39m,\u001b[38;5;124m'\u001b[39m\u001b[38;5;124m#\u001b[39m\u001b[38;5;124m'\u001b[39m,out)\n\u001b[1;32m      4\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m out\n",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m: "
     ]
    }
   ],
   "source": [
    "df['error_string_reduce'] = df['error_string_reduce'].apply(reduce_errors).replace('',np.nan)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "normalize_status\n",
       "variation record was too long to normalize (#), treating as text                                                                                       61464\n",
       "not supported                                                                                                                                          10200\n",
       "error returned from variation normalizer;position, #, does not exist on nc_#.#                                                                          5832\n",
       "error returned from variation normalizer;copy_number_count mode requires `baseline_copies`                                                              1562\n",
       "unrecognized variation record                                                                                                                            431\n",
       "error returned from variation normalizer;unable to tokenize g.(#_?)_(?_#)del                                                                              16\n",
       "error returned from variation normalizer;unable to tokenize m.#_#del                                                                                      16\n",
       "error returned from variation normalizer;unable to tokenize g.(#_#)_(#_#)del                                                                               3\n",
       "error returned from variation normalizer;unable to find valid result for classifications: {'genomic substitution'}                                         3\n",
       "error returned from variation normalizer;unable to tokenize g.(#_#)_(#_#)dup                                                                               2\n",
       "error returned from variation normalizer;unable to find valid result for classifications: {'genomic duplication'}                                          2\n",
       "error returned from variation normalizer;unable to tokenize g.(?_#)_(#_?)del                                                                               2\n",
       "error returned from variation normalizer;unable to tokenize g.(#_#)_(#_?)del                                                                               2\n",
       "not status #                                                                                                                                               1\n",
       "error returned from variation normalizer;nc_#.#:g.#_#insgccgtgggccccgccaccgccgccgtgccaccgcc is not a supported hgvs genomic duplication or deletion        1\n",
       "error returned from variation normalizer;unable to find valid result for classifications: {'genomic uncertain deletion'}                                   1\n",
       "error returned from variation normalizer;unable to tokenize g.#_#instattt                                                                                  1\n",
       "error returned from variation normalizer;unable to tokenize g.#_#dup                                                                                       1\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 283,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df['error_string_reduce'].value_counts()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Mark all empty errors as TRUE (for normalize status) and all variants with an error present as FALSE"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>in.id</th>\n",
       "      <th>in.name</th>\n",
       "      <th>in.subclass_type</th>\n",
       "      <th>in.variation_type</th>\n",
       "      <th>in.xrefs</th>\n",
       "      <th>in.vrs_xform_plan.type</th>\n",
       "      <th>in.vrs_xform_plan.inputs</th>\n",
       "      <th>in.vrs_xform_plan.policy</th>\n",
       "      <th>out.id</th>\n",
       "      <th>out.type</th>\n",
       "      <th>...</th>\n",
       "      <th>out.state.seq_expr.location.start.value</th>\n",
       "      <th>out.state.seq_expr.location.end.type</th>\n",
       "      <th>out.state.seq_expr.location.end.value</th>\n",
       "      <th>out.state.seq_expr.reverse_complement</th>\n",
       "      <th>out.state.count.type</th>\n",
       "      <th>out.state.count.value</th>\n",
       "      <th>support_status</th>\n",
       "      <th>error_string</th>\n",
       "      <th>error_string_reduce</th>\n",
       "      <th>normalize_status</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>16098</td>\n",
       "      <td>SLC2A2, 1-BP INS, 793C</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>Insertion</td>\n",
       "      <td>OMIM:138160.0009</td>\n",
       "      <td>Text</td>\n",
       "      <td>[id]</td>\n",
       "      <td>No hgvs or location info</td>\n",
       "      <td>Text:clinvar:16098</td>\n",
       "      <td>Text</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>425693</td>\n",
       "      <td>NM_001204.6(BMPR2):c.77-?_247+?dup</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>Duplication</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Text</td>\n",
       "      <td>[id]</td>\n",
       "      <td>No hgvs or location info</td>\n",
       "      <td>Text:clinvar:425693</td>\n",
       "      <td>Text</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>90650</td>\n",
       "      <td>NM_000251.2(MSH2):c.1387-?_1510+?del</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Text</td>\n",
       "      <td>[id]</td>\n",
       "      <td>No hgvs or location info</td>\n",
       "      <td>Text:clinvar:90650</td>\n",
       "      <td>Text</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>2446408</td>\n",
       "      <td>CDHR1, 783G-A (rs147346345)</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>OMIM:609502.0005</td>\n",
       "      <td>Text</td>\n",
       "      <td>[id]</td>\n",
       "      <td>No hgvs or location info</td>\n",
       "      <td>Text:clinvar:2446408</td>\n",
       "      <td>Text</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>14905</td>\n",
       "      <td>HLA-DRB1, HLA-DRB1*1101</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>Variation</td>\n",
       "      <td>OMIM:142857.0001</td>\n",
       "      <td>Text</td>\n",
       "      <td>[id]</td>\n",
       "      <td>No hgvs or location info</td>\n",
       "      <td>Text:clinvar:14905</td>\n",
       "      <td>Text</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2210622</th>\n",
       "      <td>1464440</td>\n",
       "      <td>NM_001267550.2(TTN):c.29512_29513insGGCCGGGCGC...</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>Insertion</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Text</td>\n",
       "      <td>[id]</td>\n",
       "      <td>Invalid/unsupported hgvs</td>\n",
       "      <td>Text:clinvar:1464440</td>\n",
       "      <td>Text</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2210623</th>\n",
       "      <td>1453116</td>\n",
       "      <td>NM_000051.4(ATM):c.3376_3377insGGCCGGGCGCGGTGG...</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>Insertion</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Text</td>\n",
       "      <td>[id]</td>\n",
       "      <td>Invalid/unsupported hgvs</td>\n",
       "      <td>Text:clinvar:1453116</td>\n",
       "      <td>Text</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2210624</th>\n",
       "      <td>1181160</td>\n",
       "      <td>NM_030962.4(SBF2):c.55+97_55+98insCGGGCGTCGGGGC</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>Microsatellite</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Allele</td>\n",
       "      <td>[canonical_spdi]</td>\n",
       "      <td>Canonical SPDI</td>\n",
       "      <td>ga4gh:VA.x2E2GJQA5ovZFJ1SKxg3oHzkUXv5ZGfv</td>\n",
       "      <td>Allele</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2210625</th>\n",
       "      <td>2202105</td>\n",
       "      <td>NM_153676.4(USH1C):c.496+14_496+15insGTACTCCAT...</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>Microsatellite</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Allele</td>\n",
       "      <td>[canonical_spdi]</td>\n",
       "      <td>Canonical SPDI</td>\n",
       "      <td>ga4gh:VA.5jQEJZIbf1ttjbIGJVhfrHR9XcB7vqWD</td>\n",
       "      <td>Allele</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2210626</th>\n",
       "      <td>1801464</td>\n",
       "      <td>NM_176787.5(PIGN):c.2673-514_2673-513insGCAAAC...</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>Insertion</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Allele</td>\n",
       "      <td>[hgvs]</td>\n",
       "      <td>Remaining valid hgvs alleles</td>\n",
       "      <td>ga4gh:VA.3BXZ9oWeEqz4VM16yV8mgjvQ8ZTElB2J</td>\n",
       "      <td>Allele</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2210627 rows × 103 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           in.id                                            in.name  \\\n",
       "0          16098                             SLC2A2, 1-BP INS, 793C   \n",
       "1         425693                 NM_001204.6(BMPR2):c.77-?_247+?dup   \n",
       "2          90650               NM_000251.2(MSH2):c.1387-?_1510+?del   \n",
       "3        2446408                        CDHR1, 783G-A (rs147346345)   \n",
       "4          14905                            HLA-DRB1, HLA-DRB1*1101   \n",
       "...          ...                                                ...   \n",
       "2210622  1464440  NM_001267550.2(TTN):c.29512_29513insGGCCGGGCGC...   \n",
       "2210623  1453116  NM_000051.4(ATM):c.3376_3377insGGCCGGGCGCGGTGG...   \n",
       "2210624  1181160    NM_030962.4(SBF2):c.55+97_55+98insCGGGCGTCGGGGC   \n",
       "2210625  2202105  NM_153676.4(USH1C):c.496+14_496+15insGTACTCCAT...   \n",
       "2210626  1801464  NM_176787.5(PIGN):c.2673-514_2673-513insGCAAAC...   \n",
       "\n",
       "        in.subclass_type          in.variation_type          in.xrefs  \\\n",
       "0           SimpleAllele                  Insertion  OMIM:138160.0009   \n",
       "1           SimpleAllele                Duplication               NaN   \n",
       "2           SimpleAllele                   Deletion               NaN   \n",
       "3           SimpleAllele  single nucleotide variant  OMIM:609502.0005   \n",
       "4           SimpleAllele                  Variation  OMIM:142857.0001   \n",
       "...                  ...                        ...               ...   \n",
       "2210622     SimpleAllele                  Insertion               NaN   \n",
       "2210623     SimpleAllele                  Insertion               NaN   \n",
       "2210624     SimpleAllele             Microsatellite               NaN   \n",
       "2210625     SimpleAllele             Microsatellite               NaN   \n",
       "2210626     SimpleAllele                  Insertion               NaN   \n",
       "\n",
       "        in.vrs_xform_plan.type in.vrs_xform_plan.inputs  \\\n",
       "0                         Text                     [id]   \n",
       "1                         Text                     [id]   \n",
       "2                         Text                     [id]   \n",
       "3                         Text                     [id]   \n",
       "4                         Text                     [id]   \n",
       "...                        ...                      ...   \n",
       "2210622                   Text                     [id]   \n",
       "2210623                   Text                     [id]   \n",
       "2210624                 Allele         [canonical_spdi]   \n",
       "2210625                 Allele         [canonical_spdi]   \n",
       "2210626                 Allele                   [hgvs]   \n",
       "\n",
       "             in.vrs_xform_plan.policy  \\\n",
       "0            No hgvs or location info   \n",
       "1            No hgvs or location info   \n",
       "2            No hgvs or location info   \n",
       "3            No hgvs or location info   \n",
       "4            No hgvs or location info   \n",
       "...                               ...   \n",
       "2210622      Invalid/unsupported hgvs   \n",
       "2210623      Invalid/unsupported hgvs   \n",
       "2210624                Canonical SPDI   \n",
       "2210625                Canonical SPDI   \n",
       "2210626  Remaining valid hgvs alleles   \n",
       "\n",
       "                                            out.id out.type  ...  \\\n",
       "0                               Text:clinvar:16098     Text  ...   \n",
       "1                              Text:clinvar:425693     Text  ...   \n",
       "2                               Text:clinvar:90650     Text  ...   \n",
       "3                             Text:clinvar:2446408     Text  ...   \n",
       "4                               Text:clinvar:14905     Text  ...   \n",
       "...                                            ...      ...  ...   \n",
       "2210622                       Text:clinvar:1464440     Text  ...   \n",
       "2210623                       Text:clinvar:1453116     Text  ...   \n",
       "2210624  ga4gh:VA.x2E2GJQA5ovZFJ1SKxg3oHzkUXv5ZGfv   Allele  ...   \n",
       "2210625  ga4gh:VA.5jQEJZIbf1ttjbIGJVhfrHR9XcB7vqWD   Allele  ...   \n",
       "2210626  ga4gh:VA.3BXZ9oWeEqz4VM16yV8mgjvQ8ZTElB2J   Allele  ...   \n",
       "\n",
       "        out.state.seq_expr.location.start.value  \\\n",
       "0                                           NaN   \n",
       "1                                           NaN   \n",
       "2                                           NaN   \n",
       "3                                           NaN   \n",
       "4                                           NaN   \n",
       "...                                         ...   \n",
       "2210622                                     NaN   \n",
       "2210623                                     NaN   \n",
       "2210624                                     NaN   \n",
       "2210625                                     NaN   \n",
       "2210626                                     NaN   \n",
       "\n",
       "        out.state.seq_expr.location.end.type  \\\n",
       "0                                        NaN   \n",
       "1                                        NaN   \n",
       "2                                        NaN   \n",
       "3                                        NaN   \n",
       "4                                        NaN   \n",
       "...                                      ...   \n",
       "2210622                                  NaN   \n",
       "2210623                                  NaN   \n",
       "2210624                                  NaN   \n",
       "2210625                                  NaN   \n",
       "2210626                                  NaN   \n",
       "\n",
       "        out.state.seq_expr.location.end.value  \\\n",
       "0                                         NaN   \n",
       "1                                         NaN   \n",
       "2                                         NaN   \n",
       "3                                         NaN   \n",
       "4                                         NaN   \n",
       "...                                       ...   \n",
       "2210622                                   NaN   \n",
       "2210623                                   NaN   \n",
       "2210624                                   NaN   \n",
       "2210625                                   NaN   \n",
       "2210626                                   NaN   \n",
       "\n",
       "        out.state.seq_expr.reverse_complement out.state.count.type  \\\n",
       "0                                         NaN                  NaN   \n",
       "1                                         NaN                  NaN   \n",
       "2                                         NaN                  NaN   \n",
       "3                                         NaN                  NaN   \n",
       "4                                         NaN                  NaN   \n",
       "...                                       ...                  ...   \n",
       "2210622                                   NaN                  NaN   \n",
       "2210623                                   NaN                  NaN   \n",
       "2210624                                   NaN                  NaN   \n",
       "2210625                                   NaN                  NaN   \n",
       "2210626                                   NaN                  NaN   \n",
       "\n",
       "        out.state.count.value support_status  error_string  \\\n",
       "0                         NaN          False                 \n",
       "1                         NaN          False                 \n",
       "2                         NaN          False                 \n",
       "3                         NaN          False                 \n",
       "4                         NaN          False                 \n",
       "...                       ...            ...           ...   \n",
       "2210622                   NaN          False                 \n",
       "2210623                   NaN          False                 \n",
       "2210624                   NaN           True                 \n",
       "2210625                   NaN           True                 \n",
       "2210626                   NaN           True                 \n",
       "\n",
       "        error_string_reduce  normalize_status  \n",
       "0                                       False  \n",
       "1                                       False  \n",
       "2                                       False  \n",
       "3                                       False  \n",
       "4                                       False  \n",
       "...                     ...               ...  \n",
       "2210622                                 False  \n",
       "2210623                                 False  \n",
       "2210624                                  True  \n",
       "2210625                                  True  \n",
       "2210626                                  True  \n",
       "\n",
       "[2210627 rows x 103 columns]"
      ]
     },
     "execution_count": 284,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df[\"normalize_status\"] =  df[\"error_string_reduce\"].isna()\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "in.variation_type\n",
       "single nucleotide variant    1935386\n",
       "Deletion                      110936\n",
       "Duplication                    51808\n",
       "copy number loss               30643\n",
       "copy number gain               29863\n",
       "Microsatellite                 27160\n",
       "Indel                          11504\n",
       "Insertion                       9745\n",
       "Inversion                       1143\n",
       "Diplotype                        596\n",
       "Haplotype                        565\n",
       "Variation                        547\n",
       "Translocation                    273\n",
       "CompoundHeterozygote             249\n",
       "protein only                      95\n",
       "Complex                           77\n",
       "Haplotype, single variant         21\n",
       "Phase unknown                      8\n",
       "fusion                             6\n",
       "Distinct chromosomes               1\n",
       "Tandem duplication                 1\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 285,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df[\"in.variation_type\"].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_text = df.loc[df['out.type'] == 'Text']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>in.id</th>\n",
       "      <th>in.name</th>\n",
       "      <th>in.subclass_type</th>\n",
       "      <th>in.variation_type</th>\n",
       "      <th>in.xrefs</th>\n",
       "      <th>in.vrs_xform_plan.type</th>\n",
       "      <th>in.vrs_xform_plan.inputs</th>\n",
       "      <th>in.vrs_xform_plan.policy</th>\n",
       "      <th>out.id</th>\n",
       "      <th>out.type</th>\n",
       "      <th>...</th>\n",
       "      <th>out.state.seq_expr.location.start.value</th>\n",
       "      <th>out.state.seq_expr.location.end.type</th>\n",
       "      <th>out.state.seq_expr.location.end.value</th>\n",
       "      <th>out.state.seq_expr.reverse_complement</th>\n",
       "      <th>out.state.count.type</th>\n",
       "      <th>out.state.count.value</th>\n",
       "      <th>support_status</th>\n",
       "      <th>error_string</th>\n",
       "      <th>error_string_reduce</th>\n",
       "      <th>normalize_status</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>0 rows × 103 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [in.id, in.name, in.subclass_type, in.variation_type, in.xrefs, in.vrs_xform_plan.type, in.vrs_xform_plan.inputs, in.vrs_xform_plan.policy, out.id, out.type, out.definition, in.hgvs.assembly, in.hgvs.nucleotide, out.location.id, out.location.type, out.location.sequence_id, out.location.start.type, out.location.start.value, out.location.end.type, out.location.end.value, out.state.type, out.state.sequence, out.subject.id, out.subject.type, out.subject.sequence_id, out.subject.start.type, out.subject.start.value, out.subject.end.type, out.subject.end.value, out.copy_change, in.seq.assembly, in.seq.accession, in.seq.chr, in.seq.disp_start, in.seq.disp_stop, in.seq.inner_start, in.seq.inner_stop, in.absolute_copies, in.derived_hgvs, in.cyto, out.errors, in.seq.outer_stop, in.seq.outer_start, out.subject.start.min, out.subject.start.max, out.subject.end.min, out.subject.end.max, in.seq.variant_length, out.name, out.subclass_type, out.variation_type, out.hgvs.assembly, out.hgvs.nucleotide, out.vrs_xform_plan.type, out.vrs_xform_plan.inputs, out.vrs_xform_plan.policy, out.xrefs, out.subject.start.comparator, out.subject.end.comparator, out.copies.type, out.copies.value, in.seq.start, in.seq.stop, out.seq.assembly, out.seq.accession, out.seq.chr, out.seq.disp_start, out.seq.disp_stop, out.seq.start, out.seq.stop, out.seq.variant_length, out.absolute_copies, out.derived_hgvs, out.cyto, out.seq.inner_start, out.seq.inner_stop, out.seq.outer_start, out.seq.outer_stop, in.seq.ref_allele_vcf, in.seq.position_vcf, in.seq.alt_allele_vcf, in.canonical_spdi, out.seq.ref_allele_vcf, out.seq.position_vcf, out.seq.alt_allele_vcf, out.canonical_spdi, in.max_copies, in.min_copies, out.state.seq_expr.type, out.state.seq_expr.location.id, out.state.seq_expr.location.type, out.state.seq_expr.location.sequence_id, out.state.seq_expr.location.start.type, out.state.seq_expr.location.start.value, out.state.seq_expr.location.end.type, out.state.seq_expr.location.end.value, out.state.seq_expr.reverse_complement, out.state.count.type, out.state.count.value, support_status, ...]\n",
       "Index: []\n",
       "\n",
       "[0 rows x 103 columns]"
      ]
     },
     "execution_count": 287,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_text.loc[df_text['normalize_status'] == \"NaN\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### The cells are the number of variants with each expected bahvior and how they actually ended up performing. \n",
    "##### So if a variant was in an \"expected to pass\" category and ends up as text, that is an instance of a normalizer failure on a supported variant"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead tr th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe thead tr:last-of-type th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th colspan=\"5\" halign=\"left\">in.id</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th></th>\n",
       "      <th>out.type</th>\n",
       "      <th>Allele</th>\n",
       "      <th>CopyNumberChange</th>\n",
       "      <th>CopyNumberCount</th>\n",
       "      <th>NONE</th>\n",
       "      <th>Text</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>support_status</th>\n",
       "      <th>in.vrs_xform_plan.policy</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th rowspan=\"5\" valign=\"top\">False</th>\n",
       "      <th>Genotype/Haplotype</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1440</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Invalid/unsupported hgvs</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>1331</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Min/max copy count range not supported</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NCBI36 genomic only</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4771</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>No hgvs or location info</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3089</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"4\" valign=\"top\">True</th>\n",
       "      <th>Absolute copy count</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4880</td>\n",
       "      <td>1573</td>\n",
       "      <td>46810</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Canonical SPDI</th>\n",
       "      <td>2112837</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>5832</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Copy number change (cn loss|del and cn gain|dup)</th>\n",
       "      <td>0</td>\n",
       "      <td>12424</td>\n",
       "      <td>0</td>\n",
       "      <td>31</td>\n",
       "      <td>14649</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Remaining valid hgvs alleles</th>\n",
       "      <td>936</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                   in.id  \\\n",
       "out.type                                                          Allele   \n",
       "support_status in.vrs_xform_plan.policy                                    \n",
       "False          Genotype/Haplotype                                      0   \n",
       "               Invalid/unsupported hgvs                                0   \n",
       "               Min/max copy count range not supported                  0   \n",
       "               NCBI36 genomic only                                     0   \n",
       "               No hgvs or location info                                0   \n",
       "True           Absolute copy count                                     0   \n",
       "               Canonical SPDI                                    2112837   \n",
       "               Copy number change (cn loss|del and cn gain|dup)        0   \n",
       "               Remaining valid hgvs alleles                          936   \n",
       "\n",
       "                                                                                  \\\n",
       "out.type                                                        CopyNumberChange   \n",
       "support_status in.vrs_xform_plan.policy                                            \n",
       "False          Genotype/Haplotype                                              0   \n",
       "               Invalid/unsupported hgvs                                        0   \n",
       "               Min/max copy count range not supported                          0   \n",
       "               NCBI36 genomic only                                             0   \n",
       "               No hgvs or location info                                        0   \n",
       "True           Absolute copy count                                             0   \n",
       "               Canonical SPDI                                                  0   \n",
       "               Copy number change (cn loss|del and cn gain|dup)            12424   \n",
       "               Remaining valid hgvs alleles                                    0   \n",
       "\n",
       "                                                                                 \\\n",
       "out.type                                                        CopyNumberCount   \n",
       "support_status in.vrs_xform_plan.policy                                           \n",
       "False          Genotype/Haplotype                                             0   \n",
       "               Invalid/unsupported hgvs                                       0   \n",
       "               Min/max copy count range not supported                         0   \n",
       "               NCBI36 genomic only                                            0   \n",
       "               No hgvs or location info                                       0   \n",
       "True           Absolute copy count                                         4880   \n",
       "               Canonical SPDI                                                 0   \n",
       "               Copy number change (cn loss|del and cn gain|dup)               0   \n",
       "               Remaining valid hgvs alleles                                   0   \n",
       "\n",
       "                                                                              \n",
       "out.type                                                         NONE   Text  \n",
       "support_status in.vrs_xform_plan.policy                                       \n",
       "False          Genotype/Haplotype                                   0   1440  \n",
       "               Invalid/unsupported hgvs                             5   1331  \n",
       "               Min/max copy count range not supported               0     14  \n",
       "               NCBI36 genomic only                                  0   4771  \n",
       "               No hgvs or location info                             0   3089  \n",
       "True           Absolute copy count                               1573  46810  \n",
       "               Canonical SPDI                                    5832      0  \n",
       "               Copy number change (cn loss|del and cn gain|dup)    31  14649  \n",
       "               Remaining valid hgvs alleles                         4      1  "
      ]
     },
     "execution_count": 288,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df[['in.id','support_status','in.vrs_xform_plan.policy','out.type']].fillna('NONE').groupby(['support_status','in.vrs_xform_plan.policy','out.type']).count().unstack(level=2).fillna(0).astype(int)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Create groups of variants based on Supported and Normalized Status"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "supported_df= df.copy()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>in.id</th>\n",
       "      <th>in.name</th>\n",
       "      <th>in.subclass_type</th>\n",
       "      <th>in.variation_type</th>\n",
       "      <th>in.xrefs</th>\n",
       "      <th>in.vrs_xform_plan.type</th>\n",
       "      <th>in.vrs_xform_plan.inputs</th>\n",
       "      <th>in.vrs_xform_plan.policy</th>\n",
       "      <th>out.id</th>\n",
       "      <th>out.type</th>\n",
       "      <th>...</th>\n",
       "      <th>out.state.seq_expr.location.start.value</th>\n",
       "      <th>out.state.seq_expr.location.end.type</th>\n",
       "      <th>out.state.seq_expr.location.end.value</th>\n",
       "      <th>out.state.seq_expr.reverse_complement</th>\n",
       "      <th>out.state.count.type</th>\n",
       "      <th>out.state.count.value</th>\n",
       "      <th>support_status</th>\n",
       "      <th>error_string</th>\n",
       "      <th>error_string_reduce</th>\n",
       "      <th>normalize_status</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>149</th>\n",
       "      <td>1676330</td>\n",
       "      <td>NM_054027.6(ANKH):c.259G&gt;A (p.Val87Ile)</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Allele</td>\n",
       "      <td>[hgvs]</td>\n",
       "      <td>Remaining valid hgvs alleles</td>\n",
       "      <td>ga4gh:VA.T3hLVZajyx7AGQKV7la2RgE9CO0Wiv2y</td>\n",
       "      <td>Allele</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>150</th>\n",
       "      <td>1676476</td>\n",
       "      <td>NM_012309.5(SHANK2):c.460C&gt;T (p.Gln154Ter)</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Allele</td>\n",
       "      <td>[hgvs]</td>\n",
       "      <td>Remaining valid hgvs alleles</td>\n",
       "      <td>ga4gh:VA.dZAOK2Sy6aJlXan36nZYUjEJpk5WX8pF</td>\n",
       "      <td>Allele</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>151</th>\n",
       "      <td>1676638</td>\n",
       "      <td>NM_000094.4(COL7A1):c.8729G&gt;T (p.Gly2910Val)</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Allele</td>\n",
       "      <td>[hgvs]</td>\n",
       "      <td>Remaining valid hgvs alleles</td>\n",
       "      <td>ga4gh:VA.aUAN4LCHVZ_CPwAQfkVuX42g_Gi6WlBu</td>\n",
       "      <td>Allele</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>152</th>\n",
       "      <td>1676567</td>\n",
       "      <td>NM_005445.4(SMC3):c.1343dup (p.Glu449fs)</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>Duplication</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CopyNumberChange</td>\n",
       "      <td>[hgvs]</td>\n",
       "      <td>Copy number change (cn loss|del and cn gain|dup)</td>\n",
       "      <td>ga4gh:CX.-9T00EYMyHQKC7DVOenHi-QH8aIOcc0D</td>\n",
       "      <td>CopyNumberChange</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>153</th>\n",
       "      <td>1325687</td>\n",
       "      <td>NM_000059.4(BRCA2):c.3692C&gt;G (p.Thr1231Ser)</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Allele</td>\n",
       "      <td>[hgvs]</td>\n",
       "      <td>Remaining valid hgvs alleles</td>\n",
       "      <td>ga4gh:VA.MLNaXFddWSZRL0vRCHJceHBF09R_JgBt</td>\n",
       "      <td>Allele</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2210620</th>\n",
       "      <td>1496502</td>\n",
       "      <td>NM_007255.3(B4GALT7):c.881_882insTGAGGTGGATTAA...</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>Insertion</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Allele</td>\n",
       "      <td>[canonical_spdi]</td>\n",
       "      <td>Canonical SPDI</td>\n",
       "      <td>ga4gh:VA.P-rdDkBZMl4kwW-ZHmsMOy0ik5MKa4LI</td>\n",
       "      <td>Allele</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2210621</th>\n",
       "      <td>715720</td>\n",
       "      <td>NM_033026.6(PCLO):c.3300+7_3300+8insTTTATATATA...</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>Insertion</td>\n",
       "      <td>dbSNP:746054139</td>\n",
       "      <td>Allele</td>\n",
       "      <td>[canonical_spdi]</td>\n",
       "      <td>Canonical SPDI</td>\n",
       "      <td>ga4gh:VA._iakrjaN6VkKr1AGuegIUY4vKZxnQVkT</td>\n",
       "      <td>Allele</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2210624</th>\n",
       "      <td>1181160</td>\n",
       "      <td>NM_030962.4(SBF2):c.55+97_55+98insCGGGCGTCGGGGC</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>Microsatellite</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Allele</td>\n",
       "      <td>[canonical_spdi]</td>\n",
       "      <td>Canonical SPDI</td>\n",
       "      <td>ga4gh:VA.x2E2GJQA5ovZFJ1SKxg3oHzkUXv5ZGfv</td>\n",
       "      <td>Allele</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2210625</th>\n",
       "      <td>2202105</td>\n",
       "      <td>NM_153676.4(USH1C):c.496+14_496+15insGTACTCCAT...</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>Microsatellite</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Allele</td>\n",
       "      <td>[canonical_spdi]</td>\n",
       "      <td>Canonical SPDI</td>\n",
       "      <td>ga4gh:VA.5jQEJZIbf1ttjbIGJVhfrHR9XcB7vqWD</td>\n",
       "      <td>Allele</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2210626</th>\n",
       "      <td>1801464</td>\n",
       "      <td>NM_176787.5(PIGN):c.2673-514_2673-513insGCAAAC...</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>Insertion</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Allele</td>\n",
       "      <td>[hgvs]</td>\n",
       "      <td>Remaining valid hgvs alleles</td>\n",
       "      <td>ga4gh:VA.3BXZ9oWeEqz4VM16yV8mgjvQ8ZTElB2J</td>\n",
       "      <td>Allele</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2131087 rows × 103 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           in.id                                            in.name  \\\n",
       "149      1676330            NM_054027.6(ANKH):c.259G>A (p.Val87Ile)   \n",
       "150      1676476         NM_012309.5(SHANK2):c.460C>T (p.Gln154Ter)   \n",
       "151      1676638       NM_000094.4(COL7A1):c.8729G>T (p.Gly2910Val)   \n",
       "152      1676567           NM_005445.4(SMC3):c.1343dup (p.Glu449fs)   \n",
       "153      1325687        NM_000059.4(BRCA2):c.3692C>G (p.Thr1231Ser)   \n",
       "...          ...                                                ...   \n",
       "2210620  1496502  NM_007255.3(B4GALT7):c.881_882insTGAGGTGGATTAA...   \n",
       "2210621   715720  NM_033026.6(PCLO):c.3300+7_3300+8insTTTATATATA...   \n",
       "2210624  1181160    NM_030962.4(SBF2):c.55+97_55+98insCGGGCGTCGGGGC   \n",
       "2210625  2202105  NM_153676.4(USH1C):c.496+14_496+15insGTACTCCAT...   \n",
       "2210626  1801464  NM_176787.5(PIGN):c.2673-514_2673-513insGCAAAC...   \n",
       "\n",
       "        in.subclass_type          in.variation_type         in.xrefs  \\\n",
       "149         SimpleAllele  single nucleotide variant              NaN   \n",
       "150         SimpleAllele  single nucleotide variant              NaN   \n",
       "151         SimpleAllele  single nucleotide variant              NaN   \n",
       "152         SimpleAllele                Duplication              NaN   \n",
       "153         SimpleAllele  single nucleotide variant              NaN   \n",
       "...                  ...                        ...              ...   \n",
       "2210620     SimpleAllele                  Insertion              NaN   \n",
       "2210621     SimpleAllele                  Insertion  dbSNP:746054139   \n",
       "2210624     SimpleAllele             Microsatellite              NaN   \n",
       "2210625     SimpleAllele             Microsatellite              NaN   \n",
       "2210626     SimpleAllele                  Insertion              NaN   \n",
       "\n",
       "        in.vrs_xform_plan.type in.vrs_xform_plan.inputs  \\\n",
       "149                     Allele                   [hgvs]   \n",
       "150                     Allele                   [hgvs]   \n",
       "151                     Allele                   [hgvs]   \n",
       "152           CopyNumberChange                   [hgvs]   \n",
       "153                     Allele                   [hgvs]   \n",
       "...                        ...                      ...   \n",
       "2210620                 Allele         [canonical_spdi]   \n",
       "2210621                 Allele         [canonical_spdi]   \n",
       "2210624                 Allele         [canonical_spdi]   \n",
       "2210625                 Allele         [canonical_spdi]   \n",
       "2210626                 Allele                   [hgvs]   \n",
       "\n",
       "                                 in.vrs_xform_plan.policy  \\\n",
       "149                          Remaining valid hgvs alleles   \n",
       "150                          Remaining valid hgvs alleles   \n",
       "151                          Remaining valid hgvs alleles   \n",
       "152      Copy number change (cn loss|del and cn gain|dup)   \n",
       "153                          Remaining valid hgvs alleles   \n",
       "...                                                   ...   \n",
       "2210620                                    Canonical SPDI   \n",
       "2210621                                    Canonical SPDI   \n",
       "2210624                                    Canonical SPDI   \n",
       "2210625                                    Canonical SPDI   \n",
       "2210626                      Remaining valid hgvs alleles   \n",
       "\n",
       "                                            out.id          out.type  ...  \\\n",
       "149      ga4gh:VA.T3hLVZajyx7AGQKV7la2RgE9CO0Wiv2y            Allele  ...   \n",
       "150      ga4gh:VA.dZAOK2Sy6aJlXan36nZYUjEJpk5WX8pF            Allele  ...   \n",
       "151      ga4gh:VA.aUAN4LCHVZ_CPwAQfkVuX42g_Gi6WlBu            Allele  ...   \n",
       "152      ga4gh:CX.-9T00EYMyHQKC7DVOenHi-QH8aIOcc0D  CopyNumberChange  ...   \n",
       "153      ga4gh:VA.MLNaXFddWSZRL0vRCHJceHBF09R_JgBt            Allele  ...   \n",
       "...                                            ...               ...  ...   \n",
       "2210620  ga4gh:VA.P-rdDkBZMl4kwW-ZHmsMOy0ik5MKa4LI            Allele  ...   \n",
       "2210621  ga4gh:VA._iakrjaN6VkKr1AGuegIUY4vKZxnQVkT            Allele  ...   \n",
       "2210624  ga4gh:VA.x2E2GJQA5ovZFJ1SKxg3oHzkUXv5ZGfv            Allele  ...   \n",
       "2210625  ga4gh:VA.5jQEJZIbf1ttjbIGJVhfrHR9XcB7vqWD            Allele  ...   \n",
       "2210626  ga4gh:VA.3BXZ9oWeEqz4VM16yV8mgjvQ8ZTElB2J            Allele  ...   \n",
       "\n",
       "        out.state.seq_expr.location.start.value  \\\n",
       "149                                         NaN   \n",
       "150                                         NaN   \n",
       "151                                         NaN   \n",
       "152                                         NaN   \n",
       "153                                         NaN   \n",
       "...                                         ...   \n",
       "2210620                                     NaN   \n",
       "2210621                                     NaN   \n",
       "2210624                                     NaN   \n",
       "2210625                                     NaN   \n",
       "2210626                                     NaN   \n",
       "\n",
       "        out.state.seq_expr.location.end.type  \\\n",
       "149                                      NaN   \n",
       "150                                      NaN   \n",
       "151                                      NaN   \n",
       "152                                      NaN   \n",
       "153                                      NaN   \n",
       "...                                      ...   \n",
       "2210620                                  NaN   \n",
       "2210621                                  NaN   \n",
       "2210624                                  NaN   \n",
       "2210625                                  NaN   \n",
       "2210626                                  NaN   \n",
       "\n",
       "        out.state.seq_expr.location.end.value  \\\n",
       "149                                       NaN   \n",
       "150                                       NaN   \n",
       "151                                       NaN   \n",
       "152                                       NaN   \n",
       "153                                       NaN   \n",
       "...                                       ...   \n",
       "2210620                                   NaN   \n",
       "2210621                                   NaN   \n",
       "2210624                                   NaN   \n",
       "2210625                                   NaN   \n",
       "2210626                                   NaN   \n",
       "\n",
       "        out.state.seq_expr.reverse_complement out.state.count.type  \\\n",
       "149                                       NaN                  NaN   \n",
       "150                                       NaN                  NaN   \n",
       "151                                       NaN                  NaN   \n",
       "152                                       NaN                  NaN   \n",
       "153                                       NaN                  NaN   \n",
       "...                                       ...                  ...   \n",
       "2210620                                   NaN                  NaN   \n",
       "2210621                                   NaN                  NaN   \n",
       "2210624                                   NaN                  NaN   \n",
       "2210625                                   NaN                  NaN   \n",
       "2210626                                   NaN                  NaN   \n",
       "\n",
       "        out.state.count.value support_status  error_string  \\\n",
       "149                       NaN           True                 \n",
       "150                       NaN           True                 \n",
       "151                       NaN           True                 \n",
       "152                       NaN           True                 \n",
       "153                       NaN           True                 \n",
       "...                       ...            ...           ...   \n",
       "2210620                   NaN           True                 \n",
       "2210621                   NaN           True                 \n",
       "2210624                   NaN           True                 \n",
       "2210625                   NaN           True                 \n",
       "2210626                   NaN           True                 \n",
       "\n",
       "        error_string_reduce  normalize_status  \n",
       "149                                      True  \n",
       "150                                      True  \n",
       "151                                      True  \n",
       "152                                      True  \n",
       "153                                      True  \n",
       "...                     ...               ...  \n",
       "2210620                                  True  \n",
       "2210621                                  True  \n",
       "2210624                                  True  \n",
       "2210625                                  True  \n",
       "2210626                                  True  \n",
       "\n",
       "[2131087 rows x 103 columns]"
      ]
     },
     "execution_count": 290,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "supported_df=supported_df.loc[(supported_df['support_status'] == True) & \n",
    "                         (supported_df['normalize_status'] == True)]\n",
    "supported_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "support_status\n",
       "True    2131087\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 291,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "supported_df['support_status'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "supported_not_normalized_df =  df.copy()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>in.id</th>\n",
       "      <th>in.name</th>\n",
       "      <th>in.subclass_type</th>\n",
       "      <th>in.variation_type</th>\n",
       "      <th>in.xrefs</th>\n",
       "      <th>in.vrs_xform_plan.type</th>\n",
       "      <th>in.vrs_xform_plan.inputs</th>\n",
       "      <th>in.vrs_xform_plan.policy</th>\n",
       "      <th>out.id</th>\n",
       "      <th>out.type</th>\n",
       "      <th>...</th>\n",
       "      <th>out.state.seq_expr.location.start.value</th>\n",
       "      <th>out.state.seq_expr.location.end.type</th>\n",
       "      <th>out.state.seq_expr.location.end.value</th>\n",
       "      <th>out.state.seq_expr.reverse_complement</th>\n",
       "      <th>out.state.count.type</th>\n",
       "      <th>out.state.count.value</th>\n",
       "      <th>support_status</th>\n",
       "      <th>error_string</th>\n",
       "      <th>error_string_reduce</th>\n",
       "      <th>normalize_status</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>166</th>\n",
       "      <td>1340674</td>\n",
       "      <td>GRCh37/hg19 7q31.33-32.1(chr7:127050634-127826...</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>copy number gain</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CopyNumberCount</td>\n",
       "      <td>[hgvs, absolute_copies]</td>\n",
       "      <td>Absolute copy count</td>\n",
       "      <td>Text:clinvar:1340674</td>\n",
       "      <td>Text</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>Variation record was too long to normalize (77...</td>\n",
       "      <td>variation record was too long to normalize (#)...</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>169</th>\n",
       "      <td>1340493</td>\n",
       "      <td>GRCh37/hg19 12p12.3(chr12:17595624-18236175)x3</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>copy number gain</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CopyNumberCount</td>\n",
       "      <td>[hgvs, absolute_copies]</td>\n",
       "      <td>Absolute copy count</td>\n",
       "      <td>Text:clinvar:1340493</td>\n",
       "      <td>Text</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>Variation record was too long to normalize (64...</td>\n",
       "      <td>variation record was too long to normalize (#)...</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>171</th>\n",
       "      <td>814630</td>\n",
       "      <td>GRCh37/hg19 4q33-34.1(chr4:171505226-173149981)x3</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>copy number gain</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CopyNumberCount</td>\n",
       "      <td>[hgvs, absolute_copies]</td>\n",
       "      <td>Absolute copy count</td>\n",
       "      <td>Text:clinvar:814630</td>\n",
       "      <td>Text</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>Variation record was too long to normalize (16...</td>\n",
       "      <td>variation record was too long to normalize (#)...</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>172</th>\n",
       "      <td>1808660</td>\n",
       "      <td>GRCh37/hg19 2p16.3(chr2:51009232-51492867)x1</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>copy number loss</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CopyNumberCount</td>\n",
       "      <td>[hgvs, absolute_copies]</td>\n",
       "      <td>Absolute copy count</td>\n",
       "      <td>Text:clinvar:1808660</td>\n",
       "      <td>Text</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>Variation record was too long to normalize (48...</td>\n",
       "      <td>variation record was too long to normalize (#)...</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>173</th>\n",
       "      <td>980505</td>\n",
       "      <td>GRCh37/hg19 3p14.2(chr3:62894703-63320384)x3</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>copy number gain</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CopyNumberCount</td>\n",
       "      <td>[hgvs, absolute_copies]</td>\n",
       "      <td>Absolute copy count</td>\n",
       "      <td>Text:clinvar:980505</td>\n",
       "      <td>Text</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>Variation record was too long to normalize (42...</td>\n",
       "      <td>variation record was too long to normalize (#)...</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2210611</th>\n",
       "      <td>591342</td>\n",
       "      <td>NM_001377.3(DYNC2H1):c.8243_8244insCTAATTCTTA ...</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>Insertion</td>\n",
       "      <td>dbSNP:1565393163</td>\n",
       "      <td>Allele</td>\n",
       "      <td>[canonical_spdi]</td>\n",
       "      <td>Canonical SPDI</td>\n",
       "      <td>591342</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>Error returned from variation normalizer;Posit...</td>\n",
       "      <td>error returned from variation normalizer;posit...</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2210612</th>\n",
       "      <td>1235939</td>\n",
       "      <td>NM_002693.3(POLG):c.151_152insCAGCAG (p.Gln51d...</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>Insertion</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Allele</td>\n",
       "      <td>[canonical_spdi]</td>\n",
       "      <td>Canonical SPDI</td>\n",
       "      <td>1235939</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>Error returned from variation normalizer;Posit...</td>\n",
       "      <td>error returned from variation normalizer;posit...</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2210615</th>\n",
       "      <td>1801441</td>\n",
       "      <td>NM_015189.3(EXOC6B):c.2197-66917_2197-66916ins...</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>Insertion</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Allele</td>\n",
       "      <td>[canonical_spdi]</td>\n",
       "      <td>Canonical SPDI</td>\n",
       "      <td>1801441</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>Error returned from variation normalizer;Posit...</td>\n",
       "      <td>error returned from variation normalizer;posit...</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2210616</th>\n",
       "      <td>491194</td>\n",
       "      <td>NM_000059.4(BRCA2):c.9501+7_9501+8insAGGTAAGGT...</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>Insertion</td>\n",
       "      <td>ClinGen:CA658683823,dbSNP:1555289623</td>\n",
       "      <td>Allele</td>\n",
       "      <td>[canonical_spdi]</td>\n",
       "      <td>Canonical SPDI</td>\n",
       "      <td>491194</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>Error returned from variation normalizer;Posit...</td>\n",
       "      <td>error returned from variation normalizer;posit...</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2210619</th>\n",
       "      <td>1725299</td>\n",
       "      <td>NM_012434.5(SLC17A5):c.485_486insCAAGAGAGCAAAA...</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>Insertion</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Allele</td>\n",
       "      <td>[canonical_spdi]</td>\n",
       "      <td>Canonical SPDI</td>\n",
       "      <td>1725299</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>Error returned from variation normalizer;Posit...</td>\n",
       "      <td>error returned from variation normalizer;posit...</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>68890 rows × 103 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           in.id                                            in.name  \\\n",
       "166      1340674  GRCh37/hg19 7q31.33-32.1(chr7:127050634-127826...   \n",
       "169      1340493     GRCh37/hg19 12p12.3(chr12:17595624-18236175)x3   \n",
       "171       814630  GRCh37/hg19 4q33-34.1(chr4:171505226-173149981)x3   \n",
       "172      1808660       GRCh37/hg19 2p16.3(chr2:51009232-51492867)x1   \n",
       "173       980505       GRCh37/hg19 3p14.2(chr3:62894703-63320384)x3   \n",
       "...          ...                                                ...   \n",
       "2210611   591342  NM_001377.3(DYNC2H1):c.8243_8244insCTAATTCTTA ...   \n",
       "2210612  1235939  NM_002693.3(POLG):c.151_152insCAGCAG (p.Gln51d...   \n",
       "2210615  1801441  NM_015189.3(EXOC6B):c.2197-66917_2197-66916ins...   \n",
       "2210616   491194  NM_000059.4(BRCA2):c.9501+7_9501+8insAGGTAAGGT...   \n",
       "2210619  1725299  NM_012434.5(SLC17A5):c.485_486insCAAGAGAGCAAAA...   \n",
       "\n",
       "        in.subclass_type in.variation_type  \\\n",
       "166         SimpleAllele  copy number gain   \n",
       "169         SimpleAllele  copy number gain   \n",
       "171         SimpleAllele  copy number gain   \n",
       "172         SimpleAllele  copy number loss   \n",
       "173         SimpleAllele  copy number gain   \n",
       "...                  ...               ...   \n",
       "2210611     SimpleAllele         Insertion   \n",
       "2210612     SimpleAllele         Insertion   \n",
       "2210615     SimpleAllele         Insertion   \n",
       "2210616     SimpleAllele         Insertion   \n",
       "2210619     SimpleAllele         Insertion   \n",
       "\n",
       "                                     in.xrefs in.vrs_xform_plan.type  \\\n",
       "166                                       NaN        CopyNumberCount   \n",
       "169                                       NaN        CopyNumberCount   \n",
       "171                                       NaN        CopyNumberCount   \n",
       "172                                       NaN        CopyNumberCount   \n",
       "173                                       NaN        CopyNumberCount   \n",
       "...                                       ...                    ...   \n",
       "2210611                      dbSNP:1565393163                 Allele   \n",
       "2210612                                   NaN                 Allele   \n",
       "2210615                                   NaN                 Allele   \n",
       "2210616  ClinGen:CA658683823,dbSNP:1555289623                 Allele   \n",
       "2210619                                   NaN                 Allele   \n",
       "\n",
       "        in.vrs_xform_plan.inputs in.vrs_xform_plan.policy  \\\n",
       "166      [hgvs, absolute_copies]      Absolute copy count   \n",
       "169      [hgvs, absolute_copies]      Absolute copy count   \n",
       "171      [hgvs, absolute_copies]      Absolute copy count   \n",
       "172      [hgvs, absolute_copies]      Absolute copy count   \n",
       "173      [hgvs, absolute_copies]      Absolute copy count   \n",
       "...                          ...                      ...   \n",
       "2210611         [canonical_spdi]           Canonical SPDI   \n",
       "2210612         [canonical_spdi]           Canonical SPDI   \n",
       "2210615         [canonical_spdi]           Canonical SPDI   \n",
       "2210616         [canonical_spdi]           Canonical SPDI   \n",
       "2210619         [canonical_spdi]           Canonical SPDI   \n",
       "\n",
       "                       out.id out.type  ...  \\\n",
       "166      Text:clinvar:1340674     Text  ...   \n",
       "169      Text:clinvar:1340493     Text  ...   \n",
       "171       Text:clinvar:814630     Text  ...   \n",
       "172      Text:clinvar:1808660     Text  ...   \n",
       "173       Text:clinvar:980505     Text  ...   \n",
       "...                       ...      ...  ...   \n",
       "2210611                591342      NaN  ...   \n",
       "2210612               1235939      NaN  ...   \n",
       "2210615               1801441      NaN  ...   \n",
       "2210616                491194      NaN  ...   \n",
       "2210619               1725299      NaN  ...   \n",
       "\n",
       "        out.state.seq_expr.location.start.value  \\\n",
       "166                                         NaN   \n",
       "169                                         NaN   \n",
       "171                                         NaN   \n",
       "172                                         NaN   \n",
       "173                                         NaN   \n",
       "...                                         ...   \n",
       "2210611                                     NaN   \n",
       "2210612                                     NaN   \n",
       "2210615                                     NaN   \n",
       "2210616                                     NaN   \n",
       "2210619                                     NaN   \n",
       "\n",
       "        out.state.seq_expr.location.end.type  \\\n",
       "166                                      NaN   \n",
       "169                                      NaN   \n",
       "171                                      NaN   \n",
       "172                                      NaN   \n",
       "173                                      NaN   \n",
       "...                                      ...   \n",
       "2210611                                  NaN   \n",
       "2210612                                  NaN   \n",
       "2210615                                  NaN   \n",
       "2210616                                  NaN   \n",
       "2210619                                  NaN   \n",
       "\n",
       "        out.state.seq_expr.location.end.value  \\\n",
       "166                                       NaN   \n",
       "169                                       NaN   \n",
       "171                                       NaN   \n",
       "172                                       NaN   \n",
       "173                                       NaN   \n",
       "...                                       ...   \n",
       "2210611                                   NaN   \n",
       "2210612                                   NaN   \n",
       "2210615                                   NaN   \n",
       "2210616                                   NaN   \n",
       "2210619                                   NaN   \n",
       "\n",
       "        out.state.seq_expr.reverse_complement out.state.count.type  \\\n",
       "166                                       NaN                  NaN   \n",
       "169                                       NaN                  NaN   \n",
       "171                                       NaN                  NaN   \n",
       "172                                       NaN                  NaN   \n",
       "173                                       NaN                  NaN   \n",
       "...                                       ...                  ...   \n",
       "2210611                                   NaN                  NaN   \n",
       "2210612                                   NaN                  NaN   \n",
       "2210615                                   NaN                  NaN   \n",
       "2210616                                   NaN                  NaN   \n",
       "2210619                                   NaN                  NaN   \n",
       "\n",
       "        out.state.count.value support_status  \\\n",
       "166                       NaN           True   \n",
       "169                       NaN           True   \n",
       "171                       NaN           True   \n",
       "172                       NaN           True   \n",
       "173                       NaN           True   \n",
       "...                       ...            ...   \n",
       "2210611                   NaN           True   \n",
       "2210612                   NaN           True   \n",
       "2210615                   NaN           True   \n",
       "2210616                   NaN           True   \n",
       "2210619                   NaN           True   \n",
       "\n",
       "                                              error_string  \\\n",
       "166      Variation record was too long to normalize (77...   \n",
       "169      Variation record was too long to normalize (64...   \n",
       "171      Variation record was too long to normalize (16...   \n",
       "172      Variation record was too long to normalize (48...   \n",
       "173      Variation record was too long to normalize (42...   \n",
       "...                                                    ...   \n",
       "2210611  Error returned from variation normalizer;Posit...   \n",
       "2210612  Error returned from variation normalizer;Posit...   \n",
       "2210615  Error returned from variation normalizer;Posit...   \n",
       "2210616  Error returned from variation normalizer;Posit...   \n",
       "2210619  Error returned from variation normalizer;Posit...   \n",
       "\n",
       "                                       error_string_reduce  normalize_status  \n",
       "166      variation record was too long to normalize (#)...             False  \n",
       "169      variation record was too long to normalize (#)...             False  \n",
       "171      variation record was too long to normalize (#)...             False  \n",
       "172      variation record was too long to normalize (#)...             False  \n",
       "173      variation record was too long to normalize (#)...             False  \n",
       "...                                                    ...               ...  \n",
       "2210611  error returned from variation normalizer;posit...             False  \n",
       "2210612  error returned from variation normalizer;posit...             False  \n",
       "2210615  error returned from variation normalizer;posit...             False  \n",
       "2210616  error returned from variation normalizer;posit...             False  \n",
       "2210619  error returned from variation normalizer;posit...             False  \n",
       "\n",
       "[68890 rows x 103 columns]"
      ]
     },
     "execution_count": 293,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "supported_not_normalized_df = supported_not_normalized_df.loc[(supported_not_normalized_df['support_status'] == True) & \n",
    "                         (supported_not_normalized_df['normalize_status'] == False)]\n",
    "supported_not_normalized_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "not_supported_df =  df.copy()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>in.id</th>\n",
       "      <th>in.name</th>\n",
       "      <th>in.subclass_type</th>\n",
       "      <th>in.variation_type</th>\n",
       "      <th>in.xrefs</th>\n",
       "      <th>in.vrs_xform_plan.type</th>\n",
       "      <th>in.vrs_xform_plan.inputs</th>\n",
       "      <th>in.vrs_xform_plan.policy</th>\n",
       "      <th>out.id</th>\n",
       "      <th>out.type</th>\n",
       "      <th>...</th>\n",
       "      <th>out.state.seq_expr.location.start.value</th>\n",
       "      <th>out.state.seq_expr.location.end.type</th>\n",
       "      <th>out.state.seq_expr.location.end.value</th>\n",
       "      <th>out.state.seq_expr.reverse_complement</th>\n",
       "      <th>out.state.count.type</th>\n",
       "      <th>out.state.count.value</th>\n",
       "      <th>support_status</th>\n",
       "      <th>error_string</th>\n",
       "      <th>error_string_reduce</th>\n",
       "      <th>normalize_status</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>16098</td>\n",
       "      <td>SLC2A2, 1-BP INS, 793C</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>Insertion</td>\n",
       "      <td>OMIM:138160.0009</td>\n",
       "      <td>Text</td>\n",
       "      <td>[id]</td>\n",
       "      <td>No hgvs or location info</td>\n",
       "      <td>Text:clinvar:16098</td>\n",
       "      <td>Text</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>425693</td>\n",
       "      <td>NM_001204.6(BMPR2):c.77-?_247+?dup</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>Duplication</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Text</td>\n",
       "      <td>[id]</td>\n",
       "      <td>No hgvs or location info</td>\n",
       "      <td>Text:clinvar:425693</td>\n",
       "      <td>Text</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>90650</td>\n",
       "      <td>NM_000251.2(MSH2):c.1387-?_1510+?del</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Text</td>\n",
       "      <td>[id]</td>\n",
       "      <td>No hgvs or location info</td>\n",
       "      <td>Text:clinvar:90650</td>\n",
       "      <td>Text</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>2446408</td>\n",
       "      <td>CDHR1, 783G-A (rs147346345)</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>OMIM:609502.0005</td>\n",
       "      <td>Text</td>\n",
       "      <td>[id]</td>\n",
       "      <td>No hgvs or location info</td>\n",
       "      <td>Text:clinvar:2446408</td>\n",
       "      <td>Text</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>14905</td>\n",
       "      <td>HLA-DRB1, HLA-DRB1*1101</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>Variation</td>\n",
       "      <td>OMIM:142857.0001</td>\n",
       "      <td>Text</td>\n",
       "      <td>[id]</td>\n",
       "      <td>No hgvs or location info</td>\n",
       "      <td>Text:clinvar:14905</td>\n",
       "      <td>Text</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2199348</th>\n",
       "      <td>1418992</td>\n",
       "      <td>NM_002076.4(GNS):c.841_842insTTTTTTTTTTTTTTTTT...</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>Insertion</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Text</td>\n",
       "      <td>[id]</td>\n",
       "      <td>Invalid/unsupported hgvs</td>\n",
       "      <td>Text:clinvar:1418992</td>\n",
       "      <td>Text</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2199349</th>\n",
       "      <td>2134754</td>\n",
       "      <td>NM_152564.5(VPS13B):c.5627_5628insTTTTTTTTTTTT...</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>Insertion</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Text</td>\n",
       "      <td>[id]</td>\n",
       "      <td>Invalid/unsupported hgvs</td>\n",
       "      <td>Text:clinvar:2134754</td>\n",
       "      <td>Text</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2210618</th>\n",
       "      <td>1513408</td>\n",
       "      <td>NM_024928.5(STN1):c.340_352AAG[2]CTACAAGGCCGGG...</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>Insertion</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Text</td>\n",
       "      <td>[id]</td>\n",
       "      <td>Invalid/unsupported hgvs</td>\n",
       "      <td>Text:clinvar:1513408</td>\n",
       "      <td>Text</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2210622</th>\n",
       "      <td>1464440</td>\n",
       "      <td>NM_001267550.2(TTN):c.29512_29513insGGCCGGGCGC...</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>Insertion</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Text</td>\n",
       "      <td>[id]</td>\n",
       "      <td>Invalid/unsupported hgvs</td>\n",
       "      <td>Text:clinvar:1464440</td>\n",
       "      <td>Text</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2210623</th>\n",
       "      <td>1453116</td>\n",
       "      <td>NM_000051.4(ATM):c.3376_3377insGGCCGGGCGCGGTGG...</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>Insertion</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Text</td>\n",
       "      <td>[id]</td>\n",
       "      <td>Invalid/unsupported hgvs</td>\n",
       "      <td>Text:clinvar:1453116</td>\n",
       "      <td>Text</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>10650 rows × 103 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           in.id                                            in.name  \\\n",
       "0          16098                             SLC2A2, 1-BP INS, 793C   \n",
       "1         425693                 NM_001204.6(BMPR2):c.77-?_247+?dup   \n",
       "2          90650               NM_000251.2(MSH2):c.1387-?_1510+?del   \n",
       "3        2446408                        CDHR1, 783G-A (rs147346345)   \n",
       "4          14905                            HLA-DRB1, HLA-DRB1*1101   \n",
       "...          ...                                                ...   \n",
       "2199348  1418992  NM_002076.4(GNS):c.841_842insTTTTTTTTTTTTTTTTT...   \n",
       "2199349  2134754  NM_152564.5(VPS13B):c.5627_5628insTTTTTTTTTTTT...   \n",
       "2210618  1513408  NM_024928.5(STN1):c.340_352AAG[2]CTACAAGGCCGGG...   \n",
       "2210622  1464440  NM_001267550.2(TTN):c.29512_29513insGGCCGGGCGC...   \n",
       "2210623  1453116  NM_000051.4(ATM):c.3376_3377insGGCCGGGCGCGGTGG...   \n",
       "\n",
       "        in.subclass_type          in.variation_type          in.xrefs  \\\n",
       "0           SimpleAllele                  Insertion  OMIM:138160.0009   \n",
       "1           SimpleAllele                Duplication               NaN   \n",
       "2           SimpleAllele                   Deletion               NaN   \n",
       "3           SimpleAllele  single nucleotide variant  OMIM:609502.0005   \n",
       "4           SimpleAllele                  Variation  OMIM:142857.0001   \n",
       "...                  ...                        ...               ...   \n",
       "2199348     SimpleAllele                  Insertion               NaN   \n",
       "2199349     SimpleAllele                  Insertion               NaN   \n",
       "2210618     SimpleAllele                  Insertion               NaN   \n",
       "2210622     SimpleAllele                  Insertion               NaN   \n",
       "2210623     SimpleAllele                  Insertion               NaN   \n",
       "\n",
       "        in.vrs_xform_plan.type in.vrs_xform_plan.inputs  \\\n",
       "0                         Text                     [id]   \n",
       "1                         Text                     [id]   \n",
       "2                         Text                     [id]   \n",
       "3                         Text                     [id]   \n",
       "4                         Text                     [id]   \n",
       "...                        ...                      ...   \n",
       "2199348                   Text                     [id]   \n",
       "2199349                   Text                     [id]   \n",
       "2210618                   Text                     [id]   \n",
       "2210622                   Text                     [id]   \n",
       "2210623                   Text                     [id]   \n",
       "\n",
       "         in.vrs_xform_plan.policy                out.id out.type  ...  \\\n",
       "0        No hgvs or location info    Text:clinvar:16098     Text  ...   \n",
       "1        No hgvs or location info   Text:clinvar:425693     Text  ...   \n",
       "2        No hgvs or location info    Text:clinvar:90650     Text  ...   \n",
       "3        No hgvs or location info  Text:clinvar:2446408     Text  ...   \n",
       "4        No hgvs or location info    Text:clinvar:14905     Text  ...   \n",
       "...                           ...                   ...      ...  ...   \n",
       "2199348  Invalid/unsupported hgvs  Text:clinvar:1418992     Text  ...   \n",
       "2199349  Invalid/unsupported hgvs  Text:clinvar:2134754     Text  ...   \n",
       "2210618  Invalid/unsupported hgvs  Text:clinvar:1513408     Text  ...   \n",
       "2210622  Invalid/unsupported hgvs  Text:clinvar:1464440     Text  ...   \n",
       "2210623  Invalid/unsupported hgvs  Text:clinvar:1453116     Text  ...   \n",
       "\n",
       "        out.state.seq_expr.location.start.value  \\\n",
       "0                                           NaN   \n",
       "1                                           NaN   \n",
       "2                                           NaN   \n",
       "3                                           NaN   \n",
       "4                                           NaN   \n",
       "...                                         ...   \n",
       "2199348                                     NaN   \n",
       "2199349                                     NaN   \n",
       "2210618                                     NaN   \n",
       "2210622                                     NaN   \n",
       "2210623                                     NaN   \n",
       "\n",
       "        out.state.seq_expr.location.end.type  \\\n",
       "0                                        NaN   \n",
       "1                                        NaN   \n",
       "2                                        NaN   \n",
       "3                                        NaN   \n",
       "4                                        NaN   \n",
       "...                                      ...   \n",
       "2199348                                  NaN   \n",
       "2199349                                  NaN   \n",
       "2210618                                  NaN   \n",
       "2210622                                  NaN   \n",
       "2210623                                  NaN   \n",
       "\n",
       "        out.state.seq_expr.location.end.value  \\\n",
       "0                                         NaN   \n",
       "1                                         NaN   \n",
       "2                                         NaN   \n",
       "3                                         NaN   \n",
       "4                                         NaN   \n",
       "...                                       ...   \n",
       "2199348                                   NaN   \n",
       "2199349                                   NaN   \n",
       "2210618                                   NaN   \n",
       "2210622                                   NaN   \n",
       "2210623                                   NaN   \n",
       "\n",
       "        out.state.seq_expr.reverse_complement out.state.count.type  \\\n",
       "0                                         NaN                  NaN   \n",
       "1                                         NaN                  NaN   \n",
       "2                                         NaN                  NaN   \n",
       "3                                         NaN                  NaN   \n",
       "4                                         NaN                  NaN   \n",
       "...                                       ...                  ...   \n",
       "2199348                                   NaN                  NaN   \n",
       "2199349                                   NaN                  NaN   \n",
       "2210618                                   NaN                  NaN   \n",
       "2210622                                   NaN                  NaN   \n",
       "2210623                                   NaN                  NaN   \n",
       "\n",
       "        out.state.count.value support_status  error_string  \\\n",
       "0                         NaN          False                 \n",
       "1                         NaN          False                 \n",
       "2                         NaN          False                 \n",
       "3                         NaN          False                 \n",
       "4                         NaN          False                 \n",
       "...                       ...            ...           ...   \n",
       "2199348                   NaN          False                 \n",
       "2199349                   NaN          False                 \n",
       "2210618                   NaN          False                 \n",
       "2210622                   NaN          False                 \n",
       "2210623                   NaN          False                 \n",
       "\n",
       "        error_string_reduce  normalize_status  \n",
       "0                                       False  \n",
       "1                                       False  \n",
       "2                                       False  \n",
       "3                                       False  \n",
       "4                                       False  \n",
       "...                     ...               ...  \n",
       "2199348                                 False  \n",
       "2199349                                 False  \n",
       "2210618                                 False  \n",
       "2210622                                 False  \n",
       "2210623                                 False  \n",
       "\n",
       "[10650 rows x 103 columns]"
      ]
     },
     "execution_count": 295,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "not_supported_df = not_supported_df.loc[(not_supported_df['support_status'] == False)& (not_supported_df['normalize_status'] == False)]\n",
    "not_supported_df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Sanity check: making sure there are no supported variants that have been marked as normalized"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "not_supported_but_normalized_df =  df.copy()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>in.id</th>\n",
       "      <th>in.name</th>\n",
       "      <th>in.subclass_type</th>\n",
       "      <th>in.variation_type</th>\n",
       "      <th>in.xrefs</th>\n",
       "      <th>in.vrs_xform_plan.type</th>\n",
       "      <th>in.vrs_xform_plan.inputs</th>\n",
       "      <th>in.vrs_xform_plan.policy</th>\n",
       "      <th>out.id</th>\n",
       "      <th>out.type</th>\n",
       "      <th>...</th>\n",
       "      <th>out.state.seq_expr.location.start.value</th>\n",
       "      <th>out.state.seq_expr.location.end.type</th>\n",
       "      <th>out.state.seq_expr.location.end.value</th>\n",
       "      <th>out.state.seq_expr.reverse_complement</th>\n",
       "      <th>out.state.count.type</th>\n",
       "      <th>out.state.count.value</th>\n",
       "      <th>support_status</th>\n",
       "      <th>error_string</th>\n",
       "      <th>error_string_reduce</th>\n",
       "      <th>normalize_status</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>0 rows × 103 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [in.id, in.name, in.subclass_type, in.variation_type, in.xrefs, in.vrs_xform_plan.type, in.vrs_xform_plan.inputs, in.vrs_xform_plan.policy, out.id, out.type, out.definition, in.hgvs.assembly, in.hgvs.nucleotide, out.location.id, out.location.type, out.location.sequence_id, out.location.start.type, out.location.start.value, out.location.end.type, out.location.end.value, out.state.type, out.state.sequence, out.subject.id, out.subject.type, out.subject.sequence_id, out.subject.start.type, out.subject.start.value, out.subject.end.type, out.subject.end.value, out.copy_change, in.seq.assembly, in.seq.accession, in.seq.chr, in.seq.disp_start, in.seq.disp_stop, in.seq.inner_start, in.seq.inner_stop, in.absolute_copies, in.derived_hgvs, in.cyto, out.errors, in.seq.outer_stop, in.seq.outer_start, out.subject.start.min, out.subject.start.max, out.subject.end.min, out.subject.end.max, in.seq.variant_length, out.name, out.subclass_type, out.variation_type, out.hgvs.assembly, out.hgvs.nucleotide, out.vrs_xform_plan.type, out.vrs_xform_plan.inputs, out.vrs_xform_plan.policy, out.xrefs, out.subject.start.comparator, out.subject.end.comparator, out.copies.type, out.copies.value, in.seq.start, in.seq.stop, out.seq.assembly, out.seq.accession, out.seq.chr, out.seq.disp_start, out.seq.disp_stop, out.seq.start, out.seq.stop, out.seq.variant_length, out.absolute_copies, out.derived_hgvs, out.cyto, out.seq.inner_start, out.seq.inner_stop, out.seq.outer_start, out.seq.outer_stop, in.seq.ref_allele_vcf, in.seq.position_vcf, in.seq.alt_allele_vcf, in.canonical_spdi, out.seq.ref_allele_vcf, out.seq.position_vcf, out.seq.alt_allele_vcf, out.canonical_spdi, in.max_copies, in.min_copies, out.state.seq_expr.type, out.state.seq_expr.location.id, out.state.seq_expr.location.type, out.state.seq_expr.location.sequence_id, out.state.seq_expr.location.start.type, out.state.seq_expr.location.start.value, out.state.seq_expr.location.end.type, out.state.seq_expr.location.end.value, out.state.seq_expr.reverse_complement, out.state.count.type, out.state.count.value, support_status, ...]\n",
       "Index: []\n",
       "\n",
       "[0 rows x 103 columns]"
      ]
     },
     "execution_count": 299,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "not_supported_but_normalized_df = not_supported_but_normalized_df.loc[(not_supported_but_normalized_df['support_status'] == False) & \n",
    "                         (not_supported_but_normalized_df['normalize_status'] == True)]\n",
    "not_supported_but_normalized_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "num_supported = len(supported_df)\n",
    "num_supported_not_normalized = len(supported_not_normalized_df)\n",
    "num_not_supported_but_normalized = len(not_supported_but_normalized_df)\n",
    "num_not_supported = len(not_supported_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2131087\n",
      "68890\n",
      "0\n",
      "10650\n"
     ]
    }
   ],
   "source": [
    "print(num_supported)\n",
    "print(num_supported_not_normalized)\n",
    "print(num_not_supported_but_normalized)\n",
    "print(num_not_supported)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3.10.9 ('.venv': venv)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.9"
  },
  "orig_nbformat": 4,
  "vscode": {
   "interpreter": {
    "hash": "60bedfc6dd95a1d756c285a8080496495ec396023cb257f004626f94ae504b38"
   }
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
