{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# <a id='toc1_'></a>[ClinVar Analysis](#toc0_)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Table of contents**<a id='toc0_'></a>    \n",
    "- [ClinVar Analysis](#toc1_)    \n",
    "  - [Initialize](#toc1_1_)    \n",
    "    - [Import necessary libraries](#toc1_1_1_)    \n",
    "    - [Import variant file from downloads- change to personal username](#toc1_1_2_)    \n",
    "  - [Add Supported Status of Variant based on in.vrs_xform_plan.policy](#toc1_2_)    \n",
    "  - [Add Normalization Status of Variant based on out.errors](#toc1_3_)    \n",
    "    - [Set Normalize Status of Variant as T/F](#toc1_3_1_)    \n",
    "      - [Summary Table](#toc1_3_1_1_)    \n",
    "  - [Create subgroups based on Variant Status](#toc1_4_)    \n",
    "    - [Supported and Normalized Variants](#toc1_4_1_)    \n",
    "    - [Supported and Not Normalized Variants](#toc1_4_2_)    \n",
    "    - [Not Supported Variants](#toc1_4_3_)    \n",
    "  - [Counting variants from each group](#toc1_5_)    \n",
    "\n",
    "<!-- vscode-jupyter-toc-config\n",
    "\tnumbering=false\n",
    "\tanchor=true\n",
    "\tflat=false\n",
    "\tminLevel=1\n",
    "\tmaxLevel=6\n",
    "\t/vscode-jupyter-toc-config -->\n",
    "<!-- THIS CELL WILL BE REPLACED ON TOC UPDATE. DO NOT WRITE YOUR TEXT IN THIS CELL -->"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## <a id='toc1_1_'></a>[Initialize](#toc0_)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### <a id='toc1_1_1_'></a>[Import necessary libraries](#toc0_)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import ndjson\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import re\n",
    "from dotenv import load_dotenv"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### <a id='toc1_1_2_'></a>[Import variant file from downloads- change to personal username](#toc0_)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "file = open(\"/Users/rsaxs014/Downloads/output-variation_identity-nolimits.ndjson\")\n",
    "records = ndjson.load(file)\n",
    "\n",
    "df0 = pd.json_normalize(records)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = df0.copy()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## <a id='toc1_2_'></a>[Add Supported Status of Variant based on in.vrs_xform_plan.policy](#toc0_)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Checking for blanks"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "df[\"in.vrs_xform_plan.policy\"] = df[\"in.vrs_xform_plan.policy\"].fillna(\"None\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "in.vrs_xform_plan.policy\n",
       "Canonical SPDI                                      2118669\n",
       "Absolute copy count                                   53263\n",
       "Copy number change (cn loss|del and cn gain|dup)      27104\n",
       "NCBI36 genomic only                                    4771\n",
       "No hgvs or location info                               3089\n",
       "Genotype/Haplotype                                     1440\n",
       "Invalid/unsupported hgvs                               1336\n",
       "Remaining valid hgvs alleles                            941\n",
       "Min/max copy count range not supported                   14\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df[\"in.vrs_xform_plan.policy\"].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "df[\"support_status\"] = df[\"in.vrs_xform_plan.policy\"].copy()\n",
    "\n",
    "df.loc[df[\"support_status\"] == \"Canonical SPDI\", \"support_status\"] = True\n",
    "df.loc[df[\"support_status\"] == \"Absolute copy count\", \"support_status\"] = True\n",
    "df.loc[df[\"support_status\"] == \"Copy number change (cn loss|del and cn gain|dup)\",\n",
    "    \"support_status\"] = True\n",
    "df.loc[df[\"support_status\"] == \"NCBI36 genomic only\", \"support_status\"] = False\n",
    "df.loc[df[\"support_status\"] == \"No hgvs or location info\", \"support_status\"] = False\n",
    "df.loc[df[\"support_status\"] == \"Genotype/Haplotype\", \"support_status\"] = False\n",
    "df.loc[df[\"support_status\"] == \"Invalid/unsupported hgvs\", \"support_status\"] = False\n",
    "df.loc[df[\"support_status\"] == \"Remaining valid hgvs alleles\", \"support_status\"] = True\n",
    "df.loc[df[\"support_status\"] == \"Min/max copy count range not supported\", \n",
    "    \"support_status\"] = False"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "support_status\n",
       "True     2199977\n",
       "False      10650\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df[\"support_status\"].value_counts()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## <a id='toc1_3_'></a>[Add Normalization Status of Variant based on out.errors](#toc0_)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The errors are stored as a list of values, some of which are strings and other of which are dictionaries (determined by whether error was handled at the level of Variation Normalizer or after the normalizer)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The \"get_errors\" function extracts the text error responses for better readability and ease string processing"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_errors(errors):\n",
    "    errors_out = []\n",
    "    for e in errors:\n",
    "        if type(e) == str:\n",
    "            errors_out.append(e)\n",
    "        elif type(e) == dict:\n",
    "            for k, v in e.items():\n",
    "                if k not in [\n",
    "                    \"msg\",\n",
    "                    \"response-errors\",]:\n",
    "                ## only get these keys from normalizer response\n",
    "                    continue\n",
    "                if type(v) == str:\n",
    "                    errors_out.append(v)\n",
    "                elif type(v) == list:\n",
    "                    errors_out.append(\";\".join(v))\n",
    "    return \";\".join(errors_out)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "df[\"error_string\"] = df[\"out.errors\"].fillna(\"\").apply(get_errors)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This is the number of unique error strings"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "There are many different strings because many of the errors contain specific genomic coordinates, which are unlikely to occur more than once"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "505"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df[\"error_string\"].nunique()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "To get the core error message, the numeric values are replaced with \"#\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "def reduce_errors(error_string):\n",
    "    out = error_string.lower()\n",
    "    out = re.sub(\"\\d+\", \"#\", out)\n",
    "    return out"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "def reduce_errors_more(error_string):\n",
    "    errs = error_string.split(\";\")\n",
    "    new_errs = [re.sub(\"\\:[ ]?[^\\s]+[\\s]?\", \"\", err) for err in errs]\n",
    "    return \";\".join(new_errs)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "df[\"error_string_reduce\"] = df[\"error_string\"].apply(reduce_errors)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "There are Not Supported variants that have no error because they were labeled \"Not Supported\" manually."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "An error (\"Not Supported\") is entered manually for those variants so that they are not categorized as normalized"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "df.loc[\n",
    "    (df[\"support_status\"] == False) & (df[\"error_string_reduce\"] == \"\"),\n",
    "    \"error_string_reduce\",\n",
    "] = \"Not Supported\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The error strings had to be reduced further"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "df[\"error_string_reduce_2\"] = df[\"error_string_reduce\"].apply(reduce_errors_more)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "error_string_reduce_2\n",
       "not status #                                                                                              2118672\n",
       "                                                                                                            80814\n",
       "Not Supported                                                                                               10200\n",
       "unrecognized variation record                                                                                 431\n",
       "error returned from variation normalizer;unable to find classification for                                    403\n",
       "error returned from variation normalizer;unable to tokenize;unable to find classification for                  51\n",
       "error returned from variation normalizer;unable to find classification for;unable to tokenize                  44\n",
       "error returned from variation normalizer;unable to find valid result for classification                         8\n",
       "error returned from variation normalizer;nc_#.#is not a supported hgvs genomic duplication or deletion          2\n",
       "error returned from variation normalizer;unable to tokenize                                                     2\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df[\"error_string_reduce_2\"].value_counts()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### <a id='toc1_3_1_'></a>[Set Normalize Status of Variant as T/F](#toc0_)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "If an error is present, the variant was not normalized and therefore has a False Normalize Status"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>in.id</th>\n",
       "      <th>in.name</th>\n",
       "      <th>in.subclass_type</th>\n",
       "      <th>in.variation_type</th>\n",
       "      <th>in.xrefs</th>\n",
       "      <th>in.vrs_xform_plan.type</th>\n",
       "      <th>in.vrs_xform_plan.inputs</th>\n",
       "      <th>in.vrs_xform_plan.policy</th>\n",
       "      <th>out.id</th>\n",
       "      <th>out.type</th>\n",
       "      <th>...</th>\n",
       "      <th>out.canonical_spdi</th>\n",
       "      <th>in.max_copies</th>\n",
       "      <th>in.min_copies</th>\n",
       "      <th>out.seq.inner_start</th>\n",
       "      <th>out.seq.inner_stop</th>\n",
       "      <th>support_status</th>\n",
       "      <th>error_string</th>\n",
       "      <th>error_string_reduce</th>\n",
       "      <th>error_string_reduce_2</th>\n",
       "      <th>normalize_status</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>16098</td>\n",
       "      <td>SLC2A2, 1-BP INS, 793C</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>Insertion</td>\n",
       "      <td>OMIM:138160.0009</td>\n",
       "      <td>Text</td>\n",
       "      <td>[id]</td>\n",
       "      <td>No hgvs or location info</td>\n",
       "      <td>Text:clinvar:16098</td>\n",
       "      <td>Text</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td></td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>425693</td>\n",
       "      <td>NM_001204.6(BMPR2):c.77-?_247+?dup</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>Duplication</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Text</td>\n",
       "      <td>[id]</td>\n",
       "      <td>No hgvs or location info</td>\n",
       "      <td>Text:clinvar:425693</td>\n",
       "      <td>Text</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td></td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2446408</td>\n",
       "      <td>CDHR1, 783G-A (rs147346345)</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>OMIM:609502.0005</td>\n",
       "      <td>Text</td>\n",
       "      <td>[id]</td>\n",
       "      <td>No hgvs or location info</td>\n",
       "      <td>Text:clinvar:2446408</td>\n",
       "      <td>Text</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td></td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>90650</td>\n",
       "      <td>NM_000251.2(MSH2):c.1387-?_1510+?del</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Text</td>\n",
       "      <td>[id]</td>\n",
       "      <td>No hgvs or location info</td>\n",
       "      <td>Text:clinvar:90650</td>\n",
       "      <td>Text</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td></td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>14905</td>\n",
       "      <td>HLA-DRB1, HLA-DRB1*1101</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>Variation</td>\n",
       "      <td>OMIM:142857.0001</td>\n",
       "      <td>Text</td>\n",
       "      <td>[id]</td>\n",
       "      <td>No hgvs or location info</td>\n",
       "      <td>Text:clinvar:14905</td>\n",
       "      <td>Text</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td></td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2210622</th>\n",
       "      <td>1801441</td>\n",
       "      <td>NM_015189.3(EXOC6B):c.2197-66917_2197-66916ins...</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>Insertion</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Allele</td>\n",
       "      <td>[canonical_spdi]</td>\n",
       "      <td>Canonical SPDI</td>\n",
       "      <td>1801441</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NC_000002.12:72251103::CTGATTGATATTTAATAATGTAA...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>Not status 200</td>\n",
       "      <td>not status #</td>\n",
       "      <td>not status #</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2210623</th>\n",
       "      <td>1464440</td>\n",
       "      <td>NM_001267550.2(TTN):c.29512_29513insGGCCGGGCGC...</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>Insertion</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Text</td>\n",
       "      <td>[id]</td>\n",
       "      <td>Invalid/unsupported hgvs</td>\n",
       "      <td>Text:clinvar:1464440</td>\n",
       "      <td>Text</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td></td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2210624</th>\n",
       "      <td>1513408</td>\n",
       "      <td>NM_024928.5(STN1):c.340_352AAG[2]CTACAAGGCCGGG...</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>Insertion</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Text</td>\n",
       "      <td>[id]</td>\n",
       "      <td>Invalid/unsupported hgvs</td>\n",
       "      <td>Text:clinvar:1513408</td>\n",
       "      <td>Text</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td></td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2210625</th>\n",
       "      <td>1496502</td>\n",
       "      <td>NM_007255.3(B4GALT7):c.881_882insTGAGGTGGATTAA...</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>Insertion</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Allele</td>\n",
       "      <td>[canonical_spdi]</td>\n",
       "      <td>Canonical SPDI</td>\n",
       "      <td>1496502</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NC_000005.10:177609590:GT:GTTGAGGTGGATTAAACCAA...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>Not status 200</td>\n",
       "      <td>not status #</td>\n",
       "      <td>not status #</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2210626</th>\n",
       "      <td>1801464</td>\n",
       "      <td>NM_176787.5(PIGN):c.2673-514_2673-513insGCAAAC...</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>Insertion</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Allele</td>\n",
       "      <td>[hgvs]</td>\n",
       "      <td>Remaining valid hgvs alleles</td>\n",
       "      <td>ga4gh:VA.3BXZ9oWeEqz4VM16yV8mgjvQ8ZTElB2J</td>\n",
       "      <td>Allele</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2210627 rows Ã— 93 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           in.id                                            in.name  \\\n",
       "0          16098                             SLC2A2, 1-BP INS, 793C   \n",
       "1         425693                 NM_001204.6(BMPR2):c.77-?_247+?dup   \n",
       "2        2446408                        CDHR1, 783G-A (rs147346345)   \n",
       "3          90650               NM_000251.2(MSH2):c.1387-?_1510+?del   \n",
       "4          14905                            HLA-DRB1, HLA-DRB1*1101   \n",
       "...          ...                                                ...   \n",
       "2210622  1801441  NM_015189.3(EXOC6B):c.2197-66917_2197-66916ins...   \n",
       "2210623  1464440  NM_001267550.2(TTN):c.29512_29513insGGCCGGGCGC...   \n",
       "2210624  1513408  NM_024928.5(STN1):c.340_352AAG[2]CTACAAGGCCGGG...   \n",
       "2210625  1496502  NM_007255.3(B4GALT7):c.881_882insTGAGGTGGATTAA...   \n",
       "2210626  1801464  NM_176787.5(PIGN):c.2673-514_2673-513insGCAAAC...   \n",
       "\n",
       "        in.subclass_type          in.variation_type          in.xrefs  \\\n",
       "0           SimpleAllele                  Insertion  OMIM:138160.0009   \n",
       "1           SimpleAllele                Duplication               NaN   \n",
       "2           SimpleAllele  single nucleotide variant  OMIM:609502.0005   \n",
       "3           SimpleAllele                   Deletion               NaN   \n",
       "4           SimpleAllele                  Variation  OMIM:142857.0001   \n",
       "...                  ...                        ...               ...   \n",
       "2210622     SimpleAllele                  Insertion               NaN   \n",
       "2210623     SimpleAllele                  Insertion               NaN   \n",
       "2210624     SimpleAllele                  Insertion               NaN   \n",
       "2210625     SimpleAllele                  Insertion               NaN   \n",
       "2210626     SimpleAllele                  Insertion               NaN   \n",
       "\n",
       "        in.vrs_xform_plan.type in.vrs_xform_plan.inputs  \\\n",
       "0                         Text                     [id]   \n",
       "1                         Text                     [id]   \n",
       "2                         Text                     [id]   \n",
       "3                         Text                     [id]   \n",
       "4                         Text                     [id]   \n",
       "...                        ...                      ...   \n",
       "2210622                 Allele         [canonical_spdi]   \n",
       "2210623                   Text                     [id]   \n",
       "2210624                   Text                     [id]   \n",
       "2210625                 Allele         [canonical_spdi]   \n",
       "2210626                 Allele                   [hgvs]   \n",
       "\n",
       "             in.vrs_xform_plan.policy  \\\n",
       "0            No hgvs or location info   \n",
       "1            No hgvs or location info   \n",
       "2            No hgvs or location info   \n",
       "3            No hgvs or location info   \n",
       "4            No hgvs or location info   \n",
       "...                               ...   \n",
       "2210622                Canonical SPDI   \n",
       "2210623      Invalid/unsupported hgvs   \n",
       "2210624      Invalid/unsupported hgvs   \n",
       "2210625                Canonical SPDI   \n",
       "2210626  Remaining valid hgvs alleles   \n",
       "\n",
       "                                            out.id out.type  ...  \\\n",
       "0                               Text:clinvar:16098     Text  ...   \n",
       "1                              Text:clinvar:425693     Text  ...   \n",
       "2                             Text:clinvar:2446408     Text  ...   \n",
       "3                               Text:clinvar:90650     Text  ...   \n",
       "4                               Text:clinvar:14905     Text  ...   \n",
       "...                                            ...      ...  ...   \n",
       "2210622                                    1801441      NaN  ...   \n",
       "2210623                       Text:clinvar:1464440     Text  ...   \n",
       "2210624                       Text:clinvar:1513408     Text  ...   \n",
       "2210625                                    1496502      NaN  ...   \n",
       "2210626  ga4gh:VA.3BXZ9oWeEqz4VM16yV8mgjvQ8ZTElB2J   Allele  ...   \n",
       "\n",
       "                                        out.canonical_spdi in.max_copies  \\\n",
       "0                                                      NaN           NaN   \n",
       "1                                                      NaN           NaN   \n",
       "2                                                      NaN           NaN   \n",
       "3                                                      NaN           NaN   \n",
       "4                                                      NaN           NaN   \n",
       "...                                                    ...           ...   \n",
       "2210622  NC_000002.12:72251103::CTGATTGATATTTAATAATGTAA...           NaN   \n",
       "2210623                                                NaN           NaN   \n",
       "2210624                                                NaN           NaN   \n",
       "2210625  NC_000005.10:177609590:GT:GTTGAGGTGGATTAAACCAA...           NaN   \n",
       "2210626                                                NaN           NaN   \n",
       "\n",
       "        in.min_copies out.seq.inner_start out.seq.inner_stop support_status  \\\n",
       "0                 NaN                 NaN                NaN          False   \n",
       "1                 NaN                 NaN                NaN          False   \n",
       "2                 NaN                 NaN                NaN          False   \n",
       "3                 NaN                 NaN                NaN          False   \n",
       "4                 NaN                 NaN                NaN          False   \n",
       "...               ...                 ...                ...            ...   \n",
       "2210622           NaN                 NaN                NaN           True   \n",
       "2210623           NaN                 NaN                NaN          False   \n",
       "2210624           NaN                 NaN                NaN          False   \n",
       "2210625           NaN                 NaN                NaN           True   \n",
       "2210626           NaN                 NaN                NaN           True   \n",
       "\n",
       "           error_string  error_string_reduce error_string_reduce_2  \\\n",
       "0                              Not Supported         Not Supported   \n",
       "1                              Not Supported         Not Supported   \n",
       "2                              Not Supported         Not Supported   \n",
       "3                              Not Supported         Not Supported   \n",
       "4                              Not Supported         Not Supported   \n",
       "...                 ...                  ...                   ...   \n",
       "2210622  Not status 200         not status #          not status #   \n",
       "2210623                        Not Supported         Not Supported   \n",
       "2210624                        Not Supported         Not Supported   \n",
       "2210625  Not status 200         not status #          not status #   \n",
       "2210626                                                              \n",
       "\n",
       "         normalize_status  \n",
       "0                    True  \n",
       "1                    True  \n",
       "2                    True  \n",
       "3                    True  \n",
       "4                    True  \n",
       "...                   ...  \n",
       "2210622              True  \n",
       "2210623              True  \n",
       "2210624              True  \n",
       "2210625              True  \n",
       "2210626             False  \n",
       "\n",
       "[2210627 rows x 93 columns]"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df[\"normalize_status\"] = df[\"error_string_reduce\"] != \"\"\n",
    "df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### <a id='toc1_3_1_1_'></a>[Summary Table](#toc0_)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "In the table below, the cells show the number of variants with each expected behavior and how they actually ended up performing."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "If a variant was in an \"expected to pass\" category and ends up as text, that is an instance of a normalizer failure on a supported variant"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead tr th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe thead tr:last-of-type th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th colspan=\"5\" halign=\"left\">in.id</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th></th>\n",
       "      <th>out.type</th>\n",
       "      <th>Allele</th>\n",
       "      <th>CopyNumberChange</th>\n",
       "      <th>CopyNumberCount</th>\n",
       "      <th>NONE</th>\n",
       "      <th>Text</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>support_status</th>\n",
       "      <th>in.vrs_xform_plan.policy</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th rowspan=\"5\" valign=\"top\">False</th>\n",
       "      <th>Genotype/Haplotype</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1440</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Invalid/unsupported hgvs</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>19</td>\n",
       "      <td>1317</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Min/max copy count range not supported</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NCBI36 genomic only</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4771</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>No hgvs or location info</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3089</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"4\" valign=\"top\">True</th>\n",
       "      <th>Absolute copy count</th>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>52880</td>\n",
       "      <td>379</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Canonical SPDI</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2118669</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Copy number change (cn loss|del and cn gain|dup)</th>\n",
       "      <td>0</td>\n",
       "      <td>26988</td>\n",
       "      <td>0</td>\n",
       "      <td>110</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Remaining valid hgvs alleles</th>\n",
       "      <td>935</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                 in.id  \\\n",
       "out.type                                                        Allele   \n",
       "support_status in.vrs_xform_plan.policy                                  \n",
       "False          Genotype/Haplotype                                    0   \n",
       "               Invalid/unsupported hgvs                              0   \n",
       "               Min/max copy count range not supported                0   \n",
       "               NCBI36 genomic only                                   0   \n",
       "               No hgvs or location info                              0   \n",
       "True           Absolute copy count                                   0   \n",
       "               Canonical SPDI                                        0   \n",
       "               Copy number change (cn loss|del and cn gain|dup)      0   \n",
       "               Remaining valid hgvs alleles                        935   \n",
       "\n",
       "                                                                                  \\\n",
       "out.type                                                        CopyNumberChange   \n",
       "support_status in.vrs_xform_plan.policy                                            \n",
       "False          Genotype/Haplotype                                              0   \n",
       "               Invalid/unsupported hgvs                                        0   \n",
       "               Min/max copy count range not supported                          0   \n",
       "               NCBI36 genomic only                                             0   \n",
       "               No hgvs or location info                                        0   \n",
       "True           Absolute copy count                                             1   \n",
       "               Canonical SPDI                                                  0   \n",
       "               Copy number change (cn loss|del and cn gain|dup)            26988   \n",
       "               Remaining valid hgvs alleles                                    0   \n",
       "\n",
       "                                                                                 \\\n",
       "out.type                                                        CopyNumberCount   \n",
       "support_status in.vrs_xform_plan.policy                                           \n",
       "False          Genotype/Haplotype                                             0   \n",
       "               Invalid/unsupported hgvs                                       0   \n",
       "               Min/max copy count range not supported                         0   \n",
       "               NCBI36 genomic only                                            0   \n",
       "               No hgvs or location info                                       0   \n",
       "True           Absolute copy count                                        52880   \n",
       "               Canonical SPDI                                                 0   \n",
       "               Copy number change (cn loss|del and cn gain|dup)               0   \n",
       "               Remaining valid hgvs alleles                                   0   \n",
       "\n",
       "                                                                                \n",
       "out.type                                                            NONE  Text  \n",
       "support_status in.vrs_xform_plan.policy                                         \n",
       "False          Genotype/Haplotype                                      0  1440  \n",
       "               Invalid/unsupported hgvs                               19  1317  \n",
       "               Min/max copy count range not supported                  0    14  \n",
       "               NCBI36 genomic only                                     0  4771  \n",
       "               No hgvs or location info                                0  3089  \n",
       "True           Absolute copy count                                   379     3  \n",
       "               Canonical SPDI                                    2118669     0  \n",
       "               Copy number change (cn loss|del and cn gain|dup)      110     6  \n",
       "               Remaining valid hgvs alleles                            5     1  "
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df[[\"in.id\", \"support_status\", \"in.vrs_xform_plan.policy\", \"out.type\"]].fillna(\n",
    "    \"NONE\"\n",
    ").groupby([\"support_status\", \"in.vrs_xform_plan.policy\", \"out.type\"]).count().unstack(\n",
    "    level=2\n",
    ").fillna(\n",
    "    0\n",
    ").astype(\n",
    "    int\n",
    ")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## <a id='toc1_4_'></a>[Create subgroups based on Variant Status](#toc0_)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### <a id='toc1_4_1_'></a>[Supported and Normalized Variants](#toc0_)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "supported_df = df.copy()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>in.id</th>\n",
       "      <th>in.name</th>\n",
       "      <th>in.subclass_type</th>\n",
       "      <th>in.variation_type</th>\n",
       "      <th>in.xrefs</th>\n",
       "      <th>in.vrs_xform_plan.type</th>\n",
       "      <th>in.vrs_xform_plan.inputs</th>\n",
       "      <th>in.vrs_xform_plan.policy</th>\n",
       "      <th>out.id</th>\n",
       "      <th>out.type</th>\n",
       "      <th>...</th>\n",
       "      <th>out.canonical_spdi</th>\n",
       "      <th>in.max_copies</th>\n",
       "      <th>in.min_copies</th>\n",
       "      <th>out.seq.inner_start</th>\n",
       "      <th>out.seq.inner_stop</th>\n",
       "      <th>support_status</th>\n",
       "      <th>error_string</th>\n",
       "      <th>error_string_reduce</th>\n",
       "      <th>error_string_reduce_2</th>\n",
       "      <th>normalize_status</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>170</th>\n",
       "      <td>10342</td>\n",
       "      <td>NG_011403.2:g.(80027_96047)_(131648_164496)del</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>OMIM:300841.0259</td>\n",
       "      <td>CopyNumberChange</td>\n",
       "      <td>[hgvs]</td>\n",
       "      <td>Copy number change (cn loss|del and cn gain|dup)</td>\n",
       "      <td>10342</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>Error returned from variation normalizer;Unabl...</td>\n",
       "      <td>error returned from variation normalizer;unabl...</td>\n",
       "      <td>error returned from variation normalizer;unabl...</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>177</th>\n",
       "      <td>989220</td>\n",
       "      <td>NC_000015.9:g.(44884528_44881613)_(44877833_44...</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>Duplication</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CopyNumberChange</td>\n",
       "      <td>[hgvs]</td>\n",
       "      <td>Copy number change (cn loss|del and cn gain|dup)</td>\n",
       "      <td>989220</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>Error returned from variation normalizer;Unabl...</td>\n",
       "      <td>error returned from variation normalizer;unabl...</td>\n",
       "      <td>error returned from variation normalizer;unabl...</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>283</th>\n",
       "      <td>1706497</td>\n",
       "      <td>GRCh37/hg19 16p11.2(chr16:29432212-30177807)x1</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>copy number loss</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CopyNumberCount</td>\n",
       "      <td>[hgvs, absolute_copies]</td>\n",
       "      <td>Absolute copy count</td>\n",
       "      <td>1706497</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>Error returned from variation normalizer;Unabl...</td>\n",
       "      <td>error returned from variation normalizer;unabl...</td>\n",
       "      <td>error returned from variation normalizer;unabl...</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>310</th>\n",
       "      <td>1705934</td>\n",
       "      <td>GRCh37/hg19 Xp22.33(chrX:566009-1356042)x3</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>copy number gain</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CopyNumberCount</td>\n",
       "      <td>[hgvs, absolute_copies]</td>\n",
       "      <td>Absolute copy count</td>\n",
       "      <td>1705934</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>Error returned from variation normalizer;Unabl...</td>\n",
       "      <td>error returned from variation normalizer;unabl...</td>\n",
       "      <td>error returned from variation normalizer;unabl...</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>336</th>\n",
       "      <td>1711099</td>\n",
       "      <td>GRCh37/hg19 10q26.13-26.3(chr10:127198625-1354...</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>copy number gain</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CopyNumberCount</td>\n",
       "      <td>[hgvs, absolute_copies]</td>\n",
       "      <td>Absolute copy count</td>\n",
       "      <td>1711099</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>Error returned from variation normalizer;Unabl...</td>\n",
       "      <td>error returned from variation normalizer;unabl...</td>\n",
       "      <td>error returned from variation normalizer;unabl...</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2210618</th>\n",
       "      <td>1972383</td>\n",
       "      <td>NM_003470.3(USP7):c.383+10_383+11insGTTTAAATGA...</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>Insertion</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Allele</td>\n",
       "      <td>[canonical_spdi]</td>\n",
       "      <td>Canonical SPDI</td>\n",
       "      <td>1972383</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NC_000016.10:8923204:AC:ACAACGATGTGGGGGTTTGTAG...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>Not status 200</td>\n",
       "      <td>not status #</td>\n",
       "      <td>not status #</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2210619</th>\n",
       "      <td>1181160</td>\n",
       "      <td>NM_030962.4(SBF2):c.55+97_55+98insCGGGCGTCGGGGC</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>Microsatellite</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Allele</td>\n",
       "      <td>[canonical_spdi]</td>\n",
       "      <td>Canonical SPDI</td>\n",
       "      <td>1181160</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NC_000011.10:10293917:CCCCGACGCCCG:CCCCGACGCCC...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>Not status 200</td>\n",
       "      <td>not status #</td>\n",
       "      <td>not status #</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2210620</th>\n",
       "      <td>455365</td>\n",
       "      <td>NM_000249.4(MLH1):c.1039-8_1039-7insTTTTTTTTTT...</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>Insertion</td>\n",
       "      <td>ClinGen:CA658655821,dbSNP:535965616</td>\n",
       "      <td>Allele</td>\n",
       "      <td>[canonical_spdi]</td>\n",
       "      <td>Canonical SPDI</td>\n",
       "      <td>455365</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NC_000003.12:37025629::TTTTTTTTTTTTTTTTTTA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>Not status 200</td>\n",
       "      <td>not status #</td>\n",
       "      <td>not status #</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2210622</th>\n",
       "      <td>1801441</td>\n",
       "      <td>NM_015189.3(EXOC6B):c.2197-66917_2197-66916ins...</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>Insertion</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Allele</td>\n",
       "      <td>[canonical_spdi]</td>\n",
       "      <td>Canonical SPDI</td>\n",
       "      <td>1801441</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NC_000002.12:72251103::CTGATTGATATTTAATAATGTAA...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>Not status 200</td>\n",
       "      <td>not status #</td>\n",
       "      <td>not status #</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2210625</th>\n",
       "      <td>1496502</td>\n",
       "      <td>NM_007255.3(B4GALT7):c.881_882insTGAGGTGGATTAA...</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>Insertion</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Allele</td>\n",
       "      <td>[canonical_spdi]</td>\n",
       "      <td>Canonical SPDI</td>\n",
       "      <td>1496502</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NC_000005.10:177609590:GT:GTTGAGGTGGATTAAACCAA...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>Not status 200</td>\n",
       "      <td>not status #</td>\n",
       "      <td>not status #</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2119163 rows Ã— 93 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           in.id                                            in.name  \\\n",
       "170        10342     NG_011403.2:g.(80027_96047)_(131648_164496)del   \n",
       "177       989220  NC_000015.9:g.(44884528_44881613)_(44877833_44...   \n",
       "283      1706497     GRCh37/hg19 16p11.2(chr16:29432212-30177807)x1   \n",
       "310      1705934         GRCh37/hg19 Xp22.33(chrX:566009-1356042)x3   \n",
       "336      1711099  GRCh37/hg19 10q26.13-26.3(chr10:127198625-1354...   \n",
       "...          ...                                                ...   \n",
       "2210618  1972383  NM_003470.3(USP7):c.383+10_383+11insGTTTAAATGA...   \n",
       "2210619  1181160    NM_030962.4(SBF2):c.55+97_55+98insCGGGCGTCGGGGC   \n",
       "2210620   455365  NM_000249.4(MLH1):c.1039-8_1039-7insTTTTTTTTTT...   \n",
       "2210622  1801441  NM_015189.3(EXOC6B):c.2197-66917_2197-66916ins...   \n",
       "2210625  1496502  NM_007255.3(B4GALT7):c.881_882insTGAGGTGGATTAA...   \n",
       "\n",
       "        in.subclass_type in.variation_type  \\\n",
       "170         SimpleAllele          Deletion   \n",
       "177         SimpleAllele       Duplication   \n",
       "283         SimpleAllele  copy number loss   \n",
       "310         SimpleAllele  copy number gain   \n",
       "336         SimpleAllele  copy number gain   \n",
       "...                  ...               ...   \n",
       "2210618     SimpleAllele         Insertion   \n",
       "2210619     SimpleAllele    Microsatellite   \n",
       "2210620     SimpleAllele         Insertion   \n",
       "2210622     SimpleAllele         Insertion   \n",
       "2210625     SimpleAllele         Insertion   \n",
       "\n",
       "                                    in.xrefs in.vrs_xform_plan.type  \\\n",
       "170                         OMIM:300841.0259       CopyNumberChange   \n",
       "177                                      NaN       CopyNumberChange   \n",
       "283                                      NaN        CopyNumberCount   \n",
       "310                                      NaN        CopyNumberCount   \n",
       "336                                      NaN        CopyNumberCount   \n",
       "...                                      ...                    ...   \n",
       "2210618                                  NaN                 Allele   \n",
       "2210619                                  NaN                 Allele   \n",
       "2210620  ClinGen:CA658655821,dbSNP:535965616                 Allele   \n",
       "2210622                                  NaN                 Allele   \n",
       "2210625                                  NaN                 Allele   \n",
       "\n",
       "        in.vrs_xform_plan.inputs  \\\n",
       "170                       [hgvs]   \n",
       "177                       [hgvs]   \n",
       "283      [hgvs, absolute_copies]   \n",
       "310      [hgvs, absolute_copies]   \n",
       "336      [hgvs, absolute_copies]   \n",
       "...                          ...   \n",
       "2210618         [canonical_spdi]   \n",
       "2210619         [canonical_spdi]   \n",
       "2210620         [canonical_spdi]   \n",
       "2210622         [canonical_spdi]   \n",
       "2210625         [canonical_spdi]   \n",
       "\n",
       "                                 in.vrs_xform_plan.policy   out.id out.type  \\\n",
       "170      Copy number change (cn loss|del and cn gain|dup)    10342      NaN   \n",
       "177      Copy number change (cn loss|del and cn gain|dup)   989220      NaN   \n",
       "283                                   Absolute copy count  1706497      NaN   \n",
       "310                                   Absolute copy count  1705934      NaN   \n",
       "336                                   Absolute copy count  1711099      NaN   \n",
       "...                                                   ...      ...      ...   \n",
       "2210618                                    Canonical SPDI  1972383      NaN   \n",
       "2210619                                    Canonical SPDI  1181160      NaN   \n",
       "2210620                                    Canonical SPDI   455365      NaN   \n",
       "2210622                                    Canonical SPDI  1801441      NaN   \n",
       "2210625                                    Canonical SPDI  1496502      NaN   \n",
       "\n",
       "         ...                                 out.canonical_spdi in.max_copies  \\\n",
       "170      ...                                                NaN           NaN   \n",
       "177      ...                                                NaN           NaN   \n",
       "283      ...                                                NaN           NaN   \n",
       "310      ...                                                NaN           NaN   \n",
       "336      ...                                                NaN           NaN   \n",
       "...      ...                                                ...           ...   \n",
       "2210618  ...  NC_000016.10:8923204:AC:ACAACGATGTGGGGGTTTGTAG...           NaN   \n",
       "2210619  ...  NC_000011.10:10293917:CCCCGACGCCCG:CCCCGACGCCC...           NaN   \n",
       "2210620  ...         NC_000003.12:37025629::TTTTTTTTTTTTTTTTTTA           NaN   \n",
       "2210622  ...  NC_000002.12:72251103::CTGATTGATATTTAATAATGTAA...           NaN   \n",
       "2210625  ...  NC_000005.10:177609590:GT:GTTGAGGTGGATTAAACCAA...           NaN   \n",
       "\n",
       "        in.min_copies out.seq.inner_start out.seq.inner_stop support_status  \\\n",
       "170               NaN                 NaN                NaN           True   \n",
       "177               NaN                 NaN                NaN           True   \n",
       "283               NaN                 NaN                NaN           True   \n",
       "310               NaN                 NaN                NaN           True   \n",
       "336               NaN                 NaN                NaN           True   \n",
       "...               ...                 ...                ...            ...   \n",
       "2210618           NaN                 NaN                NaN           True   \n",
       "2210619           NaN                 NaN                NaN           True   \n",
       "2210620           NaN                 NaN                NaN           True   \n",
       "2210622           NaN                 NaN                NaN           True   \n",
       "2210625           NaN                 NaN                NaN           True   \n",
       "\n",
       "                                              error_string  \\\n",
       "170      Error returned from variation normalizer;Unabl...   \n",
       "177      Error returned from variation normalizer;Unabl...   \n",
       "283      Error returned from variation normalizer;Unabl...   \n",
       "310      Error returned from variation normalizer;Unabl...   \n",
       "336      Error returned from variation normalizer;Unabl...   \n",
       "...                                                    ...   \n",
       "2210618                                     Not status 200   \n",
       "2210619                                     Not status 200   \n",
       "2210620                                     Not status 200   \n",
       "2210622                                     Not status 200   \n",
       "2210625                                     Not status 200   \n",
       "\n",
       "                                       error_string_reduce  \\\n",
       "170      error returned from variation normalizer;unabl...   \n",
       "177      error returned from variation normalizer;unabl...   \n",
       "283      error returned from variation normalizer;unabl...   \n",
       "310      error returned from variation normalizer;unabl...   \n",
       "336      error returned from variation normalizer;unabl...   \n",
       "...                                                    ...   \n",
       "2210618                                       not status #   \n",
       "2210619                                       not status #   \n",
       "2210620                                       not status #   \n",
       "2210622                                       not status #   \n",
       "2210625                                       not status #   \n",
       "\n",
       "                                     error_string_reduce_2  normalize_status  \n",
       "170      error returned from variation normalizer;unabl...              True  \n",
       "177      error returned from variation normalizer;unabl...              True  \n",
       "283      error returned from variation normalizer;unabl...              True  \n",
       "310      error returned from variation normalizer;unabl...              True  \n",
       "336      error returned from variation normalizer;unabl...              True  \n",
       "...                                                    ...               ...  \n",
       "2210618                                       not status #              True  \n",
       "2210619                                       not status #              True  \n",
       "2210620                                       not status #              True  \n",
       "2210622                                       not status #              True  \n",
       "2210625                                       not status #              True  \n",
       "\n",
       "[2119163 rows x 93 columns]"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "supported_df = supported_df.loc[\n",
    "    (supported_df[\"support_status\"] == True)\n",
    "    & (supported_df[\"normalize_status\"] == True)\n",
    "]\n",
    "supported_df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### <a id='toc1_4_2_'></a>[Supported and Not Normalized Variants](#toc0_)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [],
   "source": [
    "supported_not_normalized_df = df.copy()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>in.id</th>\n",
       "      <th>in.name</th>\n",
       "      <th>in.subclass_type</th>\n",
       "      <th>in.variation_type</th>\n",
       "      <th>in.xrefs</th>\n",
       "      <th>in.vrs_xform_plan.type</th>\n",
       "      <th>in.vrs_xform_plan.inputs</th>\n",
       "      <th>in.vrs_xform_plan.policy</th>\n",
       "      <th>out.id</th>\n",
       "      <th>out.type</th>\n",
       "      <th>...</th>\n",
       "      <th>out.canonical_spdi</th>\n",
       "      <th>in.max_copies</th>\n",
       "      <th>in.min_copies</th>\n",
       "      <th>out.seq.inner_start</th>\n",
       "      <th>out.seq.inner_stop</th>\n",
       "      <th>support_status</th>\n",
       "      <th>error_string</th>\n",
       "      <th>error_string_reduce</th>\n",
       "      <th>error_string_reduce_2</th>\n",
       "      <th>normalize_status</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>149</th>\n",
       "      <td>1676330</td>\n",
       "      <td>NM_054027.6(ANKH):c.259G&gt;A (p.Val87Ile)</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Allele</td>\n",
       "      <td>[hgvs]</td>\n",
       "      <td>Remaining valid hgvs alleles</td>\n",
       "      <td>ga4gh:VA.T3hLVZajyx7AGQKV7la2RgE9CO0Wiv2y</td>\n",
       "      <td>Allele</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>150</th>\n",
       "      <td>1676394</td>\n",
       "      <td>NM_000786.4(CYP51A1):c.1291C&gt;T (p.Arg431Cys)</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Allele</td>\n",
       "      <td>[hgvs]</td>\n",
       "      <td>Remaining valid hgvs alleles</td>\n",
       "      <td>ga4gh:VA.fQvrcvvCsScmvBvq4d6nxm09Q7avAmbX</td>\n",
       "      <td>Allele</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>151</th>\n",
       "      <td>1676371</td>\n",
       "      <td>NM_000088.4(COL1A1):c.1037C&gt;T (p.Pro346Leu)</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Allele</td>\n",
       "      <td>[hgvs]</td>\n",
       "      <td>Remaining valid hgvs alleles</td>\n",
       "      <td>ga4gh:VA.2d9Gsax-wmwO0zsQ-ZIhCExbACqjVxLO</td>\n",
       "      <td>Allele</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>152</th>\n",
       "      <td>1676476</td>\n",
       "      <td>NM_012309.5(SHANK2):c.460C&gt;T (p.Gln154Ter)</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Allele</td>\n",
       "      <td>[hgvs]</td>\n",
       "      <td>Remaining valid hgvs alleles</td>\n",
       "      <td>ga4gh:VA.dZAOK2Sy6aJlXan36nZYUjEJpk5WX8pF</td>\n",
       "      <td>Allele</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>153</th>\n",
       "      <td>1676377</td>\n",
       "      <td>NM_012064.4(MIP):c.20C&gt;T (p.Ala7Val)</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Allele</td>\n",
       "      <td>[hgvs]</td>\n",
       "      <td>Remaining valid hgvs alleles</td>\n",
       "      <td>ga4gh:VA.-FEJu4eN_baUv2BWb4BxuoSyW4r8FBub</td>\n",
       "      <td>Allele</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2199359</th>\n",
       "      <td>1430299</td>\n",
       "      <td>NM_001012759.3(CTU2):c.737+9_737+10insTGAGAGCC...</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>Insertion</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Allele</td>\n",
       "      <td>[hgvs]</td>\n",
       "      <td>Remaining valid hgvs alleles</td>\n",
       "      <td>ga4gh:VA.dm7V9TRmkiQSkvMWrtt75FRyyBZBNApj</td>\n",
       "      <td>Allele</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2210490</th>\n",
       "      <td>993055</td>\n",
       "      <td>NC_000013.10:g.32889619_32890666dup</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>Duplication</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CopyNumberChange</td>\n",
       "      <td>[hgvs]</td>\n",
       "      <td>Copy number change (cn loss|del and cn gain|dup)</td>\n",
       "      <td>ga4gh:CX.eR37N88iZGBhrAbApaJ8svu_dimnjyWD</td>\n",
       "      <td>CopyNumberChange</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2210491</th>\n",
       "      <td>1755741</td>\n",
       "      <td>NM_000059.4(BRCA2):c.6842-587_7007+2347dup</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>Duplication</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CopyNumberChange</td>\n",
       "      <td>[hgvs]</td>\n",
       "      <td>Copy number change (cn loss|del and cn gain|dup)</td>\n",
       "      <td>ga4gh:CX.wAuyGaZw-vC4Xx4hFVf6574KhJ9w4nnv</td>\n",
       "      <td>CopyNumberChange</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2210553</th>\n",
       "      <td>1508452</td>\n",
       "      <td>NM_001903.5(CTNNA1):c.2709_*44dup (p.Met903_Te...</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>Duplication</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CopyNumberChange</td>\n",
       "      <td>[hgvs]</td>\n",
       "      <td>Copy number change (cn loss|del and cn gain|dup)</td>\n",
       "      <td>ga4gh:CX.JmTrwmY_R9nyZ6Z-nGYfwk02VTmd8OpW</td>\n",
       "      <td>CopyNumberChange</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2210626</th>\n",
       "      <td>1801464</td>\n",
       "      <td>NM_176787.5(PIGN):c.2673-514_2673-513insGCAAAC...</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>Insertion</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Allele</td>\n",
       "      <td>[hgvs]</td>\n",
       "      <td>Remaining valid hgvs alleles</td>\n",
       "      <td>ga4gh:VA.3BXZ9oWeEqz4VM16yV8mgjvQ8ZTElB2J</td>\n",
       "      <td>Allele</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>80814 rows Ã— 93 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           in.id                                            in.name  \\\n",
       "149      1676330            NM_054027.6(ANKH):c.259G>A (p.Val87Ile)   \n",
       "150      1676394       NM_000786.4(CYP51A1):c.1291C>T (p.Arg431Cys)   \n",
       "151      1676371        NM_000088.4(COL1A1):c.1037C>T (p.Pro346Leu)   \n",
       "152      1676476         NM_012309.5(SHANK2):c.460C>T (p.Gln154Ter)   \n",
       "153      1676377               NM_012064.4(MIP):c.20C>T (p.Ala7Val)   \n",
       "...          ...                                                ...   \n",
       "2199359  1430299  NM_001012759.3(CTU2):c.737+9_737+10insTGAGAGCC...   \n",
       "2210490   993055                NC_000013.10:g.32889619_32890666dup   \n",
       "2210491  1755741         NM_000059.4(BRCA2):c.6842-587_7007+2347dup   \n",
       "2210553  1508452  NM_001903.5(CTNNA1):c.2709_*44dup (p.Met903_Te...   \n",
       "2210626  1801464  NM_176787.5(PIGN):c.2673-514_2673-513insGCAAAC...   \n",
       "\n",
       "        in.subclass_type          in.variation_type in.xrefs  \\\n",
       "149         SimpleAllele  single nucleotide variant      NaN   \n",
       "150         SimpleAllele  single nucleotide variant      NaN   \n",
       "151         SimpleAllele  single nucleotide variant      NaN   \n",
       "152         SimpleAllele  single nucleotide variant      NaN   \n",
       "153         SimpleAllele  single nucleotide variant      NaN   \n",
       "...                  ...                        ...      ...   \n",
       "2199359     SimpleAllele                  Insertion      NaN   \n",
       "2210490     SimpleAllele                Duplication      NaN   \n",
       "2210491     SimpleAllele                Duplication      NaN   \n",
       "2210553     SimpleAllele                Duplication      NaN   \n",
       "2210626     SimpleAllele                  Insertion      NaN   \n",
       "\n",
       "        in.vrs_xform_plan.type in.vrs_xform_plan.inputs  \\\n",
       "149                     Allele                   [hgvs]   \n",
       "150                     Allele                   [hgvs]   \n",
       "151                     Allele                   [hgvs]   \n",
       "152                     Allele                   [hgvs]   \n",
       "153                     Allele                   [hgvs]   \n",
       "...                        ...                      ...   \n",
       "2199359                 Allele                   [hgvs]   \n",
       "2210490       CopyNumberChange                   [hgvs]   \n",
       "2210491       CopyNumberChange                   [hgvs]   \n",
       "2210553       CopyNumberChange                   [hgvs]   \n",
       "2210626                 Allele                   [hgvs]   \n",
       "\n",
       "                                 in.vrs_xform_plan.policy  \\\n",
       "149                          Remaining valid hgvs alleles   \n",
       "150                          Remaining valid hgvs alleles   \n",
       "151                          Remaining valid hgvs alleles   \n",
       "152                          Remaining valid hgvs alleles   \n",
       "153                          Remaining valid hgvs alleles   \n",
       "...                                                   ...   \n",
       "2199359                      Remaining valid hgvs alleles   \n",
       "2210490  Copy number change (cn loss|del and cn gain|dup)   \n",
       "2210491  Copy number change (cn loss|del and cn gain|dup)   \n",
       "2210553  Copy number change (cn loss|del and cn gain|dup)   \n",
       "2210626                      Remaining valid hgvs alleles   \n",
       "\n",
       "                                            out.id          out.type  ...  \\\n",
       "149      ga4gh:VA.T3hLVZajyx7AGQKV7la2RgE9CO0Wiv2y            Allele  ...   \n",
       "150      ga4gh:VA.fQvrcvvCsScmvBvq4d6nxm09Q7avAmbX            Allele  ...   \n",
       "151      ga4gh:VA.2d9Gsax-wmwO0zsQ-ZIhCExbACqjVxLO            Allele  ...   \n",
       "152      ga4gh:VA.dZAOK2Sy6aJlXan36nZYUjEJpk5WX8pF            Allele  ...   \n",
       "153      ga4gh:VA.-FEJu4eN_baUv2BWb4BxuoSyW4r8FBub            Allele  ...   \n",
       "...                                            ...               ...  ...   \n",
       "2199359  ga4gh:VA.dm7V9TRmkiQSkvMWrtt75FRyyBZBNApj            Allele  ...   \n",
       "2210490  ga4gh:CX.eR37N88iZGBhrAbApaJ8svu_dimnjyWD  CopyNumberChange  ...   \n",
       "2210491  ga4gh:CX.wAuyGaZw-vC4Xx4hFVf6574KhJ9w4nnv  CopyNumberChange  ...   \n",
       "2210553  ga4gh:CX.JmTrwmY_R9nyZ6Z-nGYfwk02VTmd8OpW  CopyNumberChange  ...   \n",
       "2210626  ga4gh:VA.3BXZ9oWeEqz4VM16yV8mgjvQ8ZTElB2J            Allele  ...   \n",
       "\n",
       "        out.canonical_spdi in.max_copies in.min_copies out.seq.inner_start  \\\n",
       "149                    NaN           NaN           NaN                 NaN   \n",
       "150                    NaN           NaN           NaN                 NaN   \n",
       "151                    NaN           NaN           NaN                 NaN   \n",
       "152                    NaN           NaN           NaN                 NaN   \n",
       "153                    NaN           NaN           NaN                 NaN   \n",
       "...                    ...           ...           ...                 ...   \n",
       "2199359                NaN           NaN           NaN                 NaN   \n",
       "2210490                NaN           NaN           NaN                 NaN   \n",
       "2210491                NaN           NaN           NaN                 NaN   \n",
       "2210553                NaN           NaN           NaN                 NaN   \n",
       "2210626                NaN           NaN           NaN                 NaN   \n",
       "\n",
       "        out.seq.inner_stop support_status error_string  error_string_reduce  \\\n",
       "149                    NaN           True                                     \n",
       "150                    NaN           True                                     \n",
       "151                    NaN           True                                     \n",
       "152                    NaN           True                                     \n",
       "153                    NaN           True                                     \n",
       "...                    ...            ...          ...                  ...   \n",
       "2199359                NaN           True                                     \n",
       "2210490                NaN           True                                     \n",
       "2210491                NaN           True                                     \n",
       "2210553                NaN           True                                     \n",
       "2210626                NaN           True                                     \n",
       "\n",
       "        error_string_reduce_2  normalize_status  \n",
       "149                                       False  \n",
       "150                                       False  \n",
       "151                                       False  \n",
       "152                                       False  \n",
       "153                                       False  \n",
       "...                       ...               ...  \n",
       "2199359                                   False  \n",
       "2210490                                   False  \n",
       "2210491                                   False  \n",
       "2210553                                   False  \n",
       "2210626                                   False  \n",
       "\n",
       "[80814 rows x 93 columns]"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "supported_not_normalized_df = supported_not_normalized_df.loc[\n",
    "    (supported_not_normalized_df[\"support_status\"] == True)\n",
    "    & (supported_not_normalized_df[\"normalize_status\"] == False)\n",
    "]\n",
    "supported_not_normalized_df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### <a id='toc1_4_3_'></a>[Not Supported Variants](#toc0_)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [],
   "source": [
    "not_supported_df = df.copy()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>in.id</th>\n",
       "      <th>in.name</th>\n",
       "      <th>in.subclass_type</th>\n",
       "      <th>in.variation_type</th>\n",
       "      <th>in.xrefs</th>\n",
       "      <th>in.vrs_xform_plan.type</th>\n",
       "      <th>in.vrs_xform_plan.inputs</th>\n",
       "      <th>in.vrs_xform_plan.policy</th>\n",
       "      <th>out.id</th>\n",
       "      <th>out.type</th>\n",
       "      <th>...</th>\n",
       "      <th>out.canonical_spdi</th>\n",
       "      <th>in.max_copies</th>\n",
       "      <th>in.min_copies</th>\n",
       "      <th>out.seq.inner_start</th>\n",
       "      <th>out.seq.inner_stop</th>\n",
       "      <th>support_status</th>\n",
       "      <th>error_string</th>\n",
       "      <th>error_string_reduce</th>\n",
       "      <th>error_string_reduce_2</th>\n",
       "      <th>normalize_status</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>0 rows Ã— 93 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [in.id, in.name, in.subclass_type, in.variation_type, in.xrefs, in.vrs_xform_plan.type, in.vrs_xform_plan.inputs, in.vrs_xform_plan.policy, out.id, out.type, out.definition, in.hgvs.assembly, in.hgvs.nucleotide, out.location.id, out.location.type, out.location.sequence_id, out.location.start.type, out.location.start.value, out.location.end.type, out.location.end.value, out.state.type, out.state.sequence, out.subject.id, out.subject.type, out.subject.sequence_id, out.subject.start.type, out.subject.start.value, out.subject.end.type, out.subject.end.value, out.copy_change, in.seq.disp_start, in.seq.inner_start, in.seq.chr, in.seq.inner_stop, in.seq.accession, in.seq.assembly, in.seq.disp_stop, in.seq.outer_stop, in.seq.outer_start, in.absolute_copies, in.derived_hgvs, in.cyto, out.subject.start.min, out.subject.start.max, out.subject.end.min, out.subject.end.max, out.name, out.subclass_type, out.variation_type, out.hgvs.assembly, out.hgvs.nucleotide, out.xrefs, out.vrs_xform_plan.type, out.vrs_xform_plan.inputs, out.vrs_xform_plan.policy, out.errors, out.subject.start.comparator, out.subject.end.comparator, out.copies.type, out.copies.value, in.seq.variant_length, in.seq.start, in.seq.stop, out.seq.assembly, out.seq.accession, out.seq.chr, out.seq.disp_start, out.seq.disp_stop, out.seq.outer_start, out.seq.outer_stop, out.absolute_copies, out.derived_hgvs, out.cyto, out.seq.variant_length, out.seq.start, out.seq.stop, in.seq.ref_allele_vcf, in.seq.position_vcf, in.seq.alt_allele_vcf, in.canonical_spdi, out.seq.ref_allele_vcf, out.seq.position_vcf, out.seq.alt_allele_vcf, out.canonical_spdi, in.max_copies, in.min_copies, out.seq.inner_start, out.seq.inner_stop, support_status, error_string, error_string_reduce, error_string_reduce_2, normalize_status]\n",
       "Index: []\n",
       "\n",
       "[0 rows x 93 columns]"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "not_supported_df = not_supported_df.loc[\n",
    "    (not_supported_df[\"support_status\"] == False)\n",
    "    & (not_supported_df[\"normalize_status\"] == False)\n",
    "]\n",
    "not_supported_df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Sanity check: making sure there are no supported variants that have been marked as normalized"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [],
   "source": [
    "not_supported_but_normalized_df = df.copy()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>in.id</th>\n",
       "      <th>in.name</th>\n",
       "      <th>in.subclass_type</th>\n",
       "      <th>in.variation_type</th>\n",
       "      <th>in.xrefs</th>\n",
       "      <th>in.vrs_xform_plan.type</th>\n",
       "      <th>in.vrs_xform_plan.inputs</th>\n",
       "      <th>in.vrs_xform_plan.policy</th>\n",
       "      <th>out.id</th>\n",
       "      <th>out.type</th>\n",
       "      <th>...</th>\n",
       "      <th>out.canonical_spdi</th>\n",
       "      <th>in.max_copies</th>\n",
       "      <th>in.min_copies</th>\n",
       "      <th>out.seq.inner_start</th>\n",
       "      <th>out.seq.inner_stop</th>\n",
       "      <th>support_status</th>\n",
       "      <th>error_string</th>\n",
       "      <th>error_string_reduce</th>\n",
       "      <th>error_string_reduce_2</th>\n",
       "      <th>normalize_status</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>16098</td>\n",
       "      <td>SLC2A2, 1-BP INS, 793C</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>Insertion</td>\n",
       "      <td>OMIM:138160.0009</td>\n",
       "      <td>Text</td>\n",
       "      <td>[id]</td>\n",
       "      <td>No hgvs or location info</td>\n",
       "      <td>Text:clinvar:16098</td>\n",
       "      <td>Text</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td></td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>425693</td>\n",
       "      <td>NM_001204.6(BMPR2):c.77-?_247+?dup</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>Duplication</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Text</td>\n",
       "      <td>[id]</td>\n",
       "      <td>No hgvs or location info</td>\n",
       "      <td>Text:clinvar:425693</td>\n",
       "      <td>Text</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td></td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2446408</td>\n",
       "      <td>CDHR1, 783G-A (rs147346345)</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>OMIM:609502.0005</td>\n",
       "      <td>Text</td>\n",
       "      <td>[id]</td>\n",
       "      <td>No hgvs or location info</td>\n",
       "      <td>Text:clinvar:2446408</td>\n",
       "      <td>Text</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td></td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>90650</td>\n",
       "      <td>NM_000251.2(MSH2):c.1387-?_1510+?del</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Text</td>\n",
       "      <td>[id]</td>\n",
       "      <td>No hgvs or location info</td>\n",
       "      <td>Text:clinvar:90650</td>\n",
       "      <td>Text</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td></td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>14905</td>\n",
       "      <td>HLA-DRB1, HLA-DRB1*1101</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>Variation</td>\n",
       "      <td>OMIM:142857.0001</td>\n",
       "      <td>Text</td>\n",
       "      <td>[id]</td>\n",
       "      <td>No hgvs or location info</td>\n",
       "      <td>Text:clinvar:14905</td>\n",
       "      <td>Text</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td></td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2199348</th>\n",
       "      <td>1418992</td>\n",
       "      <td>NM_002076.4(GNS):c.841_842insTTTTTTTTTTTTTTTTT...</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>Insertion</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Text</td>\n",
       "      <td>[id]</td>\n",
       "      <td>Invalid/unsupported hgvs</td>\n",
       "      <td>Text:clinvar:1418992</td>\n",
       "      <td>Text</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td></td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2199349</th>\n",
       "      <td>2134754</td>\n",
       "      <td>NM_152564.5(VPS13B):c.5627_5628insTTTTTTTTTTTT...</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>Insertion</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Text</td>\n",
       "      <td>[id]</td>\n",
       "      <td>Invalid/unsupported hgvs</td>\n",
       "      <td>Text:clinvar:2134754</td>\n",
       "      <td>Text</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td></td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2210621</th>\n",
       "      <td>1453116</td>\n",
       "      <td>NM_000051.4(ATM):c.3376_3377insGGCCGGGCGCGGTGG...</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>Insertion</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Text</td>\n",
       "      <td>[id]</td>\n",
       "      <td>Invalid/unsupported hgvs</td>\n",
       "      <td>Text:clinvar:1453116</td>\n",
       "      <td>Text</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td></td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2210623</th>\n",
       "      <td>1464440</td>\n",
       "      <td>NM_001267550.2(TTN):c.29512_29513insGGCCGGGCGC...</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>Insertion</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Text</td>\n",
       "      <td>[id]</td>\n",
       "      <td>Invalid/unsupported hgvs</td>\n",
       "      <td>Text:clinvar:1464440</td>\n",
       "      <td>Text</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td></td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2210624</th>\n",
       "      <td>1513408</td>\n",
       "      <td>NM_024928.5(STN1):c.340_352AAG[2]CTACAAGGCCGGG...</td>\n",
       "      <td>SimpleAllele</td>\n",
       "      <td>Insertion</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Text</td>\n",
       "      <td>[id]</td>\n",
       "      <td>Invalid/unsupported hgvs</td>\n",
       "      <td>Text:clinvar:1513408</td>\n",
       "      <td>Text</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td></td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>Not Supported</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>10650 rows Ã— 93 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           in.id                                            in.name  \\\n",
       "0          16098                             SLC2A2, 1-BP INS, 793C   \n",
       "1         425693                 NM_001204.6(BMPR2):c.77-?_247+?dup   \n",
       "2        2446408                        CDHR1, 783G-A (rs147346345)   \n",
       "3          90650               NM_000251.2(MSH2):c.1387-?_1510+?del   \n",
       "4          14905                            HLA-DRB1, HLA-DRB1*1101   \n",
       "...          ...                                                ...   \n",
       "2199348  1418992  NM_002076.4(GNS):c.841_842insTTTTTTTTTTTTTTTTT...   \n",
       "2199349  2134754  NM_152564.5(VPS13B):c.5627_5628insTTTTTTTTTTTT...   \n",
       "2210621  1453116  NM_000051.4(ATM):c.3376_3377insGGCCGGGCGCGGTGG...   \n",
       "2210623  1464440  NM_001267550.2(TTN):c.29512_29513insGGCCGGGCGC...   \n",
       "2210624  1513408  NM_024928.5(STN1):c.340_352AAG[2]CTACAAGGCCGGG...   \n",
       "\n",
       "        in.subclass_type          in.variation_type          in.xrefs  \\\n",
       "0           SimpleAllele                  Insertion  OMIM:138160.0009   \n",
       "1           SimpleAllele                Duplication               NaN   \n",
       "2           SimpleAllele  single nucleotide variant  OMIM:609502.0005   \n",
       "3           SimpleAllele                   Deletion               NaN   \n",
       "4           SimpleAllele                  Variation  OMIM:142857.0001   \n",
       "...                  ...                        ...               ...   \n",
       "2199348     SimpleAllele                  Insertion               NaN   \n",
       "2199349     SimpleAllele                  Insertion               NaN   \n",
       "2210621     SimpleAllele                  Insertion               NaN   \n",
       "2210623     SimpleAllele                  Insertion               NaN   \n",
       "2210624     SimpleAllele                  Insertion               NaN   \n",
       "\n",
       "        in.vrs_xform_plan.type in.vrs_xform_plan.inputs  \\\n",
       "0                         Text                     [id]   \n",
       "1                         Text                     [id]   \n",
       "2                         Text                     [id]   \n",
       "3                         Text                     [id]   \n",
       "4                         Text                     [id]   \n",
       "...                        ...                      ...   \n",
       "2199348                   Text                     [id]   \n",
       "2199349                   Text                     [id]   \n",
       "2210621                   Text                     [id]   \n",
       "2210623                   Text                     [id]   \n",
       "2210624                   Text                     [id]   \n",
       "\n",
       "         in.vrs_xform_plan.policy                out.id out.type  ...  \\\n",
       "0        No hgvs or location info    Text:clinvar:16098     Text  ...   \n",
       "1        No hgvs or location info   Text:clinvar:425693     Text  ...   \n",
       "2        No hgvs or location info  Text:clinvar:2446408     Text  ...   \n",
       "3        No hgvs or location info    Text:clinvar:90650     Text  ...   \n",
       "4        No hgvs or location info    Text:clinvar:14905     Text  ...   \n",
       "...                           ...                   ...      ...  ...   \n",
       "2199348  Invalid/unsupported hgvs  Text:clinvar:1418992     Text  ...   \n",
       "2199349  Invalid/unsupported hgvs  Text:clinvar:2134754     Text  ...   \n",
       "2210621  Invalid/unsupported hgvs  Text:clinvar:1453116     Text  ...   \n",
       "2210623  Invalid/unsupported hgvs  Text:clinvar:1464440     Text  ...   \n",
       "2210624  Invalid/unsupported hgvs  Text:clinvar:1513408     Text  ...   \n",
       "\n",
       "        out.canonical_spdi in.max_copies in.min_copies out.seq.inner_start  \\\n",
       "0                      NaN           NaN           NaN                 NaN   \n",
       "1                      NaN           NaN           NaN                 NaN   \n",
       "2                      NaN           NaN           NaN                 NaN   \n",
       "3                      NaN           NaN           NaN                 NaN   \n",
       "4                      NaN           NaN           NaN                 NaN   \n",
       "...                    ...           ...           ...                 ...   \n",
       "2199348                NaN           NaN           NaN                 NaN   \n",
       "2199349                NaN           NaN           NaN                 NaN   \n",
       "2210621                NaN           NaN           NaN                 NaN   \n",
       "2210623                NaN           NaN           NaN                 NaN   \n",
       "2210624                NaN           NaN           NaN                 NaN   \n",
       "\n",
       "        out.seq.inner_stop support_status error_string  error_string_reduce  \\\n",
       "0                      NaN          False                     Not Supported   \n",
       "1                      NaN          False                     Not Supported   \n",
       "2                      NaN          False                     Not Supported   \n",
       "3                      NaN          False                     Not Supported   \n",
       "4                      NaN          False                     Not Supported   \n",
       "...                    ...            ...          ...                  ...   \n",
       "2199348                NaN          False                     Not Supported   \n",
       "2199349                NaN          False                     Not Supported   \n",
       "2210621                NaN          False                     Not Supported   \n",
       "2210623                NaN          False                     Not Supported   \n",
       "2210624                NaN          False                     Not Supported   \n",
       "\n",
       "        error_string_reduce_2  normalize_status  \n",
       "0               Not Supported              True  \n",
       "1               Not Supported              True  \n",
       "2               Not Supported              True  \n",
       "3               Not Supported              True  \n",
       "4               Not Supported              True  \n",
       "...                       ...               ...  \n",
       "2199348         Not Supported              True  \n",
       "2199349         Not Supported              True  \n",
       "2210621         Not Supported              True  \n",
       "2210623         Not Supported              True  \n",
       "2210624         Not Supported              True  \n",
       "\n",
       "[10650 rows x 93 columns]"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "not_supported_but_normalized_df = not_supported_but_normalized_df.loc[\n",
    "    (not_supported_but_normalized_df[\"support_status\"] == False)\n",
    "    & (not_supported_but_normalized_df[\"normalize_status\"] == True)\n",
    "]\n",
    "not_supported_but_normalized_df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## <a id='toc1_5_'></a>[Counting variants from each group](#toc0_)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [],
   "source": [
    "num_supported = len(supported_df)\n",
    "num_supported_not_normalized = len(supported_not_normalized_df)\n",
    "num_not_supported_but_normalized = len(not_supported_but_normalized_df)\n",
    "num_not_supported = len(not_supported_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2119163\n",
      "80814\n",
      "10650\n",
      "0\n"
     ]
    }
   ],
   "source": [
    "print(num_supported)\n",
    "print(num_supported_not_normalized)\n",
    "print(num_not_supported_but_normalized)\n",
    "print(num_not_supported)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3.10.9 ('.venv': venv)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.9"
  },
  "orig_nbformat": 4,
  "vscode": {
   "interpreter": {
    "hash": "60bedfc6dd95a1d756c285a8080496495ec396023cb257f004626f94ae504b38"
   }
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
